// @alfresco/js-api/_esm2015/src/api/content-rest-api/api/groups.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { GroupEntry } from '../model/groupEntry';
import { GroupMemberEntry } from '../model/groupMemberEntry';
import { GroupMemberPaging } from '../model/groupMemberPaging';
import { GroupPaging } from '../model/groupPaging';
import { BaseApi } from './base.api';
export class GroupsApi extends BaseApi {
    createGroup(groupBodyCreate, opts) {
        opts = opts || {};
        let postBody = groupBodyCreate;
        if (groupBodyCreate === undefined || groupBodyCreate === null) {
            throw new Error("Required param 'groupBodyCreate' in createGroup");
        }
        let pathParams = {};
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    }
    createGroupMembership(groupId, groupMembershipBodyCreate, opts) {
        opts = opts || {};
        let postBody = groupMembershipBodyCreate;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in createGroupMembership");
        }
        if (groupMembershipBodyCreate === undefined || groupMembershipBodyCreate === null) {
            throw new Error("Required param 'groupMembershipBodyCreate' in createGroupMembership");
        }
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupMemberEntry);
    }
    deleteGroup(groupId, opts) {
        opts = opts || {};
        let postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in deleteGroup");
        }
        let cascadeDelete = opts['cascade'] ? opts['cascade'] : false;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'cascade': cascadeDelete
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteGroupMembership(groupId, groupMemberId) {
        let postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in deleteGroupMembership");
        }
        if (groupMemberId === undefined || groupMemberId === null) {
            throw new Error("Required param 'groupMemberId' in deleteGroupMembership");
        }
        let pathParams = {
            'groupId': groupId, 'groupMemberId': groupMemberId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members/{groupMemberId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getGroup(groupId, opts) {
        opts = opts || {};
        let postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in getGroup");
        }
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    }
    listGroupMemberships(groupId, opts) {
        opts = opts || {};
        let postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in listGroupMemberships");
        }
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupMemberPaging);
    }
    listGroupMembershipsForPerson(personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listGroupMembershipsForPerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupPaging);
    }
    listGroups(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupPaging);
    }
    updateGroup(groupId, groupBodyUpdate, opts) {
        opts = opts || {};
        let postBody = groupBodyUpdate;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in updateGroup");
        }
        if (groupBodyUpdate === undefined || groupBodyUpdate === null) {
            throw new Error("Required param 'groupBodyUpdate' in updateGroup");
        }
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    }
}
