// @alfresco/js-api/_esm2015/src/api/content-rest-api/api/people.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { PersonEntry } from '../model/personEntry';
import { PersonPaging } from '../model/personPaging';
import { BaseApi } from './base.api';
export class PeopleApi extends BaseApi {
    createPerson(personBodyCreate, opts) {
        opts = opts || {};
        let postBody = personBodyCreate;
        if (personBodyCreate === undefined || personBodyCreate === null) {
            throw new Error("Required param 'personBodyCreate' in createPerson");
        }
        let pathParams = {};
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    }
    deleteAvatarImage(personId) {
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteAvatarImage");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getAvatarImage(personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getAvatarImage");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'attachment': opts['attachment'],
            'placeholder': opts['placeholder']
        };
        let headerParams = {
            'If-Modified-Since': opts['ifModifiedSince']
        };
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getPerson(personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getPerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    }
    listPeople(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonPaging);
    }
    requestPasswordReset(personId, clientBody) {
        let postBody = clientBody;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in requestPasswordReset");
        }
        if (clientBody === undefined || clientBody === null) {
            throw new Error("Required param 'clientBody' in requestPasswordReset");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/request-password-reset', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    resetPassword(personId, passwordResetBody) {
        let postBody = passwordResetBody;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in resetPassword");
        }
        if (passwordResetBody === undefined || passwordResetBody === null) {
            throw new Error("Required param 'passwordResetBody' in resetPassword");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/reset-password', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    updateAvatarImage(personId, contentBodyUpdate) {
        let postBody = contentBodyUpdate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updateAvatarImage");
        }
        if (contentBodyUpdate === undefined || contentBodyUpdate === null) {
            throw new Error("Required param 'contentBodyUpdate' in updateAvatarImage");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/octet-stream'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    updatePerson(personId, personBodyUpdate, opts) {
        opts = opts || {};
        let postBody = personBodyUpdate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updatePerson");
        }
        if (personBodyUpdate === undefined || personBodyUpdate === null) {
            throw new Error("Required param 'personBodyUpdate' in updatePerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    }
}
