// @alfresco/js-api/_esm2015/src/api/content-rest-api/api/versions.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { VersionEntry } from '../model/versionEntry';
import { VersionPaging } from '../model/versionPaging';
import { BaseApi } from './base.api';
export class VersionsApi extends BaseApi {
    deleteVersion(nodeId, versionId) {
        let postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in deleteVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in deleteVersion");
        }
        let pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getVersion(nodeId, versionId) {
        let postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in getVersion");
        }
        let pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionEntry);
    }
    getVersionContent(nodeId, versionId, opts) {
        opts = opts || {};
        let postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getVersionContent");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in getVersionContent");
        }
        let pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        let queryParams = {
            'attachment': opts['attachment']
        };
        let headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    listVersionHistory(nodeId, opts) {
        opts = opts || {};
        let postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listVersionHistory");
        }
        let pathParams = {
            'nodeId': nodeId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionPaging);
    }
    revertVersion(nodeId, versionId, revertBody, opts) {
        opts = opts || {};
        let postBody = revertBody;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in revertVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in revertVersion");
        }
        if (revertBody === undefined || revertBody === null) {
            throw new Error("Required param 'revertBody' in revertVersion");
        }
        let pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/revert', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionEntry);
    }
}
