// @alfresco/js-api/_esm2015/src/api/activiti-rest-api/api/appDefinitions.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { AppDefinitionRepresentation } from '../model/appDefinitionRepresentation';
import { AppDefinitionUpdateResultRepresentation } from '../model/appDefinitionUpdateResultRepresentation';
import { BaseApi } from './base.api';
export class AppDefinitionsApi extends BaseApi {
    deleteAppDefinition(appDefinitionId) {
        let postBody = null;
        if (appDefinitionId === undefined || appDefinitionId === null) {
            throw new Error("Required param 'appDefinitionId' in deleteAppDefinition");
        }
        let pathParams = {
            'appDefinitionId': appDefinitionId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/{appDefinitionId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    exportAppDefinition(modelId) {
        let postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in exportAppDefinition");
        }
        let pathParams = {
            'modelId': modelId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json', 'application/zip'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/{modelId}/export', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getAppDefinition(modelId) {
        let postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in getAppDefinition");
        }
        let pathParams = {
            'modelId': modelId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/{modelId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDefinitionRepresentation);
    }
    importAndPublishApp(file) {
        let postBody = null;
        if (file === undefined || file === null) {
            throw new Error("Required param 'file' in importAndPublishApp");
        }
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {
            'file': file
        };
        let contentTypes = ['multipart/form-data'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/publish-app', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDefinitionUpdateResultRepresentation);
    }
    importAppDefinition(file, opts) {
        opts = opts || {};
        let postBody = null;
        if (file === undefined || file === null) {
            throw new Error("Required param 'file' in importAppDefinition");
        }
        let pathParams = {};
        let queryParams = {
            'renewIdmEntries': opts['renewIdmEntries']
        };
        let headerParams = {};
        let formParams = {
            'file': file
        };
        let contentTypes = ['multipart/form-data'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/import', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDefinitionRepresentation);
    }
    publishAppDefinition(modelId, publishModel) {
        let postBody = publishModel;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in publishAppDefinition");
        }
        if (publishModel === undefined || publishModel === null) {
            throw new Error("Required param 'publishModel' in publishAppDefinition");
        }
        let pathParams = {
            'modelId': modelId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/app-definitions/{modelId}/publish', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDefinitionUpdateResultRepresentation);
    }
    updateAppDefinition(modelId, updatedModel) {
        let postBody = updatedModel;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in updateAppDefinition");
        }
        if (updatedModel === undefined || updatedModel === null) {
            throw new Error("Required param 'updatedModel' in updateAppDefinition");
        }
        let pathParams = {
            'modelId': modelId
        };
        let queryParams = {};
        let headerParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        if (!updatedModel['appDefinition']) {
            contentTypes = ['multipart/form-data'];
            let formParams = {
                'file': updatedModel
            };
            return this.apiClient.callApi('/api/enterprise/app-definitions/{modelId}/import', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
        }
        else {
            let formParams = {};
            return this.apiClient.callApi('/api/enterprise/app-definitions/{modelId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
        }
    }
}
