// @alfresco/js-api/src/api/content-rest-api/api/audit.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
const auditApp_1 = require("../model/auditApp");
const auditAppPaging_1 = require("../model/auditAppPaging");
const auditEntryEntry_1 = require("../model/auditEntryEntry");
const auditEntryPaging_1 = require("../model/auditEntryPaging");
const base_api_1 = require("./base.api");
class AuditApi extends base_api_1.BaseApi {
    deleteAuditEntriesForAuditApp(auditApplicationId, where) {
        let postBody = null;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in deleteAuditEntriesForAuditApp");
        }
        if (where === undefined || where === null) {
            throw new Error("Required param 'where' in deleteAuditEntriesForAuditApp");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId
        };
        let queryParams = {
            'where': where
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}/audit-entries', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteAuditEntry(auditApplicationId, auditEntryId) {
        let postBody = null;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in deleteAuditEntry");
        }
        if (auditEntryId === undefined || auditEntryId === null) {
            throw new Error("Required param 'auditEntryId' in deleteAuditEntry");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId, 'auditEntryId': auditEntryId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}/audit-entries/{auditEntryId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getAuditApp(auditApplicationId, opts) {
        opts = opts || {};
        let postBody = null;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in getAuditApp");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditApp_1.AuditApp);
    }
    getAuditEntry(auditApplicationId, auditEntryId, opts) {
        opts = opts || {};
        let postBody = null;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in getAuditEntry");
        }
        if (auditEntryId === undefined || auditEntryId === null) {
            throw new Error("Required param 'auditEntryId' in getAuditEntry");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId, 'auditEntryId': auditEntryId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}/audit-entries/{auditEntryId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditEntryEntry_1.AuditEntryEntry);
    }
    listAuditApps(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditAppPaging_1.AuditAppPaging);
    }
    listAuditEntriesForAuditApp(auditApplicationId, opts) {
        opts = opts || {};
        let postBody = null;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in listAuditEntriesForAuditApp");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}/audit-entries', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditEntryPaging_1.AuditEntryPaging);
    }
    listAuditEntriesForNode(nodeId, opts) {
        opts = opts || {};
        let postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listAuditEntriesForNode");
        }
        let pathParams = {
            'nodeId': nodeId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/audit-entries', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditEntryPaging_1.AuditEntryPaging);
    }
    updateAuditApp(auditApplicationId, auditAppBodyUpdate, opts) {
        opts = opts || {};
        let postBody = auditAppBodyUpdate;
        if (auditApplicationId === undefined || auditApplicationId === null) {
            throw new Error("Required param 'auditApplicationId' in updateAuditApp");
        }
        if (auditAppBodyUpdate === undefined || auditAppBodyUpdate === null) {
            throw new Error("Required param 'auditAppBodyUpdate' in updateAuditApp");
        }
        let pathParams = {
            'auditApplicationId': auditApplicationId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/audit-applications/{auditApplicationId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, auditApp_1.AuditApp);
    }
}
exports.AuditApi = AuditApi;
