// @alfresco/js-api/src/api/content-rest-api/api/sites.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
const siteContainerEntry_1 = require("../model/siteContainerEntry");
const siteContainerPaging_1 = require("../model/siteContainerPaging");
const siteEntry_1 = require("../model/siteEntry");
const siteMemberEntry_1 = require("../model/siteMemberEntry");
const siteMemberPaging_1 = require("../model/siteMemberPaging");
const siteMembershipRequestEntry_1 = require("../model/siteMembershipRequestEntry");
const siteMembershipRequestPaging_1 = require("../model/siteMembershipRequestPaging");
const siteMembershipRequestWithPersonPaging_1 = require("../model/siteMembershipRequestWithPersonPaging");
const sitePaging_1 = require("../model/sitePaging");
const siteRoleEntry_1 = require("../model/siteRoleEntry");
const siteRolePaging_1 = require("../model/siteRolePaging");
const base_api_1 = require("./base.api");
class SitesApi extends base_api_1.BaseApi {
    approveSiteMembershipRequest(siteId, inviteeId, opts) {
        opts = opts || {};
        let postBody = opts['siteMembershipApprovalBody'];
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in approveSiteMembershipRequest");
        }
        if (inviteeId === undefined || inviteeId === null) {
            throw new Error("Required param 'inviteeId' in approveSiteMembershipRequest");
        }
        let pathParams = {
            'siteId': siteId, 'inviteeId': inviteeId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/site-membership-requests/{inviteeId}/approve', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    createSite(siteBodyCreate, opts) {
        opts = opts || {};
        let postBody = siteBodyCreate;
        if (siteBodyCreate === undefined || siteBodyCreate === null) {
            throw new Error("Required param 'siteBodyCreate' in createSite");
        }
        let pathParams = {};
        let queryParams = {
            'skipConfiguration': opts['skipConfiguration'],
            'skipAddToFavorites': opts['skipAddToFavorites'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteEntry_1.SiteEntry);
    }
    createSiteMembership(siteId, siteMembershipBodyCreate, opts) {
        opts = opts || {};
        let postBody = siteMembershipBodyCreate;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in createSiteMembership");
        }
        if (siteMembershipBodyCreate === undefined || siteMembershipBodyCreate === null) {
            throw new Error("Required param 'siteMembershipBodyCreate' in createSiteMembership");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/members', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMemberEntry_1.SiteMemberEntry);
    }
    createSiteMembershipRequestForPerson(personId, siteMembershipRequestBodyCreate, opts) {
        opts = opts || {};
        let postBody = siteMembershipRequestBodyCreate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in createSiteMembershipRequestForPerson");
        }
        if (siteMembershipRequestBodyCreate === undefined || siteMembershipRequestBodyCreate === null) {
            throw new Error("Required param 'siteMembershipRequestBodyCreate' in createSiteMembershipRequestForPerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/site-membership-requests', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMembershipRequestEntry_1.SiteMembershipRequestEntry);
    }
    deleteSite(siteId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in deleteSite");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'permanent': opts['permanent']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteSiteMembership(siteId, personId) {
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in deleteSiteMembership");
        }
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteSiteMembership");
        }
        let pathParams = {
            'siteId': siteId, 'personId': personId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/members/{personId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteSiteMembershipForPerson(personId, siteId) {
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteSiteMembershipForPerson");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in deleteSiteMembershipForPerson");
        }
        let pathParams = {
            'personId': personId, 'siteId': siteId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/sites/{siteId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteSiteMembershipRequestForPerson(personId, siteId) {
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteSiteMembershipRequestForPerson");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in deleteSiteMembershipRequestForPerson");
        }
        let pathParams = {
            'personId': personId, 'siteId': siteId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/site-membership-requests/{siteId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getSite(siteId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getSite");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'relations': this.apiClient.buildCollectionParam(opts['relations'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteEntry_1.SiteEntry);
    }
    getSiteContainer(siteId, containerId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getSiteContainer");
        }
        if (containerId === undefined || containerId === null) {
            throw new Error("Required param 'containerId' in getSiteContainer");
        }
        let pathParams = {
            'siteId': siteId, 'containerId': containerId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/containers/{containerId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteContainerEntry_1.SiteContainerEntry);
    }
    getSiteMembership(siteId, personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getSiteMembership");
        }
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getSiteMembership");
        }
        let pathParams = {
            'siteId': siteId, 'personId': personId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/members/{personId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMemberEntry_1.SiteMemberEntry);
    }
    getSiteMembershipForPerson(personId, siteId) {
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getSiteMembershipForPerson");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getSiteMembershipForPerson");
        }
        let pathParams = {
            'personId': personId, 'siteId': siteId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/sites/{siteId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteRoleEntry_1.SiteRoleEntry);
    }
    getSiteMembershipRequestForPerson(personId, siteId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getSiteMembershipRequestForPerson");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getSiteMembershipRequestForPerson");
        }
        let pathParams = {
            'personId': personId, 'siteId': siteId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/site-membership-requests/{siteId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMembershipRequestEntry_1.SiteMembershipRequestEntry);
    }
    getSiteMembershipRequests(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/site-membership-requests', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMembershipRequestWithPersonPaging_1.SiteMembershipRequestWithPersonPaging);
    }
    listSiteContainers(siteId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in listSiteContainers");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/containers', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteContainerPaging_1.SiteContainerPaging);
    }
    listSiteMembershipRequestsForPerson(personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listSiteMembershipRequestsForPerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/site-membership-requests', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMembershipRequestPaging_1.SiteMembershipRequestPaging);
    }
    listSiteMemberships(siteId, opts) {
        opts = opts || {};
        let postBody = null;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in listSiteMemberships");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/members', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMemberPaging_1.SiteMemberPaging);
    }
    listSiteMembershipsForPerson(personId, opts) {
        opts = opts || {};
        let postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listSiteMembershipsForPerson");
        }
        let pathParams = {
            'personId': personId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'relations': this.apiClient.buildCollectionParam(opts['relations'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv'),
            'where': opts['where']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/sites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteRolePaging_1.SiteRolePaging);
    }
    listSites(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'relations': this.apiClient.buildCollectionParam(opts['relations'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv'),
            'where': opts['where']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, sitePaging_1.SitePaging);
    }
    rejectSiteMembershipRequest(siteId, inviteeId, opts) {
        opts = opts || {};
        let postBody = opts['siteMembershipRejectionBody'];
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in rejectSiteMembershipRequest");
        }
        if (inviteeId === undefined || inviteeId === null) {
            throw new Error("Required param 'inviteeId' in rejectSiteMembershipRequest");
        }
        let pathParams = {
            'siteId': siteId, 'inviteeId': inviteeId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/site-membership-requests/{inviteeId}/reject', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    updateSite(siteId, siteBodyUpdate, opts) {
        opts = opts || {};
        let postBody = siteBodyUpdate;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in updateSite");
        }
        if (siteBodyUpdate === undefined || siteBodyUpdate === null) {
            throw new Error("Required param 'siteBodyUpdate' in updateSite");
        }
        let pathParams = {
            'siteId': siteId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteEntry_1.SiteEntry);
    }
    updateSiteMembership(siteId, personId, siteMembershipBodyUpdate, opts) {
        opts = opts || {};
        let postBody = siteMembershipBodyUpdate;
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in updateSiteMembership");
        }
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updateSiteMembership");
        }
        if (siteMembershipBodyUpdate === undefined || siteMembershipBodyUpdate === null) {
            throw new Error("Required param 'siteMembershipBodyUpdate' in updateSiteMembership");
        }
        let pathParams = {
            'siteId': siteId, 'personId': personId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/sites/{siteId}/members/{personId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMemberEntry_1.SiteMemberEntry);
    }
    updateSiteMembershipRequestForPerson(personId, siteId, siteMembershipRequestBodyUpdate, opts) {
        opts = opts || {};
        let postBody = siteMembershipRequestBodyUpdate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updateSiteMembershipRequestForPerson");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in updateSiteMembershipRequestForPerson");
        }
        if (siteMembershipRequestBodyUpdate === undefined || siteMembershipRequestBodyUpdate === null) {
            throw new Error("Required param 'siteMembershipRequestBodyUpdate' in updateSiteMembershipRequestForPerson");
        }
        let pathParams = {
            'personId': personId, 'siteId': siteId
        };
        let queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/site-membership-requests/{siteId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, siteMembershipRequestEntry_1.SiteMembershipRequestEntry);
    }
}
exports.SitesApi = SitesApi;
