// @alfresco/adf-core/esm5/context-menu/context-menu.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:no-input-rename  */
import { Directive, HostListener, Input } from '@angular/core';
import { ContextMenuOverlayService } from './context-menu-overlay.service';
var ContextMenuDirective = /** @class */ (function () {
    function ContextMenuDirective(contextMenuService) {
        this.contextMenuService = contextMenuService;
        /**
         * Is the menu enabled?
         */
        this.enabled = false;
    }
    /**
     * @param {?=} event
     * @return {?}
     */
    ContextMenuDirective.prototype.onShowContextMenu = /**
     * @param {?=} event
     * @return {?}
     */
    function (event) {
        if (this.enabled) {
            if (event) {
                event.preventDefault();
            }
            if (this.links && this.links.length > 0) {
                this.contextMenuService.open({
                    source: event,
                    data: this.links
                });
            }
        }
    };
    ContextMenuDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-context-menu], [context-menu]'
                },] }
    ];
    /** @nocollapse */
    ContextMenuDirective.ctorParameters = function () { return [
        { type: ContextMenuOverlayService }
    ]; };
    ContextMenuDirective.propDecorators = {
        links: [{ type: Input, args: ['adf-context-menu',] }],
        enabled: [{ type: Input, args: ['adf-context-menu-enabled',] }],
        onShowContextMenu: [{ type: HostListener, args: ['contextmenu', ['$event'],] }]
    };
    return ContextMenuDirective;
}());
export { ContextMenuDirective };
if (false) {
    /**
     * Items for the menu.
     * @type {?}
     */
    ContextMenuDirective.prototype.links;
    /**
     * Is the menu enabled?
     * @type {?}
     */
    ContextMenuDirective.prototype.enabled;
    /**
     * @type {?}
     * @private
     */
    ContextMenuDirective.prototype.contextMenuService;
}
