// @alfresco/adf-core/esm5/context-menu/context-menu-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, HostListener, Optional, Inject, QueryList, ViewChildren } from '@angular/core';
import { trigger } from '@angular/animations';
import { DOWN_ARROW, UP_ARROW } from '@angular/cdk/keycodes';
import { FocusKeyManager } from '@angular/cdk/a11y';
import { MatMenuItem } from '@angular/material';
import { ContextMenuOverlayRef } from './context-menu-overlay';
import { contextMenuAnimation } from './animations';
import { CONTEXT_MENU_DATA } from './context-menu.tokens';
var ContextMenuListComponent = /** @class */ (function () {
    function ContextMenuListComponent(contextMenuOverlayRef, data) {
        this.contextMenuOverlayRef = contextMenuOverlayRef;
        this.data = data;
        this.links = this.data;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    ContextMenuListComponent.prototype.handleKeydownEscape = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (event) {
            this.contextMenuOverlayRef.close();
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ContextMenuListComponent.prototype.handleKeydownEvent = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (event) {
            /** @type {?} */
            var keyCode = event.keyCode;
            if (keyCode === UP_ARROW || keyCode === DOWN_ARROW) {
                this.keyManager.onKeydown(event);
            }
        }
    };
    /**
     * @param {?} event
     * @param {?} menuItem
     * @return {?}
     */
    ContextMenuListComponent.prototype.onMenuItemClick = /**
     * @param {?} event
     * @param {?} menuItem
     * @return {?}
     */
    function (event, menuItem) {
        if (menuItem && menuItem.model && menuItem.model.disabled) {
            event.preventDefault();
            event.stopImmediatePropagation();
            return;
        }
        menuItem.subject.next(menuItem);
        this.contextMenuOverlayRef.close();
    };
    /**
     * @return {?}
     */
    ContextMenuListComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.keyManager = new FocusKeyManager(this.items);
        this.keyManager.setFirstItemActive();
    };
    ContextMenuListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-context-menu',
                    template: "\n        <div mat-menu class=\"mat-menu-panel\" @panelAnimation>\n            <div id=\"adf-context-menu-content\" class=\"mat-menu-content\">\n                <ng-container *ngFor=\"let link of links\">\n                    <button *ngIf=\"link.model?.visible\"\n                            [attr.data-automation-id]=\"'context-'+((link.title || link.model?.title) | translate)\"\n                            mat-menu-item\n                            [disabled]=\"link.model?.disabled\"\n                            (click)=\"onMenuItemClick($event, link)\">\n                        <mat-icon *ngIf=\"link.model?.icon\">{{ link.model.icon }}</mat-icon>\n                        <span>{{ (link.title || link.model?.title) | translate }}</span>\n                    </button>\n                </ng-container>\n            </div>\n        </div>\n    ",
                    host: {
                        role: 'menu',
                        class: 'adf-context-menu'
                    },
                    encapsulation: ViewEncapsulation.None,
                    animations: [
                        trigger('panelAnimation', contextMenuAnimation)
                    ]
                }] }
    ];
    /** @nocollapse */
    ContextMenuListComponent.ctorParameters = function () { return [
        { type: ContextMenuOverlayRef, decorators: [{ type: Inject, args: [ContextMenuOverlayRef,] }] },
        { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [CONTEXT_MENU_DATA,] }] }
    ]; };
    ContextMenuListComponent.propDecorators = {
        items: [{ type: ViewChildren, args: [MatMenuItem,] }],
        handleKeydownEscape: [{ type: HostListener, args: ['document:keydown.Escape', ['$event'],] }],
        handleKeydownEvent: [{ type: HostListener, args: ['document:keydown', ['$event'],] }]
    };
    return ContextMenuListComponent;
}());
export { ContextMenuListComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ContextMenuListComponent.prototype.keyManager;
    /** @type {?} */
    ContextMenuListComponent.prototype.items;
    /** @type {?} */
    ContextMenuListComponent.prototype.links;
    /**
     * @type {?}
     * @private
     */
    ContextMenuListComponent.prototype.contextMenuOverlayRef;
    /**
     * @type {?}
     * @private
     */
    ContextMenuListComponent.prototype.data;
}
