// @alfresco/adf-content-services/esm5/search/components/search-chip-list/search-chip-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input } from '@angular/core';
import { SearchFilterComponent } from '../../components/search-filter/search-filter.component';
var SearchChipListComponent = /** @class */ (function () {
    function SearchChipListComponent() {
        /**
         * Flag used to enable the display of a clear-all-filters button.
         */
        this.clearAll = false;
    }
    SearchChipListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-search-chip-list',
                    template: "<mat-chip-list>\n    <ng-container *ngIf=\"searchFilter && searchFilter.selectedBuckets.length\">\n        <mat-chip *ngIf=\"clearAll && searchFilter.selectedBuckets.length > 1\"\n                  color=\"primary\"\n                  selected\n                  matTooltip=\"{{ 'SEARCH.FILTER.BUTTONS.CLEAR-ALL.TOOLTIP' | translate }}\"\n                  matTooltipPosition=\"right\"\n                  (click)=\"searchFilter.resetAllSelectedBuckets()\">\n            {{ 'SEARCH.FILTER.BUTTONS.CLEAR-ALL.LABEL' | translate }}\n        </mat-chip>\n\n        <mat-chip\n            *ngFor=\"let selection of searchFilter.selectedBuckets\"\n            [removable]=\"true\"\n            (removed)=\"searchFilter.unselectFacetBucket(selection.field, selection.bucket)\">\n            {{ (selection.bucket.display || selection.bucket.label) | translate }}\n            <mat-icon matChipRemove>cancel</mat-icon>\n        </mat-chip>\n    </ng-container>\n</mat-chip-list>\n",
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-search-chip-list' }
                }] }
    ];
    SearchChipListComponent.propDecorators = {
        searchFilter: [{ type: Input }],
        clearAll: [{ type: Input }]
    };
    return SearchChipListComponent;
}());
export { SearchChipListComponent };
if (false) {
    /**
     * Search filter to supply the data for the chips.
     * @type {?}
     */
    SearchChipListComponent.prototype.searchFilter;
    /**
     * Flag used to enable the display of a clear-all-filters button.
     * @type {?}
     */
    SearchChipListComponent.prototype.clearAll;
}
