// @alfresco/adf-content-services/esm5/content-node-share/content-node-share.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreModule } from '@alfresco/adf-core';
import { MaterialModule } from '../material.module';
import { ShareDialogComponent } from './content-node-share.dialog';
import { NodeSharedDirective } from './content-node-share.directive';
var ContentNodeShareModule = /** @class */ (function () {
    function ContentNodeShareModule() {
    }
    /**
     * @return {?}
     */
    ContentNodeShareModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: ContentNodeShareModule
        };
    };
    /**
     * @return {?}
     */
    ContentNodeShareModule.forChild = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: ContentNodeShareModule
        };
    };
    ContentNodeShareModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CoreModule.forChild(),
                        CommonModule,
                        MaterialModule
                    ],
                    declarations: [
                        ShareDialogComponent,
                        NodeSharedDirective
                    ],
                    exports: [
                        ShareDialogComponent,
                        NodeSharedDirective
                    ],
                    entryComponents: [
                        ShareDialogComponent
                    ]
                },] }
    ];
    return ContentNodeShareModule;
}());
export { ContentNodeShareModule };
