// @alfresco/adf-content-services/esm5/content-metadata/services/property-descriptors.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { AlfrescoApiService } from '@alfresco/adf-core';
import { defer, forkJoin } from 'rxjs';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
var PropertyDescriptorsService = /** @class */ (function () {
    function PropertyDescriptorsService(alfrescoApiService) {
        this.alfrescoApiService = alfrescoApiService;
    }
    /**
     * @param {?} groupNames
     * @return {?}
     */
    PropertyDescriptorsService.prototype.load = /**
     * @param {?} groupNames
     * @return {?}
     */
    function (groupNames) {
        var _this = this;
        /** @type {?} */
        var groupFetchStreams = groupNames
            .map((/**
         * @param {?} groupName
         * @return {?}
         */
        function (groupName) { return groupName.replace(':', '_'); }))
            .map((/**
         * @param {?} groupName
         * @return {?}
         */
        function (groupName) { return defer((/**
         * @return {?}
         */
        function () { return _this.alfrescoApiService.classesApi.getClass(groupName); })); }));
        return forkJoin(groupFetchStreams).pipe(map(this.convertToObject));
    };
    /**
     * @private
     * @param {?} propertyGroupsArray
     * @return {?}
     */
    PropertyDescriptorsService.prototype.convertToObject = /**
     * @private
     * @param {?} propertyGroupsArray
     * @return {?}
     */
    function (propertyGroupsArray) {
        return propertyGroupsArray.reduce((/**
         * @param {?} propertyGroups
         * @param {?} propertyGroup
         * @return {?}
         */
        function (propertyGroups, propertyGroup) {
            var _a;
            return Object.assign({}, propertyGroups, (_a = {},
                _a[propertyGroup.name] = propertyGroup,
                _a));
        }), {});
    };
    PropertyDescriptorsService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    PropertyDescriptorsService.ctorParameters = function () { return [
        { type: AlfrescoApiService }
    ]; };
    /** @nocollapse */ PropertyDescriptorsService.ngInjectableDef = i0.defineInjectable({ factory: function PropertyDescriptorsService_Factory() { return new PropertyDescriptorsService(i0.inject(i1.AlfrescoApiService)); }, token: PropertyDescriptorsService, providedIn: "root" });
    return PropertyDescriptorsService;
}());
export { PropertyDescriptorsService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    PropertyDescriptorsService.prototype.alfrescoApiService;
}
