// @alfresco/adf-core/esm5/card-view/components/card-view-keyvaluepairsitem/card-view-keyvaluepairsitem.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { CardViewUpdateService } from '../../services/card-view-update.service';
import { CardViewKeyValuePairsItemModel } from '../../models/card-view.models';
import { MatTableDataSource } from '@angular/material';
var CardViewKeyValuePairsItemComponent = /** @class */ (function () {
    function CardViewKeyValuePairsItemComponent(cardViewUpdateService) {
        this.cardViewUpdateService = cardViewUpdateService;
        this.editable = false;
    }
    /**
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.values = this.property.value || [];
        this.matTableValues = new MatTableDataSource(this.values);
    };
    /**
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.isEditable = /**
     * @return {?}
     */
    function () {
        return this.editable && this.property.editable;
    };
    /**
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.add = /**
     * @return {?}
     */
    function () {
        this.values.push({ name: '', value: '' });
    };
    /**
     * @param {?} index
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.remove = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        this.values.splice(index, 1);
        this.save(true);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.onBlur = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value.length) {
            this.save();
        }
    };
    /**
     * @param {?=} remove
     * @return {?}
     */
    CardViewKeyValuePairsItemComponent.prototype.save = /**
     * @param {?=} remove
     * @return {?}
     */
    function (remove) {
        /** @type {?} */
        var validValues = this.values.filter((/**
         * @param {?} i
         * @return {?}
         */
        function (i) { return i.name.length && i.value.length; }));
        if (remove || validValues.length) {
            this.cardViewUpdateService.update(this.property, validValues);
            this.property.value = validValues;
        }
    };
    CardViewKeyValuePairsItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-card-view-boolitem',
                    template: "<div [attr.data-automation-id]=\"'card-key-value-pairs-label-' + property.key\" class=\"adf-property-label\">{{ property.label | translate }}</div>\n<div class=\"adf-property-value\">\n\n    <div *ngIf=\"isEditable()\">\n        {{ 'CORE.CARDVIEW.KEYVALUEPAIRS.ADD' | translate }}\n        <button (click)=\"add()\" mat-icon-button class=\"adf-card-view__key-value-pairs__add-btn\" [attr.data-automation-id]=\"'card-key-value-pairs-button-' + property.key\">\n            <mat-icon>add</mat-icon>\n        </button>\n    </div>\n\n    <div *ngIf=\"!isEditable()\" class=\"adf-card-view__key-value-pairs__read-only\">\n        <mat-table #table [dataSource]=\"matTableValues\" class=\"mat-elevation-z8\">\n            <ng-container matColumnDef=\"name\">\n                <mat-header-cell *matHeaderCellDef>{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.NAME' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let item\">{{item.name}}</mat-cell>\n            </ng-container>\n            <ng-container matColumnDef=\"value\">\n                <mat-header-cell *matHeaderCellDef>{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.VALUE' | translate }}</mat-header-cell>\n                <mat-cell *matCellDef=\"let item\">{{item.value}}</mat-cell>\n            </ng-container>\n\n            <mat-header-row *matHeaderRowDef=\"['name', 'value']\"></mat-header-row>\n            <mat-row *matRowDef=\"let row; columns: ['name', 'value'];\"></mat-row>\n        </mat-table>\n    </div>\n\n\n    <div class=\"adf-card-view__key-value-pairs\" *ngIf=\"isEditable() && values && values.length\">\n        <div class=\"adf-card-view__key-value-pairs__row\">\n            <div class=\"adf-card-view__key-value-pairs__col\">{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.NAME' | translate }}</div>\n            <div class=\"adf-card-view__key-value-pairs__col\">{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.VALUE' | translate }}</div>\n        </div>\n\n        <div class=\"adf-card-view__key-value-pairs__row\" *ngFor=\"let item of values; let i = index\">\n            <div class=\"adf-card-view__key-value-pairs__col\">\n                <mat-form-field class=\"adf-example-full-width\">\n                    <input matInput\n                           placeholder=\"{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.NAME' | translate }}\"\n                           (blur)=\"onBlur(item.value)\"\n                           [attr.data-automation-id]=\"'card-'+ property.key +'-name-input-' + i\"\n                           [(ngModel)]=\"values[i].name\">\n                </mat-form-field>\n            </div>\n            <div class=\"adf-card-view__key-value-pairs__col\">\n                <mat-form-field class=\"adf-example-full-width\">\n                    <input matInput\n                           placeholder=\"{{ 'CORE.CARDVIEW.KEYVALUEPAIRS.VALUE' | translate }}\"\n                           (blur)=\"onBlur(item.value)\"\n                           [attr.data-automation-id]=\"'card-'+ property.key +'-value-input-' + i\"\n                           [(ngModel)]=\"values[i].value\">\n                </mat-form-field>\n            </div>\n            <button mat-icon-button (click)=\"remove(i)\" class=\"adf-card-view__key-value-pairs__remove-btn\">\n                <mat-icon>close</mat-icon>\n            </button>\n        </div>\n    </div>\n</div>\n",
                    styles: [".adf-card-view__key-value-pairs__col{display:inline-block;width:39%}.adf-card-view__key-value-pairs__col .mat-form-field{width:100%}.adf-card-view__key-value-pairs__read-only .mat-table{box-shadow:none}.adf-card-view__key-value-pairs__read-only .mat-header-row,.adf-card-view__key-value-pairs__read-only .mat-row{padding:0}"]
                }] }
    ];
    /** @nocollapse */
    CardViewKeyValuePairsItemComponent.ctorParameters = function () { return [
        { type: CardViewUpdateService }
    ]; };
    CardViewKeyValuePairsItemComponent.propDecorators = {
        property: [{ type: Input }],
        editable: [{ type: Input }]
    };
    return CardViewKeyValuePairsItemComponent;
}());
export { CardViewKeyValuePairsItemComponent };
if (false) {
    /** @type {?} */
    CardViewKeyValuePairsItemComponent.prototype.property;
    /** @type {?} */
    CardViewKeyValuePairsItemComponent.prototype.editable;
    /** @type {?} */
    CardViewKeyValuePairsItemComponent.prototype.values;
    /** @type {?} */
    CardViewKeyValuePairsItemComponent.prototype.matTableValues;
    /**
     * @type {?}
     * @private
     */
    CardViewKeyValuePairsItemComponent.prototype.cardViewUpdateService;
}
