// @alfresco/adf-insights/esm5/analytics-process/components/widgets/dropdown/dropdown.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-access-missing-member no-input-rename  */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { FormGroup, Validators } from '@angular/forms';
import { WidgetComponent } from './../widget.component';
var DropdownWidgetAnalyticsComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DropdownWidgetAnalyticsComponent, _super);
    function DropdownWidgetAnalyticsComponent() {
        var _this = _super.call(this) || this;
        _this.fieldChanged = new EventEmitter();
        _this.showDefaultOption = true;
        _this.required = false;
        _this.defaultOptionText = 'Choose One';
        return _this;
    }
    /**
     * @return {?}
     */
    DropdownWidgetAnalyticsComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(Validators.compose(this.buildValidatorList()));
        }
    };
    /**
     * @param {?} controller
     * @return {?}
     */
    DropdownWidgetAnalyticsComponent.prototype.validateDropDown = /**
     * @param {?} controller
     * @return {?}
     */
    function (controller) {
        return controller.value !== 'null' ? null : { controllerName: false };
    };
    /**
     * @return {?}
     */
    DropdownWidgetAnalyticsComponent.prototype.buildValidatorList = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var validatorList = [];
        validatorList.push(Validators.required);
        if (this.showDefaultOption) {
            validatorList.push(this.validateDropDown);
        }
        return validatorList;
    };
    DropdownWidgetAnalyticsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'analytics-dropdown-widget',
                    template: "<div class=\"adf-dropdown-widget\" [formGroup]=\"formGroup\">\n    <label class=\"adf-dropdown-widget__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>\n    <select [formControlName]=\"controllerName\"\n            [attr.id]=\"'select-' + field.id\"\n            class=\"adf-dropdown-widget__select\"\n            [(ngModel)]=\"field.value\"\n            (ngModelChange)=\"changeValue($event)\">\n        <option *ngIf=\"showDefaultOption\" value=\"null\">{{defaultOptionText}}</option>\n        <option *ngFor=\"let opt of field.options\" [value]=\"opt.id\">{{opt.label}}</option>\n    </select>\n</div>\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-dropdown-widget,.adf-dropdown-widget__select{width:100%}.adf-dropdown-widget__invalid .adf-dropdown-widget__select{border-color:#d50000}.adf-dropdown-widget__invalid .adf-dropdown-widget__label{color:#d50000}.adf-dropdown-widget__invalid .adf-dropdown-widget__label::after{background-color:#d50000}"]
                }] }
    ];
    /** @nocollapse */
    DropdownWidgetAnalyticsComponent.ctorParameters = function () { return []; };
    DropdownWidgetAnalyticsComponent.propDecorators = {
        field: [{ type: Input }],
        formGroup: [{ type: Input, args: ['group',] }],
        controllerName: [{ type: Input, args: ['controllerName',] }],
        fieldChanged: [{ type: Output }],
        showDefaultOption: [{ type: Input }],
        required: [{ type: Input }],
        defaultOptionText: [{ type: Input }]
    };
    return DropdownWidgetAnalyticsComponent;
}(WidgetComponent));
export { DropdownWidgetAnalyticsComponent };
if (false) {
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.field;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.formGroup;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.controllerName;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.fieldChanged;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.showDefaultOption;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.required;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.defaultOptionText;
}
