// @alfresco/adf-extensions/esm2015/lib/components/viewer/preview-extension.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ComponentFactoryResolver, ViewChild, ViewContainerRef } from '@angular/core';
import { ExtensionService } from '../../services/extension.service';
import { Node } from '@alfresco/js-api';
export class PreviewExtensionComponent {
    /**
     * @param {?} extensionService
     * @param {?} componentFactoryResolver
     */
    constructor(extensionService, componentFactoryResolver) {
        this.extensionService = extensionService;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (!this.id) {
            return;
        }
        /** @type {?} */
        const componentType = this.extensionService.getComponentById(this.id);
        if (componentType) {
            /** @type {?} */
            const factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.updateInstance();
            }
        }
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        this.updateInstance();
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    }
    /**
     * @private
     * @return {?}
     */
    updateInstance() {
        if (this.componentRef && this.componentRef.instance) {
            /** @type {?} */
            const instance = this.componentRef.instance;
            instance.node = this.node;
            instance.url = this.url;
            instance.extension = this.extension;
        }
    }
}
PreviewExtensionComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-preview-extension',
                template: `
    <div #content></div>
  `
            }] }
];
/** @nocollapse */
PreviewExtensionComponent.ctorParameters = () => [
    { type: ExtensionService },
    { type: ComponentFactoryResolver }
];
PreviewExtensionComponent.propDecorators = {
    content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
    id: [{ type: Input }],
    url: [{ type: Input }],
    extension: [{ type: Input }],
    node: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PreviewExtensionComponent.prototype.content;
    /**
     * ID string of the component to preview.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.id;
    /**
     * URL of the content in the repository.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.url;
    /**
     * File extension (.jpg, .png, etc) for the viewer.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.extension;
    /**
     * Node containing the content to display.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.node;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.extensionService;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.componentFactoryResolver;
}
