// @alfresco/adf-insights/esm2015/diagram/services/diagram-color.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
export class DiagramColorService {
    constructor() {
    }
    /**
     * @param {?} totalColors
     * @return {?}
     */
    setTotalColors(totalColors) {
        this.totalColors = totalColors;
    }
    /**
     * @return {?}
     */
    getFillOpacity() {
        return '0.6';
    }
    /**
     * @param {?} key
     * @return {?}
     */
    getFillColour(key) {
        if (this.totalColors && this.totalColors.hasOwnProperty(key)) {
            /** @type {?} */
            const colorPercentage = this.totalColors[key];
            return this.convertColorToHsb(colorPercentage);
        }
        else {
            return DiagramColorService.ACTIVITY_FILL_COLOR;
        }
    }
    /**
     * @param {?} data
     * @param {?} defaultColor
     * @return {?}
     */
    getBpmnColor(data, defaultColor) {
        if (data.current) {
            return DiagramColorService.CURRENT_COLOR;
        }
        else if (data.completed) {
            return DiagramColorService.COMPLETED_COLOR;
        }
        else {
            return defaultColor;
        }
    }
    /**
     * @param {?} data
     * @return {?}
     */
    getBpmnStrokeWidth(data) {
        if (data.current || data.completed) {
            return DiagramColorService.TASK_HIGHLIGHT_STROKE;
        }
        else {
            return DiagramColorService.TASK_STROKE;
        }
    }
    /**
     * @param {?} colorPercentage
     * @return {?}
     */
    convertColorToHsb(colorPercentage) {
        /** @type {?} */
        const hue = (120.0 - (colorPercentage * 1.2)) / 360.0;
        return 'hsb(' + hue + ', 1, 1)';
    }
}
DiagramColorService.CURRENT_COLOR = '#017501';
DiagramColorService.COMPLETED_COLOR = '#2632aa';
DiagramColorService.ACTIVITY_STROKE_COLOR = '#bbbbbb';
DiagramColorService.MAIN_STROKE_COLOR = '#585858';
DiagramColorService.ACTIVITY_FILL_COLOR = '#f9f9f9';
DiagramColorService.TASK_STROKE = 1;
DiagramColorService.TASK_HIGHLIGHT_STROKE = 2;
DiagramColorService.CALL_ACTIVITY_STROKE = 2;
DiagramColorService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
DiagramColorService.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    DiagramColorService.CURRENT_COLOR;
    /** @type {?} */
    DiagramColorService.COMPLETED_COLOR;
    /** @type {?} */
    DiagramColorService.ACTIVITY_STROKE_COLOR;
    /** @type {?} */
    DiagramColorService.MAIN_STROKE_COLOR;
    /** @type {?} */
    DiagramColorService.ACTIVITY_FILL_COLOR;
    /** @type {?} */
    DiagramColorService.TASK_STROKE;
    /** @type {?} */
    DiagramColorService.TASK_HIGHLIGHT_STROKE;
    /** @type {?} */
    DiagramColorService.CALL_ACTIVITY_STROKE;
    /** @type {?} */
    DiagramColorService.prototype.totalColors;
}
