// @alfresco/adf-core/esm5/buttons-menu/buttons-menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChildren, QueryList } from '@angular/core';
import { MatMenuItem } from '@angular/material';
var ButtonsMenuComponent = /** @class */ (function () {
    function ButtonsMenuComponent() {
    }
    /**
     * @return {?}
     */
    ButtonsMenuComponent.prototype.ngAfterContentInit = /**
     * @return {?}
     */
    function () {
        if (this.buttons.length > 0) {
            this.isMenuEmpty = false;
        }
        else {
            this.isMenuEmpty = true;
        }
    };
    /**
     * @return {?}
     */
    ButtonsMenuComponent.prototype.isMobile = /**
     * @return {?}
     */
    function () {
        return !!/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
    };
    ButtonsMenuComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-buttons-action-menu',
                    template: "<div id=\"adf-buttons-menu\" class=\"adf-buttons-menu\" *ngIf=\"!isMenuEmpty\">\n    <div *ngIf=\"isMobile()\">\n        <button mat-icon-button [matMenuTriggerFor]=\"editReportMenu\">\n            <mat-icon>more_vert</mat-icon>\n        </button>\n        <mat-menu #editReportMenu=\"matMenu\" class=\"adf-buttons-menu-mobile\">\n            <ng-content *ngTemplateOutlet=\"desktop\">\n            </ng-content>\n        </mat-menu>\n    </div>\n\n    <div *ngIf=\"!isMobile()\" class=\"adf-buttons-menu-desktop\">\n        <ng-content *ngTemplateOutlet=\"desktop\">\n        </ng-content>\n    </div>\n</div>\n\n<ng-template #desktop>\n    <ng-content></ng-content>\n</ng-template>",
                    styles: [""]
                }] }
    ];
    ButtonsMenuComponent.propDecorators = {
        buttons: [{ type: ContentChildren, args: [MatMenuItem,] }]
    };
    return ButtonsMenuComponent;
}());
export { ButtonsMenuComponent };
if (false) {
    /** @type {?} */
    ButtonsMenuComponent.prototype.buttons;
    /** @type {?} */
    ButtonsMenuComponent.prototype.isMenuEmpty;
}
