// @alfresco/adf-extensions/esm5/lib/extensions.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DynamicExtensionComponent } from './components/dynamic-component/dynamic.component';
import { DynamicTabComponent } from './components/dynamic-tab/dynamic-tab.component';
import { DynamicColumnComponent } from './components/dynamic-column/dynamic-column.component';
import { PreviewExtensionComponent } from './components/viewer/preview-extension.component';
import { NgModule, APP_INITIALIZER } from '@angular/core';
import { AppExtensionService } from './services/app-extension.service';
import { setupExtensions } from './services/startup-extension-factory';
var ExtensionsModule = /** @class */ (function () {
    function ExtensionsModule() {
    }
    /**
     * @return {?}
     */
    ExtensionsModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: ExtensionsModule,
            providers: [
                {
                    provide: APP_INITIALIZER,
                    useFactory: setupExtensions,
                    deps: [AppExtensionService],
                    multi: true
                }
            ]
        };
    };
    /**
     * @return {?}
     */
    ExtensionsModule.forChild = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: ExtensionsModule
        };
    };
    ExtensionsModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        DynamicExtensionComponent,
                        DynamicTabComponent,
                        DynamicColumnComponent,
                        PreviewExtensionComponent
                    ],
                    exports: [
                        DynamicExtensionComponent,
                        DynamicTabComponent,
                        DynamicColumnComponent,
                        PreviewExtensionComponent
                    ]
                },] }
    ];
    return ExtensionsModule;
}());
export { ExtensionsModule };
