// @alfresco/adf-insights/esm5/diagram/models/report/reportQuery.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ReportDateRange } from './reportDateRange.model';
var ReportQuery = /** @class */ (function () {
    function ReportQuery(obj) {
        this.reportName = obj && obj.reportName || null;
        this.processDefinitionId = obj && obj.processDefinitionId || null;
        this.status = obj && obj.status || null;
        this.taskName = obj && obj.taskName || null;
        this.dateRangeInterval = obj && obj.dateRangeInterval || null;
        this.typeFiltering = obj && (typeof obj.typeFiltering !== 'undefined') ? obj.typeFiltering : true;
        this.slowProcessInstanceInteger = obj && obj.slowProcessInstanceInteger || 0;
        this.duration = obj && obj.duration || 0;
        this.dateRange = new ReportDateRange(obj);
    }
    return ReportQuery;
}());
export { ReportQuery };
if (false) {
    /** @type {?} */
    ReportQuery.prototype.reportName;
    /** @type {?} */
    ReportQuery.prototype.processDefinitionId;
    /** @type {?} */
    ReportQuery.prototype.status;
    /** @type {?} */
    ReportQuery.prototype.taskName;
    /** @type {?} */
    ReportQuery.prototype.typeFiltering;
    /** @type {?} */
    ReportQuery.prototype.dateRange;
    /** @type {?} */
    ReportQuery.prototype.dateRangeInterval;
    /** @type {?} */
    ReportQuery.prototype.slowProcessInstanceInteger;
    /** @type {?} */
    ReportQuery.prototype.duration;
}
