// @alfresco/adf-insights/esm5/diagram/models/report/reportParameterDetails.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var ReportParameterDetailsModel = /** @class */ (function () {
    function ReportParameterDetailsModel(obj) {
        this.id = obj && obj.id;
        this.name = obj && obj.name || null;
        this.nameKey = obj && obj.nameKey || null;
        this.type = obj && obj.type || null;
        this.value = obj && obj.value || null;
        this.options = obj && obj.options || null;
        this.dependsOn = obj && obj.dependsOn || null;
    }
    return ReportParameterDetailsModel;
}());
export { ReportParameterDetailsModel };
if (false) {
    /** @type {?} */
    ReportParameterDetailsModel.prototype.id;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.name;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.nameKey;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.type;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.value;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.options;
    /** @type {?} */
    ReportParameterDetailsModel.prototype.dependsOn;
}
