// @alfresco/adf-insights/esm5/diagram/models/report/reportParameters.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ReportDefinitionModel } from './reportDefinition.model';
var ReportParametersModel = /** @class */ (function () {
    function ReportParametersModel(obj) {
        this.id = obj && obj.id;
        this.name = obj && obj.name || null;
        if (obj && obj.definition) {
            this.definition = new ReportDefinitionModel(JSON.parse(obj.definition));
        }
        this.created = obj && obj.created || null;
    }
    /**
     * @return {?}
     */
    ReportParametersModel.prototype.hasParameters = /**
     * @return {?}
     */
    function () {
        return (this.definition && this.definition.parameters && this.definition.parameters.length > 0) ? true : false;
    };
    return ReportParametersModel;
}());
export { ReportParametersModel };
if (false) {
    /** @type {?} */
    ReportParametersModel.prototype.id;
    /** @type {?} */
    ReportParametersModel.prototype.name;
    /** @type {?} */
    ReportParametersModel.prototype.definition;
    /** @type {?} */
    ReportParametersModel.prototype.created;
}
