// @alfresco/adf-process-services/esm2015/process-list/models/filter-process.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class FilterProcessRepresentationModel {
    /**
     * @param {?} obj
     */
    constructor(obj) {
        if (obj) {
            this.id = obj.id || null;
            this.appId = obj.appId || null;
            this.name = obj.name || null;
            this.recent = !!obj.recent;
            this.icon = obj.icon || null;
            this.filter = obj.filter || null;
            this.index = obj.index;
        }
    }
    /**
     * @return {?}
     */
    hasFilter() {
        return !!this.filter;
    }
}
if (false) {
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.appId;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.filter;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.icon;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.id;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.index;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.name;
    /** @type {?} */
    FilterProcessRepresentationModel.prototype.recent;
}
/**
 * This object represent the parameters of a process filter.
 */
export class ProcessFilterParamRepresentationModel {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        this.processDefinitionId = obj.processDefinitionId || null;
        this.appDefinitionId = obj.appDefinitionId || null;
        this.processInstanceId = obj.processInstanceId || null;
        this.state = obj.state || null;
        this.sort = obj.sort || null;
        this.page = obj.page || null;
        this.size = obj.size || null;
    }
}
if (false) {
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.processDefinitionId;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.processInstanceId;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.appDefinitionId;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.state;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.sort;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.page;
    /** @type {?} */
    ProcessFilterParamRepresentationModel.prototype.size;
}
