// @alfresco/adf-extensions/esm2015/lib/services/component-register.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @record
 */
export function ExtensionComponent() { }
if (false) {
    /** @type {?} */
    ExtensionComponent.prototype.data;
}
export class ComponentRegisterService {
    constructor() {
        this.components = {};
    }
    /**
     * @param {?} values
     * @return {?}
     */
    setComponents(values) {
        if (values) {
            this.components = Object.assign({}, this.components, values);
        }
    }
    /**
     * @template T
     * @param {?} id
     * @return {?}
     */
    getComponentById(id) {
        return (/** @type {?} */ (this.components[id]));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    hasComponentById(id) {
        return !!this.getComponentById(id);
    }
}
ComponentRegisterService.decorators = [
    { type: Injectable, args: [{ providedIn: 'root' },] }
];
/** @nocollapse */ ComponentRegisterService.ngInjectableDef = i0.defineInjectable({ factory: function ComponentRegisterService_Factory() { return new ComponentRegisterService(); }, token: ComponentRegisterService, providedIn: "root" });
if (false) {
    /** @type {?} */
    ComponentRegisterService.prototype.components;
}
