// @alfresco/adf-extensions/esm2015/lib/services/extension.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { ExtensionLoaderService } from './extension-loader.service';
import * as core from '../evaluators/core.evaluators';
import { ComponentRegisterService } from './component-register.service';
import { RuleService } from './rule.service';
import * as i0 from "@angular/core";
import * as i1 from "./extension-loader.service";
import * as i2 from "./component-register.service";
import * as i3 from "./rule.service";
export class ExtensionService {
    /**
     * @param {?} loader
     * @param {?} componentRegister
     * @param {?} ruleService
     */
    constructor(loader, componentRegister, ruleService) {
        this.loader = loader;
        this.componentRegister = componentRegister;
        this.ruleService = ruleService;
        this.config = null;
        this.configPath = 'assets/app.extensions.json';
        this.pluginsPath = 'assets/plugins';
        this.routes = [];
        this.actions = [];
        this.features = [];
        this.authGuards = {};
    }
    /**
     * Loads and registers an extension config file and plugins (specified by path properties).
     * @return {?} The loaded config data
     */
    load() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const config = yield this.loader.load(this.configPath, this.pluginsPath);
            this.setup(config);
            return config;
        });
    }
    /**
     * Registers extensions from a config object.
     * @param {?} config Object with config data
     * @return {?}
     */
    setup(config) {
        if (!config) {
            console.warn('Extension configuration not found');
            return;
        }
        this.config = config;
        this.setEvaluators({
            'core.every': core.every,
            'core.some': core.some,
            'core.not': core.not
        });
        this.actions = this.loader.getActions(config);
        this.routes = this.loader.getRoutes(config);
        this.features = this.loader.getFeatures(config);
        this.ruleService.setup(config);
    }
    /**
     * Gets features by key.
     * @param {?} key Key string, using dot notation
     * @return {?} Features array found by key
     */
    getFeature(key) {
        /** @type {?} */
        const properties = Array.isArray(key) ? [key] : key.split('.');
        return properties.reduce((/**
         * @param {?} prev
         * @param {?} curr
         * @return {?}
         */
        (prev, curr) => prev && prev[curr]), this.features) || [];
    }
    /**
     * @template T
     * @param {?} key
     * @param {?=} fallback
     * @return {?}
     */
    getElements(key, fallback = []) {
        return this.loader.getElements(this.config, key, fallback);
    }
    /**
     * Adds one or more new rule evaluators to the existing set.
     * @param {?} values The new evaluators to add
     * @return {?}
     */
    setEvaluators(values) {
        this.ruleService.setEvaluators(values);
    }
    /**
     * Adds one or more new auth guards to the existing set.
     * @param {?} values The new auth guards to add
     * @return {?}
     */
    setAuthGuards(values) {
        if (values) {
            this.authGuards = Object.assign({}, this.authGuards, values);
        }
    }
    /**
     * Adds one or more new components to the existing set.
     * @param {?} values The new components to add
     * @return {?}
     */
    setComponents(values) {
        this.componentRegister.setComponents(values);
    }
    /**
     * Retrieves a route using its ID value.
     * @param {?} id The ID value to look for
     * @return {?} The route or null if not found
     */
    getRouteById(id) {
        return this.routes.find((/**
         * @param {?} route
         * @return {?}
         */
        (route) => route.id === id));
    }
    /**
     * Retrieves one or more auth guards using an array of ID values.
     * @param {?} ids Array of ID value to look for
     * @return {?} Array of auth guards or empty array if none were found
     */
    getAuthGuards(ids) {
        return (ids || [])
            .map((/**
         * @param {?} id
         * @return {?}
         */
        (id) => this.authGuards[id]))
            .filter((/**
         * @param {?} guard
         * @return {?}
         */
        (guard) => guard));
    }
    /**
     * Retrieves an action using its ID value.
     * @param {?} id The ID value to look for
     * @return {?} Action or null if not found
     */
    getActionById(id) {
        return this.actions.find((/**
         * @param {?} action
         * @return {?}
         */
        (action) => action.id === id));
    }
    /**
     * Retrieves a RuleEvaluator function using its key name.
     * @param {?} key Key name to look for
     * @return {?} RuleEvaluator or null if not found
     */
    getEvaluator(key) {
        return this.ruleService.getEvaluator(key);
    }
    /**
     * Evaluates a rule.
     * @param {?} ruleId ID of the rule to evaluate
     * @param {?=} context (optional) Custom rule execution context.
     * @return {?} True if the rule passed, false otherwise
     */
    evaluateRule(ruleId, context) {
        return this.ruleService.evaluateRule(ruleId, context);
    }
    /**
     * Retrieves a registered extension component using its ID value.
     * @template T
     * @param {?} id The ID value to look for
     * @return {?} The component or null if not found
     */
    getComponentById(id) {
        return this.componentRegister.getComponentById(id);
    }
    /**
     * Retrieves a rule using its ID value.
     * @param {?} id The ID value to look for
     * @return {?} The rule or null if not found
     */
    getRuleById(id) {
        return this.ruleService.getRuleById(id);
    }
    /**
     * Runs a lightweight expression stored in a string.
     * @param {?} value String containing the expression or literal value
     * @param {?=} context Parameter object for the expression with details of app state
     * @return {?} Result of evaluated expression, if found, or the literal value otherwise
     */
    runExpression(value, context) {
        /** @type {?} */
        const pattern = new RegExp(/\$\((.*\)?)\)/g);
        /** @type {?} */
        const matches = pattern.exec(value);
        if (matches && matches.length > 1) {
            /** @type {?} */
            const expression = matches[1];
            /** @type {?} */
            const fn = new Function('context', `return ${expression}`);
            /** @type {?} */
            const result = fn(context);
            return result;
        }
        return value;
    }
}
ExtensionService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
ExtensionService.ctorParameters = () => [
    { type: ExtensionLoaderService },
    { type: ComponentRegisterService },
    { type: RuleService }
];
/** @nocollapse */ ExtensionService.ngInjectableDef = i0.defineInjectable({ factory: function ExtensionService_Factory() { return new ExtensionService(i0.inject(i1.ExtensionLoaderService), i0.inject(i2.ComponentRegisterService), i0.inject(i3.RuleService)); }, token: ExtensionService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @protected
     */
    ExtensionService.prototype.config;
    /** @type {?} */
    ExtensionService.prototype.configPath;
    /** @type {?} */
    ExtensionService.prototype.pluginsPath;
    /** @type {?} */
    ExtensionService.prototype.routes;
    /** @type {?} */
    ExtensionService.prototype.actions;
    /** @type {?} */
    ExtensionService.prototype.features;
    /** @type {?} */
    ExtensionService.prototype.authGuards;
    /**
     * @type {?}
     * @protected
     */
    ExtensionService.prototype.loader;
    /**
     * @type {?}
     * @protected
     */
    ExtensionService.prototype.componentRegister;
    /**
     * @type {?}
     * @protected
     */
    ExtensionService.prototype.ruleService;
}
