// @alfresco/adf-extensions/esm2015/lib/services/rule.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { ExtensionLoaderService } from './extension-loader.service';
import * as i0 from "@angular/core";
import * as i1 from "./extension-loader.service";
export class RuleService {
    /**
     * @param {?} loader
     */
    constructor(loader) {
        this.loader = loader;
        this.context = null;
        this.rules = [];
        this.evaluators = {};
    }
    /**
     * @param {?} config
     * @return {?}
     */
    setup(config) {
        this.rules = this.loader.getRules(config);
    }
    /**
     * Adds one or more new rule evaluators to the existing set.
     * @param {?} values The new evaluators to add
     * @return {?}
     */
    setEvaluators(values) {
        if (values) {
            this.evaluators = Object.assign({}, this.evaluators, values);
        }
    }
    /**
     * Retrieves a rule using its ID value.
     * @param {?} id The ID value to look for
     * @return {?} The rule or null if not found
     */
    getRuleById(id) {
        return this.rules.find((/**
         * @param {?} ref
         * @return {?}
         */
        (ref) => ref.id === id));
    }
    /**
     * Retrieves a RuleEvaluator function using its key name.
     * @param {?} key Key name to look for
     * @return {?} RuleEvaluator or null if not found
     */
    getEvaluator(key) {
        if (key && key.startsWith('!')) {
            /** @type {?} */
            const fn = this.evaluators[key.substring(1)];
            return (/**
             * @param {?} context
             * @param {...?} args
             * @return {?}
             */
            (context, ...args) => {
                return !fn(context, ...args);
            });
        }
        return this.evaluators[key];
    }
    /**
     * Evaluates a rule.
     * @param {?} ruleId ID of the rule to evaluate
     * @param {?=} context (optional) Custom rule execution context.
     * @return {?} True if the rule passed, false otherwise
     */
    evaluateRule(ruleId, context) {
        /** @type {?} */
        const ruleRef = this.getRuleById(ruleId);
        context = context || this.context;
        if (ruleRef) {
            /** @type {?} */
            const evaluator = this.getEvaluator(ruleRef.type);
            if (evaluator) {
                return evaluator(context, ...ruleRef.parameters);
            }
        }
        else {
            /** @type {?} */
            const evaluator = this.getEvaluator(ruleId);
            if (evaluator) {
                return evaluator(context);
            }
        }
        return false;
    }
}
RuleService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
RuleService.ctorParameters = () => [
    { type: ExtensionLoaderService }
];
/** @nocollapse */ RuleService.ngInjectableDef = i0.defineInjectable({ factory: function RuleService_Factory() { return new RuleService(i0.inject(i1.ExtensionLoaderService)); }, token: RuleService, providedIn: "root" });
if (false) {
    /** @type {?} */
    RuleService.prototype.context;
    /** @type {?} */
    RuleService.prototype.rules;
    /** @type {?} */
    RuleService.prototype.evaluators;
    /**
     * @type {?}
     * @protected
     */
    RuleService.prototype.loader;
}
