// @alfresco/adf-content-services/esm5/content-metadata/services/config/indifferent-config.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var IndifferentConfigService = /** @class */ (function () {
    function IndifferentConfigService(config) {
    }
    /**
     * @param {?} groupName
     * @return {?}
     */
    IndifferentConfigService.prototype.isGroupAllowed = /**
     * @param {?} groupName
     * @return {?}
     */
    function (groupName) {
        return true;
    };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    IndifferentConfigService.prototype.reorganiseByConfig = /**
     * @param {?} propertyGroups
     * @return {?}
     */
    function (propertyGroups) {
        return Object.keys(propertyGroups)
            .map((/**
         * @param {?} groupName
         * @return {?}
         */
        function (groupName) {
            /** @type {?} */
            var propertyGroup = propertyGroups[groupName];
            /** @type {?} */
            var properties = propertyGroup.properties;
            return Object.assign({}, propertyGroup, {
                properties: Object.keys(properties).map((/**
                 * @param {?} propertyName
                 * @return {?}
                 */
                function (propertyName) { return properties[propertyName]; }))
            });
        }));
    };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    IndifferentConfigService.prototype.filterExcludedPreset = /**
     * @param {?} propertyGroups
     * @return {?}
     */
    function (propertyGroups) {
        return propertyGroups;
    };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    IndifferentConfigService.prototype.appendAllPreset = /**
     * @param {?} propertyGroups
     * @return {?}
     */
    function (propertyGroups) {
        return [];
    };
    /**
     * @return {?}
     */
    IndifferentConfigService.prototype.isIncludeAllEnabled = /**
     * @return {?}
     */
    function () {
        return true;
    };
    return IndifferentConfigService;
}());
export { IndifferentConfigService };
