// @alfresco/adf-content-services/esm5/content-metadata/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { ContentMetadataCardComponent } from './components/content-metadata-card/content-metadata-card.component';
export { BasicPropertiesService } from './services/basic-properties.service';
export { ContentMetadataService } from './services/content-metadata.service';
export { PropertyDescriptorsService } from './services/property-descriptors.service';
export { PropertyGroupTranslatorService } from './services/property-groups-translator.service';
export { ContentMetadataConfigFactory } from './services/config/content-metadata-config.factory';
export { IndifferentConfigService } from './services/config/indifferent-config.service';
export { LayoutOrientedConfigService } from './services/config/layout-oriented-config.service';
export { AspectOrientedConfigService } from './services/config/aspect-oriented-config.service';
export { ContentMetadataModule } from './content-metadata.module';
