// @alfresco/adf-insights/esm5/diagram/components/activities/diagram-box-publish-task.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramBoxPublishTaskComponent = /** @class */ (function () {
    function DiagramBoxPublishTaskComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    DiagramBoxPublishTaskComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DiagramBoxPublishTaskComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-box-publish-task',
                    template: "<diagram-task [data]=\"data\"></diagram-task>\n<diagram-icon-box-publish-task [data]=\"data\"></diagram-icon-box-publish-task>"
                }] }
    ];
    /** @nocollapse */
    DiagramBoxPublishTaskComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramBoxPublishTaskComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramBoxPublishTaskComponent;
}());
export { DiagramBoxPublishTaskComponent };
if (false) {
    /** @type {?} */
    DiagramBoxPublishTaskComponent.prototype.data;
    /** @type {?} */
    DiagramBoxPublishTaskComponent.prototype.error;
    /** @type {?} */
    DiagramBoxPublishTaskComponent.prototype.elementRef;
}
