// @alfresco/adf-insights/esm5/diagram/components/activities/diagram-task.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramTaskComponent = /** @class */ (function () {
    function DiagramTaskComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: '', radius: 4 };
    }
    /**
     * @return {?}
     */
    DiagramTaskComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.rectLeftCorner = { x: this.data.x, y: this.data.y };
        this.textPosition = { x: this.data.x + (this.data.width / 2), y: this.data.y + (this.data.height / 2) };
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.ACTIVITY_STROKE_COLOR);
        this.options.strokeWidth = this.diagramColorService.getBpmnStrokeWidth(this.data);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
    };
    DiagramTaskComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-task',
                    template: "<raphael-rect [elementId]=\"data.id\" [leftCorner]=\"rectLeftCorner\" [width]=\"data.width\" [height]=\"data.height\" [radius]=\"options.radius\"\n              [stroke]=\"options.stroke\" [strokeWidth]=\"options.strokeWidth\"\n              [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-rect>\n<raphael-multiline-text [text]=\"data.name\" [position]=\"textPosition\" [elementWidth]=\"data.width\"></raphael-multiline-text>\n<diagram-tooltip [data]=\"data\"></diagram-tooltip>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramTaskComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramTaskComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramTaskComponent;
}());
export { DiagramTaskComponent };
if (false) {
    /** @type {?} */
    DiagramTaskComponent.prototype.data;
    /** @type {?} */
    DiagramTaskComponent.prototype.error;
    /** @type {?} */
    DiagramTaskComponent.prototype.rectLeftCorner;
    /** @type {?} */
    DiagramTaskComponent.prototype.textPosition;
    /** @type {?} */
    DiagramTaskComponent.prototype.options;
    /** @type {?} */
    DiagramTaskComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramTaskComponent.prototype.diagramColorService;
}
