// @alfresco/adf-insights/esm5/diagram/components/activities/diagram-rest-call-task.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramRestCallTaskComponent = /** @class */ (function () {
    function DiagramRestCallTaskComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    DiagramRestCallTaskComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DiagramRestCallTaskComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-rest-call-task',
                    template: "<diagram-task [data]=\"data\"></diagram-task>\n<diagram-icon-rest-call-task [data]=\"data\"></diagram-icon-rest-call-task>"
                }] }
    ];
    /** @nocollapse */
    DiagramRestCallTaskComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramRestCallTaskComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramRestCallTaskComponent;
}());
export { DiagramRestCallTaskComponent };
if (false) {
    /** @type {?} */
    DiagramRestCallTaskComponent.prototype.data;
    /** @type {?} */
    DiagramRestCallTaskComponent.prototype.error;
    /** @type {?} */
    DiagramRestCallTaskComponent.prototype.elementRef;
}
