// @alfresco/adf-insights/esm5/diagram/components/activities/diagram-container-service-task.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramContainerServiceTaskComponent = /** @class */ (function () {
    function DiagramContainerServiceTaskComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    DiagramContainerServiceTaskComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DiagramContainerServiceTaskComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-container-service-task',
                    template: "<div [ngSwitch]=\"data.taskType\">\n    <div *ngSwitchCase=\"'mail'\">\n        <diagram-send-task [data]=\"data\"></diagram-send-task>\n    </div>\n    <div *ngSwitchCase=\"'camel'\">\n        <diagram-camel-task [data]=\"data\"></diagram-camel-task>\n    </div>\n    <div *ngSwitchCase=\"'mule'\">\n        <diagram-mule-task [data]=\"data\"></diagram-mule-task>\n    </div>\n    <div *ngSwitchCase=\"'alfresco_publish'\">\n        <adf-diagram-publish-task [data]=\"data\"></adf-diagram-publish-task>\n    </div>\n    <div *ngSwitchCase=\"'rest_call'\">\n        <diagram-rest-call-task [data]=\"data\"></diagram-rest-call-task>\n    </div>\n    <div *ngSwitchCase=\"'google_drive_publish'\">\n        <diagram-google-drive-publish-task [data]=\"data\"></diagram-google-drive-publish-task>\n    </div>\n    <div *ngSwitchCase=\"'box_publish'\">\n        <diagram-box-publish-task [data]=\"data\"></diagram-box-publish-task>\n    </div>\n    <div *ngSwitchDefault>\n        <diagram-service-task [data]=\"data\"></diagram-service-task>\n    </div>\n</div>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramContainerServiceTaskComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramContainerServiceTaskComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramContainerServiceTaskComponent;
}());
export { DiagramContainerServiceTaskComponent };
if (false) {
    /** @type {?} */
    DiagramContainerServiceTaskComponent.prototype.data;
    /** @type {?} */
    DiagramContainerServiceTaskComponent.prototype.error;
    /** @type {?} */
    DiagramContainerServiceTaskComponent.prototype.elementRef;
}
