// @alfresco/adf-insights/esm2015/analytics-process/components/widgets/number/number.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-access-missing-member no-input-rename */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { FormGroup, Validators } from '@angular/forms';
import { WidgetComponent } from './../widget.component';
export class NumberWidgetAnalyticsComponent extends WidgetComponent {
    /**
     * @param {?} elementRef
     */
    constructor(elementRef) {
        super();
        this.elementRef = elementRef;
        this.required = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(Validators.required);
        }
    }
}
NumberWidgetAnalyticsComponent.decorators = [
    { type: Component, args: [{
                selector: 'analytics-number-widget',
                template: "<div class=\"adf-number-widget\" [formGroup]=\"formGroup\">\n    <mat-form-field class=\"example-full-width\" floatPlaceholder=\"always\">\n        <input matInput\n            placeholder=\"{{field.nameKey | translate}}\"\n            formControlName=\"{{controllerName}}\"\n            type=\"text\"\n            pattern=\"-?[0-9]*(\\.[0-9]+)?\"\n            [id]=\"field.id\"\n            [value]=\"field.value\"\n            [(ngModel)]=\"field.value\"\n            (ngModelChange)=\"changeValue(field)\">\n    </mat-form-field>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-number-widget,.adf-number-widget .mat-form-field{width:100%}"]
            }] }
];
/** @nocollapse */
NumberWidgetAnalyticsComponent.ctorParameters = () => [
    { type: ElementRef }
];
NumberWidgetAnalyticsComponent.propDecorators = {
    field: [{ type: Input }],
    formGroup: [{ type: Input, args: ['group',] }],
    controllerName: [{ type: Input, args: ['controllerName',] }],
    required: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    NumberWidgetAnalyticsComponent.prototype.field;
    /** @type {?} */
    NumberWidgetAnalyticsComponent.prototype.formGroup;
    /** @type {?} */
    NumberWidgetAnalyticsComponent.prototype.controllerName;
    /** @type {?} */
    NumberWidgetAnalyticsComponent.prototype.required;
    /** @type {?} */
    NumberWidgetAnalyticsComponent.prototype.elementRef;
}
