// @alfresco/adf-core/esm5/events/file.event.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FileUploadStatus } from '../models/file.model';
var FileUploadEvent = /** @class */ (function () {
    function FileUploadEvent(file, status, error) {
        if (status === void 0) { status = FileUploadStatus.Pending; }
        if (error === void 0) { error = null; }
        this.file = file;
        this.status = status;
        this.error = error;
    }
    return FileUploadEvent;
}());
export { FileUploadEvent };
if (false) {
    /** @type {?} */
    FileUploadEvent.prototype.file;
    /** @type {?} */
    FileUploadEvent.prototype.status;
    /** @type {?} */
    FileUploadEvent.prototype.error;
}
var FileUploadCompleteEvent = /** @class */ (function (_super) {
    tslib_1.__extends(FileUploadCompleteEvent, _super);
    function FileUploadCompleteEvent(file, totalComplete, data, totalAborted) {
        if (totalComplete === void 0) { totalComplete = 0; }
        if (totalAborted === void 0) { totalAborted = 0; }
        var _this = _super.call(this, file, FileUploadStatus.Complete) || this;
        _this.totalComplete = totalComplete;
        _this.data = data;
        _this.totalAborted = totalAborted;
        return _this;
    }
    return FileUploadCompleteEvent;
}(FileUploadEvent));
export { FileUploadCompleteEvent };
if (false) {
    /** @type {?} */
    FileUploadCompleteEvent.prototype.totalComplete;
    /** @type {?} */
    FileUploadCompleteEvent.prototype.data;
    /** @type {?} */
    FileUploadCompleteEvent.prototype.totalAborted;
}
var FileUploadDeleteEvent = /** @class */ (function (_super) {
    tslib_1.__extends(FileUploadDeleteEvent, _super);
    function FileUploadDeleteEvent(file, totalComplete) {
        if (totalComplete === void 0) { totalComplete = 0; }
        var _this = _super.call(this, file, FileUploadStatus.Deleted) || this;
        _this.totalComplete = totalComplete;
        return _this;
    }
    return FileUploadDeleteEvent;
}(FileUploadEvent));
export { FileUploadDeleteEvent };
if (false) {
    /** @type {?} */
    FileUploadDeleteEvent.prototype.totalComplete;
}
var FileUploadErrorEvent = /** @class */ (function (_super) {
    tslib_1.__extends(FileUploadErrorEvent, _super);
    function FileUploadErrorEvent(file, error, totalError) {
        if (totalError === void 0) { totalError = 0; }
        var _this = _super.call(this, file, FileUploadStatus.Error) || this;
        _this.error = error;
        _this.totalError = totalError;
        return _this;
    }
    return FileUploadErrorEvent;
}(FileUploadEvent));
export { FileUploadErrorEvent };
if (false) {
    /** @type {?} */
    FileUploadErrorEvent.prototype.error;
    /** @type {?} */
    FileUploadErrorEvent.prototype.totalError;
}
