// @alfresco/adf-process-services/esm2015/content-widget/attach-file-widget-dialog.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { MatDialog } from '@angular/material';
import { EventEmitter, Injectable, Output } from '@angular/core';
import { Subject } from 'rxjs';
import { AttachFileWidgetDialogComponent } from './attach-file-widget-dialog.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/dialog";
export class AttachFileWidgetDialogService {
    /**
     * @param {?} dialog
     */
    constructor(dialog) {
        this.dialog = dialog;
        /**
         * Emitted when an error occurs.
         */
        this.error = new EventEmitter();
    }
    /**
     * Opens a dialog to choose a file to upload.
     * @param {?} ecmHost
     * @param {?=} actionName
     * @param {?=} context
     * @return {?} Information about the chosen file(s)
     */
    openLogin(ecmHost, actionName, context) {
        /** @type {?} */
        const titleString = `Please log in for ${ecmHost}`;
        /** @type {?} */
        const selected = new Subject();
        selected.subscribe({
            complete: this.close.bind(this)
        });
        /** @type {?} */
        const data = {
            title: titleString,
            actionName,
            selected,
            ecmHost,
            context,
            isSelectionValid: this.isNodeFile.bind(this)
        };
        this.openLoginDialog(data, 'adf-attach-file-widget-dialog', '630px');
        return selected;
    }
    /**
     * @private
     * @param {?} data
     * @param {?} currentPanelClass
     * @param {?} chosenWidth
     * @return {?}
     */
    openLoginDialog(data, currentPanelClass, chosenWidth) {
        this.dialog.open(AttachFileWidgetDialogComponent, { data, panelClass: currentPanelClass, width: chosenWidth });
    }
    /**
     * Closes the currently open dialog.
     * @return {?}
     */
    close() {
        this.dialog.closeAll();
    }
    /**
     * @private
     * @param {?} entry
     * @return {?}
     */
    isNodeFile(entry) {
        return entry.isFile;
    }
}
AttachFileWidgetDialogService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
AttachFileWidgetDialogService.ctorParameters = () => [
    { type: MatDialog }
];
AttachFileWidgetDialogService.propDecorators = {
    error: [{ type: Output }]
};
/** @nocollapse */ AttachFileWidgetDialogService.ngInjectableDef = i0.defineInjectable({ factory: function AttachFileWidgetDialogService_Factory() { return new AttachFileWidgetDialogService(i0.inject(i1.MatDialog)); }, token: AttachFileWidgetDialogService, providedIn: "root" });
if (false) {
    /**
     * Emitted when an error occurs.
     * @type {?}
     */
    AttachFileWidgetDialogService.prototype.error;
    /**
     * @type {?}
     * @private
     */
    AttachFileWidgetDialogService.prototype.dialog;
}
