// @alfresco/adf-insights/esm2015/analytics-process/components/analytics-report-heat-map.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
import { AnalyticsService } from '../services/analytics.service';
export class AnalyticsReportHeatMapComponent {
    /**
     * @param {?} analyticsService
     * @param {?} formBuilder
     */
    constructor(analyticsService, formBuilder) {
        this.analyticsService = analyticsService;
        this.formBuilder = formBuilder;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.field = {};
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.initForm();
        this.field.id = 'metrics';
        this.field.value = 'totalCount';
        this.analyticsService.getMetricValues().subscribe((/**
         * @param {?} opts
         * @return {?}
         */
        (opts) => {
            this.field.options = opts;
            this.success.emit(opts);
        }));
    }
    /**
     * @param {?} field
     * @return {?}
     */
    onMetricChanges(field) {
        if (field.value === 'totalCount') {
            this.currentMetric = this.report.totalCountValues;
            this.currentMetricColors = this.report.totalCountsPercentages;
            this.metricType = 'times';
        }
        else if (field.value === 'totalTime') {
            this.currentMetric = this.report.totalTimeValues;
            this.currentMetricColors = this.report.totalTimePercentages;
            this.metricType = 'hours';
        }
        else if (field.value === 'avgTime') {
            this.currentMetric = this.report.avgTimeValues;
            this.currentMetricColors = this.report.avgTimePercentages;
            this.metricType = 'hours';
        }
    }
    /**
     * @return {?}
     */
    initForm() {
        this.metricForm = this.formBuilder.group({
            metricGroup: new FormGroup({
                metric: new FormControl()
            })
        });
    }
    /**
     * @return {?}
     */
    hasMetric() {
        return (this.report.totalCountsPercentages ||
            this.report.totalTimePercentages ||
            this.report.avgTimePercentages) ? true : false;
    }
}
AnalyticsReportHeatMapComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-analytics-report-heat-map, analytics-report-heat-map',
                template: "<h4>Process Heat map</h4>\n<div *ngIf=\"hasMetric()\">\n    <form [formGroup]=\"metricForm\" novalidate>\n        <analytics-dropdown-widget [field]=\"field\" [group]=\"metricForm.controls.metricGroup\" [controllerName]=\"'metric'\"\n                         (fieldChanged)=\"onMetricChanges(field)\" [showDefaultOption]=\"false\"></analytics-dropdown-widget>\n    </form>\n    <adf-diagram *ngIf=\"currentMetric\" [processDefinitionId]=\"report.processDefinitionId\" [metricPercentages]=\"currentMetric\" [metricColor]=\"currentMetricColors\" [metricType]=\"metricType\"></adf-diagram>\n</div>\n<div *ngIf=\"!hasMetric()\">No metric found</div>\n"
            }] }
];
/** @nocollapse */
AnalyticsReportHeatMapComponent.ctorParameters = () => [
    { type: AnalyticsService },
    { type: FormBuilder }
];
AnalyticsReportHeatMapComponent.propDecorators = {
    report: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.report;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.success;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.error;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.field;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.metricForm;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.currentMetric;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.currentMetricColors;
    /** @type {?} */
    AnalyticsReportHeatMapComponent.prototype.metricType;
    /**
     * @type {?}
     * @private
     */
    AnalyticsReportHeatMapComponent.prototype.analyticsService;
    /**
     * @type {?}
     * @private
     */
    AnalyticsReportHeatMapComponent.prototype.formBuilder;
}
