// @alfresco/adf-core/esm2015/form/components/widgets/dynamic-table/editors/row.editor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { DynamicRowValidationSummary } from './../dynamic-row-validation-summary.model';
import { DynamicTableModel } from './../dynamic-table.widget.model';
export class RowEditorComponent {
    constructor() {
        this.save = new EventEmitter();
        this.cancel = new EventEmitter();
        this.validationSummary = new DynamicRowValidationSummary({ isValid: true, message: '' });
    }
    /**
     * @return {?}
     */
    onCancelChanges() {
        this.cancel.emit({
            table: this.table,
            row: this.row,
            column: this.column
        });
    }
    /**
     * @return {?}
     */
    onSaveChanges() {
        this.validate();
        if (this.isValid()) {
            this.save.emit({
                table: this.table,
                row: this.row,
                column: this.column
            });
        }
    }
    /**
     * @private
     * @return {?}
     */
    isValid() {
        return this.validationSummary && this.validationSummary.isValid;
    }
    /**
     * @private
     * @return {?}
     */
    validate() {
        this.validationSummary = this.table.validateRow(this.row);
    }
}
RowEditorComponent.decorators = [
    { type: Component, args: [{
                selector: 'row-editor',
                template: "<div class=\"row-editor mdl-shadow--2dp\"\n    [class.row-editor__invalid]=\"!validationSummary.isValid\">\n    <div class=\"mdl-grid\" *ngFor=\"let column of table.columns\">\n        <div class=\"mdl-cell mdl-cell--6-col\" [ngSwitch]=\"column.type\">\n            <div *ngSwitchCase=\"'Dropdown'\">\n                <adf-dropdown-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-dropdown-editor>\n                </div>\n                <div *ngSwitchCase=\"'Date'\">\n                    <adf-date-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-date-editor>\n                </div>\n                <div *ngSwitchCase=\"'Datetime'\">\n                    <adf-datetime-editor\n                        [table]=\"table\"\n                        [row]=\"row\"\n                        [column]=\"column\">\n                    </adf-datetime-editor>\n                </div>\n                <div *ngSwitchCase=\"'Boolean'\">\n                <adf-boolean-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-boolean-editor>\n                </div>\n                <div *ngSwitchDefault>\n                <adf-text-editor\n                    [table]=\"table\"\n                    [row]=\"row\"\n                    [column]=\"column\">\n                </adf-text-editor>\n            </div>\n        </div>\n    </div>\n    <error-widget [error]=\"validationSummary\"></error-widget>\n    <div>\n        <button mat-button (click)=\"onCancelChanges()\">Cancel</button>\n        <button mat-button (click)=\"onSaveChanges()\">Save</button>\n    </div>\n</div>\n",
                styles: [".row-editor{padding:8px}.row-editor__validation-summary{visibility:hidden}.row-editor__invalid .row-editor__validation-summary{color:#d50000;visibility:visible;padding:8px 16px}"]
            }] }
];
/** @nocollapse */
RowEditorComponent.ctorParameters = () => [];
RowEditorComponent.propDecorators = {
    table: [{ type: Input }],
    row: [{ type: Input }],
    column: [{ type: Input }],
    save: [{ type: Output }],
    cancel: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    RowEditorComponent.prototype.table;
    /** @type {?} */
    RowEditorComponent.prototype.row;
    /** @type {?} */
    RowEditorComponent.prototype.column;
    /** @type {?} */
    RowEditorComponent.prototype.save;
    /** @type {?} */
    RowEditorComponent.prototype.cancel;
    /** @type {?} */
    RowEditorComponent.prototype.validationSummary;
}
