// @alfresco/adf-process-services/esm5/process-list/models/process-definition.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var ProcessDefinitionRepresentation = /** @class */ (function () {
    function ProcessDefinitionRepresentation(obj) {
        this.id = obj && obj.id || null;
        this.name = obj && obj.name || null;
        this.description = obj && obj.description || null;
        this.key = obj && obj.key || null;
        this.category = obj && obj.category || null;
        this.version = obj && obj.version || 0;
        this.deploymentId = obj && obj.deploymentId || null;
        this.tenantId = obj && obj.tenantId || null;
        this.metaDataValues = obj && obj.metaDataValues || [];
        this.hasStartForm = obj && obj.hasStartForm === true ? true : false;
    }
    return ProcessDefinitionRepresentation;
}());
export { ProcessDefinitionRepresentation };
if (false) {
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.id;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.name;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.description;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.key;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.category;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.version;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.deploymentId;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.tenantId;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.metaDataValues;
    /** @type {?} */
    ProcessDefinitionRepresentation.prototype.hasStartForm;
}
