// @alfresco/adf-process-services/esm5/process-list/models/process-list.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var ProcessListModel = /** @class */ (function () {
    function ProcessListModel(obj) {
        if (obj) {
            this.size = obj.size || null;
            this.total = obj.total || null;
            this.start = obj.start || null;
            this.length = obj.length || null;
            this.data = obj.data || [];
        }
    }
    return ProcessListModel;
}());
export { ProcessListModel };
if (false) {
    /** @type {?} */
    ProcessListModel.prototype.size;
    /** @type {?} */
    ProcessListModel.prototype.total;
    /** @type {?} */
    ProcessListModel.prototype.start;
    /** @type {?} */
    ProcessListModel.prototype.length;
    /** @type {?} */
    ProcessListModel.prototype.data;
}
