// @alfresco/adf-process-services/esm5/process-list/models/process-instance-filter.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var ProcessFilterRequestRepresentation = /** @class */ (function () {
    function ProcessFilterRequestRepresentation(obj) {
        this.processDefinitionId = obj && obj.processDefinitionId || null;
        this.appDefinitionId = obj && obj.appDefinitionId || null;
        this.state = obj && obj.state || null;
        this.sort = obj && obj.sort || null;
        this.page = obj && obj.page || 0;
        this.size = obj && obj.size || 25;
    }
    return ProcessFilterRequestRepresentation;
}());
export { ProcessFilterRequestRepresentation };
if (false) {
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.processDefinitionId;
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.appDefinitionId;
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.state;
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.sort;
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.page;
    /** @type {?} */
    ProcessFilterRequestRepresentation.prototype.size;
}
