// @alfresco/adf-core/esm5/form/components/form-field/form-field.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Compiler, Component, ComponentFactoryResolver, Input, NgModule, ViewChild, ViewContainerRef, ViewEncapsulation } from '@angular/core';
import { FormRenderingService } from './../../services/form-rendering.service';
import { WidgetVisibilityService } from './../../services/widget-visibility.service';
import { FormFieldModel } from './../widgets/core/form-field.model';
var FormFieldComponent = /** @class */ (function () {
    function FormFieldComponent(formRenderingService, componentFactoryResolver, visibilityService, compiler) {
        this.formRenderingService = formRenderingService;
        this.componentFactoryResolver = componentFactoryResolver;
        this.visibilityService = visibilityService;
        this.compiler = compiler;
        /**
         * Contains all the necessary data needed to determine what UI Widget
         * to use when rendering the field in the form. You would typically not
         * create this data manually but instead create the form in APS and export
         * it to get to all the `FormFieldModel` definitions.
         */
        this.field = null;
        this.focus = false;
    }
    /**
     * @return {?}
     */
    FormFieldComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var w = window;
        if (w.adf === undefined) {
            w.adf = {};
        }
        /** @type {?} */
        var originalField = this.getField();
        if (originalField) {
            /** @type {?} */
            var customTemplate = this.field.form.customFieldTemplates[originalField.type];
            if (customTemplate && this.hasController(originalField.type)) {
                /** @type {?} */
                var factory = this.getComponentFactorySync(originalField.type, customTemplate);
                this.componentRef = this.container.createComponent(factory);
                /** @type {?} */
                var instance = this.componentRef.instance;
                if (instance) {
                    instance.field = originalField;
                }
            }
            else {
                /** @type {?} */
                var componentType = this.formRenderingService.resolveComponentType(originalField);
                if (componentType) {
                    /** @type {?} */
                    var factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
                    this.componentRef = this.container.createComponent(factory);
                    /** @type {?} */
                    var instance = (/** @type {?} */ (this.componentRef.instance));
                    instance.field = this.field;
                    instance.fieldChanged.subscribe((/**
                     * @param {?} field
                     * @return {?}
                     */
                    function (field) {
                        if (field && _this.field.form) {
                            _this.visibilityService.refreshVisibility(field.form);
                            field.form.onFormFieldChanged(field);
                        }
                    }));
                }
            }
        }
    };
    /**
     * @return {?}
     */
    FormFieldComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    };
    /**
     * @private
     * @return {?}
     */
    FormFieldComponent.prototype.getField = /**
     * @private
     * @return {?}
     */
    function () {
        if (this.field && this.field.params) {
            /** @type {?} */
            var wrappedField = this.field.params.field;
            if (wrappedField && wrappedField.type) {
                return wrappedField;
            }
        }
        return this.field;
    };
    /**
     * @private
     * @param {?} type
     * @return {?}
     */
    FormFieldComponent.prototype.hasController = /**
     * @private
     * @param {?} type
     * @return {?}
     */
    function (type) {
        return (adf && adf.components && adf.components[type]);
    };
    /**
     * @private
     * @param {?} type
     * @param {?} template
     * @return {?}
     */
    FormFieldComponent.prototype.getComponentFactorySync = /**
     * @private
     * @param {?} type
     * @param {?} template
     * @return {?}
     */
    function (type, template) {
        /** @type {?} */
        var componentInfo = adf.components[type];
        if (componentInfo.factory) {
            return componentInfo.factory;
        }
        /** @type {?} */
        var metadata = {
            selector: "runtime-component-" + type,
            template: template
        };
        /** @type {?} */
        var factory = this.createComponentFactorySync(this.compiler, metadata, componentInfo.class);
        componentInfo.factory = factory;
        return factory;
    };
    /**
     * @private
     * @param {?} compiler
     * @param {?} metadata
     * @param {?} componentClass
     * @return {?}
     */
    FormFieldComponent.prototype.createComponentFactorySync = /**
     * @private
     * @param {?} compiler
     * @param {?} metadata
     * @param {?} componentClass
     * @return {?}
     */
    function (compiler, metadata, componentClass) {
        /** @type {?} */
        var cmpClass = componentClass || /** @class */ (function () {
            function RuntimeComponent() {
            }
            return RuntimeComponent;
        }());
        /** @type {?} */
        var decoratedCmp = Component(metadata)(cmpClass);
        var RuntimeComponentModule = /** @class */ (function () {
            function RuntimeComponentModule() {
            }
            RuntimeComponentModule.decorators = [
                { type: NgModule, args: [{ imports: [], declarations: [decoratedCmp] },] },
            ];
            return RuntimeComponentModule;
        }());
        /** @type {?} */
        var module = compiler.compileModuleAndAllComponentsSync(RuntimeComponentModule);
        return module.componentFactories.find((/**
         * @param {?} x
         * @return {?}
         */
        function (x) { return x.componentType === decoratedCmp; }));
    };
    /**
     * @return {?}
     */
    FormFieldComponent.prototype.focusToggle = /**
     * @return {?}
     */
    function () {
        this.focus = !this.focus;
    };
    FormFieldComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-form-field',
                    template: "\n        <div [id]=\"'field-'+field?.id+'-container'\"\n            [hidden]=\"!field?.isVisible\"\n            [class.adf-focus]=\"focus\"\n            (focusin)=\"focusToggle()\"\n            (focusout)=\"focusToggle()\">\n            <div #container></div>\n        </div>\n    ",
                    encapsulation: ViewEncapsulation.None
                }] }
    ];
    /** @nocollapse */
    FormFieldComponent.ctorParameters = function () { return [
        { type: FormRenderingService },
        { type: ComponentFactoryResolver },
        { type: WidgetVisibilityService },
        { type: Compiler }
    ]; };
    FormFieldComponent.propDecorators = {
        container: [{ type: ViewChild, args: ['container', { read: ViewContainerRef },] }],
        field: [{ type: Input }]
    };
    return FormFieldComponent;
}());
export { FormFieldComponent };
if (false) {
    /** @type {?} */
    FormFieldComponent.prototype.container;
    /**
     * Contains all the necessary data needed to determine what UI Widget
     * to use when rendering the field in the form. You would typically not
     * create this data manually but instead create the form in APS and export
     * it to get to all the `FormFieldModel` definitions.
     * @type {?}
     */
    FormFieldComponent.prototype.field;
    /** @type {?} */
    FormFieldComponent.prototype.componentRef;
    /** @type {?} */
    FormFieldComponent.prototype.focus;
    /**
     * @type {?}
     * @private
     */
    FormFieldComponent.prototype.formRenderingService;
    /**
     * @type {?}
     * @private
     */
    FormFieldComponent.prototype.componentFactoryResolver;
    /**
     * @type {?}
     * @private
     */
    FormFieldComponent.prototype.visibilityService;
    /**
     * @type {?}
     * @private
     */
    FormFieldComponent.prototype.compiler;
}
