// @alfresco/adf-process-services/esm5/task-list/services/process-upload.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiService, AppConfigService, UploadService } from '@alfresco/adf-core';
import { Injectable } from '@angular/core';
import { throwError } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
var ProcessUploadService = /** @class */ (function (_super) {
    tslib_1.__extends(ProcessUploadService, _super);
    function ProcessUploadService(apiService, appConfigService) {
        return _super.call(this, apiService, appConfigService) || this;
    }
    /**
     * @param {?} file
     * @return {?}
     */
    ProcessUploadService.prototype.getUploadPromise = /**
     * @param {?} file
     * @return {?}
     */
    function (file) {
        var _this = this;
        /** @type {?} */
        var opts = {
            isRelatedContent: true
        };
        /** @type {?} */
        var processInstanceId = file.options.parentId;
        /** @type {?} */
        var promise = this.apiService.getInstance().activiti.contentApi.createRelatedContentOnProcessInstance(processInstanceId, file.file, opts);
        promise.catch((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); }));
        return promise;
    };
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    ProcessUploadService.prototype.handleError = /**
     * @private
     * @param {?} error
     * @return {?}
     */
    function (error) {
        return throwError(error || 'Server error');
    };
    ProcessUploadService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ProcessUploadService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: AppConfigService }
    ]; };
    /** @nocollapse */ ProcessUploadService.ngInjectableDef = i0.defineInjectable({ factory: function ProcessUploadService_Factory() { return new ProcessUploadService(i0.inject(i1.AlfrescoApiService), i0.inject(i1.AppConfigService)); }, token: ProcessUploadService, providedIn: "root" });
    return ProcessUploadService;
}(UploadService));
export { ProcessUploadService };
