// @alfresco/adf-process-services/esm5/task-list/models/filter.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { UserTaskFilterRepresentation, TaskQueryRepresentation } from '@alfresco/js-api';
var AppDefinitionRepresentationModel = /** @class */ (function () {
    function AppDefinitionRepresentationModel(obj) {
        if (obj) {
            this.defaultAppId = obj.defaultAppId ? obj.defaultAppId : null;
            this.deploymentId = obj.deploymentId ? obj.deploymentId : null;
            this.name = obj.name ? obj.name : null;
            this.description = obj.description ? obj.description : null;
            this.theme = obj.theme ? obj.theme : null;
            this.icon = obj.icon ? obj.icon : null;
            this.id = obj.id ? obj.id : null;
            this.modelId = obj.modelId ? obj.modelId : null;
            this.tenantId = obj.tenantId ? obj.tenantId : null;
        }
    }
    return AppDefinitionRepresentationModel;
}());
export { AppDefinitionRepresentationModel };
if (false) {
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.defaultAppId;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.deploymentId;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.name;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.description;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.theme;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.icon;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.id;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.modelId;
    /** @type {?} */
    AppDefinitionRepresentationModel.prototype.tenantId;
}
var FilterParamsModel = /** @class */ (function () {
    function FilterParamsModel(obj) {
        if (obj) {
            this.id = obj.id || null;
            this.name = obj.name || null;
            this.index = obj.index;
        }
    }
    return FilterParamsModel;
}());
export { FilterParamsModel };
if (false) {
    /** @type {?} */
    FilterParamsModel.prototype.id;
    /** @type {?} */
    FilterParamsModel.prototype.name;
    /** @type {?} */
    FilterParamsModel.prototype.index;
}
var FilterRepresentationModel = /** @class */ (function () {
    function FilterRepresentationModel(obj) {
        if (obj) {
            this.id = obj.id || null;
            this.appId = obj.appId || null;
            this.name = obj.name || null;
            this.recent = !!obj.recent;
            this.icon = obj.icon || null;
            this.filter = new UserTaskFilterRepresentation(obj.filter);
            this.index = obj.index;
        }
    }
    /**
     * @return {?}
     */
    FilterRepresentationModel.prototype.hasFilter = /**
     * @return {?}
     */
    function () {
        return this.filter ? true : false;
    };
    return FilterRepresentationModel;
}());
export { FilterRepresentationModel };
if (false) {
    /** @type {?} */
    FilterRepresentationModel.prototype.id;
    /** @type {?} */
    FilterRepresentationModel.prototype.appId;
    /** @type {?} */
    FilterRepresentationModel.prototype.name;
    /** @type {?} */
    FilterRepresentationModel.prototype.recent;
    /** @type {?} */
    FilterRepresentationModel.prototype.icon;
    /** @type {?} */
    FilterRepresentationModel.prototype.filter;
    /** @type {?} */
    FilterRepresentationModel.prototype.index;
}
var TaskQueryRequestRepresentationModel = /** @class */ (function (_super) {
    tslib_1.__extends(TaskQueryRequestRepresentationModel, _super);
    function TaskQueryRequestRepresentationModel() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return TaskQueryRequestRepresentationModel;
}(TaskQueryRepresentation));
export { TaskQueryRequestRepresentationModel };
