// @alfresco/adf-process-services/esm5/task-list/models/task-details.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * This object represent the details of a task.
 */
import { UserProcessModel } from '@alfresco/adf-core';
var TaskDetailsModel = /** @class */ (function () {
    function TaskDetailsModel(obj) {
        this.processDefinitionVersion = 0;
        if (obj) {
            this.id = obj.id || null;
            this.name = obj.name || null;
            this.priority = obj.priority;
            this.assignee = obj.assignee ? new UserProcessModel(obj.assignee) : null;
            this.adhocTaskCanBeReassigned = obj.adhocTaskCanBeReassigned;
            this.category = obj.category || null;
            this.created = obj.created || null;
            this.description = obj.description || null;
            this.dueDate = obj.dueDate || null;
            this.duration = obj.duration || null;
            this.endDate = obj.endDate || null;
            this.executionId = obj.executionId || null;
            this.formKey = obj.formKey || null;
            this.initiatorCanCompleteTask = !!obj.initiatorCanCompleteTask;
            this.managerOfCandidateGroup = !!obj.managerOfCandidateGroup;
            this.memberOfCandidateGroup = !!obj.memberOfCandidateGroup;
            this.memberOfCandidateUsers = !!obj.memberOfCandidateUsers;
            this.involvedGroups = obj.involvedGroups;
            this.involvedPeople = obj.involvedPeople;
            this.parentTaskId = obj.parentTaskId || null;
            this.parentTaskName = obj.parentTaskName || null;
            this.processDefinitionCategory = obj.processDefinitionCategory || null;
            this.processDefinitionDeploymentId = obj.processDefinitionDeploymentId || null;
            this.processDefinitionDescription = obj.processDefinitionDescription || null;
            this.processDefinitionId = obj.processDefinitionId || null;
            this.processDefinitionKey = obj.processDefinitionKey || null;
            this.processDefinitionName = obj.processDefinitionName || null;
            this.processDefinitionVersion = obj.processDefinitionVersion || 0;
            this.processInstanceId = obj.processInstanceId || null;
            this.processInstanceName = obj.processInstanceName || null;
            this.processInstanceStartUserId = obj.processInstanceStartUserId || null;
            this.taskDefinitionKey = obj.taskDefinitionKey || null;
        }
    }
    /**
     * @return {?}
     */
    TaskDetailsModel.prototype.getFullName = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var fullName = '';
        if (this.assignee) {
            /** @type {?} */
            var firstName = this.assignee.firstName ? this.assignee.firstName : '';
            /** @type {?} */
            var lastName = this.assignee.lastName ? this.assignee.lastName : '';
            fullName = firstName + " " + lastName;
        }
        return fullName.trim();
    };
    /**
     * @return {?}
     */
    TaskDetailsModel.prototype.isCompleted = /**
     * @return {?}
     */
    function () {
        return !!this.endDate;
    };
    return TaskDetailsModel;
}());
export { TaskDetailsModel };
if (false) {
    /** @type {?} */
    TaskDetailsModel.prototype.id;
    /** @type {?} */
    TaskDetailsModel.prototype.name;
    /** @type {?} */
    TaskDetailsModel.prototype.assignee;
    /** @type {?} */
    TaskDetailsModel.prototype.priority;
    /** @type {?} */
    TaskDetailsModel.prototype.adhocTaskCanBeReassigned;
    /** @type {?} */
    TaskDetailsModel.prototype.category;
    /** @type {?} */
    TaskDetailsModel.prototype.created;
    /** @type {?} */
    TaskDetailsModel.prototype.description;
    /** @type {?} */
    TaskDetailsModel.prototype.parentName;
    /** @type {?} */
    TaskDetailsModel.prototype.dueDate;
    /** @type {?} */
    TaskDetailsModel.prototype.duration;
    /** @type {?} */
    TaskDetailsModel.prototype.endDate;
    /** @type {?} */
    TaskDetailsModel.prototype.executionId;
    /** @type {?} */
    TaskDetailsModel.prototype.formKey;
    /** @type {?} */
    TaskDetailsModel.prototype.initiatorCanCompleteTask;
    /** @type {?} */
    TaskDetailsModel.prototype.managerOfCandidateGroup;
    /** @type {?} */
    TaskDetailsModel.prototype.memberOfCandidateGroup;
    /** @type {?} */
    TaskDetailsModel.prototype.memberOfCandidateUsers;
    /** @type {?} */
    TaskDetailsModel.prototype.involvedGroups;
    /** @type {?} */
    TaskDetailsModel.prototype.involvedPeople;
    /** @type {?} */
    TaskDetailsModel.prototype.parentTaskId;
    /** @type {?} */
    TaskDetailsModel.prototype.parentTaskName;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionCategory;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionDeploymentId;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionDescription;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionId;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionKey;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionName;
    /** @type {?} */
    TaskDetailsModel.prototype.processDefinitionVersion;
    /** @type {?} */
    TaskDetailsModel.prototype.processInstanceId;
    /** @type {?} */
    TaskDetailsModel.prototype.processInstanceName;
    /** @type {?} */
    TaskDetailsModel.prototype.processInstanceStartUserId;
    /** @type {?} */
    TaskDetailsModel.prototype.taskDefinitionKey;
}
