// @alfresco/adf-process-services/esm5/task-list/models/user-group.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * This object represent the process service user group.*
 */
var UserGroupModel = /** @class */ (function () {
    function UserGroupModel(obj) {
        this.groups = {};
        this.id = obj && obj.id;
        this.name = obj && obj.name;
        this.externalId = obj && obj.externalId;
        this.status = obj && obj.status;
        this.groups = obj && obj.groups;
    }
    return UserGroupModel;
}());
export { UserGroupModel };
if (false) {
    /** @type {?} */
    UserGroupModel.prototype.id;
    /** @type {?} */
    UserGroupModel.prototype.name;
    /** @type {?} */
    UserGroupModel.prototype.externalId;
    /** @type {?} */
    UserGroupModel.prototype.status;
    /** @type {?} */
    UserGroupModel.prototype.groups;
}
