// @alfresco/adf-process-services/esm2015/task-list/components/task-standalone.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
export class TaskStandaloneComponent {
    constructor() {
        /**
         * If true then Task completed message is shown and `Complete` and `Cancel` buttons are hidden.
         */
        this.isCompleted = false;
        /**
         * Toggles rendering of the `Complete` button.
         */
        this.hasCompletePermission = true;
        // TODO: rename all with show prefix
        /**
         * Toggles rendering of the `Cancel` button.
         */
        this.hideCancelButton = true;
        /**
         * Emitted when the "Cancel" button is clicked.
         */
        this.cancel = new EventEmitter();
        /**
         * Emitted when the form associated with the task is completed.
         */
        this.complete = new EventEmitter();
        /**
         * Emitted when the form associated with the form task is attached.
         */
        this.showAttachForm = new EventEmitter();
    }
    /**
     * @return {?}
     */
    onCancelButtonClick() {
        this.cancel.emit();
    }
    /**
     * @return {?}
     */
    onCompleteButtonClick() {
        this.complete.emit();
    }
    /**
     * @return {?}
     */
    hasCompleteButton() {
        return this.hasCompletePermission && !this.isCompleted;
    }
    /**
     * @return {?}
     */
    hasCancelButton() {
        return !this.hideCancelButton && !this.isCompleted;
    }
    /**
     * @return {?}
     */
    hasAttachFormButton() {
        return !this.isCompleted;
    }
    /**
     * @return {?}
     */
    onShowAttachForm() {
        this.showAttachForm.emit();
    }
}
TaskStandaloneComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-task-standalone',
                template: "<mat-card class=\"adf-message-card\">\n    <mat-card-content>\n        <div class=\"adf-no-form-message-container\">\n            <div class=\"adf-no-form-message-list\">\n                <div *ngIf=\"!isCompleted; else completedMessage\" class=\"adf-no-form-message\">\n                    <span id=\"adf-no-form-message\">{{'ADF_TASK_LIST.STANDALONE_TASK.NO_FORM_MESSAGE' | translate}}</span>\n                </div>\n                <ng-template #completedMessage>\n                    <div id=\"adf-completed-form-message\" class=\"adf-no-form-message\">\n                        <p>{{'ADF_TASK_LIST.STANDALONE_TASK.COMPLETE_TASK_MESSAGE' | translate : {taskName : taskName} }}</p>\n                    </div>\n                    <div class=\"adf-no-form-submessage\">\n                        {{'ADF_TASK_LIST.STANDALONE_TASK.COMPLETE_TASK_SUB_MESSAGE' | translate}}\n                    </div>\n                </ng-template>\n            </div>\n        </div>\n    </mat-card-content>\n\n    <mat-card-actions class=\"adf-no-form-mat-card-actions\">\n        <button mat-button *ngIf=\"hasAttachFormButton()\" id=\"adf-no-form-attach-form-button\" (click)=\"onShowAttachForm()\">{{ 'ADF_TASK_LIST.START_TASK.FORM.LABEL.ATTACHFORM' | translate }}</button>\n        <div>\n            <button mat-button *ngIf=\"hasCancelButton()\" id=\"adf-no-form-cancel-button\" (click)=\"onCancelButtonClick()\">{{ 'ADF_TASK_LIST.START_TASK.FORM.ACTION.CANCEL' | translate }}</button>\n            <button mat-button *ngIf=\"hasCompleteButton()\" id=\"adf-no-form-complete-button\" color=\"primary\" (click)=\"onCompleteButtonClick()\">{{ 'ADF_TASK_LIST.DETAILS.BUTTON.COMPLETE' | translate }}</button>\n        </div>\n    </mat-card-actions>\n</mat-card>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            }] }
];
/** @nocollapse */
TaskStandaloneComponent.ctorParameters = () => [];
TaskStandaloneComponent.propDecorators = {
    taskName: [{ type: Input }],
    taskId: [{ type: Input }],
    isCompleted: [{ type: Input }],
    hasCompletePermission: [{ type: Input }],
    hideCancelButton: [{ type: Input }],
    cancel: [{ type: Output }],
    complete: [{ type: Output }],
    showAttachForm: [{ type: Output }]
};
if (false) {
    /**
     * Name of the task.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.taskName;
    /**
     * Id of the task.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.taskId;
    /**
     * If true then Task completed message is shown and `Complete` and `Cancel` buttons are hidden.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.isCompleted;
    /**
     * Toggles rendering of the `Complete` button.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.hasCompletePermission;
    /**
     * Toggles rendering of the `Cancel` button.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.hideCancelButton;
    /**
     * Emitted when the "Cancel" button is clicked.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.cancel;
    /**
     * Emitted when the form associated with the task is completed.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.complete;
    /**
     * Emitted when the form associated with the form task is attached.
     * @type {?}
     */
    TaskStandaloneComponent.prototype.showAttachForm;
}
