// @alfresco/adf-core/esm5/form/components/widgets/hyperlink/hyperlink.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { baseHost, WidgetComponent } from './../widget.component';
var HyperlinkWidgetComponent = /** @class */ (function (_super) {
    tslib_1.__extends(HyperlinkWidgetComponent, _super);
    function HyperlinkWidgetComponent(formService) {
        var _this = _super.call(this, formService) || this;
        _this.formService = formService;
        _this.linkUrl = WidgetComponent.DEFAULT_HYPERLINK_URL;
        _this.linkText = null;
        return _this;
    }
    /**
     * @return {?}
     */
    HyperlinkWidgetComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.field) {
            this.linkUrl = this.getHyperlinkUrl(this.field);
            this.linkText = this.getHyperlinkText(this.field);
        }
    };
    HyperlinkWidgetComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hyperlink-widget',
                    template: "<div class=\"adf-hyperlink-widget {{field.className}}\">\n    <label class=\"adf-label\" [attr.for]=\"field.id\">{{field.name}}<span *ngIf=\"isRequired()\">*</span></label>\n    <div>\n        <a [href]=\"linkUrl\" target=\"_blank\" rel=\"nofollow\">{{linkText}}</a>\n    </div>\n</div>\n",
                    host: baseHost,
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    HyperlinkWidgetComponent.ctorParameters = function () { return [
        { type: FormService }
    ]; };
    return HyperlinkWidgetComponent;
}(WidgetComponent));
export { HyperlinkWidgetComponent };
if (false) {
    /** @type {?} */
    HyperlinkWidgetComponent.prototype.linkUrl;
    /** @type {?} */
    HyperlinkWidgetComponent.prototype.linkText;
    /** @type {?} */
    HyperlinkWidgetComponent.prototype.formService;
}
