// @alfresco/js-api/_esm5/src/api/content-rest-api/api/favorites.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { FavoriteEntry } from '../model/favoriteEntry';
import { FavoritePaging } from '../model/favoritePaging';
import { FavoriteSiteEntry } from '../model/favoriteSiteEntry';
import { SiteEntry } from '../model/siteEntry';
import { SitePaging } from '../model/sitePaging';
import { BaseApi } from './base.api';
var FavoritesApi = (function (_super) {
    tslib_1.__extends(FavoritesApi, _super);
    function FavoritesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FavoritesApi.prototype.createFavorite = function (personId, favoriteBodyCreate, opts) {
        opts = opts || {};
        var postBody = favoriteBodyCreate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in createFavorite");
        }
        if (favoriteBodyCreate === undefined || favoriteBodyCreate === null) {
            throw new Error("Required param 'favoriteBodyCreate' in createFavorite");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorites', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, FavoriteEntry);
    };
    FavoritesApi.prototype.createSiteFavorite = function (personId, favoriteSiteBodyCreate, opts) {
        opts = opts || {};
        var postBody = favoriteSiteBodyCreate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in createSiteFavorite");
        }
        if (favoriteSiteBodyCreate === undefined || favoriteSiteBodyCreate === null) {
            throw new Error("Required param 'favoriteSiteBodyCreate' in createSiteFavorite");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorite-sites', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, FavoriteSiteEntry);
    };
    FavoritesApi.prototype.deleteFavorite = function (personId, favoriteId) {
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteFavorite");
        }
        if (favoriteId === undefined || favoriteId === null) {
            throw new Error("Required param 'favoriteId' in deleteFavorite");
        }
        var pathParams = {
            'personId': personId, 'favoriteId': favoriteId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorites/{favoriteId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    FavoritesApi.prototype.deleteSiteFavorite = function (personId, siteId) {
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteSiteFavorite");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in deleteSiteFavorite");
        }
        var pathParams = {
            'personId': personId, 'siteId': siteId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorite-sites/{siteId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    FavoritesApi.prototype.getFavorite = function (personId, favoriteId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getFavorite");
        }
        if (favoriteId === undefined || favoriteId === null) {
            throw new Error("Required param 'favoriteId' in getFavorite");
        }
        var pathParams = {
            'personId': personId, 'favoriteId': favoriteId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorites/{favoriteId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, FavoriteEntry);
    };
    FavoritesApi.prototype.getFavoriteSite = function (personId, siteId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getFavoriteSite");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getFavoriteSite");
        }
        var pathParams = {
            'personId': personId, 'siteId': siteId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorite-sites/{siteId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, SiteEntry);
    };
    FavoritesApi.prototype.listFavoriteSitesForPerson = function (personId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listFavoriteSitesForPerson");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorite-sites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, SitePaging);
    };
    FavoritesApi.prototype.listFavorites = function (personId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listFavorites");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/favorites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, FavoritePaging);
    };
    return FavoritesApi;
}(BaseApi));
export { FavoritesApi };
