// @alfresco/js-api/_esm5/src/api/content-rest-api/api/groups.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { GroupEntry } from '../model/groupEntry';
import { GroupMemberEntry } from '../model/groupMemberEntry';
import { GroupMemberPaging } from '../model/groupMemberPaging';
import { GroupPaging } from '../model/groupPaging';
import { BaseApi } from './base.api';
var GroupsApi = (function (_super) {
    tslib_1.__extends(GroupsApi, _super);
    function GroupsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    GroupsApi.prototype.createGroup = function (groupBodyCreate, opts) {
        opts = opts || {};
        var postBody = groupBodyCreate;
        if (groupBodyCreate === undefined || groupBodyCreate === null) {
            throw new Error("Required param 'groupBodyCreate' in createGroup");
        }
        var pathParams = {};
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    };
    GroupsApi.prototype.createGroupMembership = function (groupId, groupMembershipBodyCreate, opts) {
        opts = opts || {};
        var postBody = groupMembershipBodyCreate;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in createGroupMembership");
        }
        if (groupMembershipBodyCreate === undefined || groupMembershipBodyCreate === null) {
            throw new Error("Required param 'groupMembershipBodyCreate' in createGroupMembership");
        }
        var pathParams = {
            'groupId': groupId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupMemberEntry);
    };
    GroupsApi.prototype.deleteGroup = function (groupId, opts) {
        opts = opts || {};
        var postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in deleteGroup");
        }
        var cascadeDelete = opts['cascade'] ? opts['cascade'] : false;
        var pathParams = {
            'groupId': groupId
        };
        var queryParams = {
            'cascade': cascadeDelete
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    GroupsApi.prototype.deleteGroupMembership = function (groupId, groupMemberId) {
        var postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in deleteGroupMembership");
        }
        if (groupMemberId === undefined || groupMemberId === null) {
            throw new Error("Required param 'groupMemberId' in deleteGroupMembership");
        }
        var pathParams = {
            'groupId': groupId, 'groupMemberId': groupMemberId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members/{groupMemberId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    GroupsApi.prototype.getGroup = function (groupId, opts) {
        opts = opts || {};
        var postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in getGroup");
        }
        var pathParams = {
            'groupId': groupId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    };
    GroupsApi.prototype.listGroupMemberships = function (groupId, opts) {
        opts = opts || {};
        var postBody = null;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in listGroupMemberships");
        }
        var pathParams = {
            'groupId': groupId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}/members', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupMemberPaging);
    };
    GroupsApi.prototype.listGroupMembershipsForPerson = function (personId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in listGroupMembershipsForPerson");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupPaging);
    };
    GroupsApi.prototype.listGroups = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'where': opts['where'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupPaging);
    };
    GroupsApi.prototype.updateGroup = function (groupId, groupBodyUpdate, opts) {
        opts = opts || {};
        var postBody = groupBodyUpdate;
        if (groupId === undefined || groupId === null) {
            throw new Error("Required param 'groupId' in updateGroup");
        }
        if (groupBodyUpdate === undefined || groupBodyUpdate === null) {
            throw new Error("Required param 'groupBodyUpdate' in updateGroup");
        }
        var pathParams = {
            'groupId': groupId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/groups/{groupId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupEntry);
    };
    return GroupsApi;
}(BaseApi));
export { GroupsApi };
