// @alfresco/js-api/_esm5/src/api/content-rest-api/api/nodes.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { AssociationEntry } from '../model/associationEntry';
import { ChildAssociationEntry } from '../model/childAssociationEntry';
import { NodeAssociationPaging } from '../model/nodeAssociationPaging';
import { NodeChildAssociationPaging } from '../model/nodeChildAssociationPaging';
import { NodeEntry } from '../model/nodeEntry';
import { BaseApi } from './base.api';
var NodesApi = (function (_super) {
    tslib_1.__extends(NodesApi, _super);
    function NodesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NodesApi.prototype.copyNode = function (nodeId, nodeBodyCopy, opts) {
        opts = opts || {};
        var postBody = nodeBodyCopy;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in copyNode");
        }
        if (nodeBodyCopy === undefined || nodeBodyCopy === null) {
            throw new Error("Required param 'nodeBodyCopy' in copyNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/copy', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.createAssociation = function (nodeId, associationBodyCreate, opts) {
        opts = opts || {};
        var postBody = associationBodyCreate;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in createAssociation");
        }
        if (associationBodyCreate === undefined || associationBodyCreate === null) {
            throw new Error("Required param 'associationBodyCreate' in createAssociation");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AssociationEntry);
    };
    NodesApi.prototype.createNode = function (nodeId, nodeBodyCreate, opts, formParams) {
        opts = opts || {};
        var postBody = nodeBodyCreate;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in createNode");
        }
        if (nodeBodyCreate === undefined || nodeBodyCreate === null) {
            throw new Error("Required param 'nodeBodyCreate' in createNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'autoRename': opts['autoRename'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        formParams = formParams || {};
        var contentTypes, returnType = null;
        if (formParams.filedata) {
            contentTypes = ['multipart/form-data'];
        }
        else {
            contentTypes = ['application/json'];
            returnType = NodeEntry;
        }
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, returnType);
    };
    NodesApi.prototype.createSecondaryChildAssociation = function (nodeId, secondaryChildAssociationBodyCreate, opts) {
        opts = opts || {};
        var postBody = secondaryChildAssociationBodyCreate;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in createSecondaryChildAssociation");
        }
        if (secondaryChildAssociationBodyCreate === undefined || secondaryChildAssociationBodyCreate === null) {
            throw new Error("Required param 'secondaryChildAssociationBodyCreate' in createSecondaryChildAssociation");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ChildAssociationEntry);
    };
    NodesApi.prototype.deleteAssociation = function (nodeId, targetId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in deleteAssociation");
        }
        if (targetId === undefined || targetId === null) {
            throw new Error("Required param 'targetId' in deleteAssociation");
        }
        var pathParams = {
            'nodeId': nodeId, 'targetId': targetId
        };
        var queryParams = {
            'assocType': opts['assocType']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets/{targetId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    NodesApi.prototype.deleteNode = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in deleteNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'permanent': opts['permanent']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    NodesApi.prototype.deleteSecondaryChildAssociation = function (nodeId, childId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in deleteSecondaryChildAssociation");
        }
        if (childId === undefined || childId === null) {
            throw new Error("Required param 'childId' in deleteSecondaryChildAssociation");
        }
        var pathParams = {
            'nodeId': nodeId, 'childId': childId
        };
        var queryParams = {
            'assocType': opts['assocType']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children/{childId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    NodesApi.prototype.getNode = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.getNodeContent = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getNodeContent");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'attachment': opts['attachment']
        };
        var headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    NodesApi.prototype.listNodeChildren = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listNodeChildren");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'includeSource': opts['includeSource'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeChildAssociationPaging);
    };
    NodesApi.prototype.listParents = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listParents");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'includeSource': opts['includeSource'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/parents', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeAssociationPaging);
    };
    NodesApi.prototype.listSecondaryChildren = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listSecondaryChildren");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'includeSource': opts['includeSource'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeChildAssociationPaging);
    };
    NodesApi.prototype.listSourceAssociations = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listSourceAssociations");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/sources', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeAssociationPaging);
    };
    NodesApi.prototype.listTargetAssociations = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listTargetAssociations");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeAssociationPaging);
    };
    NodesApi.prototype.lockNode = function (nodeId, nodeBodyLock, opts) {
        opts = opts || {};
        var postBody = nodeBodyLock;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in lockNode");
        }
        if (nodeBodyLock === undefined || nodeBodyLock === null) {
            throw new Error("Required param 'nodeBodyLock' in lockNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/lock', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.moveNode = function (nodeId, nodeBodyMove, opts) {
        opts = opts || {};
        var postBody = nodeBodyMove;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in moveNode");
        }
        if (nodeBodyMove === undefined || nodeBodyMove === null) {
            throw new Error("Required param 'nodeBodyMove' in moveNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/move', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.unlockNode = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in unlockNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/unlock', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.updateNode = function (nodeId, nodeBodyUpdate, opts) {
        opts = opts || {};
        var postBody = nodeBodyUpdate;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in updateNode");
        }
        if (nodeBodyUpdate === undefined || nodeBodyUpdate === null) {
            throw new Error("Required param 'nodeBodyUpdate' in updateNode");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    NodesApi.prototype.updateNodeContent = function (nodeId, contentBodyUpdate, opts) {
        opts = opts || {};
        var postBody = contentBodyUpdate;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in updateNodeContent");
        }
        if (contentBodyUpdate === undefined || contentBodyUpdate === null) {
            throw new Error("Required param 'contentBodyUpdate' in updateNodeContent");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'majorVersion': opts['majorVersion'],
            'comment': opts['comment'],
            'name': opts['name'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/octet-stream'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/content', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, NodeEntry);
    };
    return NodesApi;
}(BaseApi));
export { NodesApi };
