// @alfresco/js-api/_esm5/src/api/content-rest-api/api/people.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { PersonEntry } from '../model/personEntry';
import { PersonPaging } from '../model/personPaging';
import { BaseApi } from './base.api';
var PeopleApi = (function (_super) {
    tslib_1.__extends(PeopleApi, _super);
    function PeopleApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PeopleApi.prototype.createPerson = function (personBodyCreate, opts) {
        opts = opts || {};
        var postBody = personBodyCreate;
        if (personBodyCreate === undefined || personBodyCreate === null) {
            throw new Error("Required param 'personBodyCreate' in createPerson");
        }
        var pathParams = {};
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    };
    PeopleApi.prototype.deleteAvatarImage = function (personId) {
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in deleteAvatarImage");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    PeopleApi.prototype.getAvatarImage = function (personId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getAvatarImage");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'attachment': opts['attachment'],
            'placeholder': opts['placeholder']
        };
        var headerParams = {
            'If-Modified-Since': opts['ifModifiedSince']
        };
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    PeopleApi.prototype.getPerson = function (personId, opts) {
        opts = opts || {};
        var postBody = null;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in getPerson");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    };
    PeopleApi.prototype.listPeople = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonPaging);
    };
    PeopleApi.prototype.requestPasswordReset = function (personId, clientBody) {
        var postBody = clientBody;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in requestPasswordReset");
        }
        if (clientBody === undefined || clientBody === null) {
            throw new Error("Required param 'clientBody' in requestPasswordReset");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/request-password-reset', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    PeopleApi.prototype.resetPassword = function (personId, passwordResetBody) {
        var postBody = passwordResetBody;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in resetPassword");
        }
        if (passwordResetBody === undefined || passwordResetBody === null) {
            throw new Error("Required param 'passwordResetBody' in resetPassword");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/reset-password', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    PeopleApi.prototype.updateAvatarImage = function (personId, contentBodyUpdate) {
        var postBody = contentBodyUpdate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updateAvatarImage");
        }
        if (contentBodyUpdate === undefined || contentBodyUpdate === null) {
            throw new Error("Required param 'contentBodyUpdate' in updateAvatarImage");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/octet-stream'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}/avatar', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    PeopleApi.prototype.updatePerson = function (personId, personBodyUpdate, opts) {
        opts = opts || {};
        var postBody = personBodyUpdate;
        if (personId === undefined || personId === null) {
            throw new Error("Required param 'personId' in updatePerson");
        }
        if (personBodyUpdate === undefined || personBodyUpdate === null) {
            throw new Error("Required param 'personBodyUpdate' in updatePerson");
        }
        var pathParams = {
            'personId': personId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/people/{personId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, PersonEntry);
    };
    return PeopleApi;
}(BaseApi));
export { PeopleApi };
