// @alfresco/js-api/_esm5/src/api/content-rest-api/api/versions.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { VersionEntry } from '../model/versionEntry';
import { VersionPaging } from '../model/versionPaging';
import { BaseApi } from './base.api';
var VersionsApi = (function (_super) {
    tslib_1.__extends(VersionsApi, _super);
    function VersionsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    VersionsApi.prototype.deleteVersion = function (nodeId, versionId) {
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in deleteVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in deleteVersion");
        }
        var pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    VersionsApi.prototype.getVersion = function (nodeId, versionId) {
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in getVersion");
        }
        var pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionEntry);
    };
    VersionsApi.prototype.getVersionContent = function (nodeId, versionId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in getVersionContent");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in getVersionContent");
        }
        var pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        var queryParams = {
            'attachment': opts['attachment']
        };
        var headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    VersionsApi.prototype.listVersionHistory = function (nodeId, opts) {
        opts = opts || {};
        var postBody = null;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in listVersionHistory");
        }
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionPaging);
    };
    VersionsApi.prototype.revertVersion = function (nodeId, versionId, revertBody, opts) {
        opts = opts || {};
        var postBody = revertBody;
        if (nodeId === undefined || nodeId === null) {
            throw new Error("Required param 'nodeId' in revertVersion");
        }
        if (versionId === undefined || versionId === null) {
            throw new Error("Required param 'versionId' in revertVersion");
        }
        if (revertBody === undefined || revertBody === null) {
            throw new Error("Required param 'revertBody' in revertVersion");
        }
        var pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        var queryParams = {
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/revert', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, VersionEntry);
    };
    return VersionsApi;
}(BaseApi));
export { VersionsApi };
