// @alfresco/adf-core/esm2015/interface/authentication.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * @record
 */
export function AbstractAuthentication() { }
if (false) {
    /** @type {?} */
    AbstractAuthentication.prototype.TYPE;
    /** @type {?} */
    AbstractAuthentication.prototype.alfrescoApi;
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    AbstractAuthentication.prototype.login = function (username, password) { };
    /**
     * @return {?}
     */
    AbstractAuthentication.prototype.logout = function () { };
    /**
     * @return {?}
     */
    AbstractAuthentication.prototype.isLoggedIn = function () { };
    /**
     * @return {?}
     */
    AbstractAuthentication.prototype.getTicket = function () { };
    /**
     * @param {?} ticket
     * @return {?}
     */
    AbstractAuthentication.prototype.saveTicket = function (ticket) { };
}
