// @alfresco/adf-testing/esm2015/lib/core/actions/api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiCompatibility as AlfrescoApi } from '@alfresco/js-api';
export class ApiService {
    /**
     * @param {?} clientId
     * @param {?} host
     * @param {?} hostSso
     * @param {?} provider
     */
    constructor(clientId, host, hostSso, provider) {
        this.config = {
            provider: provider,
            hostBpm: host,
            hostEcm: host,
            authType: 'OAUTH',
            oauth2: {
                host: hostSso,
                clientId: clientId,
                scope: 'openid',
                secret: '',
                implicitFlow: false,
                silentLogin: false,
                redirectUri: '/',
                redirectUriLogout: '/logout'
            }
        };
        this.apiService = new AlfrescoApi(this.config);
    }
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    login(username, password) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield this.apiService.login(username, password);
        });
    }
    /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    performBpmOperation(path, method, queryParams, postBody) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const uri = this.config.hostBpm + path;
            /** @type {?} */
            const pathParams = {};
            /** @type {?} */
            const formParams = {};
            /** @type {?} */
            const contentTypes = ['application/json'];
            /** @type {?} */
            const accepts = ['application/json'];
            /** @type {?} */
            const headerParams = {
                'Authorization': 'bearer ' + this.apiService.oauth2Auth.token
            };
            return this.apiService.processClient.callCustomApi(uri, method, pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, Object)
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            (error) => {
                throw (error);
            }));
        });
    }
    /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    performIdentityOperation(path, method, queryParams, postBody) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const uri = this.config.oauth2.host.replace('/realms', '/admin/realms') + path;
            /** @type {?} */
            const pathParams = {};
            /** @type {?} */
            const formParams = {};
            /** @type {?} */
            const contentTypes = ['application/json'];
            /** @type {?} */
            const accepts = ['application/json'];
            /** @type {?} */
            const headerParams = {
                'Authorization': 'bearer ' + this.apiService.oauth2Auth.token
            };
            return this.apiService.processClient.callCustomApi(uri, method, pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, Object)
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            (error) => {
                throw (error);
            }));
        });
    }
}
if (false) {
    /** @type {?} */
    ApiService.prototype.apiService;
    /** @type {?} */
    ApiService.prototype.config;
}
