// @alfresco/adf-insights/esm5/diagram/components/intermediate-catching-events/diagram-intermediate-catching-event.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramIntermediateCatchingEventComponent = /** @class */ (function () {
    function DiagramIntermediateCatchingEventComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.center = {};
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: 1 };
    }
    /**
     * @return {?}
     */
    DiagramIntermediateCatchingEventComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
        this.circleRadiusInner = 12;
        this.circleRadiusOuter = 15;
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
    };
    DiagramIntermediateCatchingEventComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-intermediate-catching-event',
                    template: "<raphael-circle [center]=\"center\" [radius]=\"circleRadiusInner\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<raphael-circle [elementId]=\"data.id\" [center]=\"center\" [radius]=\"circleRadiusOuter\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<diagram-container-icon-event [data]=\"data\" [type]=\"data.eventDefinition.type\"></diagram-container-icon-event>\n<diagram-tooltip [data]=\"data\"></diagram-tooltip>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramIntermediateCatchingEventComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramIntermediateCatchingEventComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramIntermediateCatchingEventComponent;
}());
export { DiagramIntermediateCatchingEventComponent };
if (false) {
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.data;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.error;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.center;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.options;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.circleRadiusInner;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.circleRadiusOuter;
    /** @type {?} */
    DiagramIntermediateCatchingEventComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramIntermediateCatchingEventComponent.prototype.diagramColorService;
}
