// @alfresco/js-api/_esm5/src/api/gs-core-rest-api/api/recordCategories.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { RecordCategoryChildEntry } from '../model/recordCategoryChildEntry';
import { RecordCategoryChildPaging } from '../model/recordCategoryChildPaging';
import { RecordCategoryEntry } from '../model/recordCategoryEntry';
import { BaseApi } from './base.api';
var RecordCategoriesApi = (function (_super) {
    tslib_1.__extends(RecordCategoriesApi, _super);
    function RecordCategoriesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RecordCategoriesApi.prototype.createRecordCategoryChild = function (recordCategoryId, nodeBodyCreate, opts) {
        opts = opts || {};
        var postBody = nodeBodyCreate;
        if (recordCategoryId === undefined || recordCategoryId === null) {
            throw new Error("Required param 'recordCategoryId' in createRecordCategoryChild");
        }
        if (nodeBodyCreate === undefined || nodeBodyCreate === null) {
            throw new Error("Required param 'nodeBodyCreate' in createRecordCategoryChild");
        }
        var pathParams = {
            'recordCategoryId': recordCategoryId
        };
        var queryParams = {
            'autoRename': opts['autoRename'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json', 'multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}/children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RecordCategoryChildEntry);
    };
    RecordCategoriesApi.prototype.deleteRecordCategory = function (recordCategoryId) {
        var postBody = null;
        if (recordCategoryId === undefined || recordCategoryId === null) {
            throw new Error("Required param 'recordCategoryId' in deleteRecordCategory");
        }
        var pathParams = {
            'recordCategoryId': recordCategoryId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    RecordCategoriesApi.prototype.getRecordCategory = function (recordCategoryId, opts) {
        opts = opts || {};
        var postBody = null;
        if (recordCategoryId === undefined || recordCategoryId === null) {
            throw new Error("Required param 'recordCategoryId' in getRecordCategory");
        }
        var pathParams = {
            'recordCategoryId': recordCategoryId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RecordCategoryEntry);
    };
    RecordCategoriesApi.prototype.listRecordCategoryChildren = function (recordCategoryId, opts) {
        opts = opts || {};
        var postBody = null;
        if (recordCategoryId === undefined || recordCategoryId === null) {
            throw new Error("Required param 'recordCategoryId' in listRecordCategoryChildren");
        }
        var pathParams = {
            'recordCategoryId': recordCategoryId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'includeSource': opts['includeSource'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}/children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RecordCategoryChildPaging);
    };
    RecordCategoriesApi.prototype.updateRecordCategory = function (recordCategoryId, recordCategoryBodyUpdate, opts) {
        opts = opts || {};
        var postBody = recordCategoryBodyUpdate;
        if (recordCategoryId === undefined || recordCategoryId === null) {
            throw new Error("Required param 'recordCategoryId' in updateRecordCategory");
        }
        if (recordCategoryBodyUpdate === undefined || recordCategoryBodyUpdate === null) {
            throw new Error("Required param 'recordCategoryBodyUpdate' in updateRecordCategory");
        }
        var pathParams = {
            'recordCategoryId': recordCategoryId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RecordCategoryEntry);
    };
    return RecordCategoriesApi;
}(BaseApi));
export { RecordCategoriesApi };
