// @alfresco/adf-insights/esm5/diagram/components/events/diagram-start-event.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramStartEventComponent = /** @class */ (function () {
    function DiagramStartEventComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: '', radius: '' };
    }
    /**
     * @return {?}
     */
    DiagramStartEventComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.options.radius = 15;
        this.options.strokeWidth = 1;
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
        this.iconFillColor = 'none';
    };
    DiagramStartEventComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-start-event',
                    template: "<diagram-event [data]=\"data\" [options]=\"options\" [iconFillColor]=\"iconFillColor\"></diagram-event>"
                }] }
    ];
    /** @nocollapse */
    DiagramStartEventComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramStartEventComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramStartEventComponent;
}());
export { DiagramStartEventComponent };
if (false) {
    /** @type {?} */
    DiagramStartEventComponent.prototype.data;
    /** @type {?} */
    DiagramStartEventComponent.prototype.error;
    /** @type {?} */
    DiagramStartEventComponent.prototype.options;
    /** @type {?} */
    DiagramStartEventComponent.prototype.iconFillColor;
    /** @type {?} */
    DiagramStartEventComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramStartEventComponent.prototype.diagramColorService;
}
