// @alfresco/adf-insights/esm5/diagram/components/events/diagram-event.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramEventComponent = /** @class */ (function () {
    function DiagramEventComponent(elementRef) {
        this.elementRef = elementRef;
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: '', radius: '' };
        this.error = new EventEmitter();
        this.center = {};
    }
    /**
     * @return {?}
     */
    DiagramEventComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
    };
    DiagramEventComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-event',
                    template: "<raphael-circle [elementId]=\"data.id\" [center]=\"center\" [radius]=\"options.radius\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<diagram-container-icon-event [data]=\"data\" [type]=\"data.eventDefinition && data.eventDefinition.type\"\n                              [fillColor]=\"iconFillColor\"></diagram-container-icon-event>\n<diagram-tooltip [data]=\"data\"></diagram-tooltip>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramEventComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramEventComponent.propDecorators = {
        data: [{ type: Input }],
        options: [{ type: Input }],
        iconFillColor: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramEventComponent;
}());
export { DiagramEventComponent };
if (false) {
    /** @type {?} */
    DiagramEventComponent.prototype.data;
    /** @type {?} */
    DiagramEventComponent.prototype.options;
    /** @type {?} */
    DiagramEventComponent.prototype.iconFillColor;
    /** @type {?} */
    DiagramEventComponent.prototype.error;
    /** @type {?} */
    DiagramEventComponent.prototype.center;
    /** @type {?} */
    DiagramEventComponent.prototype.elementRef;
}
