// @alfresco/adf-insights/esm2015/diagram/components/gateways/diagram-inclusive-gateway.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
export class DiagramInclusiveGatewayComponent {
    /**
     * @param {?} elementRef
     * @param {?} diagramColorService
     */
    constructor(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.center = {};
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: 2.5, radius: 9.75 };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
    }
}
DiagramInclusiveGatewayComponent.decorators = [
    { type: Component, args: [{
                selector: 'diagram-inclusive-gateway',
                template: "<diagram-gateway [data]=\"data\"></diagram-gateway>\n<raphael-circle [center]=\"center\" [radius]=\"options.radius\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>"
            }] }
];
/** @nocollapse */
DiagramInclusiveGatewayComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: DiagramColorService }
];
DiagramInclusiveGatewayComponent.propDecorators = {
    data: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.data;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.error;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.center;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.width;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.height;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.options;
    /** @type {?} */
    DiagramInclusiveGatewayComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramInclusiveGatewayComponent.prototype.diagramColorService;
}
