// @alfresco/adf-insights/esm2015/diagram/models/chart/heatMapChart.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Chart } from './chart.model';
export class HeatMapChart extends Chart {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        super(obj);
        this.avgTimePercentages = obj && obj.avgTimePercentages || null;
        this.avgTimeValues = obj && obj.avgTimeValues || null;
        this.processDefinitionId = obj && obj.processDefinitionId || null;
        this.totalCountValues = obj && obj.totalCountValues || null;
        this.titleKey = obj && obj.titleKey || null;
        this.totalCountsPercentages = obj && obj.totalCountsPercentages || null;
        this.totalTimePercentages = obj && obj.totalTimePercentages || null;
        this.totalTimeValues = obj && obj.totalTimeValues || null;
    }
}
if (false) {
    /** @type {?} */
    HeatMapChart.prototype.avgTimePercentages;
    /** @type {?} */
    HeatMapChart.prototype.avgTimeValues;
    /** @type {?} */
    HeatMapChart.prototype.processDefinitionId;
    /** @type {?} */
    HeatMapChart.prototype.titleKey;
    /** @type {?} */
    HeatMapChart.prototype.totalCountValues;
    /** @type {?} */
    HeatMapChart.prototype.totalCountsPercentages;
    /** @type {?} */
    HeatMapChart.prototype.totalTimePercentages;
    /** @type {?} */
    HeatMapChart.prototype.totalTimeValues;
}
