// @alfresco/adf-insights/esm2015/diagram/models/chart/pieChart.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Chart } from './chart.model';
export class PieChart extends Chart {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        super(obj);
        this.labels = [];
        this.data = [];
        this.title = obj && obj.title || null;
        this.titleKey = obj && obj.titleKey || null;
        if (obj.values) {
            obj.values.forEach((/**
             * @param {?} value
             * @return {?}
             */
            (value) => {
                this.add(value.key, value.y);
            }));
        }
    }
    /**
     * @param {?} label
     * @param {?} data
     * @return {?}
     */
    add(label, data) {
        this.labels.push(label);
        this.data.push(data);
    }
    /**
     * @return {?}
     */
    hasData() {
        return this.data && this.data.length > 0 ? true : false;
    }
    /**
     * @return {?}
     */
    hasZeroValues() {
        /** @type {?} */
        let isZeroValues = false;
        if (this.hasData()) {
            isZeroValues = true;
            this.data.forEach((/**
             * @param {?} value
             * @return {?}
             */
            (value) => {
                if (value.toString() !== '0') {
                    isZeroValues = false;
                }
            }));
        }
        return isZeroValues;
    }
}
if (false) {
    /** @type {?} */
    PieChart.prototype.title;
    /** @type {?} */
    PieChart.prototype.titleKey;
    /** @type {?} */
    PieChart.prototype.labels;
    /** @type {?} */
    PieChart.prototype.data;
}
