// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/edit-task-filter-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { by, element, protractor } from 'protractor';
import { EditTaskFilterDialogPage } from './dialog/edit-task-filter-dialog.page';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
var EditTaskFilterCloudComponentPage = /** @class */ (function () {
    function EditTaskFilterCloudComponentPage() {
        this.customiseFilter = element(by.id('adf-edit-task-filter-title-id'));
        this.selectedOption = element.all(by.css('mat-option[class*="mat-selected"]')).first();
        this.assignee = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-assignee"]'));
        this.priority = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-priority"]'));
        this.taskName = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-taskName"]'));
        this.id = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-taskId"]'));
        this.processDefinitionId = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-processDefinitionId"]'));
        this.processInstanceId = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-processInstanceId"]'));
        this.lastModifiedFrom = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-lastModifiedFrom"]'));
        this.lastModifiedTo = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-lastModifiedTo"]'));
        this.parentTaskId = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-parentTaskId"]'));
        this.owner = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-owner"]'));
        this.saveButton = element(by.css('[data-automation-id="adf-filter-action-save"]'));
        this.saveAsButton = element(by.css('[data-automation-id="adf-filter-action-saveAs"]'));
        this.deleteButton = element(by.css('[data-automation-id="adf-filter-action-delete"]'));
        this.editTaskFilterDialogPage = new EditTaskFilterDialogPage();
    }
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.editTaskFilterDialog = /**
     * @return {?}
     */
    function () {
        return this.editTaskFilterDialogPage;
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.clickCustomiseFilterHeader = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).customiseFilter);
        (/** @type {?} */ (this)).customiseFilter.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setStatusFilterDropDown = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clickOnDropDownArrow('status');
        /** @type {?} */
        var statusElement = element.all(by.cssContainingText('mat-option span', option)).first();
        BrowserVisibility.waitUntilElementIsVisible(statusElement);
        BrowserVisibility.waitUntilElementIsClickable(statusElement);
        statusElement.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getStatusFilterDropDownValue = /**
     * @return {?}
     */
    function () {
        return element.all(by.css("mat-select[data-automation-id='adf-cloud-edit-task-property-status'] span")).first().getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setSortFilterDropDown = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clickOnDropDownArrow('sort');
        /** @type {?} */
        var sortElement = element.all(by.cssContainingText('mat-option span', option)).first();
        BrowserVisibility.waitUntilElementIsClickable(sortElement);
        BrowserVisibility.waitUntilElementIsVisible(sortElement);
        sortElement.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getSortFilterDropDownValue = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var elementSort = element.all(by.css("mat-select[data-automation-id='adf-cloud-edit-task-property-sort'] span")).first();
        BrowserVisibility.waitUntilElementIsVisible(elementSort);
        return elementSort.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setOrderFilterDropDown = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clickOnDropDownArrow('order');
        /** @type {?} */
        var orderElement = element.all(by.cssContainingText('mat-option span', option)).first();
        BrowserVisibility.waitUntilElementIsClickable(orderElement);
        BrowserVisibility.waitUntilElementIsVisible(orderElement);
        orderElement.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getOrderFilterDropDownValue = /**
     * @return {?}
     */
    function () {
        return element.all(by.css("mat-select[data-automation-id='adf-cloud-edit-task-property-order'] span")).first().getText();
    };
    /**
     * @param {?} option
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.clickOnDropDownArrow = /**
     * @param {?} option
     * @return {?}
     */
    function (option) {
        /** @type {?} */
        var dropDownArrow = element.all(by.css("mat-form-field[data-automation-id='" + option + "'] div[class*='arrow']")).first();
        BrowserVisibility.waitUntilElementIsVisible(dropDownArrow);
        dropDownArrow.click();
        BrowserVisibility.waitUntilElementIsVisible(this.selectedOption);
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setAssignee = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('assignee', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getAssignee = /**
     * @return {?}
     */
    function () {
        return this.assignee.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setPriority = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('priority', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getPriority = /**
     * @return {?}
     */
    function () {
        return this.priority.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setParentTaskId = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('parentTaskId', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getParentTaskId = /**
     * @return {?}
     */
    function () {
        return this.parentTaskId.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setOwner = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('owner', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getOwner = /**
     * @return {?}
     */
    function () {
        return this.owner.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setLastModifiedFrom = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clearField((/** @type {?} */ (this)).lastModifiedFrom);
        return (/** @type {?} */ (this)).setProperty('lastModifiedFrom', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getLastModifiedFrom = /**
     * @return {?}
     */
    function () {
        return this.lastModifiedFrom.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setLastModifiedTo = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clearField((/** @type {?} */ (this)).lastModifiedTo);
        return (/** @type {?} */ (this)).setProperty('lastModifiedTo', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getLastModifiedTo = /**
     * @return {?}
     */
    function () {
        return this.lastModifiedTo.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.checkSaveButtonIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).saveButton);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.checkSaveAsButtonIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).saveAsButton);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.checkDeleteButtonIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).deleteButton);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.checkSaveButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.saveButton);
        return this.saveButton.isEnabled();
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.checkSaveAsButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.saveButton);
        return this.saveAsButton.isEnabled();
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.checkDeleteButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.saveButton);
        return this.deleteButton.isEnabled();
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.clickSaveAsButton = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var disabledButton = element(by.css(("button[data-automation-id='adf-filter-action-saveAs'][disabled]")));
        BrowserVisibility.waitUntilElementIsClickable(this.saveAsButton);
        BrowserVisibility.waitUntilElementIsVisible(this.saveAsButton);
        BrowserVisibility.waitUntilElementIsNotVisible(disabledButton);
        this.saveAsButton.click();
        return this.editTaskFilterDialogPage;
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.clickDeleteButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).deleteButton);
        (/** @type {?} */ (this)).deleteButton.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.clickSaveButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).saveButton);
        (/** @type {?} */ (this)).saveButton.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.clearAssignee = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        (/** @type {?} */ (this)).clearField((/** @type {?} */ (this)).assignee);
        return (/** @type {?} */ (this));
    };
    /**
     * @param {?} locator
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.clearField = /**
     * @param {?} locator
     * @return {?}
     */
    function (locator) {
        BrowserVisibility.waitUntilElementIsVisible(locator);
        locator.getAttribute('value').then((/**
         * @param {?} result
         * @return {?}
         */
        function (result) {
            for (var i = result.length; i >= 0; i--) {
                locator.sendKeys(protractor.Key.BACK_SPACE);
            }
        }));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setAppNameDropDown = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        (/** @type {?} */ (this)).clickOnDropDownArrow('appName');
        /** @type {?} */
        var appNameElement = element.all(by.cssContainingText('mat-option span', option)).first();
        BrowserVisibility.waitUntilElementIsClickable(appNameElement);
        BrowserVisibility.waitUntilElementIsVisible(appNameElement);
        appNameElement.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getAppNameDropDownValue = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var locator = element.all(by.css("mat-select[data-automation-id='adf-cloud-edit-task-property-appName'] span")).first();
        BrowserVisibility.waitUntilElementIsVisible(locator);
        return locator.getText();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setId = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('taskId', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getId = /**
     * @return {?}
     */
    function () {
        return this.id.getAttribute('value');
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setTaskName = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('taskName', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getTaskName = /**
     * @return {?}
     */
    function () {
        return this.taskName.getAttribute('value');
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setProcessDefinitionId = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('processDefinitionId', option);
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getProcessDefinitionId = /**
     * @return {?}
     */
    function () {
        return this.processDefinitionId.getAttribute('value');
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setProcessInstanceId = /**
     * @template THIS
     * @this {THIS}
     * @param {?} option
     * @return {THIS}
     */
    function (option) {
        return (/** @type {?} */ (this)).setProperty('processInstanceId', option);
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} property
     * @param {?} option
     * @return {THIS}
     */
    EditTaskFilterCloudComponentPage.prototype.setProperty = /**
     * @template THIS
     * @this {THIS}
     * @param {?} property
     * @param {?} option
     * @return {THIS}
     */
    function (property, option) {
        /** @type {?} */
        var locator = element(by.css('input[data-automation-id="adf-cloud-edit-task-property-' + property + '"]'));
        BrowserVisibility.waitUntilElementIsVisible(locator);
        locator.clear();
        locator.sendKeys(option);
        locator.sendKeys(protractor.Key.ENTER);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditTaskFilterCloudComponentPage.prototype.getProcessInstanceId = /**
     * @return {?}
     */
    function () {
        return this.processInstanceId.getAttribute('value');
    };
    return EditTaskFilterCloudComponentPage;
}());
export { EditTaskFilterCloudComponentPage };
if (false) {
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.customiseFilter;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.selectedOption;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.assignee;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.priority;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.taskName;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.id;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.processDefinitionId;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.processInstanceId;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.lastModifiedFrom;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.lastModifiedTo;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.parentTaskId;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.owner;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.saveButton;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.saveAsButton;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.deleteButton;
    /** @type {?} */
    EditTaskFilterCloudComponentPage.prototype.editTaskFilterDialogPage;
}
