// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/process-list-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { BrowserVisibility } from '../../core/utils/browser-visibility';
import { DataTableComponentPage } from '../../core/pages/data-table-component.page';
import { element, by } from 'protractor';
var ProcessListCloudComponentPage = /** @class */ (function () {
    function ProcessListCloudComponentPage() {
        this.columns = {
            id: 'Id',
            name: 'Name'
        };
        this.processList = element(by.css('adf-cloud-process-list'));
        this.noProcessFound = element.all(by.css("div[class='adf-empty-content__title']")).first();
        this.dataTable = new DataTableComponentPage(this.processList);
    }
    /**
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.getDataTable = /**
     * @return {?}
     */
    function () {
        return this.dataTable;
    };
    /**
     * @param {?} processName
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.selectRow = /**
     * @param {?} processName
     * @return {?}
     */
    function (processName) {
        return this.dataTable.selectRow(this.columns.name, processName);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.selectRowById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.selectRow(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkRowIsSelectedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkRowIsSelected(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkRowIsNotSelectedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkRowIsNotSelected(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkRowIsCheckedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkRowIsChecked(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkRowIsNotCheckedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkRowIsNotChecked(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkCheckboxById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.clickCheckbox(this.columns.id, processId);
    };
    /**
     * @param {?} processName
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkContentIsDisplayedByName = /**
     * @param {?} processName
     * @return {?}
     */
    function (processName) {
        return this.dataTable.checkContentIsDisplayed(this.columns.name, processName);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkContentIsDisplayedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkContentIsDisplayed(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.checkContentIsNotDisplayedById = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.checkContentIsNotDisplayed(this.columns.id, processId);
    };
    /**
     * @param {?} processId
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.selectRowWithKeyboard = /**
     * @param {?} processId
     * @return {?}
     */
    function (processId) {
        return this.dataTable.selectRowWithKeyboard(this.columns.id, processId);
    };
    /**
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.getAllRowsNameColumn = /**
     * @return {?}
     */
    function () {
        return this.dataTable.getAllRowsColumnValues(this.columns.name);
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    ProcessListCloudComponentPage.prototype.checkProcessListIsLoaded = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).processList);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.getNoProcessFoundMessage = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.noProcessFound);
        return this.noProcessFound.getText();
    };
    /**
     * @param {?} column
     * @return {?}
     */
    ProcessListCloudComponentPage.prototype.getAllRowsByColumn = /**
     * @param {?} column
     * @return {?}
     */
    function (column) {
        return this.dataTable.getAllRowsColumnValues(column);
    };
    return ProcessListCloudComponentPage;
}());
export { ProcessListCloudComponentPage };
if (false) {
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.columns;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.processList;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.noProcessFound;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.dataTable;
}
