// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/start-tasks-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by, Key, protractor } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
var StartTasksCloudPage = /** @class */ (function () {
    function StartTasksCloudPage() {
        this.name = element(by.css('input[id="name_id"]'));
        this.dueDate = element(by.css('input[id="date_id"]'));
        this.description = element(by.css('textarea[id="description_id"]'));
        this.priority = element(by.css('input[formcontrolname="priority"]'));
        this.startButton = element(by.css('button[id="button-start"]'));
        this.startButtonEnabled = element(by.css('button[id="button-start"]:not(disabled)'));
        this.cancelButton = element(by.css('button[id="button-cancel"]'));
        this.form = element.all(by.css('adf-cloud-start-task form')).first();
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.checkFormIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).form);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} userName
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.addName = /**
     * @template THIS
     * @this {THIS}
     * @param {?} userName
     * @return {THIS}
     */
    function (userName) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).name);
        (/** @type {?} */ (this)).name.clear();
        (/** @type {?} */ (this)).name.sendKeys(userName);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} userDescription
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.addDescription = /**
     * @template THIS
     * @this {THIS}
     * @param {?} userDescription
     * @return {THIS}
     */
    function (userDescription) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).description);
        (/** @type {?} */ (this)).description.sendKeys(userDescription);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} userPriority
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.addPriority = /**
     * @template THIS
     * @this {THIS}
     * @param {?} userPriority
     * @return {THIS}
     */
    function (userPriority) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).priority);
        (/** @type {?} */ (this)).priority.sendKeys(userPriority);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} date
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.addDueDate = /**
     * @template THIS
     * @this {THIS}
     * @param {?} date
     * @return {THIS}
     */
    function (date) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).dueDate);
        (/** @type {?} */ (this)).clearField((/** @type {?} */ (this)).dueDate);
        (/** @type {?} */ (this)).dueDate.sendKeys(date);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    StartTasksCloudPage.prototype.clickStartButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.startButton);
        BrowserVisibility.waitUntilElementIsClickable(this.startButton);
        return this.startButton.click();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.checkStartButtonIsEnabled = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).startButtonEnabled);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.checkStartButtonIsDisabled = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).startButton.getAttribute('disabled'));
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    StartTasksCloudPage.prototype.clickCancelButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.cancelButton);
        BrowserVisibility.waitUntilElementIsClickable(this.cancelButton);
        return this.cancelButton.click();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} locator
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.blur = /**
     * @template THIS
     * @this {THIS}
     * @param {?} locator
     * @return {THIS}
     */
    function (locator) {
        locator.click();
        locator.sendKeys(Key.TAB);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @param {?=} elementRef
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.checkValidationErrorIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @param {?=} elementRef
     * @return {THIS}
     */
    function (error, elementRef) {
        if (elementRef === void 0) { elementRef = 'mat-error'; }
        /** @type {?} */
        var errorElement = element(by.cssContainingText(elementRef, error));
        BrowserVisibility.waitUntilElementIsVisible(errorElement);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.validateAssignee = /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @return {THIS}
     */
    function (error) {
        (/** @type {?} */ (this)).checkValidationErrorIsDisplayed(error, '.adf-start-task-cloud-error');
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @return {THIS}
     */
    StartTasksCloudPage.prototype.validateDate = /**
     * @template THIS
     * @this {THIS}
     * @param {?} error
     * @return {THIS}
     */
    function (error) {
        (/** @type {?} */ (this)).checkValidationErrorIsDisplayed(error, '.adf-error-text');
        return (/** @type {?} */ (this));
    };
    /**
     * @param {?} locator
     * @return {?}
     */
    StartTasksCloudPage.prototype.clearField = /**
     * @param {?} locator
     * @return {?}
     */
    function (locator) {
        BrowserVisibility.waitUntilElementIsVisible(locator);
        BrowserVisibility.waitUntilElementIsClickable(locator);
        locator.getAttribute('value').then((/**
         * @param {?} result
         * @return {?}
         */
        function (result) {
            for (var i = result.length; i >= 0; i--) {
                locator.sendKeys(protractor.Key.BACK_SPACE);
            }
        }));
    };
    return StartTasksCloudPage;
}());
export { StartTasksCloudPage };
if (false) {
    /** @type {?} */
    StartTasksCloudPage.prototype.name;
    /** @type {?} */
    StartTasksCloudPage.prototype.dueDate;
    /** @type {?} */
    StartTasksCloudPage.prototype.description;
    /** @type {?} */
    StartTasksCloudPage.prototype.priority;
    /** @type {?} */
    StartTasksCloudPage.prototype.startButton;
    /** @type {?} */
    StartTasksCloudPage.prototype.startButtonEnabled;
    /** @type {?} */
    StartTasksCloudPage.prototype.cancelButton;
    /** @type {?} */
    StartTasksCloudPage.prototype.form;
}
