// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/task-filters-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { by } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
var TaskFiltersCloudComponentPage = /** @class */ (function () {
    function TaskFiltersCloudComponentPage(filter) {
        this.taskIcon = by.xpath("ancestor::div[@class='mat-list-item-content']/mat-icon");
        this.filter = filter;
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    TaskFiltersCloudComponentPage.prototype.checkTaskFilterIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).filter);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    TaskFiltersCloudComponentPage.prototype.getTaskFilterIcon = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.filter);
        /** @type {?} */
        var icon = this.filter.element(this.taskIcon);
        BrowserVisibility.waitUntilElementIsVisible(icon);
        return icon.getText();
    };
    /**
     * @return {?}
     */
    TaskFiltersCloudComponentPage.prototype.checkTaskFilterHasNoIcon = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.filter);
        BrowserVisibility.waitUntilElementIsNotOnPage(this.filter.element(this.taskIcon));
    };
    /**
     * @return {?}
     */
    TaskFiltersCloudComponentPage.prototype.clickTaskFilter = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.filter);
        return this.filter.click();
    };
    /**
     * @return {?}
     */
    TaskFiltersCloudComponentPage.prototype.checkTaskFilterNotDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.filter);
        return this.filter;
    };
    return TaskFiltersCloudComponentPage;
}());
export { TaskFiltersCloudComponentPage };
if (false) {
    /** @type {?} */
    TaskFiltersCloudComponentPage.prototype.filter;
    /** @type {?} */
    TaskFiltersCloudComponentPage.prototype.taskIcon;
}
