// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/task-form-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
var TaskFormCloudComponent = /** @class */ (function () {
    function TaskFormCloudComponent() {
        this.cancelButton = element(by.css("button[id='adf-cloud-cancel-task']"));
        this.completeButton = element(by.css('button[adf-cloud-complete-task]'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    TaskFormCloudComponent.prototype.checkCompleteButtonIsDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).completeButton);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    TaskFormCloudComponent.prototype.checkCompleteButtonIsNotDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible((/** @type {?} */ (this)).completeButton);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    TaskFormCloudComponent.prototype.clickCompleteButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).completeButton);
        (/** @type {?} */ (this)).completeButton.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    TaskFormCloudComponent.prototype.clickCancelButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).cancelButton);
        (/** @type {?} */ (this)).cancelButton.click();
        return (/** @type {?} */ (this));
    };
    return TaskFormCloudComponent;
}());
export { TaskFormCloudComponent };
if (false) {
    /** @type {?} */
    TaskFormCloudComponent.prototype.cancelButton;
    /** @type {?} */
    TaskFormCloudComponent.prototype.completeButton;
}
