// @alfresco/adf-core/esm5/clipboard/clipboard.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, Input, HostListener, Component, ViewContainerRef, ComponentFactoryResolver, ViewEncapsulation } from '@angular/core';
import { ClipboardService } from './clipboard.service';
var ClipboardDirective = /** @class */ (function () {
    function ClipboardDirective(clipboardService, viewContainerRef, resolver) {
        this.clipboardService = clipboardService;
        this.viewContainerRef = viewContainerRef;
        this.resolver = resolver;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    ClipboardDirective.prototype.handleClickEvent = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        event.stopPropagation();
        this.copyToClipboard();
    };
    /**
     * @return {?}
     */
    ClipboardDirective.prototype.showTooltip = /**
     * @return {?}
     */
    function () {
        if (this.placeholder) {
            /** @type {?} */
            var componentFactory = this.resolver.resolveComponentFactory(ClipboardComponent);
            /** @type {?} */
            var componentRef = this.viewContainerRef.createComponent(componentFactory).instance;
            componentRef.placeholder = this.placeholder;
        }
    };
    /**
     * @return {?}
     */
    ClipboardDirective.prototype.closeTooltip = /**
     * @return {?}
     */
    function () {
        this.viewContainerRef.remove();
    };
    /**
     * @private
     * @return {?}
     */
    ClipboardDirective.prototype.copyToClipboard = /**
     * @private
     * @return {?}
     */
    function () {
        /** @type {?} */
        var isValidTarget = this.clipboardService.isTargetValid(this.target);
        if (isValidTarget) {
            this.clipboardService.copyToClipboard(this.target, this.message);
        }
        else {
            this.copyContentToClipboard(this.viewContainerRef.element.nativeElement.innerHTML);
        }
    };
    /**
     * @private
     * @param {?} content
     * @return {?}
     */
    ClipboardDirective.prototype.copyContentToClipboard = /**
     * @private
     * @param {?} content
     * @return {?}
     */
    function (content) {
        this.clipboardService.copyContentToClipboard(content, this.message);
    };
    ClipboardDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-clipboard]',
                    exportAs: 'adfClipboard'
                },] }
    ];
    /** @nocollapse */
    ClipboardDirective.ctorParameters = function () { return [
        { type: ClipboardService },
        { type: ViewContainerRef },
        { type: ComponentFactoryResolver }
    ]; };
    ClipboardDirective.propDecorators = {
        placeholder: [{ type: Input, args: ['adf-clipboard',] }],
        target: [{ type: Input }],
        message: [{ type: Input, args: ['clipboard-notification',] }],
        handleClickEvent: [{ type: HostListener, args: ['click', ['$event'],] }],
        showTooltip: [{ type: HostListener, args: ['mouseenter',] }],
        closeTooltip: [{ type: HostListener, args: ['mouseleave',] }]
    };
    return ClipboardDirective;
}());
export { ClipboardDirective };
if (false) {
    /**
     * Translation key or message for the tooltip.
     * @type {?}
     */
    ClipboardDirective.prototype.placeholder;
    /**
     * Reference to the HTML element containing the text to copy.
     * @type {?}
     */
    ClipboardDirective.prototype.target;
    /**
     * Translation key or message for snackbar notification.
     * @type {?}
     */
    ClipboardDirective.prototype.message;
    /**
     * @type {?}
     * @private
     */
    ClipboardDirective.prototype.clipboardService;
    /** @type {?} */
    ClipboardDirective.prototype.viewContainerRef;
    /**
     * @type {?}
     * @private
     */
    ClipboardDirective.prototype.resolver;
}
var ClipboardComponent = /** @class */ (function () {
    function ClipboardComponent() {
    }
    /**
     * @return {?}
     */
    ClipboardComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.placeholder = this.placeholder || 'CLIPBOARD.CLICK_TO_COPY';
    };
    ClipboardComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-copy-content-tooltip',
                    template: "\n        <span class='adf-copy-tooltip'>{{ placeholder | translate }} </span>\n        ",
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    return ClipboardComponent;
}());
export { ClipboardComponent };
if (false) {
    /** @type {?} */
    ClipboardComponent.prototype.placeholder;
}
