// @alfresco/adf-core/esm5/clipboard/clipboard.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/platform-browser';
import { LogService } from '../services/log.service';
import { NotificationService } from '../services/notification.service';
var ClipboardService = /** @class */ (function () {
    function ClipboardService(document, logService, notificationService) {
        this.document = document;
        this.logService = logService;
        this.notificationService = notificationService;
    }
    /**
     * Checks if the target element can have its text copied.
     * @param target Target HTML element
     * @returns True if the text can be copied, false otherwise
     */
    /**
     * Checks if the target element can have its text copied.
     * @param {?} target Target HTML element
     * @return {?} True if the text can be copied, false otherwise
     */
    ClipboardService.prototype.isTargetValid = /**
     * Checks if the target element can have its text copied.
     * @param {?} target Target HTML element
     * @return {?} True if the text can be copied, false otherwise
     */
    function (target) {
        if (target instanceof HTMLInputElement || target instanceof HTMLTextAreaElement) {
            return !target.hasAttribute('disabled');
        }
        return false;
    };
    /**
     * Copies text from an HTML element to the clipboard.
     * @param target HTML element to be copied
     * @param message Snackbar message to alert when copying happens
     */
    /**
     * Copies text from an HTML element to the clipboard.
     * @param {?} target HTML element to be copied
     * @param {?=} message Snackbar message to alert when copying happens
     * @return {?}
     */
    ClipboardService.prototype.copyToClipboard = /**
     * Copies text from an HTML element to the clipboard.
     * @param {?} target HTML element to be copied
     * @param {?=} message Snackbar message to alert when copying happens
     * @return {?}
     */
    function (target, message) {
        if (this.isTargetValid(target)) {
            try {
                target.select();
                target.setSelectionRange(0, target.value.length);
                this.document.execCommand('copy');
                this.notify(message);
            }
            catch (error) {
                this.logService.error(error);
            }
        }
    };
    /**
     * Copies a text string to the clipboard.
     * @param content Text to copy
     * @param message Snackbar message to alert when copying happens
     */
    /**
     * Copies a text string to the clipboard.
     * @param {?} content Text to copy
     * @param {?} message Snackbar message to alert when copying happens
     * @return {?}
     */
    ClipboardService.prototype.copyContentToClipboard = /**
     * Copies a text string to the clipboard.
     * @param {?} content Text to copy
     * @param {?} message Snackbar message to alert when copying happens
     * @return {?}
     */
    function (content, message) {
        try {
            document.addEventListener('copy', (/**
             * @param {?} e
             * @return {?}
             */
            function (e) {
                e.clipboardData.setData('text/plain', (content));
                e.preventDefault();
                document.removeEventListener('copy', null);
            }));
            document.execCommand('copy');
            this.notify(message);
        }
        catch (error) {
            this.logService.error(error);
        }
    };
    /**
     * @private
     * @param {?} message
     * @return {?}
     */
    ClipboardService.prototype.notify = /**
     * @private
     * @param {?} message
     * @return {?}
     */
    function (message) {
        if (message) {
            this.notificationService.openSnackMessage(message);
        }
    };
    ClipboardService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    ClipboardService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] },
        { type: LogService },
        { type: NotificationService }
    ]; };
    return ClipboardService;
}());
export { ClipboardService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ClipboardService.prototype.document;
    /**
     * @type {?}
     * @private
     */
    ClipboardService.prototype.logService;
    /**
     * @type {?}
     * @private
     */
    ClipboardService.prototype.notificationService;
}
