// @alfresco/adf-extensions/esm2015/lib/components/dynamic-component/dynamic.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ComponentFactoryResolver, ViewChild, ViewContainerRef } from '@angular/core';
import { ExtensionService } from '../../services/extension.service';
// cSpell:words lifecycle
export class DynamicExtensionComponent {
    /**
     * @param {?} extensions
     * @param {?} componentFactoryResolver
     */
    constructor(extensions, componentFactoryResolver) {
        this.extensions = extensions;
        this.componentFactoryResolver = componentFactoryResolver;
        this.loaded = false;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        if (!this.loaded) {
            this.loadComponent();
            this.loaded = true;
        }
        if (changes.data) {
            this.data = changes.data.currentValue;
        }
        this.updateInstance();
        this.proxy('ngOnChanges', changes);
    }
    /**
     * @private
     * @return {?}
     */
    loadComponent() {
        /** @type {?} */
        const componentType = this.extensions.getComponentById(this.id);
        if (componentType) {
            /** @type {?} */
            const factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
            }
        }
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.componentCreated()) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    }
    /**
     * @private
     * @return {?}
     */
    updateInstance() {
        if (this.componentCreated()) {
            this.componentRef.instance.data = this.data;
        }
    }
    /**
     * @private
     * @param {?} lifecycleMethod
     * @param {...?} args
     * @return {?}
     */
    proxy(lifecycleMethod, ...args) {
        if (this.componentCreated() && this.lifecycleHookIsImplemented(lifecycleMethod)) {
            this.componentRef.instance[lifecycleMethod].apply(this.componentRef.instance, args);
        }
    }
    /**
     * @private
     * @return {?}
     */
    componentCreated() {
        return !!this.componentRef && !!this.componentRef.instance;
    }
    /**
     * @private
     * @param {?} lifecycleMethod
     * @return {?}
     */
    lifecycleHookIsImplemented(lifecycleMethod) {
        return !!this.componentRef.instance[lifecycleMethod];
    }
}
DynamicExtensionComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-dynamic-component',
                template: `<div #content></div>`
            }] }
];
/** @nocollapse */
DynamicExtensionComponent.ctorParameters = () => [
    { type: ExtensionService },
    { type: ComponentFactoryResolver }
];
DynamicExtensionComponent.propDecorators = {
    content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
    id: [{ type: Input }],
    data: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DynamicExtensionComponent.prototype.content;
    /**
     * Unique ID string for the component to show.
     * @type {?}
     */
    DynamicExtensionComponent.prototype.id;
    /**
     * Data for the dynamically-loaded component instance.
     * @type {?}
     */
    DynamicExtensionComponent.prototype.data;
    /**
     * @type {?}
     * @private
     */
    DynamicExtensionComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    DynamicExtensionComponent.prototype.loaded;
    /**
     * @type {?}
     * @private
     */
    DynamicExtensionComponent.prototype.extensions;
    /**
     * @type {?}
     * @private
     */
    DynamicExtensionComponent.prototype.componentFactoryResolver;
}
