// @alfresco/adf-core/esm5/models/comment.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var CommentModel = /** @class */ (function () {
    function CommentModel(obj) {
        if (obj) {
            this.id = obj.id;
            this.message = obj.message;
            this.created = obj.created;
            this.createdBy = obj.createdBy;
            this.isSelected = obj.isSelected ? obj.isSelected : false;
        }
    }
    return CommentModel;
}());
export { CommentModel };
if (false) {
    /** @type {?} */
    CommentModel.prototype.id;
    /** @type {?} */
    CommentModel.prototype.message;
    /** @type {?} */
    CommentModel.prototype.created;
    /** @type {?} */
    CommentModel.prototype.createdBy;
    /** @type {?} */
    CommentModel.prototype.isSelected;
}
