// @alfresco/adf-process-services/esm2015/people/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { PeopleComponent } from './components/people/people.component';
export { PeopleListComponent } from './components/people-list/people-list.component';
export { PeopleSearchComponent } from './components/people-search/people-search.component';
export { PeopleSearchFieldComponent } from './components/people-search-field/people-search-field.component';
export { PeopleSelectorComponent } from './components/people-selector/people-selector.component';
export {} from './interfaces/perform-search-callback.interface';
export { PeopleSearchActionLabelDirective } from './directives/people-search-action-label.directive';
export { PeopleSearchTitleDirective } from './directives/people-search-title.directive';
export { PeopleModule } from './people.module';
