// @alfresco/adf-content-services/esm5/version-manager/version-upload.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation, Output, EventEmitter } from '@angular/core';
import { Node } from '@alfresco/js-api';
import { ContentService } from '@alfresco/adf-core';
var VersionUploadComponent = /** @class */ (function () {
    function VersionUploadComponent(contentService) {
        this.contentService = contentService;
        this.semanticVersion = 'minor';
        this.uploadVersion = false;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.cancel = new EventEmitter();
    }
    /**
     * @return {?}
     */
    VersionUploadComponent.prototype.canUpload = /**
     * @return {?}
     */
    function () {
        return this.contentService.hasAllowableOperations(this.node, 'update');
    };
    /**
     * @return {?}
     */
    VersionUploadComponent.prototype.isMajorVersion = /**
     * @return {?}
     */
    function () {
        return this.semanticVersion === 'minor' ? false : true;
    };
    /**
     * @return {?}
     */
    VersionUploadComponent.prototype.cancelUpload = /**
     * @return {?}
     */
    function () {
        this.cancel.emit();
    };
    VersionUploadComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-version-upload',
                    template: "<div class=\"adf-new-version-max-width\">\n    <mat-radio-group class=\"adf-new-version-radio-group\" [(ngModel)]=\"semanticVersion\">\n        <mat-radio-button class=\"adf-new-version-radio-button\" id=\"adf-new-version-minor\"[value]=\"'minor'\">{{\n            'ADF_VERSION_LIST.ACTIONS.UPLOAD.MINOR' |\n            translate }}\n        </mat-radio-button>\n        <mat-radio-button class=\"adf-new-version-radio-button\" id=\"adf-new-version-major\" [value]=\"'major'\">{{\n            'ADF_VERSION_LIST.ACTIONS.UPLOAD.MAJOR' |\n            translate }}\n        </mat-radio-button>\n    </mat-radio-group>\n    <mat-form-field class=\"adf-new-version-max-width\">\n                    <textarea matInput [(ngModel)]=\"comment\" class=\"adf-new-version-text-area\" id=\"adf-new-version-text-area\"\n                              placeholder=\"{{ 'ADF_VERSION_LIST.ACTIONS.UPLOAD.COMMENT' | translate }}\"></textarea>\n    </mat-form-field>\n\n</div>\n<div class=\"adf-version-upload-buttons\">\n    <adf-upload-version-button\n        data-automation-id=\"adf-new-version-file-upload\"\n        staticTitle=\"{{ 'ADF_VERSION_LIST.ACTIONS.UPLOAD.TITLE' | translate }}\"\n        [node]=\"node\"\n        [disabled]=\"!canUpload()\"\n        [rootFolderId]=\"node.parentId\"\n        tooltip=\"{{ 'ADF_VERSION_LIST.ACTIONS.UPLOAD.TOOLTIP' | translate }}\"\n        [comment]=\"comment\"\n        [versioning]=\"true\"\n        [majorVersion]=\"isMajorVersion()\"\n        (success)=\"success.emit($event)\"\n        (error)=\"error.emit($event)\">\n    </adf-upload-version-button>\n    <button mat-raised-button (click)=\"cancelUpload()\" id=\"adf-new-version-cancel\"  >{{\n        'ADF_VERSION_LIST.ACTIONS.UPLOAD.CANCEL'| translate }}\n    </button>\n</div>\n\n",
                    encapsulation: ViewEncapsulation.None,
                    host: { 'class': 'adf-version-upload' },
                    styles: [".adf-new-version-radio-group{display:inline-flex;flex-direction:column}.adf-new-version-radio-button{margin:5px}.adf-version-upload-buttons{display:flex;float:right;justify-content:space-between;width:100%}.adf-new-version-max-width{width:100%;float:right}.adf-new-version-text-area{resize:none!important}"]
                }] }
    ];
    /** @nocollapse */
    VersionUploadComponent.ctorParameters = function () { return [
        { type: ContentService }
    ]; };
    VersionUploadComponent.propDecorators = {
        node: [{ type: Input }],
        success: [{ type: Output }],
        error: [{ type: Output }],
        cancel: [{ type: Output }]
    };
    return VersionUploadComponent;
}());
export { VersionUploadComponent };
if (false) {
    /** @type {?} */
    VersionUploadComponent.prototype.semanticVersion;
    /** @type {?} */
    VersionUploadComponent.prototype.comment;
    /** @type {?} */
    VersionUploadComponent.prototype.uploadVersion;
    /** @type {?} */
    VersionUploadComponent.prototype.node;
    /** @type {?} */
    VersionUploadComponent.prototype.success;
    /** @type {?} */
    VersionUploadComponent.prototype.error;
    /** @type {?} */
    VersionUploadComponent.prototype.cancel;
    /**
     * @type {?}
     * @private
     */
    VersionUploadComponent.prototype.contentService;
}
