// @alfresco/adf-testing/esm5/lib/process-services-cloud/pages/dialog/edit-process-filter-dialog.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { by, element, protractor } from 'protractor';
import { BrowserVisibility } from '../../../core/utils/browser-visibility';
var EditProcessFilterDialogPage = /** @class */ (function () {
    function EditProcessFilterDialogPage() {
        this.componentElement = element(by.css('adf-cloud-process-filter-dialog-cloud'));
        this.title = element(by.id('adf-process-filter-dialog-title'));
        this.filterNameInput = element(by.id('adf-filter-name-id'));
        this.saveButtonLocator = by.id('adf-save-button-id');
        this.cancelButtonLocator = by.id('adf-cancel-button-id');
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditProcessFilterDialogPage.prototype.clickOnSaveButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        /** @type {?} */
        var saveButton = (/** @type {?} */ (this)).componentElement.element((/** @type {?} */ (this)).saveButtonLocator);
        BrowserVisibility.waitUntilElementIsVisible(saveButton);
        saveButton.click();
        BrowserVisibility.waitUntilElementIsNotVisible((/** @type {?} */ (this)).componentElement);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditProcessFilterDialogPage.prototype.checkSaveButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.componentElement.element(this.saveButtonLocator));
        return this.componentElement.element(this.saveButtonLocator).isEnabled();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditProcessFilterDialogPage.prototype.clickOnCancelButton = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        /** @type {?} */
        var cancelButton = (/** @type {?} */ (this)).componentElement.element((/** @type {?} */ (this)).cancelButtonLocator);
        BrowserVisibility.waitUntilElementIsVisible(cancelButton);
        cancelButton.click();
        BrowserVisibility.waitUntilElementIsNotVisible((/** @type {?} */ (this)).componentElement);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditProcessFilterDialogPage.prototype.checkCancelButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.componentElement.element(this.cancelButtonLocator));
        return this.componentElement.element(this.cancelButtonLocator).isEnabled();
    };
    /**
     * @return {?}
     */
    EditProcessFilterDialogPage.prototype.getFilterName = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.filterNameInput);
        return this.filterNameInput.getAttribute('value');
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} filterName
     * @return {THIS}
     */
    EditProcessFilterDialogPage.prototype.setFilterName = /**
     * @template THIS
     * @this {THIS}
     * @param {?} filterName
     * @return {THIS}
     */
    function (filterName) {
        (/** @type {?} */ (this)).clearFilterName();
        (/** @type {?} */ (this)).filterNameInput.sendKeys(filterName);
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    EditProcessFilterDialogPage.prototype.clearFilterName = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        var _this = this;
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).filterNameInput);
        (/** @type {?} */ (this)).filterNameInput.click();
        (/** @type {?} */ (this)).filterNameInput.getAttribute('value').then((/**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            for (var i = value.length; i >= 0; i--) {
                (/** @type {?} */ (_this)).filterNameInput.sendKeys(protractor.Key.BACK_SPACE);
            }
        }));
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    EditProcessFilterDialogPage.prototype.getTitle = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.title);
        return this.title.getText();
    };
    return EditProcessFilterDialogPage;
}());
export { EditProcessFilterDialogPage };
if (false) {
    /** @type {?} */
    EditProcessFilterDialogPage.prototype.componentElement;
    /** @type {?} */
    EditProcessFilterDialogPage.prototype.title;
    /** @type {?} */
    EditProcessFilterDialogPage.prototype.filterNameInput;
    /** @type {?} */
    EditProcessFilterDialogPage.prototype.saveButtonLocator;
    /** @type {?} */
    EditProcessFilterDialogPage.prototype.cancelButtonLocator;
}
