// @alfresco/adf-core/esm5/form/components/widgets/unknown/unknown.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { baseHost, WidgetComponent } from './../widget.component';
var UnknownWidgetComponent = /** @class */ (function (_super) {
    tslib_1.__extends(UnknownWidgetComponent, _super);
    function UnknownWidgetComponent(formService) {
        var _this = _super.call(this, formService) || this;
        _this.formService = formService;
        return _this;
    }
    UnknownWidgetComponent.decorators = [
        { type: Component, args: [{
                    selector: 'unknown-widget',
                    template: "\n            <mat-list class=\"adf-unknown-widget\">\n                <mat-list-item>\n                     <mat-icon class=\"mat-24\">error_outline</mat-icon>\n                     <span class=\"adf-unknown-text\">Unknown type: {{field.type}}</span>\n                </mat-list-item>\n            </mat-list>\n\n    ",
                    host: baseHost,
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-unknown-text{margin-left:10px;color:red}.adf-unknown-widget{margin:42px}"]
                }] }
    ];
    /** @nocollapse */
    UnknownWidgetComponent.ctorParameters = function () { return [
        { type: FormService }
    ]; };
    return UnknownWidgetComponent;
}(WidgetComponent));
export { UnknownWidgetComponent };
if (false) {
    /** @type {?} */
    UnknownWidgetComponent.prototype.formService;
}
