// @alfresco/adf-content-services/esm5/document-list/components/trashcan-name-column/trashcan-name-column.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ChangeDetectionStrategy, ViewEncapsulation, Input } from '@angular/core';
var TrashcanNameColumnComponent = /** @class */ (function () {
    function TrashcanNameColumnComponent() {
        this.isLibrary = false;
    }
    /**
     * @return {?}
     */
    TrashcanNameColumnComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.node = this.context.row.node;
        /** @type {?} */
        var rows = this.context.data.rows || [];
        if (this.node && this.node.entry) {
            this.isLibrary = this.node.entry.nodeType === 'st:site';
            if (this.isLibrary) {
                var properties = this.node.entry.properties;
                this.displayText = this.makeLibraryTitle(this.node.entry, rows);
                this.displayTooltip =
                    properties['cm:description'] || properties['cm:title'];
            }
            else {
                this.displayText = this.node.entry.name || this.node.entry.id;
            }
        }
    };
    /**
     * @param {?} library
     * @param {?} rows
     * @return {?}
     */
    TrashcanNameColumnComponent.prototype.makeLibraryTitle = /**
     * @param {?} library
     * @param {?} rows
     * @return {?}
     */
    function (library, rows) {
        /** @type {?} */
        var entries = rows.map((/**
         * @param {?} r
         * @return {?}
         */
        function (r) { return r.node.entry; }));
        var id = library.id;
        /** @type {?} */
        var title = library.properties['cm:title'];
        /** @type {?} */
        var isDuplicate = false;
        if (entries) {
            isDuplicate = entries.some((/**
             * @param {?} entry
             * @return {?}
             */
            function (entry) {
                return entry.id !== id && entry.properties['cm:title'] === title;
            }));
        }
        return isDuplicate
            ? library.properties['cm:title'] + " (" + library.name + ")"
            : "" + library.properties['cm:title'];
    };
    TrashcanNameColumnComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-trashcan-name-column',
                    template: "\n    <ng-container *ngIf=\"!isLibrary\">\n      <span class=\"adf-datatable-cell-value\" title=\"{{ node | adfNodeNameTooltip }}\">{{ displayText }}</span>\n    </ng-container>\n    <ng-container *ngIf=\"isLibrary\">\n      <span class=\"adf-datatable-cell-value\" title=\"{{ displayTooltip }}\">{{ displayText }}</span>\n    </ng-container>\n  ",
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-datatable-content-cell adf-trashcan-name-column' }
                }] }
    ];
    TrashcanNameColumnComponent.propDecorators = {
        context: [{ type: Input }]
    };
    return TrashcanNameColumnComponent;
}());
export { TrashcanNameColumnComponent };
if (false) {
    /** @type {?} */
    TrashcanNameColumnComponent.prototype.context;
    /** @type {?} */
    TrashcanNameColumnComponent.prototype.isLibrary;
    /** @type {?} */
    TrashcanNameColumnComponent.prototype.displayText;
    /** @type {?} */
    TrashcanNameColumnComponent.prototype.displayTooltip;
    /** @type {?} */
    TrashcanNameColumnComponent.prototype.node;
}
