// @alfresco/adf-content-services/esm5/search/components/search-slider/search-slider.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input } from '@angular/core';
var SearchSliderComponent = /** @class */ (function () {
    function SearchSliderComponent() {
        this.thumbLabel = false;
    }
    /**
     * @return {?}
     */
    SearchSliderComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.settings) {
            if (this.settings.hasOwnProperty('min')) {
                this.min = this.settings['min'];
            }
            if (this.settings.hasOwnProperty('max')) {
                this.max = this.settings['max'];
            }
            if (this.settings.hasOwnProperty('step')) {
                this.step = this.settings['step'];
            }
            this.thumbLabel = this.settings['thumbLabel'] ? true : false;
        }
    };
    /**
     * @return {?}
     */
    SearchSliderComponent.prototype.reset = /**
     * @return {?}
     */
    function () {
        this.value = this.min || 0;
        this.updateQuery(null);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    SearchSliderComponent.prototype.onChangedHandler = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.value = event.value;
        this.updateQuery(this.value);
    };
    /**
     * @private
     * @param {?} value
     * @return {?}
     */
    SearchSliderComponent.prototype.updateQuery = /**
     * @private
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (this.id && this.context && this.settings && this.settings.field) {
            if (value === null) {
                this.context.queryFragments[this.id] = '';
            }
            else {
                this.context.queryFragments[this.id] = this.settings.field + ":[0 TO " + value + "]";
            }
            this.context.update();
        }
    };
    SearchSliderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-search-slider',
                    template: "<mat-slider\n    [(value)]=\"value\"\n    [min]=\"min\"\n    [max]=\"max\"\n    [step]=\"step\"\n    [thumbLabel]=\"thumbLabel\"\n    (change)=\"onChangedHandler($event)\"\n    data-automation-id=\"slider-range\">\n</mat-slider>\n\n<div class=\"facet-buttons\">\n    <button mat-button color=\"primary\" (click)=\"reset()\" data-automation-id=\"slider-btn-clear\">\n        {{ 'SEARCH.FILTER.ACTIONS.CLEAR' | translate }}\n    </button>\n</div>\n",
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-search-slider' },
                    styles: [".adf-search-slider .mat-slider{width:100%}"]
                }] }
    ];
    SearchSliderComponent.propDecorators = {
        value: [{ type: Input }]
    };
    return SearchSliderComponent;
}());
export { SearchSliderComponent };
if (false) {
    /** @type {?} */
    SearchSliderComponent.prototype.id;
    /** @type {?} */
    SearchSliderComponent.prototype.settings;
    /** @type {?} */
    SearchSliderComponent.prototype.context;
    /** @type {?} */
    SearchSliderComponent.prototype.step;
    /** @type {?} */
    SearchSliderComponent.prototype.min;
    /** @type {?} */
    SearchSliderComponent.prototype.max;
    /** @type {?} */
    SearchSliderComponent.prototype.thumbLabel;
    /**
     * The numeric value represented by the slider.
     * @type {?}
     */
    SearchSliderComponent.prototype.value;
}
