// @alfresco/adf-core/esm5/form/components/widgets/core/content-link.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
var ContentLinkModel = /** @class */ (function () {
    function ContentLinkModel(obj) {
        this.contentAvailable = obj && obj.contentAvailable;
        this.created = obj && obj.created;
        this.createdBy = obj && obj.createdBy || {};
        this.id = obj && obj.id;
        this.link = obj && obj.link;
        this.mimeType = obj && obj.mimeType;
        this.name = obj && obj.name;
        this.previewStatus = obj && obj.previewStatus;
        this.relatedContent = obj && obj.relatedContent;
        this.simpleType = obj && obj.simpleType;
        this.thumbnailStatus = obj && obj.thumbnailStatus;
    }
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.hasPreviewStatus = /**
     * @return {?}
     */
    function () {
        return this.previewStatus === 'supported' ? true : false;
    };
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.isTypeImage = /**
     * @return {?}
     */
    function () {
        return this.simpleType === 'image' ? true : false;
    };
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.isTypePdf = /**
     * @return {?}
     */
    function () {
        return this.simpleType === 'pdf' ? true : false;
    };
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.isTypeDoc = /**
     * @return {?}
     */
    function () {
        return this.simpleType === 'word' || this.simpleType === 'content' ? true : false;
    };
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.isThumbnailReady = /**
     * @return {?}
     */
    function () {
        return this.thumbnailStatus === 'created';
    };
    /**
     * @return {?}
     */
    ContentLinkModel.prototype.isThumbnailSupported = /**
     * @return {?}
     */
    function () {
        return this.isTypeImage() || ((this.isTypePdf() || this.isTypeDoc()) && this.isThumbnailReady());
    };
    return ContentLinkModel;
}());
export { ContentLinkModel };
if (false) {
    /** @type {?} */
    ContentLinkModel.prototype.contentAvailable;
    /** @type {?} */
    ContentLinkModel.prototype.created;
    /** @type {?} */
    ContentLinkModel.prototype.createdBy;
    /** @type {?} */
    ContentLinkModel.prototype.id;
    /** @type {?} */
    ContentLinkModel.prototype.link;
    /** @type {?} */
    ContentLinkModel.prototype.mimeType;
    /** @type {?} */
    ContentLinkModel.prototype.name;
    /** @type {?} */
    ContentLinkModel.prototype.previewStatus;
    /** @type {?} */
    ContentLinkModel.prototype.relatedContent;
    /** @type {?} */
    ContentLinkModel.prototype.simpleType;
    /** @type {?} */
    ContentLinkModel.prototype.thumbnailUrl;
    /** @type {?} */
    ContentLinkModel.prototype.contentRawUrl;
    /** @type {?} */
    ContentLinkModel.prototype.contentBlob;
    /** @type {?} */
    ContentLinkModel.prototype.thumbnailStatus;
}
