// @alfresco/adf-core/esm5/form/components/widgets/core/container-column.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
var ContainerColumnModel = /** @class */ (function () {
    function ContainerColumnModel() {
        this.size = 12;
        this.fields = [];
        this.colspan = 1;
        this.rowspan = 1;
    }
    /**
     * @return {?}
     */
    ContainerColumnModel.prototype.hasFields = /**
     * @return {?}
     */
    function () {
        return this.fields && this.fields.length > 0;
    };
    return ContainerColumnModel;
}());
export { ContainerColumnModel };
if (false) {
    /** @type {?} */
    ContainerColumnModel.prototype.size;
    /** @type {?} */
    ContainerColumnModel.prototype.fields;
    /** @type {?} */
    ContainerColumnModel.prototype.colspan;
    /** @type {?} */
    ContainerColumnModel.prototype.rowspan;
}
