// @alfresco/adf-content-services/esm5/document-list/components/content-column/content-column.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-input-rename  */
import { LogService } from '@alfresco/adf-core';
import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
import { ContentColumnListComponent } from './content-column-list.component';
var ContentColumnComponent = /** @class */ (function () {
    function ContentColumnComponent(list, logService) {
        this.list = list;
        this.logService = logService;
        this.type = 'text';
        this.sortable = false;
        this.title = '';
        this.logService.log('ContentColumnComponent is deprecated starting with 1.7.0 and may be removed in future versions. Use DataColumnComponent instead.');
    }
    /**
     * @return {?}
     */
    ContentColumnComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.srTitle && this.key === '$thumbnail') {
            this.srTitle = 'Thumbnail';
        }
    };
    /**
     * @return {?}
     */
    ContentColumnComponent.prototype.ngAfterContentInit = /**
     * @return {?}
     */
    function () {
        this.register();
    };
    /**
     * @return {?}
     */
    ContentColumnComponent.prototype.register = /**
     * @return {?}
     */
    function () {
        if (this.list) {
            return this.list.registerColumn(this);
        }
        return false;
    };
    ContentColumnComponent.decorators = [
        { type: Component, args: [{
                    selector: 'content-column',
                    template: ''
                }] }
    ];
    /** @nocollapse */
    ContentColumnComponent.ctorParameters = function () { return [
        { type: ContentColumnListComponent },
        { type: LogService }
    ]; };
    ContentColumnComponent.propDecorators = {
        key: [{ type: Input }],
        type: [{ type: Input }],
        format: [{ type: Input }],
        sortable: [{ type: Input }],
        title: [{ type: Input }],
        template: [{ type: ContentChild, args: [TemplateRef,] }],
        srTitle: [{ type: Input, args: ['sr-title',] }],
        cssClass: [{ type: Input, args: ['class',] }]
    };
    return ContentColumnComponent;
}());
export { ContentColumnComponent };
if (false) {
    /** @type {?} */
    ContentColumnComponent.prototype.key;
    /** @type {?} */
    ContentColumnComponent.prototype.type;
    /** @type {?} */
    ContentColumnComponent.prototype.format;
    /** @type {?} */
    ContentColumnComponent.prototype.sortable;
    /** @type {?} */
    ContentColumnComponent.prototype.title;
    /** @type {?} */
    ContentColumnComponent.prototype.template;
    /**
     * Title to be used for screen readers.
     * @type {?}
     */
    ContentColumnComponent.prototype.srTitle;
    /** @type {?} */
    ContentColumnComponent.prototype.cssClass;
    /**
     * @type {?}
     * @private
     */
    ContentColumnComponent.prototype.list;
    /**
     * @type {?}
     * @private
     */
    ContentColumnComponent.prototype.logService;
}
