// @alfresco/adf-testing/esm5/lib/process-services-cloud/actions/process-instances.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var ProcessInstancesService = /** @class */ (function () {
    function ProcessInstancesService(api) {
        this.api = api;
    }
    /**
     * @param {?} processDefKey
     * @param {?} appName
     * @param {?=} options
     * @return {?}
     */
    ProcessInstancesService.prototype.createProcessInstance = /**
     * @param {?} processDefKey
     * @param {?} appName
     * @param {?=} options
     * @return {?}
     */
    function (processDefKey, appName, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = '/' + appName + '/rb/v1/process-instances';
                        method = 'POST';
                        queryParams = {};
                        postBody = tslib_1.__assign({ 'processDefinitionKey': processDefKey, 'payloadType': 'StartProcessPayload' }, options);
                        return [4 /*yield*/, this.api.performBpmOperation(path, method, queryParams, postBody)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    ProcessInstancesService.prototype.suspendProcessInstance = /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    function (processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId + '/suspend';
                        method = 'POST';
                        queryParams = {};
                        postBody = {};
                        return [4 /*yield*/, this.api.performBpmOperation(path, method, queryParams, postBody)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    ProcessInstancesService.prototype.deleteProcessInstance = /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    function (processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId;
                        method = 'DELETE';
                        queryParams = {};
                        postBody = {};
                        return [4 /*yield*/, this.api.performBpmOperation(path, method, queryParams, postBody)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    ProcessInstancesService.prototype.completeProcessInstance = /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    function (processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId + '/complete';
                        method = 'POST';
                        queryParams = {};
                        postBody = {};
                        return [4 /*yield*/, this.api.performBpmOperation(path, method, queryParams, postBody)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return ProcessInstancesService;
}());
export { ProcessInstancesService };
if (false) {
    /** @type {?} */
    ProcessInstancesService.prototype.api;
}
