// @alfresco/adf-testing/esm5/lib/process-services-cloud/actions/testing-alfresco-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiCompatibility, AlfrescoApiConfig } from '@alfresco/js-api';
import { AlfrescoApiService, AppConfigValues } from '@alfresco/adf-core';
var TestingAlfrescoApiService = /** @class */ (function (_super) {
    tslib_1.__extends(TestingAlfrescoApiService, _super);
    function TestingAlfrescoApiService(appConfig) {
        var _this = _super.call(this, null) || this;
        _this.appConfig = appConfig;
        _this.config = {};
        /** @type {?} */
        var oauth = Object.assign({}, _this.appConfig.get(AppConfigValues.OAUTHCONFIG, null));
        _this.config = new AlfrescoApiConfig({
            provider: _this.appConfig.get(AppConfigValues.PROVIDERS),
            hostEcm: _this.appConfig.get(AppConfigValues.ECMHOST),
            hostBpm: _this.appConfig.get(AppConfigValues.BPMHOST),
            authType: _this.appConfig.get(AppConfigValues.AUTHTYPE, 'BASIC'),
            contextRootBpm: _this.appConfig.get(AppConfigValues.CONTEXTROOTBPM),
            contextRoot: _this.appConfig.get(AppConfigValues.CONTEXTROOTECM),
            disableCsrf: _this.appConfig.get(AppConfigValues.DISABLECSRF),
            withCredentials: _this.appConfig.get(AppConfigValues.AUTH_WITH_CREDENTIALS, false),
            oauth2: oauth
        });
        return _this;
    }
    /**
     * @return {?}
     */
    TestingAlfrescoApiService.prototype.getInstance = /**
     * @return {?}
     */
    function () {
        if (this.alfrescoApi) {
            this.alfrescoApi.configureJsApi(this.config);
        }
        else {
            this.alfrescoApi = new AlfrescoApiCompatibility(this.config);
        }
        return this.alfrescoApi;
    };
    return TestingAlfrescoApiService;
}(AlfrescoApiService));
export { TestingAlfrescoApiService };
if (false) {
    /**
     * @type {?}
     * @protected
     */
    TestingAlfrescoApiService.prototype.alfrescoApi;
    /** @type {?} */
    TestingAlfrescoApiService.prototype.config;
    /** @type {?} */
    TestingAlfrescoApiService.prototype.appConfig;
}
