// @alfresco/adf-insights/esm5/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { AnalyticsComponent, AnalyticsGeneratorComponent, AnalyticsReportListComponent, AnalyticsReportParametersComponent, AnalyticsService, AnalyticsProcessModule } from './analytics-process/index';
export { DiagramSequenceFlowComponent, DiagramComponent, DiagramColorService, DiagramsService, DiagramModel, DiagramElementModel, DiagramElementPropertyModel, DiagramEventDefinitionModel, DiagramFlowElementModel, DiagramLaneElementModel, DiagramPoolElementModel, DiagramWayPointModel, BarChart, Chart, DetailsTableChart, HeatMapChart, LineChart, MultiBarChart, PieChart, TableChart, ParameterValueModel, ReportDateRange, ReportDefinitionModel, ReportParameterDetailsModel, ReportParametersModel, ReportQuery, DiagramsModule } from './diagram/index';
export { providers, InsightsModule, InsightsModuleLazy } from './insights.module';
