// @alfresco/adf-insights/esm5/insights.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CoreModule, TRANSLATION_PROVIDER } from '@alfresco/adf-core';
import { DiagramsModule } from './diagram/diagram.module';
import { AnalyticsProcessModule } from './analytics-process/analytics-process.module';
import { MaterialModule } from './material.module';
import { DiagramsService } from './diagram/services/diagrams.service';
import { DiagramColorService } from './diagram/services/diagram-color.service';
import { RaphaelService } from './diagram/components/raphael/raphael.service';
import { AnalyticsService } from './analytics-process/services/analytics.service';
/**
 * @return {?}
 */
export function providers() {
    return [
        AnalyticsService,
        DiagramsService,
        DiagramColorService,
        RaphaelService
    ];
}
var ɵ0 = {
    name: 'adf-insights',
    source: 'assets/adf-insights'
};
var InsightsModule = /** @class */ (function () {
    function InsightsModule() {
    }
    /**
     * @return {?}
     */
    InsightsModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: InsightsModule,
            providers: tslib_1.__spread(providers(), [
                {
                    provide: TRANSLATION_PROVIDER,
                    multi: true,
                    useValue: {
                        name: 'adf-insights',
                        source: 'assets/adf-insights'
                    }
                }
            ])
        };
    };
    /**
     * @return {?}
     */
    InsightsModule.forChild = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: InsightsModuleLazy
        };
    };
    InsightsModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CoreModule.forChild(),
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        MaterialModule,
                        DiagramsModule,
                        AnalyticsProcessModule
                    ],
                    providers: tslib_1.__spread(providers(), [
                        {
                            provide: TRANSLATION_PROVIDER,
                            multi: true,
                            useValue: ɵ0
                        }
                    ]),
                    exports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        MaterialModule,
                        DiagramsModule,
                        AnalyticsProcessModule
                    ]
                },] }
    ];
    return InsightsModule;
}());
export { InsightsModule };
var InsightsModuleLazy = /** @class */ (function () {
    function InsightsModuleLazy() {
    }
    InsightsModuleLazy.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CoreModule.forChild(),
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        MaterialModule,
                        DiagramsModule,
                        AnalyticsProcessModule
                    ],
                    exports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        MaterialModule,
                        DiagramsModule,
                        AnalyticsProcessModule
                    ]
                },] }
    ];
    return InsightsModuleLazy;
}());
export { InsightsModuleLazy };
export { ɵ0 };
