// @alfresco/adf-content-services/esm5/search/components/search-widget-container/search-widget-container.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewChild, ViewContainerRef, ComponentFactoryResolver } from '@angular/core';
import { SearchQueryBuilderService } from '../../search-query-builder.service';
import { SearchFilterService } from '../search-filter/search-filter.service';
var SearchWidgetContainerComponent = /** @class */ (function () {
    function SearchWidgetContainerComponent(searchFilterService, queryBuilder, componentFactoryResolver) {
        this.searchFilterService = searchFilterService;
        this.queryBuilder = queryBuilder;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @return {?}
     */
    SearchWidgetContainerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var componentType = this.searchFilterService.widgets[this.selector];
        if (componentType) {
            /** @type {?} */
            var factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.setupWidget(this.componentRef);
            }
        }
    };
    /**
     * @private
     * @param {?} ref
     * @return {?}
     */
    SearchWidgetContainerComponent.prototype.setupWidget = /**
     * @private
     * @param {?} ref
     * @return {?}
     */
    function (ref) {
        if (ref && ref.instance) {
            ref.instance.id = this.id;
            ref.instance.settings = tslib_1.__assign({}, this.settings);
            ref.instance.context = this.queryBuilder;
        }
    };
    /**
     * @return {?}
     */
    SearchWidgetContainerComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    };
    SearchWidgetContainerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-search-widget-container',
                    template: '<div #content></div>'
                }] }
    ];
    /** @nocollapse */
    SearchWidgetContainerComponent.ctorParameters = function () { return [
        { type: SearchFilterService },
        { type: SearchQueryBuilderService },
        { type: ComponentFactoryResolver }
    ]; };
    SearchWidgetContainerComponent.propDecorators = {
        content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
        id: [{ type: Input }],
        selector: [{ type: Input }],
        settings: [{ type: Input }],
        config: [{ type: Input }]
    };
    return SearchWidgetContainerComponent;
}());
export { SearchWidgetContainerComponent };
if (false) {
    /** @type {?} */
    SearchWidgetContainerComponent.prototype.content;
    /** @type {?} */
    SearchWidgetContainerComponent.prototype.id;
    /** @type {?} */
    SearchWidgetContainerComponent.prototype.selector;
    /** @type {?} */
    SearchWidgetContainerComponent.prototype.settings;
    /** @type {?} */
    SearchWidgetContainerComponent.prototype.config;
    /**
     * @type {?}
     * @private
     */
    SearchWidgetContainerComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    SearchWidgetContainerComponent.prototype.searchFilterService;
    /**
     * @type {?}
     * @private
     */
    SearchWidgetContainerComponent.prototype.queryBuilder;
    /**
     * @type {?}
     * @private
     */
    SearchWidgetContainerComponent.prototype.componentFactoryResolver;
}
