// @alfresco/adf-content-services/esm5/tree-view/components/tree-view.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FlatTreeControl } from '@angular/cdk/tree';
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { TreeViewDataSource } from '../data/tree-view-datasource';
import { TreeViewService } from '../services/tree-view.service';
var TreeViewComponent = /** @class */ (function () {
    function TreeViewComponent(treeViewService) {
        this.treeViewService = treeViewService;
        /**
         * Emitted when a node in the tree view is clicked.
         */
        this.nodeClicked = new EventEmitter();
        /**
         * Emitted when an invalid node id is given.
         */
        this.error = new EventEmitter();
        this.getLevel = (/**
         * @param {?} node
         * @return {?}
         */
        function (node) { return node.level; });
        this.isExpandable = (/**
         * @param {?} node
         * @return {?}
         */
        function (node) { return node.expandable; });
        this.hasChild = (/**
         * @param {?} level
         * @param {?} nodeData
         * @return {?}
         */
        function (level, nodeData) { return nodeData.expandable; });
        this.treeControl = new FlatTreeControl(this.getLevel, this.isExpandable);
        this.dataSource = new TreeViewDataSource(this.treeControl, this.treeViewService);
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    TreeViewComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes['nodeId'] && changes['nodeId'].currentValue &&
            changes['nodeId'].currentValue !== changes['nodeId'].previousValue) {
            this.loadTreeNode();
        }
        else {
            this.dataSource.data = [];
        }
    };
    /**
     * @param {?} node
     * @return {?}
     */
    TreeViewComponent.prototype.onNodeClicked = /**
     * @param {?} node
     * @return {?}
     */
    function (node) {
        this.nodeClicked.emit(node);
    };
    /**
     * @private
     * @return {?}
     */
    TreeViewComponent.prototype.loadTreeNode = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        this.treeViewService.getTreeNodes(this.nodeId)
            .subscribe((/**
         * @param {?} treeNode
         * @return {?}
         */
        function (treeNode) {
            _this.dataSource.data = treeNode;
        }), (/**
         * @param {?} error
         * @return {?}
         */
        function (error) { return _this.error.emit(error); }));
    };
    TreeViewComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-tree-view-list',
                    template: "<mat-tree class=\"adf-tree-view-main\" [dataSource]=\"dataSource\"\n          [treeControl]=\"treeControl\" *ngIf=\"nodeId; else missingNodeId\">\n    <mat-tree-node class=\"adf-tree-view-node\"\n        *matTreeNodeDef=\"let treeNode\" id=\"{{treeNode.name + '-tree-node'}}\"\n        matTreeNodePadding [matTreeNodePaddingIndent]=\"15\">\n        {{treeNode.name}}\n    </mat-tree-node>\n    <mat-tree-node class=\"adf-tree-view-node\"\n        id=\"{{treeNode.name + '-tree-child-node'}}\" *matTreeNodeDef=\"let treeNode; when: hasChild\"\n        matTreeNodePadding [matTreeNodePaddingIndent]=\"15\">\n        <button id=\"{{'button-'+treeNode.name}}\" (click)=\"onNodeClicked(treeNode.node)\"\n                mat-icon-button [attr.aria-label]=\"'toggle ' + treeNode.filename\" matTreeNodeToggle>\n            <mat-icon class=\"adf-tree-view-icon\">\n                {{treeControl.isExpanded(treeNode) ? 'folder_open' : 'folder'}}\n            </mat-icon>\n        </button>\n        {{treeNode.name}}\n    </mat-tree-node>\n</mat-tree>\n<ng-template #missingNodeId>\n    <div id=\"adf-tree-view-missing-node\">\n        {{'ADF-TREE-VIEW.MISSING-ID' | translate}}\n    </div>\n</ng-template>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    TreeViewComponent.ctorParameters = function () { return [
        { type: TreeViewService }
    ]; };
    TreeViewComponent.propDecorators = {
        nodeId: [{ type: Input }],
        nodeClicked: [{ type: Output }],
        error: [{ type: Output }]
    };
    return TreeViewComponent;
}());
export { TreeViewComponent };
if (false) {
    /**
     * Identifier of the node to display.
     * @type {?}
     */
    TreeViewComponent.prototype.nodeId;
    /**
     * Emitted when a node in the tree view is clicked.
     * @type {?}
     */
    TreeViewComponent.prototype.nodeClicked;
    /**
     * Emitted when an invalid node id is given.
     * @type {?}
     */
    TreeViewComponent.prototype.error;
    /** @type {?} */
    TreeViewComponent.prototype.treeControl;
    /** @type {?} */
    TreeViewComponent.prototype.dataSource;
    /** @type {?} */
    TreeViewComponent.prototype.getLevel;
    /** @type {?} */
    TreeViewComponent.prototype.isExpandable;
    /** @type {?} */
    TreeViewComponent.prototype.hasChild;
    /**
     * @type {?}
     * @private
     */
    TreeViewComponent.prototype.treeViewService;
}
