// @alfresco/adf-extensions/esm5/lib/services/app-extension.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { ExtensionService } from '../services/extension.service';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./extension.service";
var AppExtensionService = /** @class */ (function () {
    function AppExtensionService(extensionService) {
        this.extensionService = extensionService;
        this._references = new BehaviorSubject([]);
        this.references$ = this._references.asObservable();
    }
    /**
     * @return {?}
     */
    AppExtensionService.prototype.load = /**
     * @return {?}
     */
    function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var config;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.extensionService.load()];
                    case 1:
                        config = _a.sent();
                        this.setup(config);
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} config
     * @return {?}
     */
    AppExtensionService.prototype.setup = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        if (!config) {
            return;
        }
        /** @type {?} */
        var references = (config.$references || [])
            .filter((/**
         * @param {?} entry
         * @return {?}
         */
        function (entry) { return typeof entry === 'object'; }))
            .map((/**
         * @param {?} entry
         * @return {?}
         */
        function (entry) { return (/** @type {?} */ (entry)); }));
        this._references.next(references);
    };
    /**
     * Provides a collection of document list columns for the particular preset.
     * The result is filtered by the **disabled** state.
     * @param key Preset key.
     */
    /**
     * Provides a collection of document list columns for the particular preset.
     * The result is filtered by the **disabled** state.
     * @param {?} key Preset key.
     * @return {?}
     */
    AppExtensionService.prototype.getDocumentListPreset = /**
     * Provides a collection of document list columns for the particular preset.
     * The result is filtered by the **disabled** state.
     * @param {?} key Preset key.
     * @return {?}
     */
    function (key) {
        return this.extensionService
            .getElements("features.documentList." + key)
            .filter((/**
         * @param {?} entry
         * @return {?}
         */
        function (entry) { return !entry.disabled; }));
    };
    /**
     * Provides a list of the Viewer content extensions,
     * filtered by **disabled** state and **rules**.
     */
    /**
     * Provides a list of the Viewer content extensions,
     * filtered by **disabled** state and **rules**.
     * @return {?}
     */
    AppExtensionService.prototype.getViewerExtensions = /**
     * Provides a list of the Viewer content extensions,
     * filtered by **disabled** state and **rules**.
     * @return {?}
     */
    function () {
        var _this = this;
        return this.extensionService
            .getElements('features.viewer.content')
            .filter((/**
         * @param {?} extension
         * @return {?}
         */
        function (extension) { return !_this.isViewerExtensionDisabled(extension); }));
    };
    /**
     * @protected
     * @param {?} extension
     * @return {?}
     */
    AppExtensionService.prototype.isViewerExtensionDisabled = /**
     * @protected
     * @param {?} extension
     * @return {?}
     */
    function (extension) {
        if (extension) {
            if (extension.disabled) {
                return true;
            }
            if (extension.rules && extension.rules.disabled) {
                return this.extensionService.evaluateRule(extension.rules.disabled);
            }
        }
        return false;
    };
    AppExtensionService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    AppExtensionService.ctorParameters = function () { return [
        { type: ExtensionService }
    ]; };
    /** @nocollapse */ AppExtensionService.ngInjectableDef = i0.defineInjectable({ factory: function AppExtensionService_Factory() { return new AppExtensionService(i0.inject(i1.ExtensionService)); }, token: AppExtensionService, providedIn: "root" });
    return AppExtensionService;
}());
export { AppExtensionService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AppExtensionService.prototype._references;
    /** @type {?} */
    AppExtensionService.prototype.references$;
    /**
     * @type {?}
     * @protected
     */
    AppExtensionService.prototype.extensionService;
}
