// @alfresco/adf-extensions/esm5/lib/config/action.extensions.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/** @enum {string} */
var ContentActionType = {
    default: 'default',
    button: 'button',
    separator: 'separator',
    menu: 'menu',
    custom: 'custom',
};
export { ContentActionType };
/**
 * @record
 */
export function ContentActionRef() { }
if (false) {
    /** @type {?} */
    ContentActionRef.prototype.type;
    /** @type {?|undefined} */
    ContentActionRef.prototype.title;
    /** @type {?|undefined} */
    ContentActionRef.prototype.description;
    /** @type {?|undefined} */
    ContentActionRef.prototype.icon;
    /** @type {?|undefined} */
    ContentActionRef.prototype.children;
    /** @type {?|undefined} */
    ContentActionRef.prototype.component;
    /** @type {?|undefined} */
    ContentActionRef.prototype.actions;
    /** @type {?|undefined} */
    ContentActionRef.prototype.rules;
}
/**
 * @record
 */
export function ActionRef() { }
if (false) {
    /** @type {?} */
    ActionRef.prototype.id;
    /** @type {?} */
    ActionRef.prototype.type;
    /** @type {?|undefined} */
    ActionRef.prototype.payload;
}
