// @alfresco/adf-insights/esm2015/diagram/models/diagram/diagramPoolElement.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DiagramLaneElementModel } from './diagramLaneElement.model';
export class DiagramPoolElementModel {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        this.lanes = [];
        if (obj) {
            this.height = obj.height;
            this.id = obj.id;
            this.name = obj.name;
            this.properties = obj.properties;
            this.width = obj.width;
            this.x = obj.x;
            this.y = obj.y;
            if (obj.lanes) {
                obj.lanes.forEach((/**
                 * @param {?} lane
                 * @return {?}
                 */
                (lane) => {
                    this.lanes.push(new DiagramLaneElementModel(lane));
                }));
            }
        }
    }
}
if (false) {
    /** @type {?} */
    DiagramPoolElementModel.prototype.height;
    /** @type {?} */
    DiagramPoolElementModel.prototype.id;
    /** @type {?} */
    DiagramPoolElementModel.prototype.name;
    /** @type {?} */
    DiagramPoolElementModel.prototype.properties;
    /** @type {?} */
    DiagramPoolElementModel.prototype.lanes;
    /** @type {?} */
    DiagramPoolElementModel.prototype.width;
    /** @type {?} */
    DiagramPoolElementModel.prototype.x;
    /** @type {?} */
    DiagramPoolElementModel.prototype.y;
}
