// @alfresco/adf-insights/esm2015/diagram/models/diagram/diagramFlowElement.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DiagramWayPointModel } from './diagramWayPoint.model';
export class DiagramFlowElementModel {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        this.properties = [];
        this.waypoints = [];
        if (obj) {
            this.completed = !!obj.completed;
            this.current = !!obj.current;
            this.id = obj.id;
            this.properties = obj.properties;
            this.sourceRef = obj.sourceRef;
            this.targetRef = obj.targetRef;
            this.type = obj.type;
            if (obj.waypoints) {
                obj.waypoints.forEach((/**
                 * @param {?} waypoint
                 * @return {?}
                 */
                (waypoint) => {
                    this.waypoints.push(new DiagramWayPointModel(waypoint));
                }));
            }
        }
    }
}
if (false) {
    /** @type {?} */
    DiagramFlowElementModel.prototype.completed;
    /** @type {?} */
    DiagramFlowElementModel.prototype.current;
    /** @type {?} */
    DiagramFlowElementModel.prototype.id;
    /** @type {?} */
    DiagramFlowElementModel.prototype.properties;
    /** @type {?} */
    DiagramFlowElementModel.prototype.sourceRef;
    /** @type {?} */
    DiagramFlowElementModel.prototype.targetRef;
    /** @type {?} */
    DiagramFlowElementModel.prototype.type;
    /** @type {?} */
    DiagramFlowElementModel.prototype.waypoints;
}
