// @alfresco/adf-core/esm2015/userinfo/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { UserInfoComponent } from './components/user-info.component';
export { BpmUserService } from './services/bpm-user.service';
export { EcmUserService } from './services/ecm-user.service';
export { IdentityUserService } from './services/identity-user.service';
export { BpmUserModel } from './models/bpm-user.model';
export { EcmUserModel } from './models/ecm-user.model';
export { IdentityUserModel } from './models/identity-user.model';
export { IdentityRoleModel } from './models/identity-role.model';
export { UserInfoModule } from './userinfo.module';
