// @alfresco/adf-process-services/esm2015/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { ProcessFiltersComponent, ProcessInstanceDetailsComponent, ProcessAuditDirective, ProcessInstanceHeaderComponent, ProcessInstanceTasksComponent, ProcessInstanceListComponent, StartProcessInstanceComponent, ProcessService, ProcessFilterService, FilterProcessRepresentationModel, ProcessFilterParamRepresentationModel, ProcessDefinitionRepresentation, ProcessInstance, ProcessFilterRequestRepresentation, ProcessInstanceVariable, ProcessListModule } from './process-list/index';
export { TaskListComponent, ChecklistComponent, TaskHeaderComponent, NoTaskDetailsTemplateDirective, TaskFiltersComponent, TaskDetailsComponent, TaskAuditDirective, StartTaskComponent, TaskStandaloneComponent, AttachFormComponent, TaskListService, ProcessUploadService, TaskUploadService, TaskFilterService, AppDefinitionRepresentationModel, FilterParamsModel, FilterRepresentationModel, TaskQueryRequestRepresentationModel, Form, StartTaskModel, TaskDetailsEvent, TaskDetailsModel, TaskListModel, UserEventModel, UserGroupModel, TaskListModule } from './task-list/index';
export { AppsListComponent, SelectAppsDialogComponent, AppsListModule } from './app-list/index';
export { TaskAttachmentListComponent, ProcessAttachmentListComponent, CreateProcessAttachmentComponent, AttachmentComponent, AttachmentModule } from './attachment/index';
export { ProcessCommentsComponent, ProcessCommentsModule } from './process-comments/index';
export { PeopleComponent, PeopleListComponent, PeopleSearchComponent, PeopleSearchFieldComponent, PeopleSelectorComponent, PeopleSearchActionLabelDirective, PeopleSearchTitleDirective, PeopleModule } from './people/index';
export { AttachFileWidgetComponent, AttachFolderWidgetComponent, AttachFileWidgetDialogComponent, AttachFileWidgetDialogService, ContentWidgetModule } from './content-widget/index';
export { FormComponent, StartFormComponent, FormModule } from './form/index';
export { ProcessModule, ProcessModuleLazy } from './process.module';
