// @alfresco/adf-extensions/esm2015/lib/config/extension-utils.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContentActionType } from './action.extensions';
/**
 * @param {?} target
 * @param {?} key
 * @return {?}
 */
export function getValue(target, key) {
    if (!target) {
        return undefined;
    }
    /** @type {?} */
    const keys = key.split('.');
    key = '';
    do {
        key += keys.shift();
        /** @type {?} */
        const value = target[key];
        if (value !== undefined &&
            (typeof value === 'object' || !keys.length)) {
            target = value;
            key = '';
        }
        else if (!keys.length) {
            target = undefined;
        }
        else {
            key += '.';
        }
    } while (keys.length);
    return target;
}
/**
 * @param {?} entry
 * @return {?}
 */
export function filterEnabled(entry) {
    return !entry.disabled;
}
/**
 * @param {?} a
 * @param {?} b
 * @return {?}
 */
export function sortByOrder(a, b) {
    /** @type {?} */
    const left = a.order === undefined ? Number.MAX_SAFE_INTEGER : a.order;
    /** @type {?} */
    const right = b.order === undefined ? Number.MAX_SAFE_INTEGER : b.order;
    return left - right;
}
/**
 * @param {?} acc
 * @param {?} el
 * @param {?} i
 * @param {?} arr
 * @return {?}
 */
export function reduceSeparators(acc, el, i, arr) {
    // remove leading separator
    if (i === 0) {
        if (arr[i].type === ContentActionType.separator) {
            return acc;
        }
    }
    // remove duplicate separators
    if (i > 0) {
        /** @type {?} */
        const prev = arr[i - 1];
        if (prev.type === ContentActionType.separator &&
            el.type === ContentActionType.separator) {
            return acc;
        }
        // remove trailing separator
        if (i === arr.length - 1) {
            if (el.type === ContentActionType.separator) {
                return acc;
            }
        }
    }
    return acc.concat(el);
}
/**
 * @param {?} acc
 * @param {?} el
 * @return {?}
 */
export function reduceEmptyMenus(acc, el) {
    if (el.type === ContentActionType.menu) {
        if ((el.children || []).length === 0) {
            return acc;
        }
    }
    return acc.concat(el);
}
/**
 * @param {...?} objects
 * @return {?}
 */
export function mergeObjects(...objects) {
    /** @type {?} */
    const result = {};
    objects.forEach((/**
     * @param {?} source
     * @return {?}
     */
    (source) => {
        Object.keys(source).forEach((/**
         * @param {?} prop
         * @return {?}
         */
        (prop) => {
            if (!prop.startsWith('$')) {
                if (prop in result && Array.isArray(result[prop])) {
                    // result[prop] = result[prop].concat(source[prop]);
                    result[prop] = mergeArrays(result[prop], source[prop]);
                }
                else if (prop in result && typeof result[prop] === 'object') {
                    result[prop] = mergeObjects(result[prop], source[prop]);
                }
                else {
                    result[prop] = source[prop];
                }
            }
        }));
    }));
    return result;
}
/**
 * @param {?} left
 * @param {?} right
 * @return {?}
 */
export function mergeArrays(left, right) {
    /** @type {?} */
    const result = [];
    /** @type {?} */
    const map = {};
    (left || []).forEach((/**
     * @param {?} entry
     * @return {?}
     */
    (entry) => {
        /** @type {?} */
        const element = entry;
        if (element && element.hasOwnProperty('id')) {
            map[element.id] = element;
        }
        else {
            result.push(element);
        }
    }));
    (right || []).forEach((/**
     * @param {?} entry
     * @return {?}
     */
    (entry) => {
        /** @type {?} */
        const element = entry;
        if (element && element.hasOwnProperty('id') && map[element.id]) {
            /** @type {?} */
            const merged = mergeObjects(map[element.id], element);
            map[element.id] = merged;
        }
        else {
            result.push(element);
        }
    }));
    return Object.keys(map).map((/**
     * @param {?} key
     * @return {?}
     */
    (key) => map[key])).concat(result);
}
