// @alfresco/adf-core/esm5/pipes/node-name-tooltip.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Pipe } from '@angular/core';
var NodeNameTooltipPipe = /** @class */ (function () {
    function NodeNameTooltipPipe() {
    }
    /**
     * @param {?} node
     * @return {?}
     */
    NodeNameTooltipPipe.prototype.transform = /**
     * @param {?} node
     * @return {?}
     */
    function (node) {
        if (node) {
            return this.getNodeTooltip(node);
        }
        return null;
    };
    /**
     * @private
     * @param {?} lines
     * @param {?} line
     * @return {?}
     */
    NodeNameTooltipPipe.prototype.containsLine = /**
     * @private
     * @param {?} lines
     * @param {?} line
     * @return {?}
     */
    function (lines, line) {
        return lines.some((/**
         * @param {?} item
         * @return {?}
         */
        function (item) {
            return item.toLowerCase() === line.toLowerCase();
        }));
    };
    /**
     * @private
     * @param {?} lines
     * @return {?}
     */
    NodeNameTooltipPipe.prototype.removeDuplicateLines = /**
     * @private
     * @param {?} lines
     * @return {?}
     */
    function (lines) {
        var _this = this;
        /** @type {?} */
        var reducer = (/**
         * @param {?} acc
         * @param {?} line
         * @return {?}
         */
        function (acc, line) {
            if (!_this.containsLine(acc, line)) {
                acc.push(line);
            }
            return acc;
        });
        return lines.reduce(reducer, []);
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    NodeNameTooltipPipe.prototype.getNodeTooltip = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        if (!node || !node.entry) {
            return null;
        }
        var _a = node.entry, properties = _a.properties, name = _a.name;
        /** @type {?} */
        var lines = [name];
        if (properties) {
            var title = properties["cm:title"], description = properties["cm:description"];
            if (title && description) {
                lines[0] = title;
                lines[1] = description;
            }
            if (title) {
                lines[1] = title;
            }
            if (description) {
                lines[1] = description;
            }
        }
        return this.removeDuplicateLines(lines).join("\n");
    };
    NodeNameTooltipPipe.decorators = [
        { type: Pipe, args: [{
                    name: 'adfNodeNameTooltip'
                },] }
    ];
    return NodeNameTooltipPipe;
}());
export { NodeNameTooltipPipe };
