// @alfresco/adf-insights/esm5/analytics-process/components/analytics.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewChild, ViewEncapsulation } from '@angular/core';
import { AnalyticsGeneratorComponent } from './analytics-generator.component';
var AnalyticsComponent = /** @class */ (function () {
    function AnalyticsComponent() {
        this.hideParameters = false;
        this.editReport = new EventEmitter();
        this.reportSaved = new EventEmitter();
        this.reportDeleted = new EventEmitter();
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    AnalyticsComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        this.analyticsGenerator.reset();
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    AnalyticsComponent.prototype.showReport = /**
     * @param {?} $event
     * @return {?}
     */
    function ($event) {
        this.analyticsGenerator.generateReport(this.reportId, $event);
    };
    /**
     * @return {?}
     */
    AnalyticsComponent.prototype.reset = /**
     * @return {?}
     */
    function () {
        this.analyticsGenerator.reset();
    };
    /**
     * @param {?} name
     * @return {?}
     */
    AnalyticsComponent.prototype.onEditReport = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        this.editReport.emit(name);
    };
    /**
     * @param {?} reportId
     * @return {?}
     */
    AnalyticsComponent.prototype.onSaveReportSuccess = /**
     * @param {?} reportId
     * @return {?}
     */
    function (reportId) {
        this.reportSaved.emit(reportId);
    };
    /**
     * @return {?}
     */
    AnalyticsComponent.prototype.onDeleteReportSuccess = /**
     * @return {?}
     */
    function () {
        this.reportDeleted.emit();
    };
    AnalyticsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-analytics',
                    template: "<div class=\"adf-analytics-container\">\n    <adf-analytics-report-parameters [appId]=\"appId\" [reportId]=\"reportId\"\n                                 [hideComponent]=\"hideParameters\"\n                                 (formValueChanged)=\"reset()\"\n                                 (success)=\"showReport($event)\"\n                                 (saveReportSuccess)=\"onSaveReportSuccess($event)\"\n                                 (deleteReportSuccess)=\"onDeleteReportSuccess()\"\n                                 (edit)=\"onEditReport($event)\">\n    </adf-analytics-report-parameters>\n\n    <adf-analytics-generator [reportId]=\"reportId\"\n                                  [reportParamQuery]=\"reportParamQuery\"\n                                  #analyticsGenerator>\n    </adf-analytics-generator>\n</div>\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-chart{display:block;width:100%}"]
                }] }
    ];
    AnalyticsComponent.propDecorators = {
        appId: [{ type: Input }],
        reportId: [{ type: Input }],
        hideParameters: [{ type: Input }],
        editReport: [{ type: Output }],
        reportSaved: [{ type: Output }],
        reportDeleted: [{ type: Output }],
        analyticsGenerator: [{ type: ViewChild, args: ['analyticsGenerator',] }]
    };
    return AnalyticsComponent;
}());
export { AnalyticsComponent };
if (false) {
    /** @type {?} */
    AnalyticsComponent.prototype.appId;
    /** @type {?} */
    AnalyticsComponent.prototype.reportId;
    /** @type {?} */
    AnalyticsComponent.prototype.hideParameters;
    /** @type {?} */
    AnalyticsComponent.prototype.editReport;
    /** @type {?} */
    AnalyticsComponent.prototype.reportSaved;
    /** @type {?} */
    AnalyticsComponent.prototype.reportDeleted;
    /** @type {?} */
    AnalyticsComponent.prototype.analyticsGenerator;
    /** @type {?} */
    AnalyticsComponent.prototype.reportParamQuery;
}
