// @alfresco/adf-testing/esm2015/lib/content-services/pages/document-list.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { by, element, browser } from 'protractor';
import { DataTableComponentPage } from '../../core/pages/data-table-component.page';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
export class DocumentListPage {
    /**
     * @param {?=} rootElement
     */
    constructor(rootElement = element.all(by.css('adf-document-list')).first()) {
        this.actionMenu = element(by.css('div[role="menu"]'));
        this.optionButton = by.css('button[data-automation-id*="action_menu_"]');
        this.rootElement = rootElement;
        this.dataTable = new DataTableComponentPage(this.rootElement);
        this.tableBody = rootElement.all(by.css('div[class="adf-datatable-body"]')).first();
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} content
     * @return {THIS}
     */
    checkLockedIcon(content) {
        /** @type {?} */
        const row = (/** @type {?} */ (this)).dataTable.getRow('Display name', content);
        /** @type {?} */
        const lockIcon = row.element(by.cssContainingText('div[title="Lock"] mat-icon', 'lock'));
        BrowserVisibility.waitUntilElementIsVisible(lockIcon);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} content
     * @return {THIS}
     */
    checkUnlockedIcon(content) {
        /** @type {?} */
        const row = (/** @type {?} */ (this)).dataTable.getRow('Display name', content);
        /** @type {?} */
        const lockIcon = row.element(by.cssContainingText('div[title="Lock"] mat-icon', 'lock_open'));
        BrowserVisibility.waitUntilElementIsVisible(lockIcon);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    waitForTableBody() {
        return BrowserVisibility.waitUntilElementIsVisible(this.tableBody);
    }
    /**
     * @param {?} nodeName
     * @return {?}
     */
    getTooltip(nodeName) {
        return this.dataTable.getTooltip('Display name', nodeName);
    }
    /**
     * @param {?} nodeName
     * @return {?}
     */
    selectRow(nodeName) {
        return this.dataTable.selectRow('Display name', nodeName);
    }
    /**
     * @param {?} nodeName
     * @return {?}
     */
    rightClickOnRow(nodeName) {
        return this.dataTable.rightClickOnRow('Display name', nodeName);
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} content
     * @return {THIS}
     */
    clickOnActionMenu(content) {
        /** @type {?} */
        const row = (/** @type {?} */ (this)).dataTable.getRow('Display name', content);
        row.element((/** @type {?} */ (this)).optionButton).click();
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).actionMenu);
        browser.sleep(500);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    checkActionMenuIsNotDisplayed() {
        BrowserVisibility.waitUntilElementIsNotVisible((/** @type {?} */ (this)).actionMenu);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    dataTablePage() {
        return new DataTableComponentPage(this.rootElement);
    }
    /**
     * @param {?} column
     * @return {?}
     */
    getAllRowsColumnValues(column) {
        return this.dataTable.getAllRowsColumnValues(column);
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} nodeName
     * @return {THIS}
     */
    doubleClickRow(nodeName) {
        (/** @type {?} */ (this)).dataTable.doubleClickRow('Display name', nodeName);
        return (/** @type {?} */ (this));
    }
}
if (false) {
    /** @type {?} */
    DocumentListPage.prototype.rootElement;
    /** @type {?} */
    DocumentListPage.prototype.actionMenu;
    /** @type {?} */
    DocumentListPage.prototype.optionButton;
    /** @type {?} */
    DocumentListPage.prototype.tableBody;
    /** @type {?} */
    DocumentListPage.prototype.dataTable;
}
