// @alfresco/adf-testing/esm5/lib/core/actions/api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiCompatibility as AlfrescoApi } from '@alfresco/js-api';
var ApiService = /** @class */ (function () {
    function ApiService(clientId, host, hostSso, provider) {
        this.config = {
            provider: provider,
            hostBpm: host,
            hostEcm: host,
            authType: 'OAUTH',
            oauth2: {
                host: hostSso,
                clientId: clientId,
                scope: 'openid',
                secret: '',
                implicitFlow: false,
                silentLogin: false,
                redirectUri: '/',
                redirectUriLogout: '/logout'
            }
        };
        this.apiService = new AlfrescoApi(this.config);
    }
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    ApiService.prototype.login = /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    function (username, password) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.apiService.login(username, password)];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    ApiService.prototype.performBpmOperation = /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    function (path, method, queryParams, postBody) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var uri, pathParams, formParams, contentTypes, accepts, headerParams;
            return tslib_1.__generator(this, function (_a) {
                uri = this.config.hostBpm + path;
                pathParams = {};
                formParams = {};
                contentTypes = ['application/json'];
                accepts = ['application/json'];
                headerParams = {
                    'Authorization': 'bearer ' + this.apiService.oauth2Auth.token
                };
                return [2 /*return*/, this.apiService.processClient.callCustomApi(uri, method, pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, Object)
                        .catch((/**
                     * @param {?} error
                     * @return {?}
                     */
                    function (error) {
                        throw (error);
                    }))];
            });
        });
    };
    /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    ApiService.prototype.performIdentityOperation = /**
     * @param {?} path
     * @param {?} method
     * @param {?} queryParams
     * @param {?} postBody
     * @return {?}
     */
    function (path, method, queryParams, postBody) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var uri, pathParams, formParams, contentTypes, accepts, headerParams;
            return tslib_1.__generator(this, function (_a) {
                uri = this.config.oauth2.host.replace('/realms', '/admin/realms') + path;
                pathParams = {};
                formParams = {};
                contentTypes = ['application/json'];
                accepts = ['application/json'];
                headerParams = {
                    'Authorization': 'bearer ' + this.apiService.oauth2Auth.token
                };
                return [2 /*return*/, this.apiService.processClient.callCustomApi(uri, method, pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, Object)
                        .catch((/**
                     * @param {?} error
                     * @return {?}
                     */
                    function (error) {
                        throw (error);
                    }))];
            });
        });
    };
    return ApiService;
}());
export { ApiService };
if (false) {
    /** @type {?} */
    ApiService.prototype.apiService;
    /** @type {?} */
    ApiService.prototype.config;
}
