// @alfresco/adf-extensions/esm5/lib/components/viewer/preview-extension.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ComponentFactoryResolver, ViewChild, ViewContainerRef } from '@angular/core';
import { ExtensionService } from '../../services/extension.service';
import { Node } from '@alfresco/js-api';
var PreviewExtensionComponent = /** @class */ (function () {
    function PreviewExtensionComponent(extensionService, componentFactoryResolver) {
        this.extensionService = extensionService;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @return {?}
     */
    PreviewExtensionComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.id) {
            return;
        }
        /** @type {?} */
        var componentType = this.extensionService.getComponentById(this.id);
        if (componentType) {
            /** @type {?} */
            var factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.updateInstance();
            }
        }
    };
    /**
     * @return {?}
     */
    PreviewExtensionComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.updateInstance();
    };
    /**
     * @return {?}
     */
    PreviewExtensionComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    };
    /**
     * @private
     * @return {?}
     */
    PreviewExtensionComponent.prototype.updateInstance = /**
     * @private
     * @return {?}
     */
    function () {
        if (this.componentRef && this.componentRef.instance) {
            /** @type {?} */
            var instance = this.componentRef.instance;
            instance.node = this.node;
            instance.url = this.url;
            instance.extension = this.extension;
        }
    };
    PreviewExtensionComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-preview-extension',
                    template: "\n    <div #content></div>\n  "
                }] }
    ];
    /** @nocollapse */
    PreviewExtensionComponent.ctorParameters = function () { return [
        { type: ExtensionService },
        { type: ComponentFactoryResolver }
    ]; };
    PreviewExtensionComponent.propDecorators = {
        content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
        id: [{ type: Input }],
        url: [{ type: Input }],
        extension: [{ type: Input }],
        node: [{ type: Input }]
    };
    return PreviewExtensionComponent;
}());
export { PreviewExtensionComponent };
if (false) {
    /** @type {?} */
    PreviewExtensionComponent.prototype.content;
    /**
     * ID string of the component to preview.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.id;
    /**
     * URL of the content in the repository.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.url;
    /**
     * File extension (.jpg, .png, etc) for the viewer.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.extension;
    /**
     * Node containing the content to display.
     * @type {?}
     */
    PreviewExtensionComponent.prototype.node;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.extensionService;
    /**
     * @type {?}
     * @private
     */
    PreviewExtensionComponent.prototype.componentFactoryResolver;
}
