// @alfresco/adf-testing/esm2015/lib/process-services-cloud/pages/group-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { browser, by, element, protractor } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
export class GroupCloudComponentPage {
    constructor() {
        this.groupCloudSearch = element(by.css('input[data-automation-id="adf-cloud-group-search-input"]'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} name
     * @return {THIS}
     */
    searchGroups(name) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).groupCloudSearch);
        browser.sleep(1000);
        (/** @type {?} */ (this)).groupCloudSearch.clear().then((/**
         * @return {?}
         */
        () => {
            for (let i = 0; i < name.length; i++) {
                (/** @type {?} */ (this)).groupCloudSearch.sendKeys(name[i]);
            }
            (/** @type {?} */ (this)).groupCloudSearch.sendKeys(protractor.Key.BACK_SPACE);
            (/** @type {?} */ (this)).groupCloudSearch.sendKeys(name[name.length - 1]);
        }));
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} name
     * @return {THIS}
     */
    searchGroupsToExisting(name) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).groupCloudSearch);
        for (let i = 0; i < name.length; i++) {
            (/** @type {?} */ (this)).groupCloudSearch.sendKeys(name[i]);
        }
        (/** @type {?} */ (this)).groupCloudSearch.sendKeys(protractor.Key.BACK_SPACE);
        (/** @type {?} */ (this)).groupCloudSearch.sendKeys(name[name.length - 1]);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    getGroupsFieldContent() {
        BrowserVisibility.waitUntilElementIsVisible(this.groupCloudSearch);
        return this.groupCloudSearch.getAttribute('value');
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} name
     * @return {THIS}
     */
    selectGroupFromList(name) {
        /** @type {?} */
        const groupRow = element.all(by.cssContainingText('mat-option span', name)).first();
        BrowserVisibility.waitUntilElementIsVisible(groupRow);
        browser.sleep(1000);
        groupRow.click();
        BrowserVisibility.waitUntilElementIsNotVisible(groupRow);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} name
     * @return {THIS}
     */
    checkGroupIsDisplayed(name) {
        /** @type {?} */
        const groupRow = element.all(by.cssContainingText('mat-option span', name)).first();
        BrowserVisibility.waitUntilElementIsVisible(groupRow);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} name
     * @return {THIS}
     */
    checkGroupIsNotDisplayed(name) {
        /** @type {?} */
        const groupRow = element.all(by.cssContainingText('mat-option span', name)).first();
        BrowserVisibility.waitUntilElementIsNotVisible(groupRow);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} group
     * @return {THIS}
     */
    checkSelectedGroup(group) {
        BrowserVisibility.waitUntilElementIsVisible(element(by.cssContainingText('mat-chip[data-automation-id*="adf-cloud-group-chip-"]', group)));
        return (/** @type {?} */ (this));
    }
}
if (false) {
    /** @type {?} */
    GroupCloudComponentPage.prototype.groupCloudSearch;
}
