// @alfresco/adf-testing/esm2015/lib/process-services-cloud/pages/process-list-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { BrowserVisibility } from '../../core/utils/browser-visibility';
import { DataTableComponentPage } from '../../core/pages/data-table-component.page';
import { element, by } from 'protractor';
export class ProcessListCloudComponentPage {
    constructor() {
        this.columns = {
            id: 'Id',
            name: 'Name'
        };
        this.processList = element(by.css('adf-cloud-process-list'));
        this.noProcessFound = element.all(by.css("div[class='adf-empty-content__title']")).first();
        this.dataTable = new DataTableComponentPage(this.processList);
    }
    /**
     * @return {?}
     */
    getDataTable() {
        return this.dataTable;
    }
    /**
     * @param {?} processName
     * @return {?}
     */
    selectRow(processName) {
        return this.dataTable.selectRow(this.columns.name, processName);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    selectRowById(processId) {
        return this.dataTable.selectRow(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkRowIsSelectedById(processId) {
        return this.dataTable.checkRowIsSelected(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkRowIsNotSelectedById(processId) {
        return this.dataTable.checkRowIsNotSelected(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkRowIsCheckedById(processId) {
        return this.dataTable.checkRowIsChecked(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkRowIsNotCheckedById(processId) {
        return this.dataTable.checkRowIsNotChecked(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkCheckboxById(processId) {
        return this.dataTable.clickCheckbox(this.columns.id, processId);
    }
    /**
     * @param {?} processName
     * @return {?}
     */
    checkContentIsDisplayedByName(processName) {
        return this.dataTable.checkContentIsDisplayed(this.columns.name, processName);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkContentIsDisplayedById(processId) {
        return this.dataTable.checkContentIsDisplayed(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    checkContentIsNotDisplayedById(processId) {
        return this.dataTable.checkContentIsNotDisplayed(this.columns.id, processId);
    }
    /**
     * @param {?} processId
     * @return {?}
     */
    selectRowWithKeyboard(processId) {
        return this.dataTable.selectRowWithKeyboard(this.columns.id, processId);
    }
    /**
     * @return {?}
     */
    getAllRowsNameColumn() {
        return this.dataTable.getAllRowsColumnValues(this.columns.name);
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    checkProcessListIsLoaded() {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).processList);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    getNoProcessFoundMessage() {
        BrowserVisibility.waitUntilElementIsVisible(this.noProcessFound);
        return this.noProcessFound.getText();
    }
    /**
     * @param {?} column
     * @return {?}
     */
    getAllRowsByColumn(column) {
        return this.dataTable.getAllRowsColumnValues(column);
    }
}
if (false) {
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.columns;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.processList;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.noProcessFound;
    /** @type {?} */
    ProcessListCloudComponentPage.prototype.dataTable;
}
