// @alfresco/adf-testing/esm2015/lib/process-services-cloud/pages/process-header-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
export class ProcessHeaderCloudPage {
    constructor() {
        this.idField = element.all(by.css('span[data-automation-id*="id"] span')).first();
        this.nameField = element.all(by.css('span[data-automation-id*="name"] span')).first();
        this.statusField = element(by.css('span[data-automation-id*="status"] span'));
        this.initiatorField = element(by.css('span[data-automation-id*="initiator"] span'));
        this.startDateField = element.all(by.css('span[data-automation-id*="startDate"] span')).first();
        this.lastModifiedField = element.all(by.css('span[data-automation-id*="lastModified"] span')).first();
        this.parentIdField = element(by.css('span[data-automation-id*="parentId"] span'));
        this.businessKeyField = element.all(by.css('span[data-automation-id*="businessKey"] span')).first();
    }
    /**
     * @return {?}
     */
    getId() {
        BrowserVisibility.waitUntilElementIsVisible(this.idField);
        return this.idField.getText();
    }
    /**
     * @return {?}
     */
    getName() {
        BrowserVisibility.waitUntilElementIsVisible(this.nameField);
        return this.nameField.getText();
    }
    /**
     * @return {?}
     */
    getStatus() {
        BrowserVisibility.waitUntilElementIsVisible(this.statusField);
        return this.statusField.getText();
    }
    /**
     * @return {?}
     */
    getInitiator() {
        BrowserVisibility.waitUntilElementIsVisible(this.initiatorField);
        return this.initiatorField.getText();
    }
    /**
     * @return {?}
     */
    getStartDate() {
        BrowserVisibility.waitUntilElementIsVisible(this.startDateField);
        return this.startDateField.getText();
    }
    /**
     * @return {?}
     */
    getLastModified() {
        BrowserVisibility.waitUntilElementIsVisible(this.lastModifiedField);
        return this.lastModifiedField.getText();
    }
    /**
     * @return {?}
     */
    getParentId() {
        BrowserVisibility.waitUntilElementIsVisible(this.parentIdField);
        return this.parentIdField.getText();
    }
    /**
     * @return {?}
     */
    getBusinessKey() {
        BrowserVisibility.waitUntilElementIsVisible(this.businessKeyField);
        return this.businessKeyField.getText();
    }
}
if (false) {
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.idField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.nameField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.statusField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.initiatorField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.startDateField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.lastModifiedField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.parentIdField;
    /** @type {?} */
    ProcessHeaderCloudPage.prototype.businessKeyField;
}
