// @alfresco/adf-testing/esm2015/lib/process-services-cloud/pages/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { StartTasksCloudPage } from './start-tasks-cloud-component.page';
export { TaskHeaderCloudPage } from './task-header-cloud-component.page';
export { ProcessHeaderCloudPage } from './process-header-cloud-component.page';
export { EditProcessFilterCloudComponentPage } from './edit-process-filter-cloud-component.page';
export { EditTaskFilterCloudComponentPage } from './edit-task-filter-cloud-component.page';
export { GroupCloudComponentPage } from './group-cloud-component.page';
export { PeopleCloudComponentPage } from './people-cloud-component.page';
export { ProcessFiltersCloudComponentPage } from './process-filters-cloud-component.page';
export { ProcessListCloudComponentPage } from './process-list-cloud-component.page';
export { TaskFiltersCloudComponentPage } from './task-filters-cloud-component.page';
export { TaskListCloudComponentPage } from './task-list-cloud-component.page';
export { StartProcessCloudPage } from './start-process-cloud-component.page';
export { TaskFormCloudComponent } from './task-form-cloud-component.page';
export { EditProcessFilterDialogPage, EditTaskFilterDialogPage } from './dialog/public-api';
