// @alfresco/adf-testing/esm2015/lib/process-services-cloud/pages/task-header-cloud-component.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
export class TaskHeaderCloudPage {
    constructor() {
        this.assigneeField = element(by.css('span[data-automation-id*="assignee"] span'));
        this.statusField = element(by.css('span[data-automation-id*="status"] span'));
        this.priorityField = element(by.css('span[data-automation-id*="priority"] span'));
        this.dueDateField = element.all(by.css('span[data-automation-id*="dueDate"] span')).first();
        this.categoryField = element(by.css('span[data-automation-id*="category"] span'));
        this.createdField = element(by.css('span[data-automation-id="card-dateitem-created"] span'));
        this.parentNameField = element(by.css('span[data-automation-id*="parentName"] span'));
        this.parentTaskIdField = element(by.css('span[data-automation-id*="parentTaskId"] span'));
        this.endDateField = element.all(by.css('span[data-automation-id*="endDate"] span')).first();
        this.idField = element.all(by.css('span[data-automation-id*="id"] span')).first();
        this.descriptionField = element(by.css('span[data-automation-id*="description"] span'));
        this.taskPropertyList = element(by.css('adf-cloud-task-header adf-card-view div[class="adf-property-list"]'));
    }
    /**
     * @return {?}
     */
    getAssignee() {
        BrowserVisibility.waitUntilElementIsVisible(this.assigneeField);
        return this.assigneeField.getText();
    }
    /**
     * @return {?}
     */
    getStatus() {
        BrowserVisibility.waitUntilElementIsVisible(this.statusField);
        return this.statusField.getText();
    }
    /**
     * @return {?}
     */
    getPriority() {
        BrowserVisibility.waitUntilElementIsVisible(this.priorityField);
        return this.priorityField.getText();
    }
    /**
     * @return {?}
     */
    getCategory() {
        BrowserVisibility.waitUntilElementIsVisible(this.categoryField);
        return this.categoryField.getText();
    }
    /**
     * @return {?}
     */
    getParentName() {
        BrowserVisibility.waitUntilElementIsVisible(this.parentNameField);
        return this.parentNameField.getText();
    }
    /**
     * @return {?}
     */
    getParentTaskId() {
        BrowserVisibility.waitUntilElementIsVisible(this.parentTaskIdField);
        return this.parentTaskIdField.getText();
    }
    /**
     * @return {?}
     */
    getEndDate() {
        BrowserVisibility.waitUntilElementIsVisible(this.endDateField);
        return this.endDateField.getText();
    }
    /**
     * @return {?}
     */
    getCreated() {
        BrowserVisibility.waitUntilElementIsVisible(this.createdField);
        return this.createdField.getText();
    }
    /**
     * @return {?}
     */
    getId() {
        BrowserVisibility.waitUntilElementIsVisible(this.idField);
        return this.idField.getText();
    }
    /**
     * @return {?}
     */
    getDescription() {
        BrowserVisibility.waitUntilElementIsVisible(this.descriptionField);
        return this.descriptionField.getText();
    }
    /**
     * @return {?}
     */
    getDueDate() {
        BrowserVisibility.waitUntilElementIsVisible(this.dueDateField);
        return this.dueDateField.getText();
    }
}
if (false) {
    /** @type {?} */
    TaskHeaderCloudPage.prototype.assigneeField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.statusField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.priorityField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.dueDateField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.categoryField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.createdField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.parentNameField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.parentTaskIdField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.endDateField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.idField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.descriptionField;
    /** @type {?} */
    TaskHeaderCloudPage.prototype.taskPropertyList;
}
