// @alfresco/adf-content-services/esm5/permission-manager/models/permission.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var PermissionDisplayModel = /** @class */ (function () {
    function PermissionDisplayModel(obj) {
        this.isInherited = false;
        if (obj) {
            this.authorityId = obj.authorityId;
            this.name = obj.name;
            this.accessStatus = obj.accessStatus;
            this.isInherited = obj.isInherited !== null && obj.isInherited !== undefined ? obj.isInherited : false;
            this.icon = obj.icon ? obj.icon : 'vpn_key';
        }
    }
    return PermissionDisplayModel;
}());
export { PermissionDisplayModel };
if (false) {
    /** @type {?} */
    PermissionDisplayModel.prototype.authorityId;
    /** @type {?} */
    PermissionDisplayModel.prototype.name;
    /** @type {?} */
    PermissionDisplayModel.prototype.accessStatus;
    /** @type {?} */
    PermissionDisplayModel.prototype.isInherited;
    /** @type {?} */
    PermissionDisplayModel.prototype.icon;
}
