// @alfresco/adf-insights/esm5/diagram/components/tooltip/diagram-tooltip.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
/** @type {?} */
var POSITION = { BOTTOM: 'bottom', LEFT: 'left', RIGHT: 'right', TOP: 'top' };
/** @type {?} */
var STRATEGY = { CURSOR: 'cursor', ELEMENT: 'element' };
/** @type {?} */
var IS_ACTIVE_CLASS = 'adf-is-active';
var DiagramTooltipComponent = /** @class */ (function () {
    function DiagramTooltipComponent() {
        this.position = 'bottom';
        this.strategy = 'cursor';
    }
    /**
     * Set up event listeners for the target element (defined in the data.id)
     */
    /**
     * Set up event listeners for the target element (defined in the data.id)
     * @return {?}
     */
    DiagramTooltipComponent.prototype.ngAfterViewInit = /**
     * Set up event listeners for the target element (defined in the data.id)
     * @return {?}
     */
    function () {
        this.tooltipElement = this.tooltipContent.nativeElement;
        if (this.data.id) {
            this.targetElement = document.getElementById(this.data.id);
        }
        if (this.targetElement) {
            if (!this.targetElement.hasAttribute('tabindex')) {
                this.targetElement.setAttribute('tabindex', '0');
            }
            this.boundMouseEnterHandler = this.handleMouseEnter.bind(this);
            this.boundMouseLeaveAndScrollHandler = this.hideTooltip.bind(this);
            this.targetElement.addEventListener('mouseenter', this.boundMouseEnterHandler, false);
            this.targetElement.addEventListener('touchend', this.boundMouseEnterHandler, false);
            this.targetElement.addEventListener('mouseleave', this.boundMouseLeaveAndScrollHandler, false);
            window.addEventListener('scroll', this.boundMouseLeaveAndScrollHandler, true);
            window.addEventListener('touchstart', this.boundMouseLeaveAndScrollHandler);
        }
    };
    /**
     * Clear all bound event listeners
     */
    /**
     * Clear all bound event listeners
     * @return {?}
     */
    DiagramTooltipComponent.prototype.ngOnDestroy = /**
     * Clear all bound event listeners
     * @return {?}
     */
    function () {
        window.removeEventListener('scroll', this.boundMouseLeaveAndScrollHandler, true);
        window.removeEventListener('touchstart', this.boundMouseLeaveAndScrollHandler);
    };
    /**
     * Hides the tooltip
     */
    /**
     * Hides the tooltip
     * @private
     * @return {?}
     */
    DiagramTooltipComponent.prototype.hideTooltip = /**
     * Hides the tooltip
     * @private
     * @return {?}
     */
    function () {
        this.tooltipElement.classList.remove(IS_ACTIVE_CLASS);
    };
    /**
     * Shows the tooltip
     */
    /**
     * Shows the tooltip
     * @private
     * @return {?}
     */
    DiagramTooltipComponent.prototype.showTooltip = /**
     * Shows the tooltip
     * @private
     * @return {?}
     */
    function () {
        this.tooltipElement.classList.add(IS_ACTIVE_CLASS);
    };
    /**
     * Calculates the tooltip position and displays it
     *
     * @param event mouseenter/touchend event
     */
    /**
     * Calculates the tooltip position and displays it
     *
     * @private
     * @param {?} event mouseenter/touchend event
     * @return {?}
     */
    DiagramTooltipComponent.prototype.handleMouseEnter = /**
     * Calculates the tooltip position and displays it
     *
     * @private
     * @param {?} event mouseenter/touchend event
     * @return {?}
     */
    function (event) {
        /** @type {?} */
        var props;
        if (this.strategy === STRATEGY.ELEMENT) {
            props = event.target.getBoundingClientRect();
        }
        else {
            props = { top: (event.pageY - 150), left: event.pageX, width: event.layerX, height: 50 };
        }
        /** @type {?} */
        var top = props.top + (props.height / 2);
        /** @type {?} */
        var marginLeft = -1 * (this.tooltipElement.offsetWidth / 2);
        /** @type {?} */
        var marginTop = -1 * (this.tooltipElement.offsetHeight / 2);
        /** @type {?} */
        var left = props.left + (props.width / 2);
        if (this.position === POSITION.LEFT || this.position === POSITION.RIGHT) {
            left = (props.width / 2);
            if (top + marginTop < 0) {
                this.tooltipElement.style.top = '0';
                this.tooltipElement.style.marginTop = '0';
            }
            else {
                this.tooltipElement.style.top = top + 'px';
                this.tooltipElement.style.marginTop = marginTop + 'px';
            }
        }
        else {
            if (left + marginLeft < 0) {
                this.tooltipElement.style.left = '0';
                this.tooltipElement.style.marginLeft = '0';
            }
            else {
                this.tooltipElement.style.left = left + 'px';
                this.tooltipElement.style.marginLeft = marginLeft + 'px';
            }
        }
        if (this.position === POSITION.TOP) {
            this.tooltipElement.style.top = props.top - this.tooltipElement.offsetHeight - 10 + 'px';
        }
        else if (this.position === POSITION.RIGHT) {
            this.tooltipElement.style.left = props.left + props.width + 10 + 'px';
        }
        else if (this.position === POSITION.LEFT) {
            this.tooltipElement.style.left = props.left - this.tooltipElement.offsetWidth - 10 + 'px';
        }
        else {
            this.tooltipElement.style.top = props.top + props.height + 10 + 'px';
        }
        this.showTooltip();
    };
    DiagramTooltipComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-tooltip',
                    template: "<div #tooltipContent class=\"adf-diagram-tooltip\">\n    <div class=\"adf-diagram-tooltip-header\">{{ data.type }} {{ data.name || data.id }}</div>\n    <div class=\"adf-diagram-tooltip-body\">\n        <div *ngIf=\"data.value\" class=\"adf-diagram-heat-value\">\n            <span class=\"adf-diagram-value\">{{ data.value }}</span>\n            <span class=\"adf-diagram-valuetype\"> {{ data.dataType }}</span>\n        </div>\n        <div *ngIf=\"data.name\" class=\"adf-diagram-name-property\">\n            <span class=\"adf-diagram-propertyName\">Name:</span>\n            <span class=\"adf-diagram-propertyValue\">{{ data.name }}</span>\n        </div>\n        <div *ngFor=\"let property of data.properties\" class=\"adf-diagram-general-property\">\n            <span class=\"adf-diagram-propertyName\">{{ property.name }}:</span>\n            <span class=\"adf-diagram-propertyValue\">{{ property.value }}</span>\n        </div>\n    </div>\n</div>\n",
                    styles: [""]
                }] }
    ];
    DiagramTooltipComponent.propDecorators = {
        tooltipContent: [{ type: ViewChild, args: ['tooltipContent',] }],
        data: [{ type: Input }],
        position: [{ type: Input }],
        strategy: [{ type: Input }]
    };
    return DiagramTooltipComponent;
}());
export { DiagramTooltipComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DiagramTooltipComponent.prototype.tooltipElement;
    /**
     * @type {?}
     * @private
     */
    DiagramTooltipComponent.prototype.targetElement;
    /**
     * @type {?}
     * @private
     */
    DiagramTooltipComponent.prototype.boundMouseEnterHandler;
    /**
     * @type {?}
     * @private
     */
    DiagramTooltipComponent.prototype.boundMouseLeaveAndScrollHandler;
    /** @type {?} */
    DiagramTooltipComponent.prototype.tooltipContent;
    /** @type {?} */
    DiagramTooltipComponent.prototype.data;
    /** @type {?} */
    DiagramTooltipComponent.prototype.position;
    /** @type {?} */
    DiagramTooltipComponent.prototype.strategy;
}
