// @alfresco/adf-insights/esm5/diagram/models/diagram/diagram.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { DiagramElementModel } from './diagramElement.model';
import { DiagramFlowElementModel } from './diagramFlowElement.model';
import { DiagramPoolElementModel } from './diagramPoolElement.model';
var DiagramModel = /** @class */ (function () {
    function DiagramModel(obj) {
        var _this = this;
        this.elements = [];
        this.flows = [];
        this.pools = [];
        if (obj) {
            this.diagramBeginX = obj.diagramBeginX;
            this.diagramBeginY = obj.diagramBeginY;
            this.diagramHeight = obj.diagramHeight;
            this.diagramWidth = obj.diagramWidth;
            if (obj.elements) {
                obj.elements.forEach((/**
                 * @param {?} element
                 * @return {?}
                 */
                function (element) {
                    _this.elements.push(new DiagramElementModel(element));
                }));
            }
            if (obj.flows) {
                obj.flows.forEach((/**
                 * @param {?} flow
                 * @return {?}
                 */
                function (flow) {
                    _this.flows.push(new DiagramFlowElementModel(flow));
                }));
            }
            if (obj.pools) {
                obj.pools.forEach((/**
                 * @param {?} pool
                 * @return {?}
                 */
                function (pool) {
                    _this.pools.push(new DiagramPoolElementModel(pool));
                }));
            }
        }
    }
    return DiagramModel;
}());
export { DiagramModel };
if (false) {
    /** @type {?} */
    DiagramModel.prototype.diagramBeginX;
    /** @type {?} */
    DiagramModel.prototype.diagramBeginY;
    /** @type {?} */
    DiagramModel.prototype.diagramHeight;
    /** @type {?} */
    DiagramModel.prototype.diagramWidth;
    /** @type {?} */
    DiagramModel.prototype.elements;
    /** @type {?} */
    DiagramModel.prototype.flows;
    /** @type {?} */
    DiagramModel.prototype.pools;
}
