// @alfresco/adf-insights/esm5/analytics-process/components/widgets/duration/duration.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-access-missing-member no-input-rename   */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { ParameterValueModel } from '../../../../diagram/models/report/parameterValue.model';
import { ReportParameterDetailsModel } from '../../../../diagram/models/report/reportParameterDetails.model';
import { NumberWidgetAnalyticsComponent } from './../number/number.widget';
var DurationWidgetComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DurationWidgetComponent, _super);
    function DurationWidgetComponent(elementRef) {
        var _this = _super.call(this, elementRef) || this;
        _this.elementRef = elementRef;
        _this.required = false;
        return _this;
    }
    /**
     * @return {?}
     */
    DurationWidgetComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var timeType = new FormControl();
        this.formGroup.addControl('timeType', timeType);
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(Validators.required);
        }
        if (this.field.value === null) {
            this.field.value = 0;
        }
        /** @type {?} */
        var paramOptions = [];
        paramOptions.push(new ParameterValueModel({ id: '1', name: 'Seconds' }));
        paramOptions.push(new ParameterValueModel({ id: '60', name: 'Minutes' }));
        paramOptions.push(new ParameterValueModel({ id: '3600', name: 'Hours' }));
        paramOptions.push(new ParameterValueModel({ id: '86400', name: 'Days', selected: true }));
        this.duration = new ReportParameterDetailsModel({ id: 'duration', name: 'duration', options: paramOptions });
        this.duration.value = paramOptions[0].id;
    };
    /**
     * @return {?}
     */
    DurationWidgetComponent.prototype.calculateDuration = /**
     * @return {?}
     */
    function () {
        if (this.field && this.duration.value) {
            this.currentValue = parseInt(this.field.value, 10) * parseInt(this.duration.value, 10);
            this.formGroup.get(this.controllerName).setValue(this.currentValue);
            this.fieldChanged.emit({ value: this.currentValue });
        }
    };
    DurationWidgetComponent.decorators = [
        { type: Component, args: [{
                    selector: 'duration-widget',
                    template: "<div class=\"adf-duration-widget-grid\">\n    <div class=\"adf-duration-widget-cell\">\n        <div class=\"adf-number-widget\">\n            <mat-form-field class=\"example-full-width\" floatPlaceholder=\"always\">\n                <input matInput\n                    placeholder=\"{{field.nameKey | translate}}\"\n                    type=\"text\"\n                    pattern=\"-?[0-9]*(\\.[0-9]+)?\"\n                    [id]=\"field.id\"\n                    [value]=\"field.value\"\n                    [(ngModel)]=\"field.value\"\n                    (ngModelChange)=\"calculateDuration()\">\n             </mat-form-field>\n        </div>\n    </div>\n    <div class=\"adf-duration-widget-cell\">\n        <div class=\"adf-dropdown-container\">\n            <analytics-dropdown-widget  [field]=\"duration\" [group]=\"formGroup\" [controllerName]=\"'timeType'\"\n                             [showDefaultOption]=\"false\"\n                             (fieldChanged)=\"calculateDuration()\"></analytics-dropdown-widget>\n        </div>\n    </div>\n</div>\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-duration-widget-grid{display:flex;justify-content:space-between}.adf-duration-widget-cell{width:49%}.adf-number-widget,.adf-number-widget .mat-form-field{width:100%}.adf-dropdown-container{margin-top:30px}"]
                }] }
    ];
    /** @nocollapse */
    DurationWidgetComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DurationWidgetComponent.propDecorators = {
        field: [{ type: Input }],
        formGroup: [{ type: Input, args: ['group',] }],
        controllerName: [{ type: Input, args: ['controllerName',] }],
        required: [{ type: Input }]
    };
    return DurationWidgetComponent;
}(NumberWidgetAnalyticsComponent));
export { DurationWidgetComponent };
if (false) {
    /** @type {?} */
    DurationWidgetComponent.prototype.field;
    /** @type {?} */
    DurationWidgetComponent.prototype.formGroup;
    /** @type {?} */
    DurationWidgetComponent.prototype.controllerName;
    /** @type {?} */
    DurationWidgetComponent.prototype.required;
    /** @type {?} */
    DurationWidgetComponent.prototype.duration;
    /** @type {?} */
    DurationWidgetComponent.prototype.currentValue;
    /** @type {?} */
    DurationWidgetComponent.prototype.selectionGroup;
    /** @type {?} */
    DurationWidgetComponent.prototype.elementRef;
}
