// @alfresco/adf-core/esm5/viewer/components/mediaPlayer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { ContentService } from '../../services/content.service';
var MediaPlayerComponent = /** @class */ (function () {
    function MediaPlayerComponent(contentService) {
        this.contentService = contentService;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    MediaPlayerComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        /** @type {?} */
        var blobFile = changes['blobFile'];
        if (blobFile && blobFile.currentValue) {
            this.urlFile = this.contentService.createTrustedUrl(this.blobFile);
            return;
        }
        if (!this.urlFile && !this.blobFile) {
            throw new Error('Attribute urlFile or blobFile is required');
        }
    };
    MediaPlayerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-media-player',
                    template: "<video controls>\n    <source [src]=\"urlFile\" [type]=\"mimeType\" />\n</video>\n",
                    host: { 'class': 'adf-media-player' },
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-media-player{display:flex}.adf-media-player video{display:flex;flex:1;max-height:90vh;max-width:100%}"]
                }] }
    ];
    /** @nocollapse */
    MediaPlayerComponent.ctorParameters = function () { return [
        { type: ContentService }
    ]; };
    MediaPlayerComponent.propDecorators = {
        urlFile: [{ type: Input }],
        blobFile: [{ type: Input }],
        mimeType: [{ type: Input }],
        nameFile: [{ type: Input }]
    };
    return MediaPlayerComponent;
}());
export { MediaPlayerComponent };
if (false) {
    /** @type {?} */
    MediaPlayerComponent.prototype.urlFile;
    /** @type {?} */
    MediaPlayerComponent.prototype.blobFile;
    /** @type {?} */
    MediaPlayerComponent.prototype.mimeType;
    /** @type {?} */
    MediaPlayerComponent.prototype.nameFile;
    /**
     * @type {?}
     * @private
     */
    MediaPlayerComponent.prototype.contentService;
}
