// @alfresco/adf-core/esm5/viewer/components/pdfViewer-thumb.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var PdfThumbComponent = /** @class */ (function () {
    function PdfThumbComponent(sanitizer) {
        this.sanitizer = sanitizer;
        this.page = null;
    }
    /**
     * @return {?}
     */
    PdfThumbComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.image$ = this.page.getPage().then((/**
         * @param {?} page
         * @return {?}
         */
        function (page) { return _this.getThumb(page); }));
    };
    /**
     * @private
     * @param {?} page
     * @return {?}
     */
    PdfThumbComponent.prototype.getThumb = /**
     * @private
     * @param {?} page
     * @return {?}
     */
    function (page) {
        var _this = this;
        /** @type {?} */
        var viewport = page.getViewport(1);
        /** @type {?} */
        var pageRatio = viewport.width / viewport.height;
        /** @type {?} */
        var canvas = this.getCanvas(pageRatio);
        /** @type {?} */
        var scale = Math.min((canvas.height / viewport.height), (canvas.width / viewport.width));
        return page.render({
            canvasContext: canvas.getContext('2d'),
            viewport: page.getViewport(scale)
        })
            .then((/**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var imageSource = canvas.toDataURL();
            return _this.sanitizer.bypassSecurityTrustUrl(imageSource);
        }));
    };
    /**
     * @private
     * @param {?} pageRatio
     * @return {?}
     */
    PdfThumbComponent.prototype.getCanvas = /**
     * @private
     * @param {?} pageRatio
     * @return {?}
     */
    function (pageRatio) {
        /** @type {?} */
        var canvas = document.createElement('canvas');
        canvas.width = this.page.getWidth();
        canvas.height = this.page.getHeight();
        return canvas;
    };
    PdfThumbComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-pdf-thumb',
                    template: "<ng-container *ngIf=\"image$ | async as image\">\n    <img [src]=\"image\" [alt]=\"'ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE' | translate: { pageNum: page.id }\"\n        title=\"{{ 'ADF_VIEWER.SIDEBAR.THUMBNAILS.PAGE' | translate: { pageNum: page.id } }}\">\n</ng-container>\n",
                    encapsulation: ViewEncapsulation.None
                }] }
    ];
    /** @nocollapse */
    PdfThumbComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    PdfThumbComponent.propDecorators = {
        page: [{ type: Input }]
    };
    return PdfThumbComponent;
}());
export { PdfThumbComponent };
if (false) {
    /** @type {?} */
    PdfThumbComponent.prototype.page;
    /** @type {?} */
    PdfThumbComponent.prototype.image$;
    /**
     * @type {?}
     * @private
     */
    PdfThumbComponent.prototype.sanitizer;
}
