// @alfresco/adf-core/esm5/viewer/components/pdfViewer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, TemplateRef, HostListener, Output, Input, ViewEncapsulation, EventEmitter } from '@angular/core';
import { MatDialog } from '@angular/material';
import { LogService } from '../../services/log.service';
import { RenderingQueueServices } from '../services/rendering-queue.services';
import { PdfPasswordDialogComponent } from './pdfViewer-password-dialog';
import { AppConfigService } from './../../app-config/app-config.service';
/**
 * @record
 */
export function PdfDocumentOptions() { }
if (false) {
    /** @type {?|undefined} */
    PdfDocumentOptions.prototype.url;
    /** @type {?|undefined} */
    PdfDocumentOptions.prototype.data;
    /** @type {?|undefined} */
    PdfDocumentOptions.prototype.withCredentials;
}
var PdfViewerComponent = /** @class */ (function () {
    function PdfViewerComponent(dialog, renderingQueueServices, logService, appConfigService) {
        this.dialog = dialog;
        this.renderingQueueServices = renderingQueueServices;
        this.logService = logService;
        this.appConfigService = appConfigService;
        this.showToolbar = true;
        this.allowThumbnails = false;
        this.thumbnailsTemplate = null;
        this.rendered = new EventEmitter();
        this.error = new EventEmitter();
        this.currentScaleMode = 'auto';
        this.currentScale = 1;
        this.MAX_AUTO_SCALE = 1.25;
        this.DEFAULT_SCALE_DELTA = 1.1;
        this.MIN_SCALE = 0.25;
        this.MAX_SCALE = 10.0;
        this.isPanelDisabled = true;
        this.showThumbnails = false;
        this.pdfThumbnailsContext = { viewer: null };
        // needed to preserve "this" context
        this.onPageChange = this.onPageChange.bind(this);
        this.onPagesLoaded = this.onPagesLoaded.bind(this);
        this.onPageRendered = this.onPageRendered.bind(this);
        this.randomPdfId = this.generateUuid();
        this.currentScale = this.getUserScaling();
    }
    Object.defineProperty(PdfViewerComponent.prototype, "currentScaleText", {
        get: /**
         * @return {?}
         */
        function () {
            return Math.round(this.currentScale * 100) + '%';
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.getUserScaling = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var scaleConfig = this.appConfigService.get('adf-viewer.pdf-viewer-scaling', undefined) / 100;
        if (scaleConfig) {
            return this.checkLimits(scaleConfig);
        }
        else {
            return 1;
        }
    };
    /**
     * @param {?} scaleConfig
     * @return {?}
     */
    PdfViewerComponent.prototype.checkLimits = /**
     * @param {?} scaleConfig
     * @return {?}
     */
    function (scaleConfig) {
        if (scaleConfig > this.MAX_SCALE) {
            return this.MAX_SCALE;
        }
        else if (scaleConfig < this.MIN_SCALE) {
            return this.MIN_SCALE;
        }
        else {
            return scaleConfig;
        }
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    PdfViewerComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        var _this = this;
        /** @type {?} */
        var blobFile = changes['blobFile'];
        if (blobFile && blobFile.currentValue) {
            /** @type {?} */
            var reader_1 = new FileReader();
            reader_1.onload = (/**
             * @return {?}
             */
            function () {
                /** @type {?} */
                var options = {
                    data: reader_1.result,
                    withCredentials: _this.appConfigService.get('auth.withCredentials', undefined)
                };
                _this.executePdf(options);
            });
            reader_1.readAsArrayBuffer(blobFile.currentValue);
        }
        /** @type {?} */
        var urlFile = changes['urlFile'];
        if (urlFile && urlFile.currentValue) {
            /** @type {?} */
            var options = {
                url: urlFile.currentValue,
                withCredentials: this.appConfigService.get('auth.withCredentials', undefined)
            };
            this.executePdf(options);
        }
        if (!this.urlFile && !this.blobFile) {
            throw new Error('Attribute urlFile or blobFile is required');
        }
    };
    /**
     * @param {?} pdfOptions
     * @return {?}
     */
    PdfViewerComponent.prototype.executePdf = /**
     * @param {?} pdfOptions
     * @return {?}
     */
    function (pdfOptions) {
        var _this = this;
        pdfjsLib.GlobalWorkerOptions.workerSrc = 'pdf.worker.min.js';
        this.loadingTask = pdfjsLib.getDocument(pdfOptions);
        this.loadingTask.onPassword = (/**
         * @param {?} callback
         * @param {?} reason
         * @return {?}
         */
        function (callback, reason) {
            _this.onPdfPassword(callback, reason);
        });
        this.loadingTask.onProgress = (/**
         * @param {?} progressData
         * @return {?}
         */
        function (progressData) {
            /** @type {?} */
            var level = progressData.loaded / progressData.total;
            _this.loadingPercent = Math.round(level * 100);
        });
        this.loadingTask.then((/**
         * @param {?} pdfDocument
         * @return {?}
         */
        function (pdfDocument) {
            _this.currentPdfDocument = pdfDocument;
            _this.totalPages = pdfDocument.numPages;
            _this.page = 1;
            _this.displayPage = 1;
            _this.initPDFViewer(_this.currentPdfDocument);
            _this.currentPdfDocument.getPage(1).then((/**
             * @return {?}
             */
            function () {
                _this.scalePage('auto');
            }), (/**
             * @param {?} error
             * @return {?}
             */
            function (error) {
                _this.error.emit();
            }));
        }), (/**
         * @param {?} error
         * @return {?}
         */
        function (error) {
            _this.error.emit();
        }));
    };
    /**
     * @param {?} pdfDocument
     * @return {?}
     */
    PdfViewerComponent.prototype.initPDFViewer = /**
     * @param {?} pdfDocument
     * @return {?}
     */
    function (pdfDocument) {
        /** @type {?} */
        var viewer = document.getElementById(this.randomPdfId + "-viewer-viewerPdf");
        /** @type {?} */
        var container = document.getElementById(this.randomPdfId + "-viewer-pdf-viewer");
        if (viewer && container) {
            this.documentContainer = container;
            // cspell: disable-next
            this.documentContainer.addEventListener('pagechange', this.onPageChange, true);
            // cspell: disable-next
            this.documentContainer.addEventListener('pagesloaded', this.onPagesLoaded, true);
            // cspell: disable-next
            this.documentContainer.addEventListener('textlayerrendered', this.onPageRendered, true);
            this.pdfViewer = new pdfjsViewer.PDFViewer({
                container: this.documentContainer,
                viewer: viewer,
                renderingQueue: this.renderingQueueServices
            });
            this.renderingQueueServices.setViewer(this.pdfViewer);
            this.pdfViewer.setDocument(pdfDocument);
            this.pdfThumbnailsContext.viewer = this.pdfViewer;
        }
    };
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.documentContainer) {
            // cspell: disable-next
            this.documentContainer.removeEventListener('pagechange', this.onPageChange, true);
            // cspell: disable-next
            this.documentContainer.removeEventListener('pagesloaded', this.onPagesLoaded, true);
            // cspell: disable-next
            this.documentContainer.removeEventListener('textlayerrendered', this.onPageRendered, true);
        }
        if (this.loadingTask) {
            try {
                this.loadingTask.destroy();
            }
            catch (_a) {
            }
            this.loadingTask = null;
        }
    };
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.toggleThumbnails = /**
     * @return {?}
     */
    function () {
        this.showThumbnails = !this.showThumbnails;
    };
    /**
     * Method to scale the page current support implementation
     *
     * @param scaleMode - new scale mode
     */
    /**
     * Method to scale the page current support implementation
     *
     * @param {?} scaleMode - new scale mode
     * @return {?}
     */
    PdfViewerComponent.prototype.scalePage = /**
     * Method to scale the page current support implementation
     *
     * @param {?} scaleMode - new scale mode
     * @return {?}
     */
    function (scaleMode) {
        this.currentScaleMode = scaleMode;
        /** @type {?} */
        var viewerContainer = document.getElementById(this.randomPdfId + "-viewer-main-container");
        /** @type {?} */
        var documentContainer = document.getElementById(this.randomPdfId + "-viewer-pdf-viewer");
        if (this.pdfViewer && documentContainer) {
            /** @type {?} */
            var widthContainer = void 0;
            /** @type {?} */
            var heightContainer = void 0;
            if (viewerContainer && viewerContainer.clientWidth <= documentContainer.clientWidth) {
                widthContainer = viewerContainer.clientWidth;
                heightContainer = viewerContainer.clientHeight;
            }
            else {
                widthContainer = documentContainer.clientWidth;
                heightContainer = documentContainer.clientHeight;
            }
            /** @type {?} */
            var currentPage = this.pdfViewer._pages[this.pdfViewer._currentPageNumber - 1];
            /** @type {?} */
            var padding = 20;
            /** @type {?} */
            var pageWidthScale = (widthContainer - padding) / currentPage.width * currentPage.scale;
            /** @type {?} */
            var pageHeightScale = (heightContainer - padding) / currentPage.width * currentPage.scale;
            /** @type {?} */
            var scale = this.getUserScaling();
            if (!scale) {
                switch (this.currentScaleMode) {
                    case 'page-actual':
                        scale = 1;
                        break;
                    case 'page-width':
                        scale = pageWidthScale;
                        break;
                    case 'page-height':
                        scale = pageHeightScale;
                        break;
                    case 'page-fit':
                        scale = Math.min(pageWidthScale, pageHeightScale);
                        break;
                    case 'auto':
                        /** @type {?} */
                        var horizontalScale = void 0;
                        if (this.isLandscape) {
                            horizontalScale = Math.min(pageHeightScale, pageWidthScale);
                        }
                        else {
                            horizontalScale = pageWidthScale;
                        }
                        horizontalScale = Math.round(horizontalScale);
                        scale = Math.min(this.MAX_AUTO_SCALE, horizontalScale);
                        break;
                    default:
                        this.logService.error('pdfViewSetScale: \'' + scaleMode + '\' is an unknown zoom value.');
                        return;
                }
                this.setScaleUpdatePages(scale);
            }
            else {
                this.currentScale = 0;
                this.setScaleUpdatePages(scale);
            }
        }
    };
    /**
     * Update all the pages with the newScale scale
     *
     * @param newScale - new scale page
     */
    /**
     * Update all the pages with the newScale scale
     *
     * @param {?} newScale - new scale page
     * @return {?}
     */
    PdfViewerComponent.prototype.setScaleUpdatePages = /**
     * Update all the pages with the newScale scale
     *
     * @param {?} newScale - new scale page
     * @return {?}
     */
    function (newScale) {
        if (this.pdfViewer) {
            if (!this.isSameScale(this.currentScale, newScale)) {
                this.currentScale = newScale;
                this.pdfViewer._pages.forEach((/**
                 * @param {?} currentPage
                 * @return {?}
                 */
                function (currentPage) {
                    currentPage.update(newScale);
                }));
            }
            this.pdfViewer.update();
        }
    };
    /**
     * Check if the request scale of the page is the same for avoid useless re-rendering
     *
     * @param oldScale - old scale page
     * @param newScale - new scale page
     *
     */
    /**
     * Check if the request scale of the page is the same for avoid useless re-rendering
     *
     * @param {?} oldScale - old scale page
     * @param {?} newScale - new scale page
     *
     * @return {?}
     */
    PdfViewerComponent.prototype.isSameScale = /**
     * Check if the request scale of the page is the same for avoid useless re-rendering
     *
     * @param {?} oldScale - old scale page
     * @param {?} newScale - new scale page
     *
     * @return {?}
     */
    function (oldScale, newScale) {
        return (newScale === oldScale);
    };
    /**
     * Check if is a land scape view
     *
     * @param width
     * @param height
     */
    /**
     * Check if is a land scape view
     *
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    PdfViewerComponent.prototype.isLandscape = /**
     * Check if is a land scape view
     *
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    function (width, height) {
        return (width > height);
    };
    /**
     * Method triggered when the page is resized
     */
    /**
     * Method triggered when the page is resized
     * @return {?}
     */
    PdfViewerComponent.prototype.onResize = /**
     * Method triggered when the page is resized
     * @return {?}
     */
    function () {
        this.scalePage(this.currentScaleMode);
    };
    /**
     * toggle the fit page pdf
     */
    /**
     * toggle the fit page pdf
     * @return {?}
     */
    PdfViewerComponent.prototype.pageFit = /**
     * toggle the fit page pdf
     * @return {?}
     */
    function () {
        if (this.currentScaleMode !== 'page-fit') {
            this.scalePage('page-fit');
        }
        else {
            this.scalePage('auto');
        }
    };
    /**
     * zoom in page pdf
     *
     * @param ticks
     */
    /**
     * zoom in page pdf
     *
     * @param {?=} ticks
     * @return {?}
     */
    PdfViewerComponent.prototype.zoomIn = /**
     * zoom in page pdf
     *
     * @param {?=} ticks
     * @return {?}
     */
    function (ticks) {
        /** @type {?} */
        var newScale = this.currentScale;
        do {
            newScale = (newScale * this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.ceil(newScale * 10) / 10;
            newScale = Math.min(this.MAX_SCALE, newScale);
        } while (--ticks > 0 && newScale < this.MAX_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    };
    /**
     * zoom out page pdf
     *
     * @param ticks
     */
    /**
     * zoom out page pdf
     *
     * @param {?=} ticks
     * @return {?}
     */
    PdfViewerComponent.prototype.zoomOut = /**
     * zoom out page pdf
     *
     * @param {?=} ticks
     * @return {?}
     */
    function (ticks) {
        /** @type {?} */
        var newScale = this.currentScale;
        do {
            newScale = (newScale / this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.floor(newScale * 10) / 10;
            newScale = Math.max(this.MIN_SCALE, newScale);
        } while (--ticks > 0 && newScale > this.MIN_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    };
    /**
     * load the previous page
     */
    /**
     * load the previous page
     * @return {?}
     */
    PdfViewerComponent.prototype.previousPage = /**
     * load the previous page
     * @return {?}
     */
    function () {
        if (this.pdfViewer && this.page > 1) {
            this.page--;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    };
    /**
     * load the next page
     */
    /**
     * load the next page
     * @return {?}
     */
    PdfViewerComponent.prototype.nextPage = /**
     * load the next page
     * @return {?}
     */
    function () {
        if (this.pdfViewer && this.page < this.totalPages) {
            this.page++;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    };
    /**
     * load the page in input
     *
     * @param page to load
     */
    /**
     * load the page in input
     *
     * @param {?} page to load
     * @return {?}
     */
    PdfViewerComponent.prototype.inputPage = /**
     * load the page in input
     *
     * @param {?} page to load
     * @return {?}
     */
    function (page) {
        /** @type {?} */
        var pageInput = parseInt(page, 10);
        if (!isNaN(pageInput) && pageInput > 0 && pageInput <= this.totalPages) {
            this.page = pageInput;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
        else {
            this.displayPage = this.page;
        }
    };
    /**
     * Page Change Event
     *
     * @param event
     */
    /**
     * Page Change Event
     *
     * @param {?} event
     * @return {?}
     */
    PdfViewerComponent.prototype.onPageChange = /**
     * Page Change Event
     *
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.page = event.pageNumber;
        this.displayPage = event.pageNumber;
    };
    /**
     * @param {?} callback
     * @param {?} reason
     * @return {?}
     */
    PdfViewerComponent.prototype.onPdfPassword = /**
     * @param {?} callback
     * @param {?} reason
     * @return {?}
     */
    function (callback, reason) {
        this.dialog
            .open(PdfPasswordDialogComponent, {
            width: '400px',
            disableClose: true,
            data: { reason: reason }
        })
            .afterClosed().subscribe((/**
         * @param {?} password
         * @return {?}
         */
        function (password) {
            if (password) {
                callback(password);
            }
        }));
    };
    /**
     * Page Rendered Event
     */
    /**
     * Page Rendered Event
     * @return {?}
     */
    PdfViewerComponent.prototype.onPageRendered = /**
     * Page Rendered Event
     * @return {?}
     */
    function () {
        this.rendered.emit();
    };
    /**
     * Pages Loaded Event
     *
     * @param event
     */
    /**
     * Pages Loaded Event
     *
     * @param {?} event
     * @return {?}
     */
    PdfViewerComponent.prototype.onPagesLoaded = /**
     * Pages Loaded Event
     *
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.isPanelDisabled = false;
    };
    /**
     * Keyboard Event Listener
     * @param KeyboardEvent event
     */
    /**
     * Keyboard Event Listener
     * @param {?} event
     * @return {?}
     */
    PdfViewerComponent.prototype.handleKeyboardEvent = /**
     * Keyboard Event Listener
     * @param {?} event
     * @return {?}
     */
    function (event) {
        /** @type {?} */
        var key = event.keyCode;
        if (key === 39) { // right arrow
            this.nextPage();
        }
        else if (key === 37) { // left arrow
            this.previousPage();
        }
    };
    /**
     * @private
     * @return {?}
     */
    PdfViewerComponent.prototype.generateUuid = /**
     * @private
     * @return {?}
     */
    function () {
        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (/**
         * @param {?} c
         * @return {?}
         */
        function (c) {
            /** @type {?} */
            var r = Math.random() * 16 | 0;
            /** @type {?} */
            var v = c === 'x' ? r : (r & 0x3 | 0x8);
            return v.toString(16);
        }));
    };
    PdfViewerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-pdf-viewer',
                    template: "<div class=\"adf-pdf-viewer__container\">\n    <ng-container *ngIf=\"showThumbnails\">\n        <div class=\"adf-pdf-viewer__thumbnails\">\n            <div class=\"adf-thumbnails-template__container\">\n                <div class=\"adf-thumbnails-template__buttons\">\n                    <button mat-icon-button data-automation-id='adf-thumbnails-close' (click)=\"toggleThumbnails()\">\n                        <mat-icon>close</mat-icon>\n                    </button>\n                </div>\n\n                <ng-container *ngIf=\"thumbnailsTemplate\">\n                    <ng-container *ngTemplateOutlet=\"thumbnailsTemplate;context:pdfThumbnailsContext\"></ng-container>\n                </ng-container>\n\n                <ng-container *ngIf=\"!thumbnailsTemplate\">\n                    <adf-pdf-thumbnails [pdfViewer]=\"pdfViewer\"></adf-pdf-thumbnails>\n                </ng-container>\n            </div>\n        </div>\n    </ng-container>\n\n    <div class=\"adf-pdf-viewer__content\">\n        <div [id]=\"randomPdfId+'-viewer-pdf-viewer'\" class=\"adf-viewer-pdf-viewer\" (window:resize)=\"onResize()\">\n            <div [id]=\"randomPdfId+'-viewer-viewerPdf'\" class=\"adf-pdfViewer\" role=\"document\" tabindex=\"0\" aria-expanded=\"true\">\n                <div id=\"loader-container\" class=\"adf-loader-container\">\n                    <div class=\"adf-loader-item\">\n                        <mat-progress-bar mode=\"indeterminate\"></mat-progress-bar>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </div>\n</div>\n\n<div class=\"adf-pdf-viewer__toolbar\" *ngIf=\"showToolbar\">\n    <adf-toolbar>\n\n        <ng-container *ngIf=\"allowThumbnails\">\n            <button mat-icon-button\n                    data-automation-id=\"adf-thumbnails-button\"\n                    [disabled]=\"isPanelDisabled\"\n                    (click)=\"toggleThumbnails()\">\n                <mat-icon>dashboard</mat-icon>\n            </button>\n            <adf-toolbar-divider></adf-toolbar-divider>\n        </ng-container>\n\n        <button\n            id=\"viewer-previous-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.PREVIOUS_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.PREVIOUS_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"previousPage()\">\n            <mat-icon>keyboard_arrow_up</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-next-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.NEXT_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.NEXT_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"nextPage()\">\n            <mat-icon>keyboard_arrow_down</mat-icon>\n        </button>\n\n        <div class=\"adf-pdf-viewer__toolbar-page-selector\">\n            <span>{{ 'ADF_VIEWER.PAGE_LABEL.SHOWING' | translate }}</span>\n            <input #page\n                   type=\"text\"\n                   data-automation-id=\"adf-page-selector\"\n                   pattern=\"-?[0-9]*(\\.[0-9]+)?\"\n                   value=\"{{ displayPage }}\"\n                   (keyup.enter)=\"inputPage(page.value)\">\n            <span>{{ 'ADF_VIEWER.PAGE_LABEL.OF' | translate }} {{ totalPages }}</span>\n        </div>\n\n        <div class=\"adf-viewer__toolbar-page-scale\" data-automation-id=\"adf-page-scale\">\n            {{ currentScaleText }}\n        </div>\n\n        <button\n            id=\"viewer-zoom-in-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.ZOOM_IN' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.ZOOM_IN' | translate }}\"\n            mat-icon-button\n            (click)=\"zoomIn()\">\n            <mat-icon>zoom_in</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-zoom-out-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.ZOOM_OUT' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.ZOOM_OUT' | translate }}\"\n            mat-icon-button\n            (click)=\"zoomOut()\">\n            <mat-icon>zoom_out</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-scale-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.FIT_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.FIT_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"pageFit()\">\n            <mat-icon>zoom_out_map</mat-icon>\n        </button>\n\n    </adf-toolbar>\n</div>\n",
                    providers: [RenderingQueueServices],
                    host: { 'class': 'adf-pdf-viewer' },
                    encapsulation: ViewEncapsulation.None,
                    styles: ["", ".adf-pdf-viewer .textLayer{position:absolute;left:0;top:0;right:0;bottom:0;overflow:hidden;opacity:.2;line-height:1;border:1px solid gray}.adf-pdf-viewer .textLayer>div{color:transparent;position:absolute;white-space:pre;cursor:text;-webkit-transform-origin:0 0;transform-origin:0 0}.adf-pdf-viewer .textLayer .adf-highlight{margin:-1px;padding:1px;background-color:#b400aa;border-radius:4px}.adf-pdf-viewer .textLayer .adf-highlight.adf-begin{border-radius:4px 0 0 4px}.adf-pdf-viewer .textLayer .adf-highlight.adf-end{border-radius:0 4px 4px 0}.adf-pdf-viewer .textLayer .adf-highlight.adf-middle{border-radius:0}.adf-pdf-viewer .textLayer .adf-highlight.adf-selected{background-color:#006400}.adf-pdf-viewer .textLayer::selection{background:#00f}.adf-pdf-viewer .textLayer::-moz-selection{background:#00f}.adf-pdf-viewer .textLayer .adf-endOfContent{display:block;position:absolute;left:0;top:100%;right:0;bottom:0;z-index:-1;cursor:default;user-select:none;-webkit-user-select:none;-ms-user-select:none;-moz-user-select:none}.adf-pdf-viewer .textLayer .adf-endOfContent.adf-active{top:0}.adf-pdf-viewer .adf-annotationLayer section{position:absolute}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a{position:absolute;font-size:1em;top:0;left:0;width:100%;height:100%;background:url(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a:hover{opacity:.2;background:#ff0;box-shadow:0 2px 10px #ff0}.adf-pdf-viewer .adf-annotationLayer .adf-textAnnotation img{position:absolute;cursor:pointer}.adf-pdf-viewer .adf-annotationLayer .adf-popupWrapper{position:absolute;width:20em}.adf-pdf-viewer .adf-annotationLayer .adf-popup{position:absolute;z-index:200;max-width:20em;background-color:#ff9;box-shadow:0 2px 5px #333;border-radius:2px;padding:.6em;margin-left:5px;cursor:pointer;word-wrap:break-word}.adf-pdf-viewer .adf-annotationLayer .adf-popup h1{font-size:1em;border-bottom:1px solid #000;padding-bottom:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-popup p{padding-top:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-fileAttachmentAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-highlightAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-squigglyAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-strikeoutAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-underlineAnnotation{cursor:pointer}.adf-pdf-viewer .adf-pdfViewer .canvasWrapper{overflow:hidden}.adf-pdf-viewer .adf-pdfViewer .page{direction:ltr;width:816px;height:1056px;margin:1px auto -8px;position:relative;overflow:visible;border:9px solid transparent;background-clip:content-box;background-color:#fff}.adf-pdf-viewer .adf-pdfViewer .page canvas{margin:0;display:block}.adf-pdf-viewer .adf-pdfViewer .page .adf-loadingIcon{position:absolute;display:block;left:0;top:0;right:0;bottom:0}.adf-pdf-viewer .adf-pdfViewer .page *{padding:0;margin:0}.adf-pdf-viewer .adf-pdfViewer.adf-removePageBorders .adf-page{margin:0 auto 10px;border:none}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon{width:100px;height:100px;left:50%!important;top:50%!important;margin-top:-50px;margin-left:-50px;font-size:5px;text-indent:-9999em;border-top:1.1em solid rgba(3,0,2,.2);border-right:1.1em solid rgba(3,0,2,.2);border-bottom:1.1em solid rgba(3,0,2,.2);border-left:1.1em solid #030002;-webkit-transform:translateZ(0);transform:translateZ(0);-webkit-animation:1.1s linear infinite load8;animation:1.1s linear infinite load8;border-radius:50%}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon::after{border-radius:50%}.adf-pdf-viewer .adf-hidden,.adf-pdf-viewer [hidden]{display:none!important}@-webkit-keyframes load8{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes load8{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}.adf-viewer-pdf-viewer{overflow:auto;-webkit-overflow-scrolling:touch;position:absolute;top:0;right:0;bottom:0;left:0;outline:0}html[dir=ltr] .adf-viewer-pdf-viewer{box-shadow:inset 1px 0 0 rgba(255,255,255,.05)}html[dir=rtl] .adf-viewer-pdf-viewer{box-shadow:inset -1px 0 0 rgba(255,255,255,.05)}"]
                }] }
    ];
    /** @nocollapse */
    PdfViewerComponent.ctorParameters = function () { return [
        { type: MatDialog },
        { type: RenderingQueueServices },
        { type: LogService },
        { type: AppConfigService }
    ]; };
    PdfViewerComponent.propDecorators = {
        urlFile: [{ type: Input }],
        blobFile: [{ type: Input }],
        nameFile: [{ type: Input }],
        showToolbar: [{ type: Input }],
        allowThumbnails: [{ type: Input }],
        thumbnailsTemplate: [{ type: Input }],
        rendered: [{ type: Output }],
        error: [{ type: Output }],
        handleKeyboardEvent: [{ type: HostListener, args: ['document:keydown', ['$event'],] }]
    };
    return PdfViewerComponent;
}());
export { PdfViewerComponent };
if (false) {
    /** @type {?} */
    PdfViewerComponent.prototype.urlFile;
    /** @type {?} */
    PdfViewerComponent.prototype.blobFile;
    /** @type {?} */
    PdfViewerComponent.prototype.nameFile;
    /** @type {?} */
    PdfViewerComponent.prototype.showToolbar;
    /** @type {?} */
    PdfViewerComponent.prototype.allowThumbnails;
    /** @type {?} */
    PdfViewerComponent.prototype.thumbnailsTemplate;
    /** @type {?} */
    PdfViewerComponent.prototype.rendered;
    /** @type {?} */
    PdfViewerComponent.prototype.error;
    /** @type {?} */
    PdfViewerComponent.prototype.loadingTask;
    /** @type {?} */
    PdfViewerComponent.prototype.currentPdfDocument;
    /** @type {?} */
    PdfViewerComponent.prototype.page;
    /** @type {?} */
    PdfViewerComponent.prototype.displayPage;
    /** @type {?} */
    PdfViewerComponent.prototype.totalPages;
    /** @type {?} */
    PdfViewerComponent.prototype.loadingPercent;
    /** @type {?} */
    PdfViewerComponent.prototype.pdfViewer;
    /** @type {?} */
    PdfViewerComponent.prototype.documentContainer;
    /** @type {?} */
    PdfViewerComponent.prototype.currentScaleMode;
    /** @type {?} */
    PdfViewerComponent.prototype.currentScale;
    /** @type {?} */
    PdfViewerComponent.prototype.MAX_AUTO_SCALE;
    /** @type {?} */
    PdfViewerComponent.prototype.DEFAULT_SCALE_DELTA;
    /** @type {?} */
    PdfViewerComponent.prototype.MIN_SCALE;
    /** @type {?} */
    PdfViewerComponent.prototype.MAX_SCALE;
    /** @type {?} */
    PdfViewerComponent.prototype.isPanelDisabled;
    /** @type {?} */
    PdfViewerComponent.prototype.showThumbnails;
    /** @type {?} */
    PdfViewerComponent.prototype.pdfThumbnailsContext;
    /** @type {?} */
    PdfViewerComponent.prototype.randomPdfId;
    /**
     * @type {?}
     * @private
     */
    PdfViewerComponent.prototype.dialog;
    /**
     * @type {?}
     * @private
     */
    PdfViewerComponent.prototype.renderingQueueServices;
    /**
     * @type {?}
     * @private
     */
    PdfViewerComponent.prototype.logService;
    /**
     * @type {?}
     * @private
     */
    PdfViewerComponent.prototype.appConfigService;
}
