// @alfresco/adf-core/esm5/viewer/components/viewer-open-with.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
var ViewerOpenWithComponent = /** @class */ (function () {
    function ViewerOpenWithComponent() {
    }
    ViewerOpenWithComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-viewer-open-with',
                    encapsulation: ViewEncapsulation.None,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'adf-viewer-open-with' },
                    template: "<ng-content></ng-content>"
                }] }
    ];
    return ViewerOpenWithComponent;
}());
export { ViewerOpenWithComponent };
