// @alfresco/adf-extensions/esm2015/lib/components/dynamic-tab/dynamic-tab.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewChild, ViewContainerRef, ComponentFactoryResolver } from '@angular/core';
import { Node } from '@alfresco/js-api';
import { ExtensionService } from '../../services/extension.service';
export class DynamicTabComponent {
    /**
     * @param {?} extensions
     * @param {?} componentFactoryResolver
     */
    constructor(extensions, componentFactoryResolver) {
        this.extensions = extensions;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const componentType = this.extensions.getComponentById(this.id);
        if (componentType) {
            /** @type {?} */
            const factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.updateInstance();
            }
        }
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        if (changes.node) {
            this.updateInstance();
        }
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    }
    /**
     * @private
     * @return {?}
     */
    updateInstance() {
        if (this.componentRef && this.componentRef.instance) {
            this.componentRef.instance.node = this.node;
        }
    }
}
DynamicTabComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-dynamic-tab',
                template: `<div #content></div>`
            }] }
];
/** @nocollapse */
DynamicTabComponent.ctorParameters = () => [
    { type: ExtensionService },
    { type: ComponentFactoryResolver }
];
DynamicTabComponent.propDecorators = {
    content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
    id: [{ type: Input }],
    node: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DynamicTabComponent.prototype.content;
    /**
     * Unique ID string for the component to show.
     * @type {?}
     */
    DynamicTabComponent.prototype.id;
    /** @type {?} */
    DynamicTabComponent.prototype.node;
    /**
     * @type {?}
     * @private
     */
    DynamicTabComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    DynamicTabComponent.prototype.extensions;
    /**
     * @type {?}
     * @private
     */
    DynamicTabComponent.prototype.componentFactoryResolver;
}
