// @alfresco/adf-testing/esm5/lib/process-services-cloud/app/app-list-cloud.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by } from 'protractor';
import { BrowserVisibility } from '../../core/utils/browser-visibility';
var AppListCloudPage = /** @class */ (function () {
    function AppListCloudPage() {
        this.apsAppsContainer = element(by.css('adf-cloud-app-list'));
    }
    /**
     * @return {?}
     */
    AppListCloudPage.prototype.checkApsContainer = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.apsAppsContainer);
    };
    /**
     * @param {?} applicationName
     * @return {?}
     */
    AppListCloudPage.prototype.goToApp = /**
     * @param {?} applicationName
     * @return {?}
     */
    function (applicationName) {
        /** @type {?} */
        var app = element(by.css('mat-card[title="' + applicationName + '"]'));
        BrowserVisibility.waitUntilElementIsVisible(app);
        app.click();
    };
    /**
     * @param {?} applicationName
     * @return {?}
     */
    AppListCloudPage.prototype.checkAppIsNotDisplayed = /**
     * @param {?} applicationName
     * @return {?}
     */
    function (applicationName) {
        /** @type {?} */
        var app = element(by.css('mat-card[title="' + applicationName + '"]'));
        return BrowserVisibility.waitUntilElementIsNotOnPage(app);
    };
    /**
     * @param {?} applicationName
     * @return {?}
     */
    AppListCloudPage.prototype.checkAppIsDisplayed = /**
     * @param {?} applicationName
     * @return {?}
     */
    function (applicationName) {
        /** @type {?} */
        var app = element(by.css('mat-card[title="' + applicationName + '"]'));
        return BrowserVisibility.waitUntilElementIsVisible(app);
    };
    return AppListCloudPage;
}());
export { AppListCloudPage };
if (false) {
    /** @type {?} */
    AppListCloudPage.prototype.apsAppsContainer;
}
