// @alfresco/adf-insights/esm5/analytics-process/components/widgets/widget.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventEmitter, Input, Output } from '@angular/core';
var WidgetComponent = /** @class */ (function () {
    function WidgetComponent() {
        this.fieldChanged = new EventEmitter();
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    WidgetComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        /** @type {?} */
        var field = changes['field'];
        if (field && field.currentValue) {
            this.fieldChanged.emit(field.currentValue.value);
            return;
        }
    };
    /**
     * @return {?}
     */
    WidgetComponent.prototype.hasField = /**
     * @return {?}
     */
    function () {
        return this.field ? true : false;
    };
    /**
     * @return {?}
     */
    WidgetComponent.prototype.hasValue = /**
     * @return {?}
     */
    function () {
        return this.field &&
            this.field.value !== null &&
            this.field.value !== undefined;
    };
    /**
     * @param {?} field
     * @return {?}
     */
    WidgetComponent.prototype.changeValue = /**
     * @param {?} field
     * @return {?}
     */
    function (field) {
        this.fieldChanged.emit(field);
    };
    WidgetComponent.propDecorators = {
        field: [{ type: Input }],
        fieldChanged: [{ type: Output }]
    };
    return WidgetComponent;
}());
export { WidgetComponent };
if (false) {
    /** @type {?} */
    WidgetComponent.prototype.field;
    /** @type {?} */
    WidgetComponent.prototype.fieldChanged;
}
