// @alfresco/adf-insights/esm2015/diagram/components/swimlanes/diagram-lane.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
export class DiagramLaneComponent {
    /**
     * @param {?} elementRef
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
        this.options = { stroke: '#000000', fillColors: 'none', fillOpacity: '', strokeWidth: '1', radius: 0 };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.rectLeftCorner = { x: this.lane.x, y: this.lane.y };
        this.width = this.lane.width;
        this.height = this.lane.height;
        this.textPosition = { x: this.lane.x + 10, y: this.lane.y + (this.lane.height / 2) };
        this.text = this.lane.name;
        this.textTransform = 'r270';
    }
}
DiagramLaneComponent.decorators = [
    { type: Component, args: [{
                selector: 'diagram-lane',
                template: "<raphael-rect [leftCorner]=\"rectLeftCorner\" [width]=\"width\" [height]=\"height\" [radius]=\"options.radius\"\n              [stroke]=\"options.stroke\" [strokeWidth]=\"options.strokeWidth\"\n              [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-rect>\n<raphael-text [text]=\"text\" [position]=\"textPosition\" [transform]=\"textTransform\"></raphael-text>"
            }] }
];
/** @nocollapse */
DiagramLaneComponent.ctorParameters = () => [
    { type: ElementRef }
];
DiagramLaneComponent.propDecorators = {
    lane: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    DiagramLaneComponent.prototype.lane;
    /** @type {?} */
    DiagramLaneComponent.prototype.error;
    /** @type {?} */
    DiagramLaneComponent.prototype.rectLeftCorner;
    /** @type {?} */
    DiagramLaneComponent.prototype.width;
    /** @type {?} */
    DiagramLaneComponent.prototype.height;
    /** @type {?} */
    DiagramLaneComponent.prototype.textPosition;
    /** @type {?} */
    DiagramLaneComponent.prototype.text;
    /** @type {?} */
    DiagramLaneComponent.prototype.textTransform;
    /** @type {?} */
    DiagramLaneComponent.prototype.options;
    /** @type {?} */
    DiagramLaneComponent.prototype.elementRef;
}
