// @alfresco/adf-insights/esm2015/diagram/components/swimlanes/diagram-lanes.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
export class DiagramLanesComponent {
    /**
     * @param {?} elementRef
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
DiagramLanesComponent.decorators = [
    { type: Component, args: [{
                selector: 'diagram-lanes',
                template: "<div *ngIf=\"lanes\">\n    <div *ngFor=\"let lane of lanes\">\n        <diagram-lane [lane]=\"lane\"></diagram-lane>\n    </div>\n</div>\n"
            }] }
];
/** @nocollapse */
DiagramLanesComponent.ctorParameters = () => [
    { type: ElementRef }
];
DiagramLanesComponent.propDecorators = {
    lanes: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    DiagramLanesComponent.prototype.lanes;
    /** @type {?} */
    DiagramLanesComponent.prototype.error;
    /** @type {?} */
    DiagramLanesComponent.prototype.elementRef;
}
