// @alfresco/adf-content-services/esm2015/tree-view/models/tree-view.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NodeEntry } from '@alfresco/js-api';
export class TreeBaseNode {
    /**
     * @param {?} nodeEntry
     * @param {?=} level
     * @param {?=} expandable
     */
    constructor(nodeEntry, level, expandable) {
        this.expandable = true;
        this.name = nodeEntry.entry.name;
        this.nodeId = nodeEntry.entry.id;
        this.level = level ? level : 0;
        this.expandable = expandable ? expandable : true;
        this.node = nodeEntry ? new NodeEntry(nodeEntry) : undefined;
    }
}
if (false) {
    /** @type {?} */
    TreeBaseNode.prototype.name;
    /** @type {?} */
    TreeBaseNode.prototype.nodeId;
    /** @type {?} */
    TreeBaseNode.prototype.level;
    /** @type {?} */
    TreeBaseNode.prototype.expandable;
    /** @type {?} */
    TreeBaseNode.prototype.node;
}
