// @alfresco/adf-insights/esm5/diagram/components/swimlanes/diagram-pool.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramPoolComponent = /** @class */ (function () {
    function DiagramPoolComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
        this.options = { stroke: '#000000', fillColors: 'none', fillOpacity: '', strokeWidth: '1', radius: 0 };
    }
    /**
     * @return {?}
     */
    DiagramPoolComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.rectLeftCorner = { x: this.pool.x, y: this.pool.y };
        this.width = this.pool.width;
        this.height = this.pool.height;
        this.textPosition = { x: this.pool.x + 14, y: this.pool.y + (this.pool.height / 2) };
        this.text = this.pool.name;
        this.textTransform = 'r270';
    };
    DiagramPoolComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-pool',
                    template: "<raphael-rect [leftCorner]=\"rectLeftCorner\" [width]=\"width\" [height]=\"height\" [radius]=\"options.radius\"\n              [stroke]=\"options.stroke\" [strokeWidth]=\"options.strokeWidth\"\n              [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-rect>\n<raphael-text [text]=\"text\" [position]=\"textPosition\" [transform]=\"textTransform\"></raphael-text>"
                }] }
    ];
    /** @nocollapse */
    DiagramPoolComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramPoolComponent.propDecorators = {
        pool: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramPoolComponent;
}());
export { DiagramPoolComponent };
if (false) {
    /** @type {?} */
    DiagramPoolComponent.prototype.pool;
    /** @type {?} */
    DiagramPoolComponent.prototype.error;
    /** @type {?} */
    DiagramPoolComponent.prototype.rectLeftCorner;
    /** @type {?} */
    DiagramPoolComponent.prototype.width;
    /** @type {?} */
    DiagramPoolComponent.prototype.height;
    /** @type {?} */
    DiagramPoolComponent.prototype.textPosition;
    /** @type {?} */
    DiagramPoolComponent.prototype.text;
    /** @type {?} */
    DiagramPoolComponent.prototype.textTransform;
    /** @type {?} */
    DiagramPoolComponent.prototype.options;
    /** @type {?} */
    DiagramPoolComponent.prototype.elementRef;
}
