// @alfresco/adf-insights/esm5/diagram/components/swimlanes/diagram-pools.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramPoolsComponent = /** @class */ (function () {
    function DiagramPoolsComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    DiagramPoolsComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DiagramPoolsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-pools',
                    template: "<div *ngIf=\"pools\">\n    <div *ngFor=\"let pool of pools\">\n        <diagram-pool [pool]=\"pool\"></diagram-pool>\n        <diagram-lanes [lanes]=\"pool.lanes\"></diagram-lanes>\n    </div>\n</div>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramPoolsComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramPoolsComponent.propDecorators = {
        pools: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramPoolsComponent;
}());
export { DiagramPoolsComponent };
if (false) {
    /** @type {?} */
    DiagramPoolsComponent.prototype.pools;
    /** @type {?} */
    DiagramPoolsComponent.prototype.error;
    /** @type {?} */
    DiagramPoolsComponent.prototype.elementRef;
}
