// @alfresco/adf-testing/esm5/lib/core/pages/error.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { BrowserVisibility } from '../utils/browser-visibility';
import { element, by } from 'protractor';
var ErrorPage = /** @class */ (function () {
    function ErrorPage() {
        this.errorPageCode = element(by.css('adf-error-content .adf-error-content-code'));
        this.errorPageTitle = element(by.css('adf-error-content .adf-error-content-title'));
        this.errorPageDescription = element(by.css('adf-error-content .adf-error-content-description'));
        this.backButton = element(by.id('adf-return-button'));
        this.secondButton = element(by.id('adf-secondary-button'));
    }
    /**
     * @return {?}
     */
    ErrorPage.prototype.clickBackButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.backButton);
        this.backButton.click();
    };
    /**
     * @return {?}
     */
    ErrorPage.prototype.clickSecondButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.secondButton);
        this.secondButton.click();
    };
    /**
     * @return {?}
     */
    ErrorPage.prototype.checkErrorCode = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageCode);
    };
    /**
     * @return {?}
     */
    ErrorPage.prototype.getErrorCode = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageCode);
        return this.errorPageCode.getText();
    };
    /**
     * @return {?}
     */
    ErrorPage.prototype.getErrorTitle = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageTitle);
        return this.errorPageTitle.getText();
    };
    /**
     * @return {?}
     */
    ErrorPage.prototype.getErrorDescription = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageDescription);
        return this.errorPageDescription.getText();
    };
    return ErrorPage;
}());
export { ErrorPage };
if (false) {
    /** @type {?} */
    ErrorPage.prototype.errorPageCode;
    /** @type {?} */
    ErrorPage.prototype.errorPageTitle;
    /** @type {?} */
    ErrorPage.prototype.errorPageDescription;
    /** @type {?} */
    ErrorPage.prototype.backButton;
    /** @type {?} */
    ErrorPage.prototype.secondButton;
}
