// @alfresco/adf-testing/esm5/lib/core/pages/login.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FormControllersPage } from './form-controller.page';
import { browser, by, element, protractor } from 'protractor';
import { BrowserVisibility } from '../utils/browser-visibility';
import { LocalStorageUtil } from '../utils/local-storage.util';
var LoginPage = /** @class */ (function () {
    function LoginPage() {
        this.loginURL = browser.baseUrl + '/login';
        this.formControllersPage = new FormControllersPage();
        this.txtUsername = element(by.css('input[id="username"]'));
        this.txtPassword = element(by.css('input[id="password"]'));
        this.logoImg = element(by.css('img[id="adf-login-img-logo"]'));
        this.successRouteTxt = element(by.css('input[data-automation-id="adf-success-route"]'));
        this.logoTxt = element(by.css('input[data-automation-id="adf-url-logo"]'));
        this.usernameTooltip = element(by.css('span[data-automation-id="username-error"]'));
        this.passwordTooltip = element(by.css('span[data-automation-id="password-required"]'));
        this.loginTooltip = element(by.css('span[class="adf-login-error-message"]'));
        this.usernameInactive = element(by.css('input[id="username"][aria-invalid="false"]'));
        this.passwordInactive = element(by.css('input[id="password"][aria-invalid="false"]'));
        this.adfLogo = element(by.css('img[class="adf-img-logo ng-star-inserted"]'));
        this.usernameHighlighted = element(by.css('input[id="username"][aria-invalid="true"]'));
        this.passwordHighlighted = element(by.css('input[id="password"][aria-invalid="true"]'));
        this.signInButton = element(by.id('login-button'));
        this.showPasswordElement = element(by.css('mat-icon[data-automation-id="show_password"]'));
        this.hidePasswordElement = element(by.css('mat-icon[data-automation-id="hide_password"]'));
        this.rememberMe = element(by.css('mat-checkbox[id="adf-login-remember"]'));
        this.needHelp = element(by.css('div[id="adf-login-action-left"]'));
        this.register = element(by.css('div[id="adf-login-action-right"]'));
        this.footerSwitch = element(by.id('switch4'));
        this.rememberMeSwitch = element(by.id('adf-toggle-show-rememberme'));
        this.successRouteSwitch = element(by.id('adf-toggle-show-successRoute'));
        this.logoSwitch = element(by.id('adf-toggle-logo'));
        this.header = element(by.id('adf-header'));
        this.settingsIcon = element(by.cssContainingText('a[data-automation-id="settings"] mat-icon', 'settings'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    LoginPage.prototype.goToLoginPage = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        browser.waitForAngularEnabled(true);
        browser.driver.get((/** @type {?} */ (this)).loginURL);
        (/** @type {?} */ (this)).waitForElements();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    LoginPage.prototype.waitForElements = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).txtUsername);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).txtPassword);
        return (/** @type {?} */ (this));
    };
    /**
     * @param {?} username
     * @return {?}
     */
    LoginPage.prototype.enterUsername = /**
     * @param {?} username
     * @return {?}
     */
    function (username) {
        BrowserVisibility.waitUntilElementIsVisible(this.txtUsername);
        this.txtUsername.sendKeys('');
        this.txtUsername.clear();
        return this.txtUsername.sendKeys(username);
    };
    /**
     * @param {?} password
     * @return {?}
     */
    LoginPage.prototype.enterPassword = /**
     * @param {?} password
     * @return {?}
     */
    function (password) {
        BrowserVisibility.waitUntilElementIsVisible(this.txtPassword);
        this.txtPassword.clear();
        return this.txtPassword.sendKeys(password);
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    LoginPage.prototype.clearUsername = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        var _this = this;
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).txtUsername);
        (/** @type {?} */ (this)).txtUsername.click();
        (/** @type {?} */ (this)).txtUsername.getAttribute('value').then((/**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            for (var i = value.length; i >= 0; i--) {
                (/** @type {?} */ (_this)).txtUsername.sendKeys(protractor.Key.BACK_SPACE);
            }
        }));
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.clearPassword = /**
     * @return {?}
     */
    function () {
        var _this = this;
        BrowserVisibility.waitUntilElementIsVisible(this.txtPassword);
        this.txtPassword.getAttribute('value').then((/**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            for (var i = value.length; i >= 0; i--) {
                _this.txtPassword.sendKeys(protractor.Key.BACK_SPACE);
            }
        }));
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.getUsernameTooltip = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.usernameTooltip);
        return this.usernameTooltip.getText();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.getPasswordTooltip = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.passwordTooltip);
        return this.passwordTooltip.getText();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.getLoginError = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.loginTooltip);
        return this.loginTooltip.getText();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkLoginImgURL = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.logoImg);
        return this.logoImg.getAttribute('src');
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkUsernameInactive = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.usernameInactive);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkPasswordInactive = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.passwordInactive);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkUsernameHighlighted = /**
     * @return {?}
     */
    function () {
        this.adfLogo.click();
        BrowserVisibility.waitUntilElementIsVisible(this.usernameHighlighted);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkPasswordHighlighted = /**
     * @return {?}
     */
    function () {
        this.adfLogo.click();
        BrowserVisibility.waitUntilElementIsVisible(this.passwordHighlighted);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkUsernameTooltipIsNotVisible = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.usernameTooltip);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkPasswordTooltipIsNotVisible = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.passwordTooltip);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.getSignInButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.signInButton);
        return this.signInButton.isEnabled();
    };
    /**
     * @param {?} userModel
     * @return {?}
     */
    LoginPage.prototype.loginToProcessServicesUsingUserModel = /**
     * @param {?} userModel
     * @return {?}
     */
    function (userModel) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.goToLoginPage();
                        return [4 /*yield*/, LocalStorageUtil.clearStorage()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.setStorageItem('providers', 'BPM')];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.apiReset()];
                    case 3:
                        _a.sent();
                        this.login(userModel.email, userModel.password);
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} userModel
     * @return {?}
     */
    LoginPage.prototype.loginToContentServicesUsingUserModel = /**
     * @param {?} userModel
     * @return {?}
     */
    function (userModel) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.goToLoginPage();
                        return [4 /*yield*/, LocalStorageUtil.clearStorage()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.setStorageItem('providers', 'ECM')];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.apiReset()];
                    case 3:
                        _a.sent();
                        this.login(userModel.getId(), userModel.getPassword());
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    LoginPage.prototype.loginToContentServices = /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    function (username, password) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.goToLoginPage();
                        return [4 /*yield*/, LocalStorageUtil.clearStorage()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.setStorageItem('providers', 'ECM')];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, LocalStorageUtil.apiReset()];
                    case 3:
                        _a.sent();
                        this.waitForElements();
                        this.login(username, password);
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.clickSignInButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.signInButton);
        this.signInButton.click();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.clickSettingsIcon = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.settingsIcon);
        this.settingsIcon.click();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.showPassword = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.showPasswordElement);
        this.showPasswordElement.click();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.hidePassword = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.hidePasswordElement);
        this.hidePasswordElement.click();
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.getShownPassword = /**
     * @return {?}
     */
    function () {
        return this.txtPassword.getAttribute('value');
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkPasswordIsHidden = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.txtPassword);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkRememberIsDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.rememberMe);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkRememberIsNotDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.rememberMe);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkNeedHelpIsDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.needHelp);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkNeedHelpIsNotDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.needHelp);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkRegisterDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.register);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.checkRegisterIsNotDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotVisible(this.register);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.enableFooter = /**
     * @return {?}
     */
    function () {
        this.formControllersPage.enableToggle(this.footerSwitch);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.disableFooter = /**
     * @return {?}
     */
    function () {
        this.formControllersPage.disableToggle(this.footerSwitch);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.disableRememberMe = /**
     * @return {?}
     */
    function () {
        this.formControllersPage.disableToggle(this.rememberMeSwitch);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.enableSuccessRouteSwitch = /**
     * @return {?}
     */
    function () {
        this.formControllersPage.enableToggle(this.successRouteSwitch);
    };
    /**
     * @return {?}
     */
    LoginPage.prototype.enableLogoSwitch = /**
     * @return {?}
     */
    function () {
        this.formControllersPage.enableToggle(this.logoSwitch);
    };
    /**
     * @param {?} route
     * @return {?}
     */
    LoginPage.prototype.enterSuccessRoute = /**
     * @param {?} route
     * @return {?}
     */
    function (route) {
        BrowserVisibility.waitUntilElementIsVisible(this.successRouteTxt);
        this.successRouteTxt.sendKeys('');
        this.successRouteTxt.clear();
        return this.successRouteTxt.sendKeys(route);
    };
    /**
     * @param {?} logo
     * @return {?}
     */
    LoginPage.prototype.enterLogo = /**
     * @param {?} logo
     * @return {?}
     */
    function (logo) {
        BrowserVisibility.waitUntilElementIsVisible(this.logoTxt);
        this.logoTxt.sendKeys('');
        this.logoTxt.clear();
        return this.logoTxt.sendKeys(logo);
    };
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    LoginPage.prototype.login = /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    function (username, password) {
        this.waitForElements();
        this.enterUsername(username);
        this.enterPassword(password);
        this.clickSignInButton();
        return BrowserVisibility.waitUntilElementIsVisible(this.header);
    };
    return LoginPage;
}());
export { LoginPage };
if (false) {
    /** @type {?} */
    LoginPage.prototype.loginURL;
    /** @type {?} */
    LoginPage.prototype.formControllersPage;
    /** @type {?} */
    LoginPage.prototype.txtUsername;
    /** @type {?} */
    LoginPage.prototype.txtPassword;
    /** @type {?} */
    LoginPage.prototype.logoImg;
    /** @type {?} */
    LoginPage.prototype.successRouteTxt;
    /** @type {?} */
    LoginPage.prototype.logoTxt;
    /** @type {?} */
    LoginPage.prototype.usernameTooltip;
    /** @type {?} */
    LoginPage.prototype.passwordTooltip;
    /** @type {?} */
    LoginPage.prototype.loginTooltip;
    /** @type {?} */
    LoginPage.prototype.usernameInactive;
    /** @type {?} */
    LoginPage.prototype.passwordInactive;
    /** @type {?} */
    LoginPage.prototype.adfLogo;
    /** @type {?} */
    LoginPage.prototype.usernameHighlighted;
    /** @type {?} */
    LoginPage.prototype.passwordHighlighted;
    /** @type {?} */
    LoginPage.prototype.signInButton;
    /** @type {?} */
    LoginPage.prototype.showPasswordElement;
    /** @type {?} */
    LoginPage.prototype.hidePasswordElement;
    /** @type {?} */
    LoginPage.prototype.rememberMe;
    /** @type {?} */
    LoginPage.prototype.needHelp;
    /** @type {?} */
    LoginPage.prototype.register;
    /** @type {?} */
    LoginPage.prototype.footerSwitch;
    /** @type {?} */
    LoginPage.prototype.rememberMeSwitch;
    /** @type {?} */
    LoginPage.prototype.successRouteSwitch;
    /** @type {?} */
    LoginPage.prototype.logoSwitch;
    /** @type {?} */
    LoginPage.prototype.header;
    /** @type {?} */
    LoginPage.prototype.settingsIcon;
}
