// @alfresco/adf-testing/esm5/lib/core/pages/pagination.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { browser, by, element, protractor } from 'protractor';
import { BrowserVisibility } from '../utils/browser-visibility';
var PaginationPage = /** @class */ (function () {
    function PaginationPage() {
        this.itemsPerPageDropdown = element(by.css('div[class*="adf-pagination__perpage-block"] button'));
        this.pageSelectorDropDown = element(by.css('div[class*="adf-pagination__page-selector"]'));
        this.pageSelectorArrow = element(by.css('button[data-automation-id="page-selector"]'));
        this.itemsPerPage = element(by.css('span[class="adf-pagination__max-items"]'));
        this.currentPage = element(by.css('span[class="adf-pagination__current-page"]'));
        this.totalPages = element(by.css('span[class="adf-pagination__total-pages"]'));
        this.paginationRange = element(by.css('span[class="adf-pagination__range"]'));
        this.nextPageButton = element(by.css('button[class*="adf-pagination__next-button"]'));
        this.nextButtonDisabled = element(by.css('button[class*="adf-pagination__next-button"][disabled]'));
        this.previousButtonDisabled = element(by.css('button[class*="adf-pagination__previous-button"][disabled]'));
        this.pageDropDown = element(by.css('div[class*="adf-pagination__actualinfo-block"] button'));
        this.pageDropDownOptions = by.css('div[class*="mat-menu-content"] button');
        this.paginationSection = element(by.css('adf-pagination'));
        this.paginationSectionEmpty = element(by.css('adf-pagination[class*="adf-pagination__empty"]'));
        this.totalFiles = element(by.css('span[class="adf-pagination__range"]'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} numberOfItem
     * @return {THIS}
     */
    PaginationPage.prototype.selectItemsPerPage = /**
     * @template THIS
     * @this {THIS}
     * @param {?} numberOfItem
     * @return {THIS}
     */
    function (numberOfItem) {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).itemsPerPageDropdown);
        BrowserVisibility.waitUntilElementIsClickable((/** @type {?} */ (this)).itemsPerPageDropdown);
        browser.actions().mouseMove((/** @type {?} */ (this)).itemsPerPageDropdown).perform();
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).itemsPerPageDropdown);
        BrowserVisibility.waitUntilElementIsClickable((/** @type {?} */ (this)).itemsPerPageDropdown);
        (/** @type {?} */ (this)).itemsPerPageDropdown.click();
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).pageSelectorDropDown);
        /** @type {?} */
        var itemsPerPage = element.all(by.cssContainingText('.mat-menu-item', numberOfItem)).first();
        BrowserVisibility.waitUntilElementIsClickable(itemsPerPage);
        BrowserVisibility.waitUntilElementIsVisible(itemsPerPage);
        itemsPerPage.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkPageSelectorIsNotDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.pageSelectorArrow);
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkPageSelectorIsDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.pageSelectorArrow);
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    PaginationPage.prototype.checkPaginationIsNotDisplayed = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsOnPage((/** @type {?} */ (this)).paginationSectionEmpty);
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getCurrentItemsPerPage = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.itemsPerPage);
        return this.itemsPerPage.getText();
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getCurrentPage = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.paginationSection);
        BrowserVisibility.waitUntilElementIsVisible(this.currentPage);
        return this.currentPage.getText();
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getTotalPages = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.totalPages);
        return this.totalPages.getText();
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getPaginationRange = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.paginationRange);
        return this.paginationRange.getText();
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.clickOnNextPage = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.nextPageButton);
        BrowserVisibility.waitUntilElementIsClickable(this.nextPageButton);
        browser.actions().mouseMove(this.nextPageButton).perform();
        BrowserVisibility.waitUntilElementIsVisible(this.nextPageButton);
        BrowserVisibility.waitUntilElementIsClickable(this.nextPageButton);
        return this.nextPageButton.click();
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.clickOnPageDropdown = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.pageDropDown);
        BrowserVisibility.waitUntilElementIsClickable(this.pageDropDown);
        return this.pageDropDown.click();
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} numberOfItemPerPage
     * @return {THIS}
     */
    PaginationPage.prototype.clickOnPageDropdownOption = /**
     * @template THIS
     * @this {THIS}
     * @param {?} numberOfItemPerPage
     * @return {THIS}
     */
    function (numberOfItemPerPage) {
        BrowserVisibility.waitUntilElementIsVisible(element.all((/** @type {?} */ (this)).pageDropDownOptions).first());
        /** @type {?} */
        var option = element(by.cssContainingText('div[class*="mat-menu-content"] button', numberOfItemPerPage));
        BrowserVisibility.waitUntilElementIsVisible(option);
        option.click();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getPageDropdownOptions = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var deferred = protractor.promise.defer();
        BrowserVisibility.waitUntilElementIsVisible(element.all(this.pageDropDownOptions).first());
        /** @type {?} */
        var initialList = [];
        element.all(this.pageDropDownOptions).each((/**
         * @param {?} currentOption
         * @return {?}
         */
        function (currentOption) {
            currentOption.getText().then((/**
             * @param {?} text
             * @return {?}
             */
            function (text) {
                if (text !== '') {
                    initialList.push(text);
                }
            }));
        })).then((/**
         * @return {?}
         */
        function () {
            deferred.fulfill(initialList);
        }));
        return deferred.promise;
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkNextPageButtonIsDisabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.nextButtonDisabled);
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkPreviousPageButtonIsDisabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.previousButtonDisabled);
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkNextPageButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.nextButtonDisabled);
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.checkPreviousPageButtonIsEnabled = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.previousButtonDisabled);
    };
    /**
     * @return {?}
     */
    PaginationPage.prototype.getTotalNumberOfFiles = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.totalFiles);
        /** @type {?} */
        var numberOfFiles = this.totalFiles.getText().then((/**
         * @param {?} totalNumber
         * @return {?}
         */
        function (totalNumber) {
            /** @type {?} */
            var totalNumberOfFiles = totalNumber.split('of ')[1];
            return totalNumberOfFiles;
        }));
        return numberOfFiles;
    };
    return PaginationPage;
}());
export { PaginationPage };
if (false) {
    /** @type {?} */
    PaginationPage.prototype.itemsPerPageDropdown;
    /** @type {?} */
    PaginationPage.prototype.pageSelectorDropDown;
    /** @type {?} */
    PaginationPage.prototype.pageSelectorArrow;
    /** @type {?} */
    PaginationPage.prototype.itemsPerPage;
    /** @type {?} */
    PaginationPage.prototype.currentPage;
    /** @type {?} */
    PaginationPage.prototype.totalPages;
    /** @type {?} */
    PaginationPage.prototype.paginationRange;
    /** @type {?} */
    PaginationPage.prototype.nextPageButton;
    /** @type {?} */
    PaginationPage.prototype.nextButtonDisabled;
    /** @type {?} */
    PaginationPage.prototype.previousButtonDisabled;
    /** @type {?} */
    PaginationPage.prototype.pageDropDown;
    /** @type {?} */
    PaginationPage.prototype.pageDropDownOptions;
    /** @type {?} */
    PaginationPage.prototype.paginationSection;
    /** @type {?} */
    PaginationPage.prototype.paginationSectionEmpty;
    /** @type {?} */
    PaginationPage.prototype.totalFiles;
}
