// @alfresco/adf-testing/esm5/lib/core/pages/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { HeaderPage } from './header.page';
export { UserInfoPage } from './user-info.page';
export { LoginPage } from './login.page';
export { SettingsPage } from './settings.page';
export { FormControllersPage } from './form-controller.page';
export { LoginSSOPage } from './login-sso.page';
export { DataTableComponentPage } from './data-table-component.page';
export { PaginationPage } from './pagination.page';
export { ErrorPage } from './error.page';
export {} from './login.page';
