// @alfresco/adf-testing/esm5/lib/core/pages/settings.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { browser, by, element, protractor } from 'protractor';
import { BrowserVisibility } from '../utils/browser-visibility';
var SettingsPage = /** @class */ (function () {
    function SettingsPage() {
        this.settingsURL = browser.baseUrl + '/settings';
        this.providerDropdown = element(by.css('mat-select[id="adf-provider-selector"] div[class="mat-select-arrow-wrapper"]'));
        this.ecmAndBpm = {
            option: element(by.xpath('//SPAN[@class="mat-option-text"][contains(text(),"ALL")]')),
            text: 'ALL'
        };
        this.bpm = {
            option: element(by.xpath('//SPAN[@class="mat-option-text"][contains(text(),"BPM") and not (contains(text(),"and"))]')),
            text: 'BPM'
        };
        this.ecm = {
            option: element(by.xpath('//SPAN[@class="mat-option-text"][contains(text(),"ECM") and not (contains(text(),"and"))]')),
            text: 'ECM'
        };
        this.oauth = {
            option: element(by.xpath('//SPAN[@class="mat-option-text"][contains(text(),"OAUTH")]')),
            text: 'OAUTH'
        };
        this.selectedOption = element(by.css('span[class*="mat-select-value-text"]'));
        this.ecmText = element(by.css('input[data-automation-id*="ecmHost"]'));
        this.bpmText = element(by.css('input[data-automation-id*="bpmHost"]'));
        this.clientIdText = element(by.css('input[id="clientId"]'));
        this.authHostText = element(by.css('input[id="oauthHost"]'));
        this.logoutUrlText = element(by.css('input[id="logout-url"]'));
        this.basicAuthRadioButton = element(by.cssContainingText('mat-radio-button[id*="mat-radio"]', 'Basic Authentication'));
        this.identityHostText = element(by.css('input[id="identityHost"]'));
        this.ssoRadioButton = element(by.cssContainingText('[id*="mat-radio"]', 'SSO'));
        this.silentLoginToggleLabel = element(by.css('mat-slide-toggle[name="silentLogin"] label'));
        this.silentLoginToggleElement = element(by.css('mat-slide-toggle[name="silentLogin"]'));
        this.implicitFlowLabel = element(by.css('mat-slide-toggle[name="implicitFlow"] label'));
        this.implicitFlowElement = element(by.css('mat-slide-toggle[name="implicitFlow"]'));
        this.applyButton = element(by.css('button[data-automation-id*="host-button"]'));
        this.backButton = element(by.cssContainingText('button span[class="mat-button-wrapper"]', 'Back'));
        this.validationMessage = element(by.cssContainingText('mat-error', 'This field is required'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.goToSettingsPage = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        browser.waitForAngularEnabled(true);
        browser.driver.get((/** @type {?} */ (this)).settingsURL);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).providerDropdown);
        return (/** @type {?} */ (this));
    };
    /**
     * @param {?} option
     * @param {?} selected
     * @return {?}
     */
    SettingsPage.prototype.setProvider = /**
     * @param {?} option
     * @param {?} selected
     * @return {?}
     */
    function (option, selected) {
        BrowserVisibility.waitUntilElementIsVisible(this.providerDropdown);
        this.providerDropdown.click();
        BrowserVisibility.waitUntilElementIsVisible(option);
        option.click();
        return expect(this.selectedOption.getText()).toEqual(selected);
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getSelectedOptionText = /**
     * @return {?}
     */
    function () {
        return this.selectedOption.getText();
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getBpmHostUrl = /**
     * @return {?}
     */
    function () {
        return this.bpmText.getAttribute('value');
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getEcmHostUrl = /**
     * @return {?}
     */
    function () {
        return this.ecmText.getAttribute('value');
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getBpmOption = /**
     * @return {?}
     */
    function () {
        return this.bpm.option;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getEcmOption = /**
     * @return {?}
     */
    function () {
        return this.ecm.option;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getEcmAndBpmOption = /**
     * @return {?}
     */
    function () {
        return this.ecmAndBpm.option;
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.setProviderEcmBpm = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        (/** @type {?} */ (this)).goToSettingsPage();
        (/** @type {?} */ (this)).setProvider((/** @type {?} */ (this)).ecmAndBpm.option, (/** @type {?} */ (this)).ecmAndBpm.text);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).bpmText);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).ecmText);
        (/** @type {?} */ (this)).clickApply();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.setProviderBpm = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        (/** @type {?} */ (this)).goToSettingsPage();
        (/** @type {?} */ (this)).setProvider((/** @type {?} */ (this)).bpm.option, (/** @type {?} */ (this)).bpm.text);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).bpmText);
        (/** @type {?} */ (this)).clickApply();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.setProviderEcm = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        (/** @type {?} */ (this)).goToSettingsPage();
        (/** @type {?} */ (this)).setProvider((/** @type {?} */ (this)).ecm.option, (/** @type {?} */ (this)).ecm.text);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).ecmText);
        expect((/** @type {?} */ (this)).bpmText.isPresent()).toBeFalsy();
        (/** @type {?} */ (this)).clickApply();
        return (/** @type {?} */ (this));
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.setProviderOauth = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        (/** @type {?} */ (this)).goToSettingsPage();
        (/** @type {?} */ (this)).setProvider((/** @type {?} */ (this)).oauth.option, (/** @type {?} */ (this)).oauth.text);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).bpmText);
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).ecmText);
        expect((/** @type {?} */ (this)).authHostText.isPresent()).toBeTruthy();
        (/** @type {?} */ (this)).clickApply();
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.clickBackButton = /**
     * @return {?}
     */
    function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        BrowserVisibility.waitUntilElementIsVisible(this.backButton);
                        return [4 /*yield*/, this.backButton.click()];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.clickSsoRadioButton = /**
     * @return {?}
     */
    function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        BrowserVisibility.waitUntilElementIsVisible(this.ssoRadioButton);
                        return [4 /*yield*/, this.ssoRadioButton.click()];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} contentServiceURL
     * @param {?} authHost
     * @param {?} identityHost
     * @param {?=} silentLogin
     * @param {?=} implicitFlow
     * @param {?=} clientId
     * @param {?=} logoutUr
     * @return {?}
     */
    SettingsPage.prototype.setProviderEcmSso = /**
     * @param {?} contentServiceURL
     * @param {?} authHost
     * @param {?} identityHost
     * @param {?=} silentLogin
     * @param {?=} implicitFlow
     * @param {?=} clientId
     * @param {?=} logoutUr
     * @return {?}
     */
    function (contentServiceURL, authHost, identityHost, silentLogin, implicitFlow, clientId, logoutUr) {
        if (silentLogin === void 0) { silentLogin = true; }
        if (implicitFlow === void 0) { implicitFlow = true; }
        if (logoutUr === void 0) { logoutUr = '/logout'; }
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.goToSettingsPage();
                        this.setProvider(this.ecm.option, this.ecm.text);
                        BrowserVisibility.waitUntilElementIsNotOnPage(this.bpmText);
                        BrowserVisibility.waitUntilElementIsVisible(this.ecmText);
                        return [4 /*yield*/, this.clickSsoRadioButton()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.setClientId(clientId)];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.setContentServicesURL(contentServiceURL)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.setAuthHost(authHost)];
                    case 4:
                        _a.sent();
                        return [4 /*yield*/, this.setIdentityHost(identityHost)];
                    case 5:
                        _a.sent();
                        return [4 /*yield*/, this.setSilentLogin(silentLogin)];
                    case 6:
                        _a.sent();
                        return [4 /*yield*/, this.setImplicitFlow(implicitFlow)];
                    case 7:
                        _a.sent();
                        return [4 /*yield*/, this.setLogoutUrl(logoutUr)];
                    case 8:
                        _a.sent();
                        return [4 /*yield*/, this.clickApply()];
                    case 9:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} processServiceURL
     * @param {?} authHost
     * @param {?} identityHost
     * @param {?=} silentLogin
     * @param {?=} implicitFlow
     * @return {?}
     */
    SettingsPage.prototype.setProviderBpmSso = /**
     * @param {?} processServiceURL
     * @param {?} authHost
     * @param {?} identityHost
     * @param {?=} silentLogin
     * @param {?=} implicitFlow
     * @return {?}
     */
    function (processServiceURL, authHost, identityHost, silentLogin, implicitFlow) {
        if (silentLogin === void 0) { silentLogin = true; }
        if (implicitFlow === void 0) { implicitFlow = true; }
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        this.goToSettingsPage();
                        this.setProvider(this.bpm.option, this.bpm.text);
                        BrowserVisibility.waitUntilElementIsVisible(this.bpmText);
                        BrowserVisibility.waitUntilElementIsNotOnPage(this.ecmText);
                        return [4 /*yield*/, this.clickSsoRadioButton()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.setClientId()];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, this.setProcessServicesURL(processServiceURL)];
                    case 3:
                        _a.sent();
                        return [4 /*yield*/, this.setAuthHost(authHost)];
                    case 4:
                        _a.sent();
                        return [4 /*yield*/, this.setIdentityHost(identityHost)];
                    case 5:
                        _a.sent();
                        return [4 /*yield*/, this.setSilentLogin(silentLogin)];
                    case 6:
                        _a.sent();
                        return [4 /*yield*/, this.setImplicitFlow(implicitFlow)];
                    case 7:
                        _a.sent();
                        return [4 /*yield*/, this.clickApply()];
                    case 8:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} logoutUrl
     * @return {?}
     */
    SettingsPage.prototype.setLogoutUrl = /**
     * @param {?} logoutUrl
     * @return {?}
     */
    function (logoutUrl) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                BrowserVisibility.waitUntilElementIsPresent(this.logoutUrlText);
                this.logoutUrlText.clear();
                this.logoutUrlText.sendKeys(logoutUrl);
                return [2 /*return*/];
            });
        });
    };
    /**
     * @param {?} processServiceURL
     * @return {?}
     */
    SettingsPage.prototype.setProcessServicesURL = /**
     * @param {?} processServiceURL
     * @return {?}
     */
    function (processServiceURL) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                BrowserVisibility.waitUntilElementIsVisible(this.bpmText);
                this.bpmText.clear();
                this.bpmText.sendKeys(processServiceURL);
                return [2 /*return*/];
            });
        });
    };
    /**
     * @param {?=} clientId
     * @return {?}
     */
    SettingsPage.prototype.setClientId = /**
     * @param {?=} clientId
     * @return {?}
     */
    function (clientId) {
        if (clientId === void 0) { clientId = browser.params.config.oauth2.clientId; }
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                BrowserVisibility.waitUntilElementIsVisible(this.clientIdText);
                this.clientIdText.clear();
                this.clientIdText.sendKeys(clientId);
                return [2 /*return*/];
            });
        });
    };
    /**
     * @param {?} contentServiceURL
     * @return {?}
     */
    SettingsPage.prototype.setContentServicesURL = /**
     * @param {?} contentServiceURL
     * @return {?}
     */
    function (contentServiceURL) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                BrowserVisibility.waitUntilElementIsClickable(this.ecmText);
                this.ecmText.clear();
                this.ecmText.sendKeys(contentServiceURL);
                return [2 /*return*/];
            });
        });
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.clearContentServicesURL = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.ecmText);
        this.ecmText.clear();
        this.ecmText.sendKeys('a');
        this.ecmText.sendKeys(protractor.Key.BACK_SPACE);
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.clearProcessServicesURL = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.bpmText);
        this.bpmText.clear();
        this.bpmText.sendKeys('a');
        this.bpmText.sendKeys(protractor.Key.BACK_SPACE);
    };
    /**
     * @param {?} authHostURL
     * @return {?}
     */
    SettingsPage.prototype.setAuthHost = /**
     * @param {?} authHostURL
     * @return {?}
     */
    function (authHostURL) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        BrowserVisibility.waitUntilElementIsVisible(this.authHostText);
                        return [4 /*yield*/, this.authHostText.clear()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.authHostText.sendKeys(authHostURL)];
                    case 2:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} identityHost
     * @return {?}
     */
    SettingsPage.prototype.setIdentityHost = /**
     * @param {?} identityHost
     * @return {?}
     */
    function (identityHost) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        BrowserVisibility.waitUntilElementIsVisible(this.identityHostText);
                        return [4 /*yield*/, this.identityHostText.clear()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.identityHostText.sendKeys(identityHost)];
                    case 2:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.clickApply = /**
     * @return {?}
     */
    function () {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        BrowserVisibility.waitUntilElementIsVisible(this.applyButton);
                        return [4 /*yield*/, this.applyButton.click()];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} enableToggle
     * @return {?}
     */
    SettingsPage.prototype.setSilentLogin = /**
     * @param {?} enableToggle
     * @return {?}
     */
    function (enableToggle) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var isChecked;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, BrowserVisibility.waitUntilElementIsVisible(this.silentLoginToggleElement)];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.silentLoginToggleElement.getAttribute('class')];
                    case 2:
                        isChecked = (_a.sent()).includes('mat-checked');
                        if (isChecked && !enableToggle || !isChecked && enableToggle) {
                            return [2 /*return*/, this.silentLoginToggleLabel.click()];
                        }
                        return [2 /*return*/, Promise.resolve()];
                }
            });
        });
    };
    /**
     * @param {?} enableToggle
     * @return {?}
     */
    SettingsPage.prototype.setImplicitFlow = /**
     * @param {?} enableToggle
     * @return {?}
     */
    function (enableToggle) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var isChecked;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, BrowserVisibility.waitUntilElementIsVisible(this.implicitFlowElement)];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.implicitFlowElement.getAttribute('class')];
                    case 2:
                        isChecked = (_a.sent()).includes('mat-checked');
                        if (isChecked && !enableToggle || !isChecked && enableToggle) {
                            return [2 /*return*/, this.implicitFlowLabel.click()];
                        }
                        return [2 /*return*/, Promise.resolve()];
                }
            });
        });
    };
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    SettingsPage.prototype.checkApplyButtonIsDisabled = /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).applyButton.getAttribute('disabled'));
        return (/** @type {?} */ (this));
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.checkProviderDropdownIsDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.providerDropdown);
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.checkValidationMessageIsDisplayed = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.validationMessage);
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.checkProviderOptions = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.providerDropdown);
        this.providerDropdown.click();
        BrowserVisibility.waitUntilElementIsVisible(this.ecmAndBpm.option);
        BrowserVisibility.waitUntilElementIsVisible(this.ecm.option);
        BrowserVisibility.waitUntilElementIsVisible(this.bpm.option);
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getBasicAuthRadioButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.basicAuthRadioButton);
        return this.basicAuthRadioButton;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getSsoRadioButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.ssoRadioButton);
        return this.ssoRadioButton;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getBackButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.backButton);
        return this.backButton;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.getApplyButton = /**
     * @return {?}
     */
    function () {
        BrowserVisibility.waitUntilElementIsVisible(this.applyButton);
        return this.applyButton;
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.checkBasicAuthRadioIsSelected = /**
     * @return {?}
     */
    function () {
        expect(this.getBasicAuthRadioButton().getAttribute('class')).toContain('mat-radio-checked');
    };
    /**
     * @return {?}
     */
    SettingsPage.prototype.checkSsoRadioIsNotSelected = /**
     * @return {?}
     */
    function () {
        expect(this.getSsoRadioButton().getAttribute('class')).not.toContain('mat-radio-checked');
    };
    return SettingsPage;
}());
export { SettingsPage };
if (false) {
    /** @type {?} */
    SettingsPage.prototype.settingsURL;
    /** @type {?} */
    SettingsPage.prototype.providerDropdown;
    /** @type {?} */
    SettingsPage.prototype.ecmAndBpm;
    /** @type {?} */
    SettingsPage.prototype.bpm;
    /** @type {?} */
    SettingsPage.prototype.ecm;
    /** @type {?} */
    SettingsPage.prototype.oauth;
    /** @type {?} */
    SettingsPage.prototype.selectedOption;
    /** @type {?} */
    SettingsPage.prototype.ecmText;
    /** @type {?} */
    SettingsPage.prototype.bpmText;
    /** @type {?} */
    SettingsPage.prototype.clientIdText;
    /** @type {?} */
    SettingsPage.prototype.authHostText;
    /** @type {?} */
    SettingsPage.prototype.logoutUrlText;
    /** @type {?} */
    SettingsPage.prototype.basicAuthRadioButton;
    /** @type {?} */
    SettingsPage.prototype.identityHostText;
    /** @type {?} */
    SettingsPage.prototype.ssoRadioButton;
    /** @type {?} */
    SettingsPage.prototype.silentLoginToggleLabel;
    /** @type {?} */
    SettingsPage.prototype.silentLoginToggleElement;
    /** @type {?} */
    SettingsPage.prototype.implicitFlowLabel;
    /** @type {?} */
    SettingsPage.prototype.implicitFlowElement;
    /** @type {?} */
    SettingsPage.prototype.applyButton;
    /** @type {?} */
    SettingsPage.prototype.backButton;
    /** @type {?} */
    SettingsPage.prototype.validationMessage;
}
