// @alfresco/adf-core/esm5/form/components/form-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { FormService } from './../services/form.service';
var FormListComponent = /** @class */ (function () {
    function FormListComponent(formService) {
        this.formService = formService;
        /**
         * The array that contains the information to show inside the list.
         */
        this.forms = [];
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    FormListComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        this.getForms();
    };
    /**
     * @return {?}
     */
    FormListComponent.prototype.isEmpty = /**
     * @return {?}
     */
    function () {
        return this.forms && this.forms.length === 0;
    };
    /**
     * @return {?}
     */
    FormListComponent.prototype.getForms = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.formService.getForms().subscribe((/**
         * @param {?} forms
         * @return {?}
         */
        function (forms) {
            var _a;
            (_a = _this.forms).push.apply(_a, tslib_1.__spread(forms));
        }));
    };
    FormListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-form-list',
                    template: "<adf-datatable *ngIf=\"!isEmpty()\"\n    [rows]=\"forms\">\n    <data-columns>\n        <data-column key=\"name\" type=\"text\" title=\"Name\" class=\"adf-ellipsis-cell\" [sortable]=\"true\"></data-column>\n        <data-column key=\"lastUpdatedByFullName\" type=\"text\" title=\"User\" class=\"adf-ellipsis-cell\" [sortable]=\"true\"></data-column>\n        <data-column key=\"lastUpdated\" type=\"date\" format=\"shortDate\" title=\"Date\"></data-column>\n    </data-columns>\n</adf-datatable>\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    FormListComponent.ctorParameters = function () { return [
        { type: FormService }
    ]; };
    FormListComponent.propDecorators = {
        forms: [{ type: Input }]
    };
    return FormListComponent;
}());
export { FormListComponent };
if (false) {
    /**
     * The array that contains the information to show inside the list.
     * @type {?}
     */
    FormListComponent.prototype.forms;
    /**
     * @type {?}
     * @protected
     */
    FormListComponent.prototype.formService;
}
