// @alfresco/adf-process-services/esm2015/attachment/create-process-attachment.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { ProcessContentService } from '@alfresco/adf-core';
export class CreateProcessAttachmentComponent {
    /**
     * @param {?} activitiContentService
     */
    constructor(activitiContentService) {
        this.activitiContentService = activitiContentService;
        /**
         * Emitted when an error occurs while creating or uploading an attachment
         * from the user within the component.
         */
        this.error = new EventEmitter();
        /**
         * Emitted when an attachment is successfully created or uploaded
         * from within the component.
         */
        this.success = new EventEmitter();
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        if (changes['processInstanceId'] && changes['processInstanceId'].currentValue) {
            this.processInstanceId = changes['processInstanceId'].currentValue;
        }
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onFileUpload(event) {
        /** @type {?} */
        const filesList = event.detail.files.map((/**
         * @param {?} obj
         * @return {?}
         */
        (obj) => obj.file));
        for (const fileInfoObj of filesList) {
            /** @type {?} */
            const file = fileInfoObj;
            /** @type {?} */
            const opts = {
                isRelatedContent: true
            };
            this.activitiContentService.createProcessRelatedContent(this.processInstanceId, file, opts).subscribe((/**
             * @param {?} res
             * @return {?}
             */
            (res) => {
                this.success.emit(res);
            }), (/**
             * @param {?} err
             * @return {?}
             */
            (err) => {
                this.error.emit(err);
            }));
        }
    }
}
CreateProcessAttachmentComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-create-process-attachment',
                template: "<button\n    id=\"add_new_process_content_button\"\n    color=\"primary\"\n    mat-button\n    mat-raised-button\n    mat-icon-button\n    class=\"adf-create-attachment\"\n    adf-upload=\"true\"\n    [mode]=\"['click']\"\n    [multiple]=\"true\"\n    (upload-files)=\"onFileUpload($event)\">\n    <mat-icon>add</mat-icon>\n</button>\n",
                styles: [".adf-create-attachment{display:inline-block;line-height:0;vertical-align:middle}"]
            }] }
];
/** @nocollapse */
CreateProcessAttachmentComponent.ctorParameters = () => [
    { type: ProcessContentService }
];
CreateProcessAttachmentComponent.propDecorators = {
    processInstanceId: [{ type: Input }],
    error: [{ type: Output }],
    success: [{ type: Output }]
};
if (false) {
    /**
     * (required) The ID of the process instance to display.
     * @type {?}
     */
    CreateProcessAttachmentComponent.prototype.processInstanceId;
    /**
     * Emitted when an error occurs while creating or uploading an attachment
     * from the user within the component.
     * @type {?}
     */
    CreateProcessAttachmentComponent.prototype.error;
    /**
     * Emitted when an attachment is successfully created or uploaded
     * from within the component.
     * @type {?}
     */
    CreateProcessAttachmentComponent.prototype.success;
    /**
     * @type {?}
     * @private
     */
    CreateProcessAttachmentComponent.prototype.activitiContentService;
}
