// @alfresco/adf-core/esm2015/form/components/form-renderer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input } from '@angular/core';
import { FormBaseModel } from './form-base.model';
export class FormRendererComponent {
    constructor() {
        /**
         * Toggle debug options.
         */
        this.showDebugButton = false;
    }
}
FormRendererComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-form-renderer',
                template: "<div class=\"{{formDefinition.className}}\" [ngClass]=\"{'adf-readonly-form': formDefinition.readOnly }\">\n    <div *ngIf=\"formDefinition.hasTabs()\">\n        <tabs-widget [tabs]=\"formDefinition.tabs\"></tabs-widget>\n    </div>\n\n    <div *ngIf=\"!formDefinition.hasTabs() && formDefinition.hasFields()\">\n        <div *ngFor=\"let field of formDefinition.fields\">\n            <adf-form-field [field]=\"field.field\"></adf-form-field>\n        </div>\n    </div>\n</div>\n<!--\nFor debugging and data visualisation purposes,\nwill be removed during future revisions\n-->\n<div *ngIf=\"showDebugButton\" class=\"adf-form-debug-container\">\n    <mat-slide-toggle [(ngModel)]=\"debugMode\">Debug mode</mat-slide-toggle>\n    <div *ngIf=\"debugMode\">\n        <h4>Values</h4>\n        <pre>{{formDefinition.values | json}}</pre>\n\n        <h4>Form</h4>\n        <pre>{{formDefinition.json | json}}</pre>\n    </div>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            }] }
];
FormRendererComponent.propDecorators = {
    showDebugButton: [{ type: Input }],
    formDefinition: [{ type: Input }]
};
if (false) {
    /**
     * Toggle debug options.
     * @type {?}
     */
    FormRendererComponent.prototype.showDebugButton;
    /** @type {?} */
    FormRendererComponent.prototype.formDefinition;
    /** @type {?} */
    FormRendererComponent.prototype.debugMode;
}
