// @alfresco/adf-insights/esm5/diagram/components/gateways/diagram-event-gateway.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramEventGatewayComponent = /** @class */ (function () {
    function DiagramEventGatewayComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.center = {};
        this.centerPentagon = {};
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: 0.5 };
        this.circleRadiusInner = 10.4;
        this.circleRadiusOuter = 11.7;
        this.pentaStrokeWidth = 1.39999998;
    }
    /**
     * @return {?}
     */
    DiagramEventGatewayComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
        this.centerPentagon.x = this.data.x;
        this.centerPentagon.y = this.data.y;
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
    };
    DiagramEventGatewayComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-event-gateway',
                    template: "<diagram-gateway [data]=\"data\"></diagram-gateway>\n<raphael-circle [center]=\"center\" [radius]=\"circleRadiusInner\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<raphael-circle [center]=\"center\" [radius]=\"circleRadiusOuter\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<raphael-pentagon [center]=\"centerPentagon\" [strokeWidth]=\"pentaStrokeWidth\" [stroke]=\"options.stroke\"\n                  [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-pentagon>"
                }] }
    ];
    /** @nocollapse */
    DiagramEventGatewayComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramEventGatewayComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramEventGatewayComponent;
}());
export { DiagramEventGatewayComponent };
if (false) {
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.data;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.error;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.center;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.centerPentagon;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.options;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.circleRadiusInner;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.circleRadiusOuter;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.pentaStrokeWidth;
    /** @type {?} */
    DiagramEventGatewayComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramEventGatewayComponent.prototype.diagramColorService;
}
