// @alfresco/adf-insights/esm5/diagram/components/gateways/diagram-parallel-gateway.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramParallelGatewayComponent = /** @class */ (function () {
    function DiagramParallelGatewayComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.center = {};
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: 3 };
    }
    /**
     * @return {?}
     */
    DiagramParallelGatewayComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.center.x = this.data.x;
        this.center.y = this.data.y;
        this.width = this.data.width;
        this.height = this.data.height;
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
    };
    DiagramParallelGatewayComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-parallel-gateway',
                    template: "<diagram-gateway [data]=\"data\"></diagram-gateway>\n<raphael-plus [center]=\"center\" [stroke]=\"options.stroke\" [strokeWidth]=\"options.strokeWidth\"\n                 [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-plus>"
                }] }
    ];
    /** @nocollapse */
    DiagramParallelGatewayComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramParallelGatewayComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramParallelGatewayComponent;
}());
export { DiagramParallelGatewayComponent };
if (false) {
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.data;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.error;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.center;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.width;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.height;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.options;
    /** @type {?} */
    DiagramParallelGatewayComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramParallelGatewayComponent.prototype.diagramColorService;
}
