// @alfresco/adf-core/esm5/services/apps-process.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { AlfrescoApiService } from './alfresco-api.service';
import { LogService } from './log.service';
import { map, catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as i2 from "./log.service";
var AppsProcessService = /** @class */ (function () {
    function AppsProcessService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    /**
     * Gets a list of deployed apps for this user.
     * @returns The list of deployed apps
     */
    /**
     * Gets a list of deployed apps for this user.
     * @return {?} The list of deployed apps
     */
    AppsProcessService.prototype.getDeployedApplications = /**
     * Gets a list of deployed apps for this user.
     * @return {?} The list of deployed apps
     */
    function () {
        var _this = this;
        return from(this.apiService.getInstance().activiti.appsApi.getAppDefinitions())
            .pipe(map((/**
         * @param {?} response
         * @return {?}
         */
        function (response) { return (/** @type {?} */ (response.data)); })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Gets a list of deployed apps for this user, where the app name is `name`.
     * @param name Name of the app
     * @returns The list of deployed apps
     */
    /**
     * Gets a list of deployed apps for this user, where the app name is `name`.
     * @param {?} name Name of the app
     * @return {?} The list of deployed apps
     */
    AppsProcessService.prototype.getDeployedApplicationsByName = /**
     * Gets a list of deployed apps for this user, where the app name is `name`.
     * @param {?} name Name of the app
     * @return {?} The list of deployed apps
     */
    function (name) {
        var _this = this;
        return from(this.apiService.getInstance().activiti.appsApi.getAppDefinitions())
            .pipe(map((/**
         * @param {?} response
         * @return {?}
         */
        function (response) { return (/** @type {?} */ (response.data.find((/**
         * @param {?} app
         * @return {?}
         */
        function (app) { return app.name === name; })))); })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Gets the details for a specific app ID number.
     * @param appId ID of the target app
     * @returns Details of the app
     */
    /**
     * Gets the details for a specific app ID number.
     * @param {?} appId ID of the target app
     * @return {?} Details of the app
     */
    AppsProcessService.prototype.getApplicationDetailsById = /**
     * Gets the details for a specific app ID number.
     * @param {?} appId ID of the target app
     * @return {?} Details of the app
     */
    function (appId) {
        var _this = this;
        return from(this.apiService.getInstance().activiti.appsApi.getAppDefinitions())
            .pipe(map((/**
         * @param {?} response
         * @return {?}
         */
        function (response) { return response.data.find((/**
         * @param {?} app
         * @return {?}
         */
        function (app) { return app.id === appId; })); })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    AppsProcessService.prototype.handleError = /**
     * @private
     * @param {?} error
     * @return {?}
     */
    function (error) {
        this.logService.error(error);
        return throwError(error || 'Server error');
    };
    AppsProcessService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    AppsProcessService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: LogService }
    ]; };
    /** @nocollapse */ AppsProcessService.ngInjectableDef = i0.defineInjectable({ factory: function AppsProcessService_Factory() { return new AppsProcessService(i0.inject(i1.AlfrescoApiService), i0.inject(i2.LogService)); }, token: AppsProcessService, providedIn: "root" });
    return AppsProcessService;
}());
export { AppsProcessService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AppsProcessService.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    AppsProcessService.prototype.logService;
}
