// @alfresco/adf-core/esm5/services/auth-guard-base.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AppConfigValues } from '../app-config/app-config.service';
/**
 * @abstract
 */
var /**
 * @abstract
 */
AuthGuardBase = /** @class */ (function () {
    function AuthGuardBase(authenticationService, router, appConfigService) {
        this.authenticationService = authenticationService;
        this.router = router;
        this.appConfigService = appConfigService;
    }
    Object.defineProperty(AuthGuardBase.prototype, "withCredentials", {
        get: /**
         * @protected
         * @return {?}
         */
        function () {
            return this.appConfigService.get('auth.withCredentials', false);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    AuthGuardBase.prototype.canActivate = /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    function (route, state) {
        return this.checkLogin(route, state.url);
    };
    /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    AuthGuardBase.prototype.canActivateChild = /**
     * @param {?} route
     * @param {?} state
     * @return {?}
     */
    function (route, state) {
        return this.canActivate(route, state);
    };
    /**
     * @protected
     * @param {?} provider
     * @param {?} url
     * @return {?}
     */
    AuthGuardBase.prototype.redirectToUrl = /**
     * @protected
     * @param {?} provider
     * @param {?} url
     * @return {?}
     */
    function (provider, url) {
        this.authenticationService.setRedirect({ provider: provider, url: url });
        /** @type {?} */
        var pathToLogin = this.getLoginRoute();
        /** @type {?} */
        var urlToRedirect = "/" + pathToLogin + "?redirectUrl=" + url;
        this.router.navigateByUrl(urlToRedirect);
    };
    /**
     * @protected
     * @return {?}
     */
    AuthGuardBase.prototype.getLoginRoute = /**
     * @protected
     * @return {?}
     */
    function () {
        return (this.appConfigService &&
            this.appConfigService.get(AppConfigValues.LOGIN_ROUTE, 'login'));
    };
    /**
     * @protected
     * @return {?}
     */
    AuthGuardBase.prototype.isOAuthWithoutSilentLogin = /**
     * @protected
     * @return {?}
     */
    function () {
        /** @type {?} */
        var oauth = this.appConfigService.get(AppConfigValues.OAUTHCONFIG, null);
        return (this.authenticationService.isOauth() && oauth.silentLogin === false);
    };
    return AuthGuardBase;
}());
/**
 * @abstract
 */
export { AuthGuardBase };
if (false) {
    /**
     * @type {?}
     * @protected
     */
    AuthGuardBase.prototype.authenticationService;
    /**
     * @type {?}
     * @protected
     */
    AuthGuardBase.prototype.router;
    /**
     * @type {?}
     * @protected
     */
    AuthGuardBase.prototype.appConfigService;
    /**
     * @abstract
     * @param {?} activeRoute
     * @param {?} redirectUrl
     * @return {?}
     */
    AuthGuardBase.prototype.checkLogin = function (activeRoute, redirectUrl) { };
}
