// @alfresco/adf-core/esm5/services/cookie.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
var CookieService = /** @class */ (function () {
    function CookieService() {
    }
    /**
     * Checks if cookies are enabled.
     * @returns True if enabled, false otherwise
     */
    /**
     * Checks if cookies are enabled.
     * @return {?} True if enabled, false otherwise
     */
    CookieService.prototype.isEnabled = /**
     * Checks if cookies are enabled.
     * @return {?} True if enabled, false otherwise
     */
    function () {
        // for certain scenarios Chrome may say 'true' but have cookies still disabled
        if (navigator.cookieEnabled === false) {
            return false;
        }
        document.cookie = 'test-cookie';
        return document.cookie.indexOf('test-cookie') >= 0;
    };
    /**
     * Retrieves a cookie by its key.
     * @param key Key to identify the cookie
     * @returns The cookie data or null if it is not found
     */
    /**
     * Retrieves a cookie by its key.
     * @param {?} key Key to identify the cookie
     * @return {?} The cookie data or null if it is not found
     */
    CookieService.prototype.getItem = /**
     * Retrieves a cookie by its key.
     * @param {?} key Key to identify the cookie
     * @return {?} The cookie data or null if it is not found
     */
    function (key) {
        /** @type {?} */
        var regexp = new RegExp('(?:' + key + '|;\s*' + key + ')=(.*?)(?:;|$)', 'g');
        /** @type {?} */
        var result = regexp.exec(document.cookie);
        return (result === null) ? null : result[1];
    };
    /**
     * Sets a cookie.
     * @param key Key to identify the cookie
     * @param data Data value to set for the cookie
     * @param expiration Expiration date of the data
     * @param path "Pathname" to store the cookie
     */
    /**
     * Sets a cookie.
     * @param {?} key Key to identify the cookie
     * @param {?} data Data value to set for the cookie
     * @param {?} expiration Expiration date of the data
     * @param {?} path "Pathname" to store the cookie
     * @return {?}
     */
    CookieService.prototype.setItem = /**
     * Sets a cookie.
     * @param {?} key Key to identify the cookie
     * @param {?} data Data value to set for the cookie
     * @param {?} expiration Expiration date of the data
     * @param {?} path "Pathname" to store the cookie
     * @return {?}
     */
    function (key, data, expiration, path) {
        document.cookie = key + "=" + data +
            (expiration ? ';expires=' + expiration.toUTCString() : '') +
            (path ? ";path=" + path : ';path=/');
    };
    /** Placeholder for testing purposes - do not use. */
    /**
     * Placeholder for testing purposes - do not use.
     * @return {?}
     */
    CookieService.prototype.clear = /**
     * Placeholder for testing purposes - do not use.
     * @return {?}
     */
    function () {
        /* placeholder for testing purposes */
    };
    CookieService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */ CookieService.ngInjectableDef = i0.defineInjectable({ factory: function CookieService_Factory() { return new CookieService(); }, token: CookieService, providedIn: "root" });
    return CookieService;
}());
export { CookieService };
