// @alfresco/adf-core/esm5/services/discovery-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { BpmProductVersionModel, EcmProductVersionModel } from '../models/product-version.model';
import { AlfrescoApiService } from './alfresco-api.service';
import { map, catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
var DiscoveryApiService = /** @class */ (function () {
    function DiscoveryApiService(apiService) {
        this.apiService = apiService;
    }
    /**
     * Gets product information for Content Services.
     * @returns ProductVersionModel containing product details
     */
    /**
     * Gets product information for Content Services.
     * @return {?} ProductVersionModel containing product details
     */
    DiscoveryApiService.prototype.getEcmProductInfo = /**
     * Gets product information for Content Services.
     * @return {?} ProductVersionModel containing product details
     */
    function () {
        return from(this.apiService.getInstance().discovery.discoveryApi.getRepositoryInformation())
            .pipe(map((/**
         * @param {?} res
         * @return {?}
         */
        function (res) { return new EcmProductVersionModel(res); })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return throwError(err); })));
    };
    /**
     * Gets product information for Process Services.
     * @returns ProductVersionModel containing product details
     */
    /**
     * Gets product information for Process Services.
     * @return {?} ProductVersionModel containing product details
     */
    DiscoveryApiService.prototype.getBpmProductInfo = /**
     * Gets product information for Process Services.
     * @return {?} ProductVersionModel containing product details
     */
    function () {
        return from(this.apiService.getInstance().activiti.aboutApi.getAppVersion())
            .pipe(map((/**
         * @param {?} res
         * @return {?}
         */
        function (res) { return new BpmProductVersionModel(res); })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return throwError(err); })));
    };
    DiscoveryApiService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    DiscoveryApiService.ctorParameters = function () { return [
        { type: AlfrescoApiService }
    ]; };
    /** @nocollapse */ DiscoveryApiService.ngInjectableDef = i0.defineInjectable({ factory: function DiscoveryApiService_Factory() { return new DiscoveryApiService(i0.inject(i1.AlfrescoApiService)); }, token: DiscoveryApiService, providedIn: "root" });
    return DiscoveryApiService;
}());
export { DiscoveryApiService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DiscoveryApiService.prototype.apiService;
}
