// @alfresco/adf-core/esm5/services/download-zip.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { LogService } from './log.service';
import { AlfrescoApiService } from './alfresco-api.service';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as i2 from "./log.service";
var DownloadZipService = /** @class */ (function () {
    function DownloadZipService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    /**
     * Creates a new download.
     * @param payload Object containing the node IDs of the items to add to the ZIP file
     * @returns Status object for the download
     */
    /**
     * Creates a new download.
     * @param {?} payload Object containing the node IDs of the items to add to the ZIP file
     * @return {?} Status object for the download
     */
    DownloadZipService.prototype.createDownload = /**
     * Creates a new download.
     * @param {?} payload Object containing the node IDs of the items to add to the ZIP file
     * @return {?} Status object for the download
     */
    function (payload) {
        var _this = this;
        return from(this.apiService.getInstance().core.downloadsApi.createDownload(payload)).pipe(catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Gets a content URL for the given node.
     * @param nodeId Node to get URL for.
     * @param attachment Toggles whether to retrieve content as an attachment for download
     * @returns URL string
     */
    /**
     * Gets a content URL for the given node.
     * @param {?} nodeId Node to get URL for.
     * @param {?=} attachment Toggles whether to retrieve content as an attachment for download
     * @return {?} URL string
     */
    DownloadZipService.prototype.getContentUrl = /**
     * Gets a content URL for the given node.
     * @param {?} nodeId Node to get URL for.
     * @param {?=} attachment Toggles whether to retrieve content as an attachment for download
     * @return {?} URL string
     */
    function (nodeId, attachment) {
        return this.apiService.getInstance().content.getContentUrl(nodeId, attachment);
    };
    /**
     * Gets a Node via its node ID.
     * @param nodeId ID of the target node
     * @returns Details of the node
     */
    /**
     * Gets a Node via its node ID.
     * @param {?} nodeId ID of the target node
     * @return {?} Details of the node
     */
    DownloadZipService.prototype.getNode = /**
     * Gets a Node via its node ID.
     * @param {?} nodeId ID of the target node
     * @return {?} Details of the node
     */
    function (nodeId) {
        return from(this.apiService.getInstance().core.nodesApi.getNode(nodeId));
    };
    /**
     * Gets status information for a download node.
     * @param downloadId ID of the download node
     * @returns Status object for the download
     */
    /**
     * Gets status information for a download node.
     * @param {?} downloadId ID of the download node
     * @return {?} Status object for the download
     */
    DownloadZipService.prototype.getDownload = /**
     * Gets status information for a download node.
     * @param {?} downloadId ID of the download node
     * @return {?} Status object for the download
     */
    function (downloadId) {
        return from(this.apiService.getInstance().core.downloadsApi.getDownload(downloadId));
    };
    /**
     * Cancels a download.
     * @param downloadId ID of the target download node
     */
    /**
     * Cancels a download.
     * @param {?} downloadId ID of the target download node
     * @return {?}
     */
    DownloadZipService.prototype.cancelDownload = /**
     * Cancels a download.
     * @param {?} downloadId ID of the target download node
     * @return {?}
     */
    function (downloadId) {
        this.apiService.getInstance().core.downloadsApi.cancelDownload(downloadId);
    };
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    DownloadZipService.prototype.handleError = /**
     * @private
     * @param {?} error
     * @return {?}
     */
    function (error) {
        this.logService.error(error);
        return throwError(error || 'Server error');
    };
    DownloadZipService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    DownloadZipService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: LogService }
    ]; };
    /** @nocollapse */ DownloadZipService.ngInjectableDef = i0.defineInjectable({ factory: function DownloadZipService_Factory() { return new DownloadZipService(i0.inject(i1.AlfrescoApiService), i0.inject(i2.LogService)); }, token: DownloadZipService, providedIn: "root" });
    return DownloadZipService;
}());
export { DownloadZipService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DownloadZipService.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    DownloadZipService.prototype.logService;
}
