// @alfresco/adf-core/esm5/services/favorites-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, of } from 'rxjs';
import { AlfrescoApiService } from './alfresco-api.service';
import { UserPreferencesService } from './user-preferences.service';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as i2 from "./user-preferences.service";
var FavoritesApiService = /** @class */ (function () {
    function FavoritesApiService(apiService, preferences) {
        this.apiService = apiService;
        this.preferences = preferences;
    }
    /**
     * @param {?} __0
     * @return {?}
     */
    FavoritesApiService.remapEntry = /**
     * @param {?} __0
     * @return {?}
     */
    function (_a) {
        var entry = _a.entry;
        entry.properties = {
            'cm:title': entry.title,
            'cm:description': entry.description
        };
        return { entry: entry };
    };
    /**
     * @param {?=} data
     * @return {?}
     */
    FavoritesApiService.prototype.remapFavoritesData = /**
     * @param {?=} data
     * @return {?}
     */
    function (data) {
        if (data === void 0) { data = {}; }
        /** @type {?} */
        var list = (data.list || {});
        /** @type {?} */
        var pagination = (list.pagination || {});
        /** @type {?} */
        var entries = this
            .remapFavoriteEntries(list.entries || []);
        return (/** @type {?} */ ({
            list: { entries: entries, pagination: pagination }
        }));
    };
    /**
     * @param {?} entries
     * @return {?}
     */
    FavoritesApiService.prototype.remapFavoriteEntries = /**
     * @param {?} entries
     * @return {?}
     */
    function (entries) {
        return entries
            .map((/**
         * @param {?} __0
         * @return {?}
         */
        function (_a) {
            var target = _a.entry.target;
            return ({
                entry: target.file || target.folder
            });
        }))
            .filter((/**
         * @param {?} __0
         * @return {?}
         */
        function (_a) {
            var entry = _a.entry;
            return (!!entry);
        }))
            .map(FavoritesApiService.remapEntry);
    };
    Object.defineProperty(FavoritesApiService.prototype, "favoritesApi", {
        get: /**
         * @private
         * @return {?}
         */
        function () {
            return this.apiService.getInstance().core.favoritesApi;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * Gets the favorites for a user.
     * @param personId ID of the user
     * @param options Options supported by JS-API
     * @returns List of favorites
     */
    /**
     * Gets the favorites for a user.
     * @param {?} personId ID of the user
     * @param {?=} options Options supported by JS-API
     * @return {?} List of favorites
     */
    FavoritesApiService.prototype.getFavorites = /**
     * Gets the favorites for a user.
     * @param {?} personId ID of the user
     * @param {?=} options Options supported by JS-API
     * @return {?} List of favorites
     */
    function (personId, options) {
        /** @type {?} */
        var defaultOptions = {
            maxItems: this.preferences.paginationSize,
            skipCount: 0,
            where: '(EXISTS(target/file) OR EXISTS(target/folder))',
            include: ['properties', 'allowableOperations']
        };
        /** @type {?} */
        var queryOptions = Object.assign(defaultOptions, options);
        /** @type {?} */
        var promise = this.favoritesApi
            .getFavorites(personId, queryOptions)
            .then(this.remapFavoritesData);
        return from(promise).pipe(catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return of(err); })));
    };
    FavoritesApiService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    FavoritesApiService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: UserPreferencesService }
    ]; };
    /** @nocollapse */ FavoritesApiService.ngInjectableDef = i0.defineInjectable({ factory: function FavoritesApiService_Factory() { return new FavoritesApiService(i0.inject(i1.AlfrescoApiService), i0.inject(i2.UserPreferencesService)); }, token: FavoritesApiService, providedIn: "root" });
    return FavoritesApiService;
}());
export { FavoritesApiService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    FavoritesApiService.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    FavoritesApiService.prototype.preferences;
}
