// @alfresco/adf-core/esm5/services/lock.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { AlfrescoApiService } from './alfresco-api.service';
import moment from 'moment-es6';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
var LockService = /** @class */ (function () {
    function LockService(alfrescoApiService) {
        this.alfrescoApiService = alfrescoApiService;
    }
    /**
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.isLocked = /**
     * @param {?} node
     * @return {?}
     */
    function (node) {
        /** @type {?} */
        var isLocked = false;
        if (this.hasLockConfigured(node)) {
            if (this.isReadOnlyLock(node)) {
                isLocked = true;
                if (this.isLockExpired(node)) {
                    isLocked = false;
                }
            }
            else if (this.isLockOwnerAllowed(node)) {
                isLocked = this.alfrescoApiService.getInstance().getEcmUsername() !== node.properties['cm:lockOwner'].id;
                if (this.isLockExpired(node)) {
                    isLocked = false;
                }
            }
        }
        return isLocked;
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.hasLockConfigured = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        return node.isFile && node.isLocked && node.properties['cm:lockType'];
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.isReadOnlyLock = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        return node.properties['cm:lockType'] === 'READ_ONLY_LOCK' && node.properties['cm:lockLifetime'] === 'PERSISTENT';
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.isLockOwnerAllowed = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        return node.properties['cm:lockType'] === 'WRITE_LOCK' && node.properties['cm:lockLifetime'] === 'PERSISTENT';
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.getLockExpiryTime = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        if (node.properties['cm:expiryDate']) {
            return moment(node.properties['cm:expiryDate'], 'yyyy-MM-ddThh:mm:ssZ');
        }
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    LockService.prototype.isLockExpired = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        /** @type {?} */
        var expiryLockTime = this.getLockExpiryTime(node);
        return moment().isAfter(expiryLockTime);
    };
    LockService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    LockService.ctorParameters = function () { return [
        { type: AlfrescoApiService }
    ]; };
    /** @nocollapse */ LockService.ngInjectableDef = i0.defineInjectable({ factory: function LockService_Factory() { return new LockService(i0.inject(i1.AlfrescoApiService)); }, token: LockService, providedIn: "root" });
    return LockService;
}());
export { LockService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    LockService.prototype.alfrescoApiService;
}
