// @alfresco/adf-core/esm5/services/people-content.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, of } from 'rxjs';
import { AlfrescoApiService } from './alfresco-api.service';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
var PeopleContentService = /** @class */ (function () {
    function PeopleContentService(apiService) {
        this.apiService = apiService;
    }
    Object.defineProperty(PeopleContentService.prototype, "peopleApi", {
        get: /**
         * @private
         * @return {?}
         */
        function () {
            return this.apiService.getInstance().core.peopleApi;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * Gets information about a user identified by their username.
     * @param personId ID of the target user
     * @returns User information
     */
    /**
     * Gets information about a user identified by their username.
     * @param {?} personId ID of the target user
     * @return {?} User information
     */
    PeopleContentService.prototype.getPerson = /**
     * Gets information about a user identified by their username.
     * @param {?} personId ID of the target user
     * @return {?} User information
     */
    function (personId) {
        /** @type {?} */
        var promise = this.peopleApi.getPerson(personId);
        return from(promise).pipe(catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return of(err); })));
    };
    /**
     * Gets information about the user who is currently logged in.
     * @returns User information
     */
    /**
     * Gets information about the user who is currently logged in.
     * @return {?} User information
     */
    PeopleContentService.prototype.getCurrentPerson = /**
     * Gets information about the user who is currently logged in.
     * @return {?} User information
     */
    function () {
        return this.getPerson('-me-');
    };
    PeopleContentService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    PeopleContentService.ctorParameters = function () { return [
        { type: AlfrescoApiService }
    ]; };
    /** @nocollapse */ PeopleContentService.ngInjectableDef = i0.defineInjectable({ factory: function PeopleContentService_Factory() { return new PeopleContentService(i0.inject(i1.AlfrescoApiService)); }, token: PeopleContentService, providedIn: "root" });
    return PeopleContentService;
}());
export { PeopleContentService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    PeopleContentService.prototype.apiService;
}
