// @alfresco/adf-core/esm5/services/previous-route.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { Router, NavigationEnd } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
var PreviousRouteService = /** @class */ (function () {
    function PreviousRouteService(router) {
        var _this = this;
        this.router = router;
        this.currentUrl = this.router.url;
        this.router.events.subscribe((/**
         * @param {?} event
         * @return {?}
         */
        function (event) {
            if (event instanceof NavigationEnd) {
                _this.previousUrl = _this.currentUrl;
                _this.currentUrl = event.url;
            }
        }));
    }
    /**
     * @return {?}
     */
    PreviousRouteService.prototype.getPreviousUrl = /**
     * @return {?}
     */
    function () {
        return this.previousUrl;
    };
    PreviousRouteService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    PreviousRouteService.ctorParameters = function () { return [
        { type: Router }
    ]; };
    /** @nocollapse */ PreviousRouteService.ngInjectableDef = i0.defineInjectable({ factory: function PreviousRouteService_Factory() { return new PreviousRouteService(i0.inject(i1.Router)); }, token: PreviousRouteService, providedIn: "root" });
    return PreviousRouteService;
}());
export { PreviousRouteService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    PreviousRouteService.prototype.previousUrl;
    /**
     * @type {?}
     * @private
     */
    PreviousRouteService.prototype.currentUrl;
    /**
     * @type {?}
     * @private
     */
    PreviousRouteService.prototype.router;
}
