// @alfresco/adf-content-services/esm2015/document-list/models/content-action.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class ContentActionModel {
    /**
     * @param {?=} obj
     */
    constructor(obj) {
        this.disableWithNoPermission = false;
        this.disabled = false;
        this.visible = true;
        if (obj) {
            this.icon = obj.icon;
            this.title = obj.title;
            this.handler = obj.handler;
            this.execute = obj.execute;
            this.target = obj.target;
            this.permission = obj.permission;
            this.disableWithNoPermission = obj.disableWithNoPermission;
            if (obj.hasOwnProperty('disabled')) {
                this.disabled = obj.disabled;
            }
            if (obj.hasOwnProperty('visible')) {
                this.visible = obj.visible;
            }
        }
    }
}
if (false) {
    /** @type {?} */
    ContentActionModel.prototype.icon;
    /** @type {?} */
    ContentActionModel.prototype.title;
    /** @type {?} */
    ContentActionModel.prototype.handler;
    /** @type {?} */
    ContentActionModel.prototype.execute;
    /** @type {?} */
    ContentActionModel.prototype.target;
    /** @type {?} */
    ContentActionModel.prototype.permission;
    /** @type {?} */
    ContentActionModel.prototype.disableWithNoPermission;
    /** @type {?} */
    ContentActionModel.prototype.disabled;
    /** @type {?} */
    ContentActionModel.prototype.visible;
}
/** @enum {string} */
const ContentActionTarget = {
    Document: 'document',
    Folder: 'folder',
    All: 'all',
};
export { ContentActionTarget };
export class DocumentActionModel extends ContentActionModel {
    /**
     * @param {?=} json
     */
    constructor(json) {
        super(json);
        this.target = 'document';
    }
}
export class FolderActionModel extends ContentActionModel {
    /**
     * @param {?=} json
     */
    constructor(json) {
        super(json);
        this.target = 'folder';
    }
}
