// @alfresco/adf-core/esm5/login/components/login-dialog.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, ViewEncapsulation, ViewChild } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material';
import { LoginDialogPanelComponent } from './login-dialog-panel.component';
var LoginDialogComponent = /** @class */ (function () {
    function LoginDialogComponent(data) {
        this.data = data;
        this.buttonActionName = '';
        this.buttonActionName = data.actionName ? "LOGIN.DIALOG." + data.actionName.toUpperCase() : 'LOGIN.DIALOG.CHOOSE';
    }
    /**
     * @return {?}
     */
    LoginDialogComponent.prototype.close = /**
     * @return {?}
     */
    function () {
        this.data.logged.complete();
    };
    /**
     * @return {?}
     */
    LoginDialogComponent.prototype.submitForm = /**
     * @return {?}
     */
    function () {
        this.loginPanel.submitForm();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    LoginDialogComponent.prototype.onLoginSuccess = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.data.logged.next(event);
        this.close();
    };
    /**
     * @return {?}
     */
    LoginDialogComponent.prototype.isFormValid = /**
     * @return {?}
     */
    function () {
        return this.loginPanel ? this.loginPanel.isValid() : false;
    };
    LoginDialogComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-login-dialog',
                    template: "<header\n    mat-dialog-title\n    data-automation-id=\"login-dialog-title\">{{data?.title}}\n</header>\n\n<mat-dialog-content class=\"adf-login-dialog-content\">\n    <adf-login-dialog-panel #adfLoginPanel\n                            (success)=\"onLoginSuccess($event)\">\n    </adf-login-dialog-panel>\n</mat-dialog-content>\n\n<mat-dialog-actions align=\"end\">\n    <button\n        mat-button\n        (click)=\"close()\"\n        data-automation-id=\"login-dialog-actions-cancel\">{{ 'LOGIN.DIALOG.CANCEL' | translate }}\n    </button>\n\n    <button mat-button\n        class=\"choose-action\"\n        data-automation-id=\"login-dialog-actions-perform\"\n        [disabled]=\"!isFormValid()\"\n        (click)=\"submitForm()\">{{ buttonActionName | translate}}\n    </button>\n</mat-dialog-actions>\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    LoginDialogComponent.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] }
    ]; };
    LoginDialogComponent.propDecorators = {
        loginPanel: [{ type: ViewChild, args: ['adfLoginPanel',] }]
    };
    return LoginDialogComponent;
}());
export { LoginDialogComponent };
if (false) {
    /** @type {?} */
    LoginDialogComponent.prototype.loginPanel;
    /** @type {?} */
    LoginDialogComponent.prototype.buttonActionName;
    /** @type {?} */
    LoginDialogComponent.prototype.data;
}
