// @alfresco/adf-core/esm2015/sorting-picker/sorting-picker.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input, EventEmitter, Output } from '@angular/core';
export class SortingPickerComponent {
    constructor() {
        /**
         * Available sorting options
         */
        this.options = [];
        /**
         * Current sorting direction
         */
        this.ascending = true;
        /**
         * Raised each time sorting key gets changed.
         */
        this.valueChange = new EventEmitter();
        /**
         * Raised each time direction gets changed.
         */
        this.sortingChange = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onOptionChanged(event) {
        this.selected = event.value;
        this.valueChange.emit(this.selected);
    }
    /**
     * @return {?}
     */
    toggleSortDirection() {
        this.ascending = !this.ascending;
        this.sortingChange.emit(this.ascending);
    }
}
SortingPickerComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-sorting-picker',
                template: "<mat-form-field>\n    <mat-select [(value)]=\"selected\" (selectionChange)=\"onOptionChanged($event)\">\n        <mat-option *ngFor=\"let option of options\" [value]=\"option.key\">\n            {{ option.label | translate }}\n        </mat-option>\n    </mat-select>\n</mat-form-field>\n\n<button *ngIf=\"selected\" mat-icon-button (click)=\"toggleSortDirection()\">\n    <mat-icon *ngIf=\"ascending\">arrow_upward</mat-icon>\n    <mat-icon *ngIf=\"!ascending\">arrow_downward</mat-icon>\n</button>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-sorting-picker' }
            }] }
];
SortingPickerComponent.propDecorators = {
    options: [{ type: Input }],
    selected: [{ type: Input }],
    ascending: [{ type: Input }],
    valueChange: [{ type: Output }],
    sortingChange: [{ type: Output }]
};
if (false) {
    /**
     * Available sorting options
     * @type {?}
     */
    SortingPickerComponent.prototype.options;
    /**
     * Currently selected option key
     * @type {?}
     */
    SortingPickerComponent.prototype.selected;
    /**
     * Current sorting direction
     * @type {?}
     */
    SortingPickerComponent.prototype.ascending;
    /**
     * Raised each time sorting key gets changed.
     * @type {?}
     */
    SortingPickerComponent.prototype.valueChange;
    /**
     * Raised each time direction gets changed.
     * @type {?}
     */
    SortingPickerComponent.prototype.sortingChange;
}
