// @alfresco/adf-process-services/esm2015/people/components/people-selector/people-selector.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewChild, ViewEncapsulation, EventEmitter, Input, Output } from '@angular/core';
import { PeopleProcessService, UserProcessModel, LogService, TranslationService } from '@alfresco/adf-core';
import { PeopleSearchFieldComponent } from '../people-search-field/people-search-field.component';
import { getDisplayUser } from '../../helpers/getDisplayUser';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
/** @type {?} */
const DEFAULT_ASSIGNEE_PLACEHOLDER = 'ADF_TASK_LIST.PEOPLE.ASSIGNEE';
export class PeopleSelectorComponent {
    /**
     * @param {?} peopleProcessService
     * @param {?} logService
     * @param {?} translationService
     */
    constructor(peopleProcessService, logService, translationService) {
        this.peopleProcessService = peopleProcessService;
        this.logService = logService;
        this.translationService = translationService;
        this.peopleIdChange = new EventEmitter();
        this.performSearch = this.searchUser.bind(this);
        this.defaultPlaceholder = this.translationService.instant(DEFAULT_ASSIGNEE_PLACEHOLDER);
    }
    /**
     * @param {?} searchWord
     * @return {?}
     */
    searchUser(searchWord) {
        return this.peopleProcessService.getWorkflowUsers(undefined, searchWord)
            .pipe(catchError(this.onSearchUserError.bind(this)));
    }
    /**
     * @private
     * @return {?}
     */
    onSearchUserError() {
        this.logService.error('getWorkflowUsers threw error');
        return of([]);
    }
    /**
     * @param {?} user
     * @return {?}
     */
    userSelected(user) {
        this.updateUserSelection(user);
    }
    /**
     * @return {?}
     */
    userDeselected() {
        this.updateUserSelection(undefined);
    }
    /**
     * @private
     * @param {?} user
     * @return {?}
     */
    updateUserSelection(user) {
        this.selectedUser = user;
        this.peopleIdChange.emit(user && user.id || undefined);
        this.searchFieldComponent.reset();
    }
    /**
     * @return {?}
     */
    get placeholder() {
        if (!this.selectedUser) {
            return this.defaultPlaceholder;
        }
        return getDisplayUser(this.selectedUser.firstName, this.selectedUser.lastName, ' ');
    }
}
PeopleSelectorComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-people-selector',
                template: "<adf-people-search-field\n    #peopleSearchField\n    class=\"adf-people-selector-field\"\n    [performSearch]=\"performSearch\"\n    [placeholder]=\"placeholder\"\n    (rowClick)=\"userSelected($event)\">\n</adf-people-search-field>\n<button\n    *ngIf=\"selectedUser\"\n    mat-icon-button\n    class=\"adf-people-selector-deselect\"\n    data-automation-id=\"adf-people-selector-deselect\"\n    (click)=\"userDeselected()\">\n        <mat-icon>cancel</mat-icon>\n</button>\n",
                host: { 'class': 'adf-people-selector' },
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            }] }
];
/** @nocollapse */
PeopleSelectorComponent.ctorParameters = () => [
    { type: PeopleProcessService },
    { type: LogService },
    { type: TranslationService }
];
PeopleSelectorComponent.propDecorators = {
    peopleId: [{ type: Input }],
    peopleIdChange: [{ type: Output }],
    searchFieldComponent: [{ type: ViewChild, args: ['peopleSearchField',] }]
};
if (false) {
    /** @type {?} */
    PeopleSelectorComponent.prototype.peopleId;
    /** @type {?} */
    PeopleSelectorComponent.prototype.peopleIdChange;
    /** @type {?} */
    PeopleSelectorComponent.prototype.searchFieldComponent;
    /** @type {?} */
    PeopleSelectorComponent.prototype.performSearch;
    /** @type {?} */
    PeopleSelectorComponent.prototype.selectedUser;
    /** @type {?} */
    PeopleSelectorComponent.prototype.defaultPlaceholder;
    /**
     * @type {?}
     * @private
     */
    PeopleSelectorComponent.prototype.peopleProcessService;
    /**
     * @type {?}
     * @private
     */
    PeopleSelectorComponent.prototype.logService;
    /**
     * @type {?}
     * @private
     */
    PeopleSelectorComponent.prototype.translationService;
}
