// @alfresco/adf-extensions/esm5/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { ExtensionsModule } from './lib/extensions.module';
export { ContentActionType } from './lib/config/action.extensions';
export {} from './lib/config/document-list.extensions';
export {} from './lib/config/extension-element';
export { getValue, filterEnabled, sortByOrder, reduceSeparators, reduceEmptyMenus, mergeObjects, mergeArrays } from './lib/config/extension-utils';
export {} from './lib/config/extension.config';
export {} from './lib/config/icon.extensions';
export {} from './lib/config/navbar.extensions';
export {} from './lib/config/permission.extensions';
export {} from './lib/config/routing.extensions';
export { RuleRef } from './lib/config/rule.extensions';
export {} from './lib/config/sidebar.extensions';
export {} from './lib/config/viewer.extensions';
export { ExtensionLoaderService } from './lib/services/extension-loader.service';
export { ExtensionService } from './lib/services/extension.service';
export { ComponentRegisterService } from './lib/services/component-register.service';
export { AppExtensionService } from './lib/services/app-extension.service';
export {} from './lib/store/states/navigation.state';
export {} from './lib/store/states/profile.state';
export {} from './lib/store/states/selection.state';
export {} from './lib/store/states/repository.state';
export { PreviewExtensionComponent, DynamicColumnComponent, DynamicExtensionComponent, DynamicTabComponent } from './lib/components/public-api';
