// @alfresco/adf-core/esm5/form/services/activiti-alfresco.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiService } from '../../services/alfresco-api.service';
import { LogService } from '../../services/log.service';
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../services/alfresco-api.service";
import * as i2 from "../../services/log.service";
var ActivitiContentService = /** @class */ (function () {
    function ActivitiContentService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    /**
     * Returns a list of child nodes below the specified folder
     *
     * @param accountId
     * @param folderId
     */
    /**
     * Returns a list of child nodes below the specified folder
     *
     * @param {?} accountId
     * @param {?} folderId
     * @return {?}
     */
    ActivitiContentService.prototype.getAlfrescoNodes = /**
     * Returns a list of child nodes below the specified folder
     *
     * @param {?} accountId
     * @param {?} folderId
     * @return {?}
     */
    function (accountId, folderId) {
        var _this = this;
        /** @type {?} */
        var apiService = this.apiService.getInstance();
        /** @type {?} */
        var accountShortId = accountId.replace('alfresco-', '');
        return from(apiService.activiti.alfrescoApi.getContentInFolder(accountShortId, folderId))
            .pipe(map(this.toJsonArray), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Returns a list of all the repositories configured
     *
     * @param accountId
     * @param folderId
     */
    /**
     * Returns a list of all the repositories configured
     *
     * @param {?} tenantId
     * @param {?} includeAccount
     * @return {?}
     */
    ActivitiContentService.prototype.getAlfrescoRepositories = /**
     * Returns a list of all the repositories configured
     *
     * @param {?} tenantId
     * @param {?} includeAccount
     * @return {?}
     */
    function (tenantId, includeAccount) {
        var _this = this;
        /** @type {?} */
        var apiService = this.apiService.getInstance();
        /** @type {?} */
        var opts = {
            tenantId: tenantId,
            includeAccounts: includeAccount
        };
        return from(apiService.activiti.alfrescoApi.getRepositories(opts))
            .pipe(map(this.toJsonArray), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Returns a list of child nodes below the specified folder
     *
     * @param accountId
     * @param node
     * @param siteId
     */
    /**
     * Returns a list of child nodes below the specified folder
     *
     * @param {?} accountId
     * @param {?} node
     * @param {?} siteId
     * @return {?}
     */
    ActivitiContentService.prototype.linkAlfrescoNode = /**
     * Returns a list of child nodes below the specified folder
     *
     * @param {?} accountId
     * @param {?} node
     * @param {?} siteId
     * @return {?}
     */
    function (accountId, node, siteId) {
        var _this = this;
        /** @type {?} */
        var apiService = this.apiService.getInstance();
        return from(apiService.activiti.contentApi.createTemporaryRelatedContent({
            link: true,
            name: node.title,
            simpleType: node.simpleType,
            source: accountId,
            sourceId: node.id + '@' + siteId
        }))
            .pipe(map(this.toJson), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * @param {?} node
     * @param {?} siteId
     * @param {?} accountId
     * @return {?}
     */
    ActivitiContentService.prototype.applyAlfrescoNode = /**
     * @param {?} node
     * @param {?} siteId
     * @param {?} accountId
     * @return {?}
     */
    function (node, siteId, accountId) {
        var _this = this;
        /** @type {?} */
        var apiService = this.apiService.getInstance();
        /** @type {?} */
        var currentSideId = siteId ? siteId : this.getSiteNameFromNodePath(node);
        /** @type {?} */
        var params = {
            source: accountId,
            mimeType: node.content.mimeType,
            sourceId: node.id + ';' + node.properties['cm:versionLabel'] + '@' + currentSideId,
            name: node.name,
            link: false
        };
        return from(apiService.activiti.contentApi.createTemporaryRelatedContent(params))
            .pipe(map(this.toJson), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    ActivitiContentService.prototype.getSiteNameFromNodePath = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        /** @type {?} */
        var siteName = '';
        if (node.path) {
            /** @type {?} */
            var foundNode = node.path
                .elements.find((/**
             * @param {?} pathNode
             * @return {?}
             */
            function (pathNode) {
                return pathNode.nodeType === 'st:site' &&
                    pathNode.name !== 'Sites';
            }));
            siteName = foundNode ? foundNode.name : '';
        }
        return siteName.toLocaleLowerCase();
    };
    /**
     * @param {?} res
     * @return {?}
     */
    ActivitiContentService.prototype.toJson = /**
     * @param {?} res
     * @return {?}
     */
    function (res) {
        if (res) {
            return res || {};
        }
        return {};
    };
    /**
     * @param {?} res
     * @return {?}
     */
    ActivitiContentService.prototype.toJsonArray = /**
     * @param {?} res
     * @return {?}
     */
    function (res) {
        if (res) {
            return res.data || [];
        }
        return [];
    };
    /**
     * @param {?} error
     * @return {?}
     */
    ActivitiContentService.prototype.handleError = /**
     * @param {?} error
     * @return {?}
     */
    function (error) {
        /** @type {?} */
        var errMsg = ActivitiContentService.UNKNOWN_ERROR_MESSAGE;
        if (error) {
            errMsg = (error.message) ? error.message :
                error.status ? error.status + " - " + error.statusText : ActivitiContentService.GENERIC_ERROR_MESSAGE;
        }
        this.logService.error(errMsg);
        return throwError(errMsg);
    };
    ActivitiContentService.UNKNOWN_ERROR_MESSAGE = 'Unknown error';
    ActivitiContentService.GENERIC_ERROR_MESSAGE = 'Server error';
    ActivitiContentService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    ActivitiContentService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: LogService }
    ]; };
    /** @nocollapse */ ActivitiContentService.ngInjectableDef = i0.defineInjectable({ factory: function ActivitiContentService_Factory() { return new ActivitiContentService(i0.inject(i1.AlfrescoApiService), i0.inject(i2.LogService)); }, token: ActivitiContentService, providedIn: "root" });
    return ActivitiContentService;
}());
export { ActivitiContentService };
if (false) {
    /** @type {?} */
    ActivitiContentService.UNKNOWN_ERROR_MESSAGE;
    /** @type {?} */
    ActivitiContentService.GENERIC_ERROR_MESSAGE;
    /**
     * @type {?}
     * @private
     */
    ActivitiContentService.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    ActivitiContentService.prototype.logService;
}
