// @alfresco/adf-insights/esm2015/diagram/components/raphael/anchor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class Anchor {
    /**
     * @param {?} uuid
     * @param {?} type
     * @param {?} x
     * @param {?} y
     */
    constructor(uuid, type, x, y) {
        this.uuid = null;
        this.x = 0;
        this.y = 0;
        this.isFirst = false;
        this.isLast = false;
        this.typeIndex = 0;
        this.type = Anchor.ANCHOR_TYPE.main;
        this.uuid = uuid;
        this.x = x;
        this.y = y;
        this.type = (type === Anchor.ANCHOR_TYPE.middle) ? Anchor.ANCHOR_TYPE.middle : Anchor.ANCHOR_TYPE.main;
    }
}
Anchor.ANCHOR_TYPE = {
    main: 'main',
    middle: 'middle',
    first: 'first',
    last: 'last'
};
if (false) {
    /** @type {?} */
    Anchor.ANCHOR_TYPE;
    /** @type {?} */
    Anchor.prototype.uuid;
    /** @type {?} */
    Anchor.prototype.x;
    /** @type {?} */
    Anchor.prototype.y;
    /** @type {?} */
    Anchor.prototype.isFirst;
    /** @type {?} */
    Anchor.prototype.isLast;
    /** @type {?} */
    Anchor.prototype.typeIndex;
    /** @type {?} */
    Anchor.prototype.type;
}
