// @alfresco/adf-insights/esm2015/diagram/components/raphael/raphael-rect.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
export class RaphaelRectDirective extends RaphaelBase {
    /**
     * @param {?} elementRef
     * @param {?} raphaelService
     */
    constructor(elementRef, raphaelService) {
        super(elementRef, raphaelService);
        this.elementRef = elementRef;
        this.radius = 0;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const opts = {
            'stroke-width': this.strokeWidth,
            'fill': this.fillColors,
            'stroke': this.stroke,
            'fill-opacity': this.fillOpacity
        };
        /** @type {?} */
        const elementDraw = this.draw(this.leftCorner, this.width, this.height, this.radius, opts);
        elementDraw.node.id = this.elementId;
    }
    /**
     * @param {?} leftCorner
     * @param {?} width
     * @param {?} height
     * @param {?} radius
     * @param {?} opts
     * @return {?}
     */
    draw(leftCorner, width, height, radius, opts) {
        return this.paper.rect(leftCorner.x, leftCorner.y, width, height, radius).attr(opts);
    }
}
RaphaelRectDirective.decorators = [
    { type: Directive, args: [{ selector: 'adf-raphael-rect, raphael-rect' },] }
];
/** @nocollapse */
RaphaelRectDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: RaphaelService }
];
RaphaelRectDirective.propDecorators = {
    paper: [{ type: Input }],
    leftCorner: [{ type: Input }],
    width: [{ type: Input }],
    height: [{ type: Input }],
    radius: [{ type: Input }],
    fillColors: [{ type: Input }],
    stroke: [{ type: Input }],
    strokeWidth: [{ type: Input }],
    fillOpacity: [{ type: Input }],
    elementId: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    RaphaelRectDirective.prototype.paper;
    /** @type {?} */
    RaphaelRectDirective.prototype.leftCorner;
    /** @type {?} */
    RaphaelRectDirective.prototype.width;
    /** @type {?} */
    RaphaelRectDirective.prototype.height;
    /** @type {?} */
    RaphaelRectDirective.prototype.radius;
    /** @type {?} */
    RaphaelRectDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelRectDirective.prototype.stroke;
    /** @type {?} */
    RaphaelRectDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelRectDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelRectDirective.prototype.elementId;
    /** @type {?} */
    RaphaelRectDirective.prototype.error;
    /** @type {?} */
    RaphaelRectDirective.prototype.elementRef;
}
