// @alfresco/adf-insights/esm2015/diagram/components/raphael/raphael-plus.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
export class RaphaelPlusDirective extends RaphaelBase {
    /**
     * @param {?} elementRef
     * @param {?} raphaelService
     */
    constructor(elementRef, raphaelService) {
        super(elementRef, raphaelService);
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const opts = { 'stroke-width': this.strokeWidth, 'fill': this.fillColors, 'stroke': this.stroke, 'fill-opacity': this.fillOpacity };
        this.draw(this.center, opts);
    }
    /**
     * @param {?} center
     * @param {?=} opts
     * @return {?}
     */
    draw(center, opts) {
        /** @type {?} */
        const path = this.paper.path('M 6.75,16 L 25.75,16 M 16,6.75 L 16,25.75').attr(opts);
        return path.transform('T' + (center.x + 4) + ',' + (center.y + 4));
    }
}
RaphaelPlusDirective.decorators = [
    { type: Directive, args: [{ selector: 'adf-raphael-plus, raphael-plus' },] }
];
/** @nocollapse */
RaphaelPlusDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: RaphaelService }
];
RaphaelPlusDirective.propDecorators = {
    center: [{ type: Input }],
    fillColors: [{ type: Input }],
    stroke: [{ type: Input }],
    strokeWidth: [{ type: Input }],
    fillOpacity: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    RaphaelPlusDirective.prototype.center;
    /** @type {?} */
    RaphaelPlusDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelPlusDirective.prototype.stroke;
    /** @type {?} */
    RaphaelPlusDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelPlusDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelPlusDirective.prototype.error;
    /** @type {?} */
    RaphaelPlusDirective.prototype.elementRef;
}
