// @alfresco/adf-insights/esm2015/diagram/components/raphael/raphael.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
export class RaphaelService {
    constructor() {
        this.width = 300;
        this.height = 400;
    }
    /**
     * @param {?} element
     * @return {?}
     */
    getInstance(element) {
        if (!this.paper) {
            this.ctx = element.nativeElement;
            this.refresh();
        }
        return this.paper;
    }
    /**
     * @private
     * @return {?}
     */
    refresh() {
        this.ngOnDestroy();
        this.paper = this.getPaperBuilder(this.ctx);
    }
    /**
     * @param {?} ctx
     * @return {?}
     */
    getPaperBuilder(ctx) {
        if (typeof Raphael === 'undefined') {
            throw new Error('insights configuration issue: Embedding Chart.js lib is mandatory');
        }
        /** @type {?} */
        const paper = new Raphael(ctx, this.width, this.height);
        return paper;
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        if (this.paper) {
            this.paper.clear();
            this.paper = void 0;
        }
    }
    /**
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    setting(width, height) {
        this.width = width;
        this.height = height;
    }
    /**
     * @return {?}
     */
    reset() {
        this.ngOnDestroy();
    }
}
RaphaelService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
RaphaelService.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    RaphaelService.prototype.paper;
    /** @type {?} */
    RaphaelService.prototype.width;
    /** @type {?} */
    RaphaelService.prototype.height;
    /**
     * @type {?}
     * @private
     */
    RaphaelService.prototype.ctx;
}
