// @alfresco/adf-insights/esm2015/diagram/components/raphael/raphael-cross.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
export class RaphaelCrossDirective extends RaphaelBase {
    /**
     * @param {?} elementRef
     * @param {?} raphaelService
     */
    constructor(elementRef, raphaelService) {
        super(elementRef, raphaelService);
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const opts = { 'stroke-width': this.strokeWidth, 'fill': this.fillColors, 'stroke': this.stroke, 'fill-opacity': this.fillOpacity };
        this.draw(this.center, this.width, this.height, opts);
    }
    /**
     * @param {?} center
     * @param {?} width
     * @param {?} height
     * @param {?=} opts
     * @return {?}
     */
    draw(center, width, height, opts) {
        /** @type {?} */
        const quarterWidth = width / 4;
        /** @type {?} */
        const quarterHeight = height / 4;
        return this.paper.path('M' + (center.x + quarterWidth + 3) + ' ' + (center.y + quarterHeight + 3) +
            'L' + (center.x + 3 * quarterWidth - 3) + ' ' + (center.y + 3 * quarterHeight - 3) +
            'M' + (center.x + quarterWidth + 3) + ' ' + (center.y + 3 * quarterHeight - 3) +
            'L' + (center.x + 3 * quarterWidth - 3) + ' ' + (center.y + quarterHeight + 3)).attr(opts);
    }
}
RaphaelCrossDirective.decorators = [
    { type: Directive, args: [{ selector: 'adf-raphael-cross, raphael-cross' },] }
];
/** @nocollapse */
RaphaelCrossDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: RaphaelService }
];
RaphaelCrossDirective.propDecorators = {
    center: [{ type: Input }],
    width: [{ type: Input }],
    height: [{ type: Input }],
    fillColors: [{ type: Input }],
    stroke: [{ type: Input }],
    strokeWidth: [{ type: Input }],
    fillOpacity: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    RaphaelCrossDirective.prototype.center;
    /** @type {?} */
    RaphaelCrossDirective.prototype.width;
    /** @type {?} */
    RaphaelCrossDirective.prototype.height;
    /** @type {?} */
    RaphaelCrossDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelCrossDirective.prototype.stroke;
    /** @type {?} */
    RaphaelCrossDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelCrossDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelCrossDirective.prototype.error;
    /** @type {?} */
    RaphaelCrossDirective.prototype.elementRef;
}
