// @alfresco/adf-insights/esm2015/diagram/components/raphael/raphael-pentagon.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
export class RaphaelPentagonDirective extends RaphaelBase {
    /**
     * @param {?} elementRef
     * @param {?} raphaelService
     */
    constructor(elementRef, raphaelService) {
        super(elementRef, raphaelService);
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        const opts = {
            'stroke-width': this.strokeWidth,
            'fill': this.fillColors,
            'stroke': this.stroke,
            'fill-opacity': this.fillOpacity,
            'stroke-linejoin': 'bevel'
        };
        this.draw(this.center, opts);
    }
    /**
     * @param {?} center
     * @param {?=} opts
     * @return {?}
     */
    draw(center, opts) {
        /** @type {?} */
        const shape = this.paper.path('M 20.327514,22.344972 L 11.259248,22.344216 L 8.4577203,13.719549' +
            ' L 15.794545,8.389969 L 23.130481,13.720774 L 20.327514,22.344972 z').attr(opts);
        shape.transform('T' + (center.x + 4) + ',' + (center.y + 4));
    }
}
RaphaelPentagonDirective.decorators = [
    { type: Directive, args: [{ selector: 'adf-raphael-pentagon, raphael-pentagon' },] }
];
/** @nocollapse */
RaphaelPentagonDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: RaphaelService }
];
RaphaelPentagonDirective.propDecorators = {
    center: [{ type: Input }],
    fillColors: [{ type: Input }],
    stroke: [{ type: Input }],
    strokeWidth: [{ type: Input }],
    fillOpacity: [{ type: Input }],
    strokeLinejoin: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    RaphaelPentagonDirective.prototype.center;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.stroke;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.strokeLinejoin;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.error;
    /** @type {?} */
    RaphaelPentagonDirective.prototype.elementRef;
}
