// @alfresco/adf-insights/esm5/diagram/components/diagram-sequence-flow.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
var DiagramSequenceFlowComponent = /** @class */ (function () {
    function DiagramSequenceFlowComponent(elementRef) {
        this.elementRef = elementRef;
        this.error = new EventEmitter();
    }
    /**
     * @return {?}
     */
    DiagramSequenceFlowComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DiagramSequenceFlowComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-diagram-sequence-flow',
                    template: "<raphael-flow-arrow [flow]=\"flow\"></raphael-flow-arrow>\n<diagram-tooltip [data]=\"flow\"></diagram-tooltip>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramSequenceFlowComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    DiagramSequenceFlowComponent.propDecorators = {
        flow: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramSequenceFlowComponent;
}());
export { DiagramSequenceFlowComponent };
if (false) {
    /** @type {?} */
    DiagramSequenceFlowComponent.prototype.flow;
    /** @type {?} */
    DiagramSequenceFlowComponent.prototype.error;
    /** @type {?} */
    DiagramSequenceFlowComponent.prototype.elementRef;
}
