// @alfresco/adf-content-services/esm2015/search/components/search-text/search-text.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, Input } from '@angular/core';
export class SearchTextComponent {
    constructor() {
        /**
         * The content of the text box.
         */
        this.value = '';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.context && this.settings && this.settings.pattern) {
            /** @type {?} */
            const pattern = new RegExp(this.settings.pattern, 'g');
            /** @type {?} */
            const match = pattern.exec(this.context.queryFragments[this.id] || '');
            if (match && match.length > 1) {
                this.value = match[1];
            }
        }
    }
    /**
     * @return {?}
     */
    reset() {
        this.value = '';
        this.updateQuery(null);
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onChangedHandler(event) {
        this.value = event.target.value;
        this.updateQuery(this.value);
    }
    /**
     * @private
     * @param {?} value
     * @return {?}
     */
    updateQuery(value) {
        if (this.context && this.settings && this.settings.field) {
            this.context.queryFragments[this.id] = value ? `${this.settings.field}:'${value}'` : '';
            this.context.update();
        }
    }
}
SearchTextComponent.decorators = [
    { type: Component, args: [{
                selector: 'adf-search-text',
                template: "<mat-form-field>\n    <input\n        matInput\n        [placeholder]=\"settings?.placeholder\"\n        [(ngModel)]=\"value\"\n        (change)=\"onChangedHandler($event)\">\n    <button mat-button *ngIf=\"value\" matSuffix mat-icon-button (click)=\"reset()\">\n        <mat-icon>close</mat-icon>\n    </button>\n</mat-form-field>\n",
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-search-text' },
                styles: [".adf-search-text .mat-form-field{width:100%}"]
            }] }
];
SearchTextComponent.propDecorators = {
    value: [{ type: Input }]
};
if (false) {
    /**
     * The content of the text box.
     * @type {?}
     */
    SearchTextComponent.prototype.value;
    /** @type {?} */
    SearchTextComponent.prototype.id;
    /** @type {?} */
    SearchTextComponent.prototype.settings;
    /** @type {?} */
    SearchTextComponent.prototype.context;
}
