// @alfresco/adf-insights/esm2015/analytics-process/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { AnalyticsComponent } from './components/analytics.component';
export { AnalyticsGeneratorComponent } from './components/analytics-generator.component';
export { AnalyticsReportListComponent } from './components/analytics-report-list.component';
export { AnalyticsReportParametersComponent } from './components/analytics-report-parameters.component';
export { AnalyticsService } from './services/analytics.service';
export { AnalyticsProcessModule } from './analytics-process.module';
