// @alfresco/adf-content-services/esm2015/content-metadata/services/config/aspect-oriented-config.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { getGroup, getProperty } from './property-group-reader';
export class AspectOrientedConfigService {
    /**
     * @param {?} config
     */
    constructor(config) {
        this.config = config;
    }
    /**
     * @param {?} groupName
     * @return {?}
     */
    isGroupAllowed(groupName) {
        if (this.isIncludeAllEnabled()) {
            return true;
        }
        /** @type {?} */
        const groupNames = Object.keys(this.config);
        return groupNames.indexOf(groupName) !== -1;
    }
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    reorganiseByConfig(propertyGroups) {
        /** @type {?} */
        const aspects = this.config;
        /** @type {?} */
        const aspectNames = Object.keys(aspects);
        return aspectNames
            .reduce((/**
         * @param {?} groupAccumulator
         * @param {?} aspectName
         * @return {?}
         */
        (groupAccumulator, aspectName) => {
            /** @type {?} */
            const newGroup = this.getOrganisedPropertyGroup(propertyGroups, aspectName);
            return groupAccumulator.concat(newGroup);
        }), [])
            .filter((/**
         * @param {?} organisedPropertyGroup
         * @return {?}
         */
        (organisedPropertyGroup) => organisedPropertyGroup.properties.length > 0));
    }
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    appendAllPreset(propertyGroups) {
        /** @type {?} */
        const groups = Object.keys(propertyGroups)
            .map((/**
         * @param {?} groupName
         * @return {?}
         */
        (groupName) => {
            /** @type {?} */
            const propertyGroup = propertyGroups[groupName];
            /** @type {?} */
            const properties = propertyGroup.properties;
            return Object.assign({}, propertyGroup, {
                properties: Object.keys(properties).map((/**
                 * @param {?} propertyName
                 * @return {?}
                 */
                (propertyName) => properties[propertyName]))
            });
        }));
        return groups;
    }
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    filterExcludedPreset(propertyGroups) {
        if (this.config.exclude) {
            return propertyGroups.filter((/**
             * @param {?} preset
             * @return {?}
             */
            (preset) => {
                return !this.config.exclude.includes(preset.name);
            }));
        }
        return propertyGroups;
    }
    /**
     * @return {?}
     */
    isIncludeAllEnabled() {
        return this.config.includeAll;
    }
    /**
     * @private
     * @param {?} propertyGroups
     * @param {?} aspectName
     * @return {?}
     */
    getOrganisedPropertyGroup(propertyGroups, aspectName) {
        /** @type {?} */
        const group = getGroup(propertyGroups, aspectName);
        /** @type {?} */
        let newGroup = [];
        if (group) {
            /** @type {?} */
            const aspectProperties = this.config[aspectName];
            /** @type {?} */
            let properties;
            if (aspectProperties === '*') {
                properties = getProperty(propertyGroups, aspectName, aspectProperties);
            }
            else {
                properties = ((/** @type {?} */ (aspectProperties)))
                    .map((/**
                 * @param {?} propertyName
                 * @return {?}
                 */
                (propertyName) => getProperty(propertyGroups, aspectName, propertyName)))
                    .filter((/**
                 * @param {?} props
                 * @return {?}
                 */
                (props) => props !== undefined));
            }
            newGroup = [{ title: group.title, properties }];
        }
        return newGroup;
    }
}
if (false) {
    /**
     * @type {?}
     * @private
     */
    AspectOrientedConfigService.prototype.config;
}
