// @alfresco/adf-content-services/esm5/document-list/models/document-library.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
// note: contains only limited subset of available fields
var NodePaging = /** @class */ (function () {
    function NodePaging() {
    }
    return NodePaging;
}());
export { NodePaging };
if (false) {
    /** @type {?} */
    NodePaging.prototype.list;
}
var NodePagingList = /** @class */ (function () {
    function NodePagingList() {
    }
    return NodePagingList;
}());
export { NodePagingList };
if (false) {
    /** @type {?} */
    NodePagingList.prototype.pagination;
    /** @type {?} */
    NodePagingList.prototype.entries;
}
var NodeMinimalEntry = /** @class */ (function () {
    function NodeMinimalEntry() {
    }
    return NodeMinimalEntry;
}());
export { NodeMinimalEntry };
if (false) {
    /** @type {?} */
    NodeMinimalEntry.prototype.entry;
}
var Pagination = /** @class */ (function () {
    function Pagination() {
    }
    return Pagination;
}());
export { Pagination };
if (false) {
    /** @type {?} */
    Pagination.prototype.count;
    /** @type {?} */
    Pagination.prototype.hasMoreItems;
    /** @type {?} */
    Pagination.prototype.totalItems;
    /** @type {?} */
    Pagination.prototype.skipCount;
    /** @type {?} */
    Pagination.prototype.maxItems;
}
var NodeMinimal = /** @class */ (function () {
    function NodeMinimal() {
        this.properties = {};
    }
    return NodeMinimal;
}());
export { NodeMinimal };
if (false) {
    /** @type {?} */
    NodeMinimal.prototype.id;
    /** @type {?} */
    NodeMinimal.prototype.parentId;
    /** @type {?} */
    NodeMinimal.prototype.name;
    /** @type {?} */
    NodeMinimal.prototype.nodeType;
    /** @type {?} */
    NodeMinimal.prototype.isFolder;
    /** @type {?} */
    NodeMinimal.prototype.isFile;
    /** @type {?} */
    NodeMinimal.prototype.modifiedAt;
    /** @type {?} */
    NodeMinimal.prototype.modifiedByUser;
    /** @type {?} */
    NodeMinimal.prototype.createdAt;
    /** @type {?} */
    NodeMinimal.prototype.createdByUser;
    /** @type {?} */
    NodeMinimal.prototype.content;
    /** @type {?} */
    NodeMinimal.prototype.path;
    /** @type {?} */
    NodeMinimal.prototype.properties;
    /** @type {?} */
    NodeMinimal.prototype.aspectNames;
}
var UserInfo = /** @class */ (function () {
    function UserInfo() {
    }
    return UserInfo;
}());
export { UserInfo };
if (false) {
    /** @type {?} */
    UserInfo.prototype.displayName;
    /** @type {?} */
    UserInfo.prototype.id;
}
var ContentInfo = /** @class */ (function () {
    function ContentInfo() {
    }
    return ContentInfo;
}());
export { ContentInfo };
if (false) {
    /** @type {?} */
    ContentInfo.prototype.mimeType;
    /** @type {?} */
    ContentInfo.prototype.mimeTypeName;
    /** @type {?} */
    ContentInfo.prototype.sizeInBytes;
    /** @type {?} */
    ContentInfo.prototype.encoding;
}
var PathInfoEntity = /** @class */ (function () {
    function PathInfoEntity() {
    }
    return PathInfoEntity;
}());
export { PathInfoEntity };
if (false) {
    /** @type {?} */
    PathInfoEntity.prototype.elements;
    /** @type {?} */
    PathInfoEntity.prototype.isComplete;
    /** @type {?} */
    PathInfoEntity.prototype.name;
}
var PathElementEntity = /** @class */ (function () {
    function PathElementEntity() {
    }
    return PathElementEntity;
}());
export { PathElementEntity };
if (false) {
    /** @type {?} */
    PathElementEntity.prototype.id;
    /** @type {?} */
    PathElementEntity.prototype.name;
}
/**
 * @record
 */
export function NodeProperties() { }
