// @alfresco/adf-core/esm5/mock/app-config.service.mock.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { AppConfigService } from '../app-config/app-config.service';
import { HttpClient } from '@angular/common/http';
var AppConfigServiceMock = /** @class */ (function (_super) {
    tslib_1.__extends(AppConfigServiceMock, _super);
    function AppConfigServiceMock(http) {
        var _this = _super.call(this, http) || this;
        _this.config = {
            application: {
                name: 'Alfresco ADF Application'
            },
            ecmHost: 'http://{hostname}{:port}/ecm',
            bpmHost: 'http://{hostname}{:port}/bpm',
            logLevel: 'silent'
        };
        return _this;
    }
    /**
     * @return {?}
     */
    AppConfigServiceMock.prototype.load = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return new Promise((/**
         * @param {?} resolve
         * @return {?}
         */
        function (resolve) {
            _this.onLoadSubject.next(_this.config);
            resolve(_this.config);
        }));
    };
    AppConfigServiceMock.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    AppConfigServiceMock.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    return AppConfigServiceMock;
}(AppConfigService));
export { AppConfigServiceMock };
if (false) {
    /** @type {?} */
    AppConfigServiceMock.prototype.config;
}
