// @alfresco/adf-core/esm5/mock/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { AlfrescoApiMock } from './AlfrescoApi.mock';
export { AppConfigServiceMock } from './app-config.service.mock';
export { fakeApps } from './apps-service.mock';
export { AuthenticationMock } from './authentication.service.mock';
export { fakeBpmUserNoImage, fakeBpmUser, fakeBpmEditedUser } from './bpm-user.service.mock';
export { fakeUser1, fakeUser2, fakeTasksComment, fakeProcessComment } from './comment-process-service.mock';
export { CookieServiceMock } from './cookie.service.mock';
export { fakeEcmCompany, fakeEcmUser, fakeEcmUserNoImage, fakeEcmEditedUser } from './ecm-user.service.mock';
export { EventMock } from './event.mock';
export { fakeRendition, fakeRenditionCreated, fakeRenditionsList, fakeRenditionsListWithACreated } from './renditionsService.mock';
export { fakeSearch, mockError, searchMockApi } from './search.service.mock';
export { TranslationMock } from './translation.service.mock';
export { AlfrescoApiServiceMock } from './alfresco-api.service.mock';
export { fakeForm } from './form/form.component.mock';
export { formDefinitionTwoTextFields, formDefinitionDropdownField, formDefinitionRequiredField } from './form/formDefinition.mock';
export { formReadonlyTwoTextFields } from './form/formDefinitionReadonly.mock';
export { formDefVisibilitiFieldDependsOnNextOne, formDefVisibilitiFieldDependsOnPreviousOne } from './form/formDefinitionVisibility.mock';
export { startFormDateWidgetMock, startFormNumberWidgetMock, startFormAmountWidgetMock, startFormRadioButtonWidgetMock, startFormTextDefinitionMock, startFormDropdownDefinitionMock, startMockForm, startMockFormWithTab } from './form/start-form.component.mock';
export { formModelTabs } from './form/form.service.mock';
export { formTest, fakeTaskProcessVariableModels, formValues, fakeFormJson } from './form/widget-visibility.service.mock';
export { mockToken } from './jwt-helper.service.spec';
