// @alfresco/js-api/_esm5/src/authentication/oauth2Auth.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import * as _Emitter from 'event-emitter';
import { AlfrescoApiClient } from '../alfrescoApiClient';
import { Authentication } from './authentication';
var Emitter = _Emitter;
var Oauth2Auth = (function (_super) {
    tslib_1.__extends(Oauth2Auth, _super);
    function Oauth2Auth(config) {
        var _this = _super.call(this) || this;
        _this.authentications = new Authentication({
            'oauth2': { accessToken: '' }, type: 'oauth2', 'basicAuth': {}
        });
        _this.className = 'Oauth2Auth';
        _this.setConfig(config);
        Emitter.call(_this);
        return _this;
    }
    Oauth2Auth.getInstance = function (config) {
        if (!Oauth2Auth.instance) {
            Oauth2Auth.instance = new Oauth2Auth(config);
        }
        else {
            Oauth2Auth.instance.setConfig(config);
        }
        return Oauth2Auth.instance;
    };
    Oauth2Auth.prototype.setConfig = function (config) {
        this.config = config;
        if (this.config.oauth2) {
            if (this.config.oauth2.host === undefined || this.config.oauth2.host === null) {
                throw 'Missing the required oauth2 host parameter';
            }
            if (this.config.oauth2.clientId === undefined || this.config.oauth2.clientId === null) {
                throw 'Missing the required oauth2 clientId parameter';
            }
            if (this.config.oauth2.scope === undefined || this.config.oauth2.scope === null) {
                throw 'Missing the required oauth2 scope parameter';
            }
            if (this.config.oauth2.secret === undefined || this.config.oauth2.secret === null) {
                this.config.oauth2.secret = '';
            }
            if ((this.config.oauth2.redirectUri === undefined || this.config.oauth2.redirectUri === null) && this.config.oauth2.implicitFlow) {
                throw 'Missing redirectUri required parameter';
            }
            if (!this.config.oauth2.refreshTokenTimeout) {
                this.config.oauth2.refreshTokenTimeout = 40000;
            }
            if (!this.config.oauth2.redirectSilentIframeUri) {
                var context_1 = '';
                if (typeof window !== 'undefined') {
                    context_1 = window.location.origin;
                }
                this.config.oauth2.redirectSilentIframeUri = context_1 + '/assets/silent-refresh.html';
            }
            this.basePath = this.config.oauth2.host;
            this.host = this.config.oauth2.host;
            this.initOauth();
        }
    };
    Oauth2Auth.prototype.initOauth = function () {
        var _this = this;
        if (!this.config.oauth2.implicitFlow && this.isValidAccessToken()) {
            var accessToken = this.storage.getItem('access_token');
            this.setToken(accessToken, null);
        }
        return Promise.resolve()
            .then(function () {
            return _this.discoveryUrls();
        })
            .then(function () {
            if (_this.config.oauth2.implicitFlow) {
                return _this.loadJwks();
            }
        })
            .then(function () {
            if (_this.config.oauth2.implicitFlow) {
                return _this.checkFragment();
            }
        });
    };
    Oauth2Auth.prototype.discoveryUrls = function () {
        var _this = this;
        return new Promise(function (resolve, reject) {
            var discoveryStore = _this.storage.getItem('discovery');
            if (discoveryStore) {
                _this.discovery = JSON.parse(discoveryStore);
            }
            if (!_this.discovery) {
                var postBody = {}, pathParams = {}, queryParams = {}, formParams = {}, headerParams = {};
                var contentTypes = ['application/json'];
                var accepts = ['application/json'];
                var url = '.well-known/openid-configuration';
                _this.callApi(url, 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts).then(function (discovery) {
                    _this.discovery = {};
                    _this.discovery.loginUrl = discovery.authorization_endpoint;
                    _this.discovery.logoutUrl = discovery.end_session_endpoint;
                    _this.discovery.grantTypesSupported = discovery.grant_types_supported;
                    _this.discovery.issuer = discovery.issuer;
                    _this.discovery.tokenEndpoint = discovery.token_endpoint;
                    _this.discovery.userinfoEndpoint = discovery.userinfo_endpoint;
                    _this.discovery.jwksUri = discovery.jwks_uri;
                    _this.discovery.sessionCheckIFrameUrl = discovery.check_session_iframe;
                    _this.emit('discovery', _this.discovery);
                    _this.storage.setItem('discovery', JSON.stringify(_this.discovery));
                    resolve(discovery);
                }, function (error) {
                    _this.emit('error', error);
                    _this.storage.removeItem('discovery');
                    reject(error.error);
                });
            }
            else {
                _this.emit('discovery', _this.discovery);
                _this.storage.setItem('discovery', JSON.stringify(_this.discovery));
                resolve(_this.discovery);
            }
        });
    };
    Oauth2Auth.prototype.loadJwks = function () {
        var _this = this;
        return new Promise(function (resolve, reject) {
            var jwksStore = _this.storage.getItem('jwks');
            if (jwksStore) {
                _this.jwks = JSON.parse(jwksStore);
            }
            if (_this.discovery.jwksUri) {
                if (!_this.jwks) {
                    var postBody = {}, pathParams = {}, queryParams = {}, formParams = {}, headerParams = {};
                    var contentTypes = ['application/json'];
                    var accepts = ['application/json'];
                    _this.callCustomApi(_this.discovery.jwksUri, 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts).then(function (jwks) {
                        _this.jwks = jwks;
                        _this.emit('jwks', jwks);
                        _this.storage.setItem('jwks', JSON.stringify(jwks));
                        resolve(jwks);
                    }, function (error) {
                        reject(error.error);
                    });
                }
                else {
                    _this.emit('jwks', _this.jwks);
                    resolve(_this.jwks);
                }
            }
            else {
                reject('jwks error');
            }
        });
    };
    Oauth2Auth.prototype.checkFragment = function (externalHash) {
        var _this = this;
        return new Promise(function (resolve, reject) {
            _this.hashFragmentParams = _this.getHashFragmentParams(externalHash);
            if (externalHash === undefined && _this.isValidAccessToken()) {
                var accessToken = _this.storage.getItem('access_token');
                _this.setToken(accessToken, null);
                _this.silentRefresh();
                resolve(accessToken);
            }
            if (_this.hashFragmentParams) {
                var accessToken_1 = _this.hashFragmentParams.access_token;
                var idToken_1 = _this.hashFragmentParams.id_token;
                var sessionState = _this.hashFragmentParams.session_state;
                var expiresIn_1 = _this.hashFragmentParams.expires_in;
                if (!sessionState) {
                    reject('session state not present');
                }
                _this.processJWTToken(idToken_1).then(function (jwt) {
                    if (jwt) {
                        _this.storeIdToken(idToken_1, jwt.payload.exp);
                        _this.storeAccessToken(accessToken_1, expiresIn_1);
                        _this.authentications.basicAuth.username = jwt.payload.preferred_username;
                        _this.saveUsername(jwt.payload.preferred_username);
                        _this.silentRefresh();
                        resolve(accessToken_1);
                    }
                }, function (error) {
                    reject('Validation JWT error' + error);
                });
            }
            else {
                if (_this.config.oauth2.silentLogin) {
                    _this.implicitLogin();
                }
            }
        });
    };
    Oauth2Auth.prototype.padBase64 = function (base64data) {
        while (base64data.length % 4 !== 0) {
            base64data += '=';
        }
        return base64data;
    };
    Oauth2Auth.prototype.processJWTToken = function (jwt) {
        var _this = this;
        return new Promise(function (resolve, reject) {
            if (jwt) {
                var jwtArray = jwt.split('.');
                var headerBase64 = _this.padBase64(jwtArray[0]);
                var headerJson = _this.b64DecodeUnicode(headerBase64);
                var header = JSON.parse(headerJson);
                var payloadBase64 = _this.padBase64(jwtArray[1]);
                var payloadJson = _this.b64DecodeUnicode(payloadBase64);
                var payload = JSON.parse(payloadJson);
                var savedNonce = _this.storage.getItem('nonce');
                if (!payload.sub) {
                    reject('Missing sub in JWT');
                }
                if (payload.nonce !== savedNonce) {
                    reject('Failing nonce JWT is not corrisponding' + payload.nonce);
                }
                if (_this.jwks) {
                    resolve({
                        idToken: jwt,
                        payload: payload,
                        header: header
                    });
                }
            }
        });
    };
    Oauth2Auth.prototype.b64DecodeUnicode = function (b64string) {
        var base64 = b64string.replace(/\-/g, '+').replace(/\_/g, '/');
        return decodeURIComponent(atob(base64)
            .split('')
            .map(function (c) {
            return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
        })
            .join(''));
    };
    Oauth2Auth.prototype.storeIdToken = function (idToken, exp) {
        this.storage.setItem('id_token', idToken);
        this.storage.setItem('id_token_expires_at', Number(exp * 1000).toString());
        this.storage.setItem('id_token_stored_at', Date.now().toString());
    };
    Oauth2Auth.prototype.storeAccessToken = function (accessToken, expiresIn, refreshToken) {
        this.storage.setItem('access_token', accessToken);
        var expiresInMilliSeconds = expiresIn * 1000;
        var now = new Date();
        var expiresAt = now.getTime() + expiresInMilliSeconds;
        this.storage.setItem('access_token_expires_in', expiresAt);
        this.storage.setItem('access_token_stored_at', Date.now().toString());
        this.setToken(accessToken, refreshToken);
    };
    Oauth2Auth.prototype.saveUsername = function (username) {
        if (this.storage.supportsStorage()) {
            this.storage.setItem('USERNAME', username);
        }
    };
    Oauth2Auth.prototype.implicitLogin = function () {
        var _this = this;
        if (!this.isValidToken() || !this.isValidAccessToken()) {
            if (this.discovery && this.discovery.loginUrl) {
                this.redirectLogin();
            }
            else {
                this.on('discovery', function () {
                    _this.redirectLogin();
                });
            }
        }
        else {
            var accessToken = this.storage.getItem('access_token');
            this.setToken(accessToken, null);
        }
    };
    Oauth2Auth.prototype.isValidToken = function () {
        var validToken = false;
        if (this.getIdToken()) {
            var expiresAt = this.storage.getItem('id_token_expires_at'), now = new Date();
            if (expiresAt && parseInt(expiresAt, 10) >= now.getTime()) {
                validToken = true;
            }
        }
        return validToken;
    };
    Oauth2Auth.prototype.isValidAccessToken = function () {
        var validAccessToken = false;
        if (this.getAccessToken()) {
            var expiresAt = this.storage.getItem('access_token_expires_in');
            var now = new Date();
            if (expiresAt && parseInt(expiresAt, 10) >= now.getTime()) {
                validAccessToken = true;
            }
        }
        return validAccessToken;
    };
    Oauth2Auth.prototype.getIdToken = function () {
        return this.storage.getItem('id_token');
    };
    Oauth2Auth.prototype.getAccessToken = function () {
        return this.storage.getItem('access_token');
    };
    Oauth2Auth.prototype.redirectLogin = function () {
        if (this.config.oauth2.implicitFlow && typeof window !== 'undefined') {
            var href = this.composeImplicitLoginUrl();
            window.location.href = href;
            this.emit('implicit_redirect', href);
        }
    };
    Oauth2Auth.prototype.genNonce = function () {
        var text = '';
        var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        for (var i = 0; i < 40; i++) {
            text += possible.charAt(Math.floor(Math.random() * possible.length));
        }
        return text;
    };
    Oauth2Auth.prototype.composeImplicitLoginUrl = function () {
        var nonce = this.genNonce();
        this.storage.setItem('nonce', nonce);
        var separation = this.discovery.loginUrl.indexOf('?') > -1 ? '&' : '?';
        return this.discovery.loginUrl +
            separation +
            'client_id=' +
            encodeURIComponent(this.config.oauth2.clientId) +
            '&redirect_uri=' +
            encodeURIComponent(this.config.oauth2.redirectUri) +
            '&scope=' +
            encodeURIComponent(this.config.oauth2.scope) +
            '&response_type=' +
            encodeURIComponent('id_token token') +
            '&nonce=' +
            encodeURIComponent(nonce);
    };
    Oauth2Auth.prototype.composeIframeLoginUrl = function () {
        var nonce = this.genNonce();
        this.storage.setItem('nonce', nonce);
        var separation = this.discovery.loginUrl.indexOf('?') > -1 ? '&' : '?';
        return this.discovery.loginUrl +
            separation +
            'client_id=' +
            encodeURIComponent(this.config.oauth2.clientId) +
            '&redirect_uri=' +
            encodeURIComponent(this.config.oauth2.redirectSilentIframeUri) +
            '&scope=' +
            encodeURIComponent(this.config.oauth2.scope) +
            '&response_type=' +
            encodeURIComponent('id_token token') +
            '&nonce=' +
            encodeURIComponent(nonce) +
            '&prompt=none';
    };
    Oauth2Auth.prototype.hasHashCharacter = function (hash) {
        return hash.indexOf('#') === 0;
    };
    Oauth2Auth.prototype.startWithHashRoute = function (hash) {
        return hash.startsWith('#/');
    };
    Oauth2Auth.prototype.getHashFragmentParams = function (externalHash) {
        var hashFragmentParams = null;
        if (typeof window !== 'undefined') {
            var hash = void 0;
            if (!externalHash) {
                hash = decodeURIComponent(window.location.hash);
                if (!this.startWithHashRoute(hash)) {
                    window.location.hash = '';
                }
            }
            else {
                hash = decodeURIComponent(externalHash);
                this.removeHashFromSilentIframe();
                this.destroyIframe();
            }
            if (this.hasHashCharacter(hash) && !this.startWithHashRoute(hash)) {
                var questionMarkPosition = hash.indexOf('?');
                if (questionMarkPosition > -1) {
                    hash = hash.substr(questionMarkPosition + 1);
                }
                else {
                    hash = hash.substr(1);
                }
                hashFragmentParams = this.parseQueryString(hash);
            }
        }
        return hashFragmentParams;
    };
    Oauth2Auth.prototype.parseQueryString = function (queryString) {
        var data = {};
        var pairs, pair, separatorIndex, escapedKey, escapedValue, key, value;
        if (queryString !== null) {
            pairs = queryString.split('&');
            for (var i = 0; i < pairs.length; i++) {
                pair = pairs[i];
                separatorIndex = pair.indexOf('=');
                if (separatorIndex === -1) {
                    escapedKey = pair;
                    escapedValue = null;
                }
                else {
                    escapedKey = pair.substr(0, separatorIndex);
                    escapedValue = pair.substr(separatorIndex + 1);
                }
                key = decodeURIComponent(escapedKey);
                value = decodeURIComponent(escapedValue);
                if (key.substr(0, 1) === '/') {
                    key = key.substr(1);
                }
                data[key] = value;
            }
        }
        return data;
    };
    Oauth2Auth.prototype.silentRefresh = function () {
        var _this = this;
        if (typeof document === 'undefined') {
            return;
        }
        setTimeout(function () {
            _this.destroyIframe();
            _this.createIframe();
        }, this.config.oauth2.refreshTokenTimeout);
    };
    Oauth2Auth.prototype.removeHashFromSilentIframe = function () {
        var iframe = document.getElementById('silent_refresh_token_iframe');
        if (iframe && iframe.contentWindow.location.hash) {
            iframe.contentWindow.location.hash = '';
        }
    };
    Oauth2Auth.prototype.createIframe = function () {
        var _this = this;
        var iframe = document.createElement('iframe');
        iframe.id = 'silent_refresh_token_iframe';
        var loginUrl = this.composeIframeLoginUrl();
        iframe.setAttribute('src', loginUrl);
        iframe.style.display = 'none';
        document.body.appendChild(iframe);
        this.iFameHashListner = function () {
            var silentRefreshTokenIframe = document.getElementById('silent_refresh_token_iframe');
            var hash = silentRefreshTokenIframe.contentWindow.location.hash;
            _this.checkFragment(hash);
        };
        iframe.addEventListener('load', this.iFameHashListner);
    };
    Oauth2Auth.prototype.destroyIframe = function () {
        var iframe = document.getElementById('silent_refresh_token_iframe');
        if (iframe) {
            iframe.removeEventListener('load', this.iFameHashListner);
            document.body.removeChild(iframe);
        }
    };
    Oauth2Auth.prototype.login = function (username, password) {
        var _this = this;
        var promise = new Promise(function (resolve, reject) {
            if (_this.discovery) {
                _this.grantPasswordLogin(username, password, resolve, reject);
            }
            else {
                _this.on('discovery', function () {
                    _this.grantPasswordLogin(username, password, resolve, reject);
                });
            }
        });
        return promise;
    };
    Oauth2Auth.prototype.grantPasswordLogin = function (username, password, resolve, reject) {
        var _this = this;
        var postBody = {}, pathParams = {}, queryParams = {};
        var headerParams = {
            'Content-Type': 'application/x-www-form-urlencoded'
        };
        var formParams = {
            username: username,
            password: password,
            grant_type: 'password',
            client_id: this.config.oauth2.clientId
        };
        var contentTypes = ['application/x-www-form-urlencoded'];
        var accepts = ['application/json'];
        var promise = this.callCustomApi(this.discovery.tokenEndpoint, 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts).then(function (data) {
            _this.saveUsername(username);
            _this.storeAccessToken(data.access_token, data.expires_in, data.refresh_token);
            _this.silentRefresh();
            resolve(data);
        }, function (error) {
            if (error.error && error.error.status === 401) {
                _this.emit('unauthorized');
            }
            _this.emit('error');
            reject(error.error);
        });
        Emitter(promise);
    };
    Oauth2Auth.prototype.refreshToken = function () {
        var _this = this;
        var postBody = {}, pathParams = {}, queryParams = {}, formParams = {};
        var auth = 'Basic ' + btoa(this.config.oauth2.clientId + ':' + this.config.oauth2.secret);
        var headerParams = {
            'Content-Type': 'application/x-www-form-urlencoded',
            'Cache-Control': 'no-cache',
            'Authorization': auth
        };
        queryParams = {
            refresh_token: this.authentications.oauth2.refreshToken,
            grant_type: 'refresh_token'
        };
        var contentTypes = ['application/x-www-form-urlencoded'];
        var accepts = ['application/json'];
        var promise = new Promise(function (resolve, reject) {
            _this.callCustomApi(_this.discovery.tokenEndpoint, 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts).then(function (data) {
                _this.setToken(data.access_token, data.refresh_token);
                resolve(data);
            }, function (error) {
                if (error.error.status === 401) {
                    _this.emit('unauthorized');
                }
                _this.emit('error');
                reject(error.error);
            });
        });
        Emitter(promise);
        return promise;
    };
    Oauth2Auth.prototype.setToken = function (token, refreshToken) {
        this.authentications.oauth2.accessToken = token;
        this.authentications.oauth2.refreshToken = refreshToken;
        this.authentications.basicAuth.password = null;
        this.token = token;
        this.emit('token_issued');
    };
    Oauth2Auth.prototype.getToken = function () {
        return this.token;
    };
    Oauth2Auth.prototype.getAuthentication = function () {
        return this.authentications;
    };
    Oauth2Auth.prototype.changeHost = function (host) {
        this.config.hostOauth2 = host;
    };
    Oauth2Auth.prototype.isLoggedIn = function () {
        return !!this.authentications.oauth2.accessToken;
    };
    Oauth2Auth.prototype.logOut = function () {
        var _this = this;
        var id_token = this.getIdToken();
        this.invalidateSession();
        this.setToken(null, null);
        var separation = this.discovery.logoutUrl.indexOf('?') > -1 ? '&' : '?';
        var redirectLogout = this.config.oauth2.redirectUriLogout || this.config.oauth2.redirectUri;
        var logoutUrl = this.discovery.logoutUrl +
            separation +
            'post_logout_redirect_uri=' +
            encodeURIComponent(redirectLogout) +
            '&id_token_hint=' +
            encodeURIComponent(id_token);
        var returnPromise = Promise.resolve().then(function () {
            if (_this.config.oauth2.implicitFlow && typeof window !== 'undefined') {
                window.location.href = logoutUrl;
            }
        });
        return returnPromise;
    };
    Oauth2Auth.prototype.invalidateSession = function () {
        this.storage.removeItem('access_token');
        this.storage.removeItem('access_token_expires_in');
        this.storage.removeItem('access_token_stored_at');
        this.storage.removeItem('id_token');
        this.storage.removeItem('id_token');
        this.storage.removeItem('id_token_claims_obj');
        this.storage.removeItem('id_token_expires_at');
        this.storage.removeItem('id_token_stored_at');
        this.storage.removeItem('nonce');
        this.storage.removeItem('jwks');
        this.storage.removeItem('discovery');
    };
    Oauth2Auth.instance = null;
    return Oauth2Auth;
}(AlfrescoApiClient));
export { Oauth2Auth };
