// @alfresco/js-api/_esm5/src/api/gs-classification-rest-api/api/classificationGuides.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { ClassificationGuideEntry } from '../model/classificationGuideEntry';
import { ClassificationGuidePaging } from '../model/classificationGuidePaging';
import { InstructionEntry } from '../model/instructionEntry';
import { SubtopicPaging } from '../model/subtopicPaging';
import { TopicEntry } from '../model/topicEntry';
import { TopicPaging } from '../model/topicPaging';
import { BaseApi } from './base.api';
var ClassificationGuidesApi = (function (_super) {
    tslib_1.__extends(ClassificationGuidesApi, _super);
    function ClassificationGuidesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ClassificationGuidesApi.prototype.combinedInstructions = function (opts) {
        opts = opts || {};
        var postBody = opts['instructions'];
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/combined-instructions', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, InstructionEntry);
    };
    ClassificationGuidesApi.prototype.createClassificationGuide = function (classificationGuide) {
        var postBody = classificationGuide;
        if (classificationGuide === undefined || classificationGuide === null) {
            throw new Error("Required param 'classificationGuide' in createClassificationGuide");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationGuideEntry);
    };
    ClassificationGuidesApi.prototype.createSubtopic = function (topicId, topic, opts) {
        opts = opts || {};
        var postBody = topic;
        if (topicId === undefined || topicId === null) {
            throw new Error("Required param 'topicId' in createSubtopic");
        }
        if (topic === undefined || topic === null) {
            throw new Error("Required param 'topic' in createSubtopic");
        }
        var pathParams = {
            'topicId': topicId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/topics/{topicId}/subtopics', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TopicEntry);
    };
    ClassificationGuidesApi.prototype.createTopic = function (classificationGuideId, topic, opts) {
        opts = opts || {};
        var postBody = topic;
        if (classificationGuideId === undefined || classificationGuideId === null) {
            throw new Error("Required param 'classificationGuideId' in createTopic");
        }
        if (topic === undefined || topic === null) {
            throw new Error("Required param 'topic' in createTopic");
        }
        var pathParams = {
            'classificationGuideId': classificationGuideId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides/{classificationGuideId}/topics', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TopicEntry);
    };
    ClassificationGuidesApi.prototype.deleteClassificationGuide = function (classificationGuideId) {
        var postBody = null;
        if (classificationGuideId === undefined || classificationGuideId === null) {
            throw new Error("Required param 'classificationGuideId' in deleteClassificationGuide");
        }
        var pathParams = {
            'classificationGuideId': classificationGuideId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides/{classificationGuideId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ClassificationGuidesApi.prototype.deleteTopic = function (topicId) {
        var postBody = null;
        if (topicId === undefined || topicId === null) {
            throw new Error("Required param 'topicId' in deleteTopic");
        }
        var pathParams = {
            'topicId': topicId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/topics/{topicId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ClassificationGuidesApi.prototype.listClassificationGuides = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationGuidePaging);
    };
    ClassificationGuidesApi.prototype.listSubtopics = function (topicId, opts) {
        opts = opts || {};
        var postBody = null;
        if (topicId === undefined || topicId === null) {
            throw new Error("Required param 'topicId' in listSubtopics");
        }
        var pathParams = {
            'topicId': topicId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'includeSource': opts['includeSource']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/topics/{topicId}/subtopics', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, SubtopicPaging);
    };
    ClassificationGuidesApi.prototype.listTopics = function (classificationGuideId, opts) {
        opts = opts || {};
        var postBody = null;
        if (classificationGuideId === undefined || classificationGuideId === null) {
            throw new Error("Required param 'classificationGuideId' in listTopics");
        }
        var pathParams = {
            'classificationGuideId': classificationGuideId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': this.apiClient.buildCollectionParam(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'includeSource': opts['includeSource']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides/{classificationGuideId}/topics', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TopicPaging);
    };
    ClassificationGuidesApi.prototype.showClassificationGuideById = function (classificationGuideId) {
        var postBody = null;
        if (classificationGuideId === undefined || classificationGuideId === null) {
            throw new Error("Required param 'classificationGuideId' in showClassificationGuideById");
        }
        var pathParams = {
            'classificationGuideId': classificationGuideId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides/{classificationGuideId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationGuideEntry);
    };
    ClassificationGuidesApi.prototype.showTopicById = function (topicId, opts) {
        opts = opts || {};
        var postBody = null;
        if (topicId === undefined || topicId === null) {
            throw new Error("Required param 'topicId' in showTopicById");
        }
        var pathParams = {
            'topicId': topicId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/topics/{topicId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TopicEntry);
    };
    ClassificationGuidesApi.prototype.updateClassificationGuide = function (classificationGuideId, classificationGuide) {
        var postBody = classificationGuide;
        if (classificationGuideId === undefined || classificationGuideId === null) {
            throw new Error("Required param 'classificationGuideId' in updateClassificationGuide");
        }
        if (classificationGuide === undefined || classificationGuide === null) {
            throw new Error("Required param 'classificationGuide' in updateClassificationGuide");
        }
        var pathParams = {
            'classificationGuideId': classificationGuideId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-guides/{classificationGuideId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationGuideEntry);
    };
    ClassificationGuidesApi.prototype.updateTopic = function (topicId, topic, opts) {
        opts = opts || {};
        var postBody = topic;
        if (topicId === undefined || topicId === null) {
            throw new Error("Required param 'topicId' in updateTopic");
        }
        if (topic === undefined || topic === null) {
            throw new Error("Required param 'topic' in updateTopic");
        }
        var pathParams = {
            'topicId': topicId
        };
        var queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/topics/{topicId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TopicEntry);
    };
    return ClassificationGuidesApi;
}(BaseApi));
export { ClassificationGuidesApi };
