// @alfresco/js-api/_esm5/src/api/gs-classification-rest-api/api/classificationReasons.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { ClassificationReasonEntry } from '../model/classificationReasonEntry';
import { ClassificationReasonsPaging } from '../model/classificationReasonsPaging';
import { BaseApi } from './base.api';
var ClassificationReasonsApi = (function (_super) {
    tslib_1.__extends(ClassificationReasonsApi, _super);
    function ClassificationReasonsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ClassificationReasonsApi.prototype.createClassificationReason = function (classificationReason) {
        var postBody = classificationReason;
        if (classificationReason === undefined || classificationReason === null) {
            throw new Error("Required param 'classificationReason' in createClassificationReason");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    };
    ClassificationReasonsApi.prototype.deleteClassificationReason = function (classificationReasonId) {
        var postBody = null;
        if (classificationReasonId === undefined || classificationReasonId === null) {
            throw new Error("Required param 'classificationReasonId' in deleteClassificationReason");
        }
        var pathParams = {
            'classificationReasonId': classificationReasonId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ClassificationReasonsApi.prototype.listClassificationReasons = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonsPaging);
    };
    ClassificationReasonsApi.prototype.showClassificationReasonById = function (classificationReasonId) {
        var postBody = null;
        if (classificationReasonId === undefined || classificationReasonId === null) {
            throw new Error("Required param 'classificationReasonId' in showClassificationReasonById");
        }
        var pathParams = {
            'classificationReasonId': classificationReasonId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    };
    ClassificationReasonsApi.prototype.updateClassificationReason = function (classificationReasonId, classificationReason) {
        var postBody = classificationReason;
        if (classificationReasonId === undefined || classificationReasonId === null) {
            throw new Error("Required param 'classificationReasonId' in updateClassificationReason");
        }
        if (classificationReason === undefined || classificationReason === null) {
            throw new Error("Required param 'classificationReason' in updateClassificationReason");
        }
        var pathParams = {
            'classificationReasonId': classificationReasonId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    };
    return ClassificationReasonsApi;
}(BaseApi));
export { ClassificationReasonsApi };
