// @alfresco/adf-insights/esm2015/diagram/components/boundary-events/diagram-boundary-event.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
export class DiagramBoundaryEventComponent {
    /**
     * @param {?} elementRef
     * @param {?} diagramColorService
     */
    constructor(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.center = {};
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: 1 };
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.center.x = this.data.x + (this.data.width / 2);
        this.center.y = this.data.y + (this.data.height / 2);
        this.circleRadiusInner = 12;
        this.circleRadiusOuter = 15;
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.fillColors = this.diagramColorService.getFillColour(this.data.id);
        this.options.fillOpacity = this.diagramColorService.getFillOpacity();
        this.signalFillColor = 'none';
    }
}
DiagramBoundaryEventComponent.decorators = [
    { type: Component, args: [{
                selector: 'diagram-boundary-event',
                template: "<raphael-circle [center]=\"center\" [radius]=\"circleRadiusInner\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<raphael-circle [elementId]=\"data.id\" [center]=\"center\" [radius]=\"circleRadiusOuter\" [strokeWidth]=\"options.strokeWidth\" [stroke]=\"options.stroke\"\n                [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-circle>\n<diagram-container-icon-event [data]=\"data\" [type]=\"data.eventDefinition.type\" [fillColor]=\"signalFillColor\"></diagram-container-icon-event>\n<diagram-tooltip [data]=\"data\"></diagram-tooltip>\n"
            }] }
];
/** @nocollapse */
DiagramBoundaryEventComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: DiagramColorService }
];
DiagramBoundaryEventComponent.propDecorators = {
    data: [{ type: Input }],
    error: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.data;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.error;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.center;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.options;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.signalFillColor;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.circleRadiusInner;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.circleRadiusOuter;
    /** @type {?} */
    DiagramBoundaryEventComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramBoundaryEventComponent.prototype.diagramColorService;
}
