// @alfresco/adf-core/esm5/form/components/widgets/display-text/display-text.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ViewEncapsulation } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { baseHost, WidgetComponent } from './../widget.component';
var DisplayTextWidgetComponentComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DisplayTextWidgetComponentComponent, _super);
    function DisplayTextWidgetComponentComponent(formService) {
        var _this = _super.call(this, formService) || this;
        _this.formService = formService;
        return _this;
    }
    DisplayTextWidgetComponentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'display-text-widget',
                    template: "<div class=\"adf-display-text-widget {{field.className}}\">{{field.value}}</div>\n",
                    host: baseHost,
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-display-text-widget{white-space:pre-wrap}"]
                }] }
    ];
    /** @nocollapse */
    DisplayTextWidgetComponentComponent.ctorParameters = function () { return [
        { type: FormService }
    ]; };
    return DisplayTextWidgetComponentComponent;
}(WidgetComponent));
export { DisplayTextWidgetComponentComponent };
if (false) {
    /** @type {?} */
    DisplayTextWidgetComponentComponent.prototype.formService;
}
