// @alfresco/adf-core/esm5/userinfo/services/bpm-user.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { AlfrescoApiService } from '../../services/alfresco-api.service';
import { LogService } from '../../services/log.service';
import { BpmUserModel } from '../models/bpm-user.model';
import { map, catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "../../services/alfresco-api.service";
import * as i2 from "../../services/log.service";
/**
 *
 * BPMUserService retrieve all the information of an Ecm user.
 *
 */
var BpmUserService = /** @class */ (function () {
    function BpmUserService(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    /**
     * Gets information about the current user.
     * @returns User information object
     */
    /**
     * Gets information about the current user.
     * @return {?} User information object
     */
    BpmUserService.prototype.getCurrentUserInfo = /**
     * Gets information about the current user.
     * @return {?} User information object
     */
    function () {
        var _this = this;
        return from(this.apiService.getInstance().activiti.profileApi.getProfile())
            .pipe(map((/**
         * @param {?} userRepresentation
         * @return {?}
         */
        function (userRepresentation) {
            return new BpmUserModel(userRepresentation);
        })), catchError((/**
         * @param {?} err
         * @return {?}
         */
        function (err) { return _this.handleError(err); })));
    };
    /**
     * Gets the current user's profile image as a URL.
     * @returns URL string
     */
    /**
     * Gets the current user's profile image as a URL.
     * @return {?} URL string
     */
    BpmUserService.prototype.getCurrentUserProfileImage = /**
     * Gets the current user's profile image as a URL.
     * @return {?} URL string
     */
    function () {
        return this.apiService.getInstance().activiti.profileApi.getProfilePictureUrl();
    };
    /**
     * Throw the error
     * @param error
     */
    /**
     * Throw the error
     * @private
     * @param {?} error
     * @return {?}
     */
    BpmUserService.prototype.handleError = /**
     * Throw the error
     * @private
     * @param {?} error
     * @return {?}
     */
    function (error) {
        // in a real world app, we may send the error to some remote logging infrastructure
        // instead of just logging it to the console
        this.logService.error(error);
        return throwError(error || 'Server error');
    };
    BpmUserService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    BpmUserService.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: LogService }
    ]; };
    /** @nocollapse */ BpmUserService.ngInjectableDef = i0.defineInjectable({ factory: function BpmUserService_Factory() { return new BpmUserService(i0.inject(i1.AlfrescoApiService), i0.inject(i2.LogService)); }, token: BpmUserService, providedIn: "root" });
    return BpmUserService;
}());
export { BpmUserService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    BpmUserService.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    BpmUserService.prototype.logService;
}
