// @alfresco/js-api/src/api/gs-core-rest-api/api/records.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
const recordEntry_1 = require("../model/recordEntry");
const base_api_1 = require("./base.api");
class RecordsApi extends base_api_1.BaseApi {
    completeRecord(recordId, opts) {
        opts = opts || {};
        let postBody = null;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in completeRecord");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}/complete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordEntry_1.RecordEntry);
    }
    deleteRecord(recordId) {
        let postBody = null;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in deleteRecord");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    fileRecord(recordId, nodeBodyFile, opts) {
        opts = opts || {};
        let postBody = nodeBodyFile;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in fileRecord");
        }
        if (nodeBodyFile === undefined || nodeBodyFile === null) {
            throw new Error("Required param 'nodeBodyFile' in fileRecord");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}/file', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordEntry_1.RecordEntry);
    }
    getRecord(recordId, opts) {
        opts = opts || {};
        let postBody = null;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in getRecord");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordEntry_1.RecordEntry);
    }
    getRecordContent(recordId, opts) {
        opts = opts || {};
        let postBody = null;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in getRecordContent");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {
            'attachment': opts['attachment']
        };
        let headerParams = {
            'If-Modified-Since': opts['ifModifiedSince']
        };
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    updateRecord(recordId, recordBodyUpdate, opts) {
        opts = opts || {};
        let postBody = recordBodyUpdate;
        if (recordId === undefined || recordId === null) {
            throw new Error("Required param 'recordId' in updateRecord");
        }
        if (recordBodyUpdate === undefined || recordBodyUpdate === null) {
            throw new Error("Required param 'recordBodyUpdate' in updateRecord");
        }
        let pathParams = {
            'recordId': recordId
        };
        let queryParams = {
            'include': this.apiClient.buildCollectionParam(opts['include'], 'csv'),
            'fields': this.apiClient.buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/records/{recordId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordEntry_1.RecordEntry);
    }
}
exports.RecordsApi = RecordsApi;
