// @alfresco/adf-testing/esm5/lib/core/actions/identity/group-identity.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { StringUtil } from '../../utils/string.util';
var GroupIdentityService = /** @class */ (function () {
    function GroupIdentityService(api) {
        this.api = api;
    }
    /**
     * @param {?=} groupName
     * @return {?}
     */
    GroupIdentityService.prototype.createIdentityGroup = /**
     * @param {?=} groupName
     * @return {?}
     */
    function (groupName) {
        if (groupName === void 0) { groupName = StringUtil.generateRandomString(5); }
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var group;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.createGroup(groupName)];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.getGroupInfoByGroupName(groupName)];
                    case 2:
                        group = _a.sent();
                        return [2 /*return*/, group];
                }
            });
        });
    };
    /**
     * @param {?} groupId
     * @return {?}
     */
    GroupIdentityService.prototype.deleteIdentityGroup = /**
     * @param {?} groupId
     * @return {?}
     */
    function (groupId) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.deleteGroup(groupId)];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} groupName
     * @return {?}
     */
    GroupIdentityService.prototype.createGroup = /**
     * @param {?} groupName
     * @return {?}
     */
    function (groupName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = '/groups';
                        method = 'POST';
                        queryParams = {};
                        postBody = {
                            'name': groupName + 'TestGroup'
                        };
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data];
                }
            });
        });
    };
    /**
     * @param {?} groupId
     * @return {?}
     */
    GroupIdentityService.prototype.deleteGroup = /**
     * @param {?} groupId
     * @return {?}
     */
    function (groupId) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = "/groups/" + groupId;
                        method = 'DELETE';
                        queryParams = {};
                        postBody = {};
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data];
                }
            });
        });
    };
    /**
     * @param {?} groupName
     * @return {?}
     */
    GroupIdentityService.prototype.getGroupInfoByGroupName = /**
     * @param {?} groupName
     * @return {?}
     */
    function (groupName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = "/groups";
                        method = 'GET';
                        queryParams = { 'search': groupName };
                        postBody = {};
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data[0]];
                }
            });
        });
    };
    /**
     * @param {?} groupId
     * @param {?} roleId
     * @param {?} roleName
     * @return {?}
     */
    GroupIdentityService.prototype.assignRole = /**
     * @param {?} groupId
     * @param {?} roleId
     * @param {?} roleName
     * @return {?}
     */
    function (groupId, roleId, roleName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = "/groups/" + groupId + "/role-mappings/realm";
                        method = 'POST';
                        queryParams = {};
                        postBody = [{ 'id': roleId, 'name': roleName }];
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data];
                }
            });
        });
    };
    /**
     * Add client roles.
     * @param groupId ID of the target group
     * @param clientId ID of the client
     * @param roleId ID of the clientRole
     * @param roleName of the clientRole
     */
    /**
     * Add client roles.
     * @param {?} groupId ID of the target group
     * @param {?} clientId ID of the client
     * @param {?} roleId ID of the clientRole
     * @param {?} roleName of the clientRole
     * @return {?}
     */
    GroupIdentityService.prototype.addClientRole = /**
     * Add client roles.
     * @param {?} groupId ID of the target group
     * @param {?} clientId ID of the client
     * @param {?} roleId ID of the clientRole
     * @param {?} roleName of the clientRole
     * @return {?}
     */
    function (groupId, clientId, roleId, roleName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = "/groups/" + groupId + "/role-mappings/clients/" + clientId;
                        method = 'POST';
                        queryParams = {};
                        postBody = [{
                                'id': roleId,
                                'name': roleName,
                                'composite': false,
                                'clientRole': true,
                                'containerId': clientId
                            }];
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data];
                }
            });
        });
    };
    /**
     * Gets the client ID using the app name.
     * @param applicationName Name of the app
     * @returns client ID string
     */
    /**
     * Gets the client ID using the app name.
     * @param {?} applicationName Name of the app
     * @return {?} client ID string
     */
    GroupIdentityService.prototype.getClientIdByApplicationName = /**
     * Gets the client ID using the app name.
     * @param {?} applicationName Name of the app
     * @return {?} client ID string
     */
    function (applicationName) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var path, method, queryParams, postBody, data;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        path = "/clients";
                        method = 'GET';
                        queryParams = { clientId: applicationName };
                        postBody = {};
                        return [4 /*yield*/, this.api.performIdentityOperation(path, method, queryParams, postBody)];
                    case 1:
                        data = _a.sent();
                        return [2 /*return*/, data[0].id];
                }
            });
        });
    };
    return GroupIdentityService;
}());
export { GroupIdentityService };
if (false) {
    /** @type {?} */
    GroupIdentityService.prototype.api;
}
