// @alfresco/adf-core/esm5/toolbar/toolbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, Input, ViewEncapsulation } from '@angular/core';
var ToolbarComponent = /** @class */ (function () {
    function ToolbarComponent() {
        /**
         * Toolbar title.
         */
        this.title = '';
    }
    ToolbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-toolbar',
                    template: "<mat-toolbar [color]=\"color\">\n    <span *ngIf=\"title\">{{ title }}</span>\n    <ng-content select=\"adf-toolbar-title\"></ng-content>\n    <ng-content></ng-content>\n</mat-toolbar>\n",
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    encapsulation: ViewEncapsulation.None,
                    host: { 'class': 'adf-toolbar' },
                    styles: [""]
                }] }
    ];
    ToolbarComponent.propDecorators = {
        title: [{ type: Input }],
        color: [{ type: Input }]
    };
    return ToolbarComponent;
}());
export { ToolbarComponent };
if (false) {
    /**
     * Toolbar title.
     * @type {?}
     */
    ToolbarComponent.prototype.title;
    /**
     * Toolbar color. Can be changed to empty value (default), `primary`, `accent` or `warn`.
     * @type {?}
     */
    ToolbarComponent.prototype.color;
}
