// @alfresco/adf-content-services/esm5/upload/components/upload-files.event.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var UploadFilesEvent = /** @class */ (function () {
    function UploadFilesEvent(files, uploadService, callback) {
        this.files = files;
        this.uploadService = uploadService;
        this.callback = callback;
        this.isDefaultPrevented = false;
    }
    Object.defineProperty(UploadFilesEvent.prototype, "defaultPrevented", {
        get: /**
         * @return {?}
         */
        function () {
            return this.isDefaultPrevented;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    UploadFilesEvent.prototype.preventDefault = /**
     * @return {?}
     */
    function () {
        this.isDefaultPrevented = true;
    };
    /**
     * @return {?}
     */
    UploadFilesEvent.prototype.pauseUpload = /**
     * @return {?}
     */
    function () {
        this.preventDefault();
    };
    /**
     * @return {?}
     */
    UploadFilesEvent.prototype.resumeUpload = /**
     * @return {?}
     */
    function () {
        var _a;
        if (this.files && this.files.length > 0) {
            (_a = this.uploadService).addToQueue.apply(_a, tslib_1.__spread(this.files));
            this.uploadService.uploadFilesInTheQueue(this.callback);
        }
    };
    return UploadFilesEvent;
}());
export { UploadFilesEvent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    UploadFilesEvent.prototype.isDefaultPrevented;
    /** @type {?} */
    UploadFilesEvent.prototype.files;
    /**
     * @type {?}
     * @private
     */
    UploadFilesEvent.prototype.uploadService;
    /**
     * @type {?}
     * @private
     */
    UploadFilesEvent.prototype.callback;
}
