// @alfresco/adf-content-services/esm5/permission-manager/components/add-permission/add-permission.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, EventEmitter, Input, Output } from '@angular/core';
import { NodePermissionService } from '../../services/node-permission.service';
import { NodesApiService, ContentService, AllowableOperationsEnum } from '@alfresco/adf-core';
var AddPermissionComponent = /** @class */ (function () {
    function AddPermissionComponent(nodePermissionService, nodeApiService, contentService) {
        var _this = this;
        this.nodePermissionService = nodePermissionService;
        this.nodeApiService = nodeApiService;
        this.contentService = contentService;
        /**
         * Emitted when the node is updated successfully.
         */
        this.success = new EventEmitter();
        /**
         * Emitted when an error occurs during the update.
         */
        this.error = new EventEmitter();
        this.selectedItems = [];
        this.nodeApiService.getNode(this.nodeId).subscribe((/**
         * @param {?} node
         * @return {?}
         */
        function (node) { return _this.currentNode = node; }));
    }
    /**
     * @param {?} selection
     * @return {?}
     */
    AddPermissionComponent.prototype.onSelect = /**
     * @param {?} selection
     * @return {?}
     */
    function (selection) {
        this.selectedItems = selection;
    };
    /**
     * @return {?}
     */
    AddPermissionComponent.prototype.isAddEnabled = /**
     * @return {?}
     */
    function () {
        return this.contentService.hasAllowableOperations(this.currentNode, AllowableOperationsEnum.UPDATEPERMISSIONS) &&
            this.selectedItems.length !== 0;
    };
    /**
     * @return {?}
     */
    AddPermissionComponent.prototype.applySelection = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.contentService.hasAllowableOperations(this.currentNode, AllowableOperationsEnum.UPDATEPERMISSIONS)) {
            this.nodePermissionService.updateNodePermissions(this.nodeId, this.selectedItems)
                .subscribe((/**
             * @param {?} node
             * @return {?}
             */
            function (node) {
                _this.success.emit(node);
            }), (/**
             * @param {?} error
             * @return {?}
             */
            function (error) {
                _this.error.emit(error);
            }));
        }
    };
    AddPermissionComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-add-permission',
                    template: "<adf-add-permission-panel\n    (select)=\"onSelect($event)\">\n</adf-add-permission-panel>\n<div id=\"adf-add-permission-actions\">\n   <button mat-button\n          id=\"adf-add-permission-action-button\"\n          class=\"adf-permission-action\"\n          [disabled]=\"!isAddEnabled()\"\n          (click)=\"applySelection()\">\n     {{'PERMISSION_MANAGER.ADD-PERMISSION.ADD-ACTION' | translate}}\n    </button>\n</div>\n\n\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    AddPermissionComponent.ctorParameters = function () { return [
        { type: NodePermissionService },
        { type: NodesApiService },
        { type: ContentService }
    ]; };
    AddPermissionComponent.propDecorators = {
        nodeId: [{ type: Input }],
        success: [{ type: Output }],
        error: [{ type: Output }]
    };
    return AddPermissionComponent;
}());
export { AddPermissionComponent };
if (false) {
    /**
     * ID of the target node.
     * @type {?}
     */
    AddPermissionComponent.prototype.nodeId;
    /**
     * Emitted when the node is updated successfully.
     * @type {?}
     */
    AddPermissionComponent.prototype.success;
    /**
     * Emitted when an error occurs during the update.
     * @type {?}
     */
    AddPermissionComponent.prototype.error;
    /** @type {?} */
    AddPermissionComponent.prototype.selectedItems;
    /** @type {?} */
    AddPermissionComponent.prototype.currentNode;
    /** @type {?} */
    AddPermissionComponent.prototype.currentNodeRoles;
    /**
     * @type {?}
     * @private
     */
    AddPermissionComponent.prototype.nodePermissionService;
    /**
     * @type {?}
     * @private
     */
    AddPermissionComponent.prototype.nodeApiService;
    /**
     * @type {?}
     * @private
     */
    AddPermissionComponent.prototype.contentService;
}
