// @alfresco/adf-insights/esm2015/analytics-process/components/widgets/dropdown/dropdown.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-access-missing-member no-input-rename  */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { FormGroup, Validators } from '@angular/forms';
import { WidgetComponent } from './../widget.component';
export class DropdownWidgetAnalyticsComponent extends WidgetComponent {
    constructor() {
        super();
        this.fieldChanged = new EventEmitter();
        this.showDefaultOption = true;
        this.required = false;
        this.defaultOptionText = 'Choose One';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.required) {
            this.formGroup.get(this.controllerName).setValidators(Validators.compose(this.buildValidatorList()));
        }
    }
    /**
     * @param {?} controller
     * @return {?}
     */
    validateDropDown(controller) {
        return controller.value !== 'null' ? null : { controllerName: false };
    }
    /**
     * @return {?}
     */
    buildValidatorList() {
        /** @type {?} */
        const validatorList = [];
        validatorList.push(Validators.required);
        if (this.showDefaultOption) {
            validatorList.push(this.validateDropDown);
        }
        return validatorList;
    }
}
DropdownWidgetAnalyticsComponent.decorators = [
    { type: Component, args: [{
                selector: 'analytics-dropdown-widget',
                template: "<div class=\"adf-dropdown-widget\" [formGroup]=\"formGroup\">\n    <label class=\"adf-dropdown-widget__label\" [attr.for]=\"field.id\">{{field.nameKey | translate}}</label>\n    <select [formControlName]=\"controllerName\"\n            [attr.id]=\"'select-' + field.id\"\n            class=\"adf-dropdown-widget__select\"\n            [(ngModel)]=\"field.value\"\n            (ngModelChange)=\"changeValue($event)\">\n        <option *ngIf=\"showDefaultOption\" value=\"null\">{{defaultOptionText}}</option>\n        <option *ngFor=\"let opt of field.options\" [value]=\"opt.id\">{{opt.label}}</option>\n    </select>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-dropdown-widget,.adf-dropdown-widget__select{width:100%}.adf-dropdown-widget__invalid .adf-dropdown-widget__select{border-color:#d50000}.adf-dropdown-widget__invalid .adf-dropdown-widget__label{color:#d50000}.adf-dropdown-widget__invalid .adf-dropdown-widget__label::after{background-color:#d50000}"]
            }] }
];
/** @nocollapse */
DropdownWidgetAnalyticsComponent.ctorParameters = () => [];
DropdownWidgetAnalyticsComponent.propDecorators = {
    field: [{ type: Input }],
    formGroup: [{ type: Input, args: ['group',] }],
    controllerName: [{ type: Input, args: ['controllerName',] }],
    fieldChanged: [{ type: Output }],
    showDefaultOption: [{ type: Input }],
    required: [{ type: Input }],
    defaultOptionText: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.field;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.formGroup;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.controllerName;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.fieldChanged;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.showDefaultOption;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.required;
    /** @type {?} */
    DropdownWidgetAnalyticsComponent.prototype.defaultOptionText;
}
