// @alfresco/adf-core/esm5/templates/empty-content/empty-content.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ChangeDetectionStrategy, ViewEncapsulation, Input } from '@angular/core';
var EmptyContentComponent = /** @class */ (function () {
    function EmptyContentComponent() {
        /**
         * Material Icon to use.
         */
        this.icon = 'cake';
        /**
         * String or Resource Key for the title.
         */
        this.title = '';
        /**
         * String or Resource Key for the subtitle.
         */
        this.subtitle = '';
    }
    EmptyContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-empty-content',
                    template: "<div class=\"adf-empty-content\">\n    <mat-icon class=\"adf-empty-content__icon\">{{ icon }}</mat-icon>\n    <div class=\"adf-empty-content__title\">{{ title | translate }}</div>\n    <div class=\"adf-empty-content__subtitle\">{{ subtitle | translate }}</div>\n    <ng-content></ng-content>\n</div>\n",
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-empty-content' },
                    styles: [""]
                }] }
    ];
    EmptyContentComponent.propDecorators = {
        icon: [{ type: Input }],
        title: [{ type: Input }],
        subtitle: [{ type: Input }]
    };
    return EmptyContentComponent;
}());
export { EmptyContentComponent };
if (false) {
    /**
     * Material Icon to use.
     * @type {?}
     */
    EmptyContentComponent.prototype.icon;
    /**
     * String or Resource Key for the title.
     * @type {?}
     */
    EmptyContentComponent.prototype.title;
    /**
     * String or Resource Key for the subtitle.
     * @type {?}
     */
    EmptyContentComponent.prototype.subtitle;
}
