// @alfresco/adf-content-services/esm5/content-metadata/interfaces/content-metadata-config.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * @record
 */
export function ContentMetadataConfig() { }
if (false) {
    /**
     * @param {?} groupName
     * @return {?}
     */
    ContentMetadataConfig.prototype.isGroupAllowed = function (groupName) { };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    ContentMetadataConfig.prototype.reorganiseByConfig = function (propertyGroups) { };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    ContentMetadataConfig.prototype.filterExcludedPreset = function (propertyGroups) { };
    /**
     * @param {?} propertyGroups
     * @return {?}
     */
    ContentMetadataConfig.prototype.appendAllPreset = function (propertyGroups) { };
    /**
     * @return {?}
     */
    ContentMetadataConfig.prototype.isIncludeAllEnabled = function () { };
}
