// @alfresco/adf-process-services/esm5/attachment/create-task-attachment.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { ProcessContentService } from '@alfresco/adf-core';
var AttachmentComponent = /** @class */ (function () {
    function AttachmentComponent(activitiContentService) {
        this.activitiContentService = activitiContentService;
        /**
         * Emitted when an error occurs while creating or uploading an
         *  attachment from the user within the component.
         */
        this.error = new EventEmitter();
        /**
         * Emitted when an attachment is created or uploaded successfully
         * from within the component.
         */
        this.success = new EventEmitter();
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    AttachmentComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes['taskId'] && changes['taskId'].currentValue) {
            this.taskId = changes['taskId'].currentValue;
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    AttachmentComponent.prototype.onFileUpload = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        var _this = this;
        var e_1, _a;
        /** @type {?} */
        var filesList = event.detail.files.map((/**
         * @param {?} obj
         * @return {?}
         */
        function (obj) { return obj.file; }));
        try {
            for (var filesList_1 = tslib_1.__values(filesList), filesList_1_1 = filesList_1.next(); !filesList_1_1.done; filesList_1_1 = filesList_1.next()) {
                var fileInfoObj = filesList_1_1.value;
                /** @type {?} */
                var file = fileInfoObj;
                /** @type {?} */
                var opts = {
                    isRelatedContent: true
                };
                this.activitiContentService.createTaskRelatedContent(this.taskId, file, opts).subscribe((/**
                 * @param {?} res
                 * @return {?}
                 */
                function (res) {
                    _this.success.emit(res);
                }), (/**
                 * @param {?} err
                 * @return {?}
                 */
                function (err) {
                    _this.error.emit(err);
                }));
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (filesList_1_1 && !filesList_1_1.done && (_a = filesList_1.return)) _a.call(filesList_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
    };
    AttachmentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-create-task-attachment',
                    template: "<button\n    color=\"primary\"\n    mat-button\n    mat-raised-button\n    mat-icon-button\n    class=\"adf-create-attachment\"\n    adf-upload=\"true\"\n    [mode]=\"['click']\"\n    [multiple]=\"true\"\n    (upload-files)=\"onFileUpload($event)\">\n    <mat-icon>add</mat-icon>\n</button>\n",
                    styles: [".adf-create-attachment{display:inline-block;line-height:0;vertical-align:middle}"]
                }] }
    ];
    /** @nocollapse */
    AttachmentComponent.ctorParameters = function () { return [
        { type: ProcessContentService }
    ]; };
    AttachmentComponent.propDecorators = {
        taskId: [{ type: Input }],
        error: [{ type: Output }],
        success: [{ type: Output }]
    };
    return AttachmentComponent;
}());
export { AttachmentComponent };
if (false) {
    /**
     * (required) The numeric ID of the task to display.
     * @type {?}
     */
    AttachmentComponent.prototype.taskId;
    /**
     * Emitted when an error occurs while creating or uploading an
     *  attachment from the user within the component.
     * @type {?}
     */
    AttachmentComponent.prototype.error;
    /**
     * Emitted when an attachment is created or uploaded successfully
     * from within the component.
     * @type {?}
     */
    AttachmentComponent.prototype.success;
    /**
     * @type {?}
     * @private
     */
    AttachmentComponent.prototype.activitiContentService;
}
