// @alfresco/adf-core/esm5/card-view/components/card-view-mapitem/card-view-mapitem.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { CardViewMapItemModel } from '../../models/card-view-mapitem.model';
import { CardViewUpdateService } from '../../services/card-view-update.service';
var CardViewMapItemComponent = /** @class */ (function () {
    function CardViewMapItemComponent(cardViewUpdateService) {
        this.cardViewUpdateService = cardViewUpdateService;
        this.displayEmpty = true;
    }
    /**
     * @return {?}
     */
    CardViewMapItemComponent.prototype.showProperty = /**
     * @return {?}
     */
    function () {
        return this.displayEmpty || !this.property.isEmpty();
    };
    /**
     * @return {?}
     */
    CardViewMapItemComponent.prototype.isClickable = /**
     * @return {?}
     */
    function () {
        return this.property.clickable;
    };
    /**
     * @return {?}
     */
    CardViewMapItemComponent.prototype.clicked = /**
     * @return {?}
     */
    function () {
        this.cardViewUpdateService.clicked(this.property);
    };
    CardViewMapItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-card-view-mapitem',
                    template: "<div [attr.data-automation-id]=\"'card-mapitem-label-' + property.key\" class=\"adf-property-label\" *ngIf=\"showProperty()\">{{ property.label | translate }}</div>\n<div class=\"adf-property-value\">\n    <div>\n        <span *ngIf=\"!isClickable(); else elseBlock\" [attr.data-automation-id]=\"'card-mapitem-value-' + property.key\">\n            <span *ngIf=\"showProperty();\">{{ property.displayValue }}</span>\n        </span>\n        <ng-template #elseBlock>\n            <span class=\"adf-mapitem-clickable-value\" (click)=\"clicked()\" [attr.data-automation-id]=\"'card-mapitem-value-' + property.key\">\n                <span *ngIf=\"showProperty(); else elseEmptyValueBlock\">{{ property.displayValue }}</span>\n            </span>\n        </ng-template>\n    </div>\n    <ng-template #elseEmptyValueBlock>\n        {{ property.default | translate }}\n    </ng-template>\n</div>\n",
                    styles: [".adf-mapitem-clickable-value{cursor:pointer!important}"]
                }] }
    ];
    /** @nocollapse */
    CardViewMapItemComponent.ctorParameters = function () { return [
        { type: CardViewUpdateService }
    ]; };
    CardViewMapItemComponent.propDecorators = {
        property: [{ type: Input }],
        displayEmpty: [{ type: Input }]
    };
    return CardViewMapItemComponent;
}());
export { CardViewMapItemComponent };
if (false) {
    /** @type {?} */
    CardViewMapItemComponent.prototype.property;
    /** @type {?} */
    CardViewMapItemComponent.prototype.displayEmpty;
    /**
     * @type {?}
     * @private
     */
    CardViewMapItemComponent.prototype.cardViewUpdateService;
}
