// @alfresco/adf-process-services/esm5/content-widget/attach-folder-widget.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector*/
import { Component, ViewEncapsulation } from '@angular/core';
import { WidgetComponent, FormService, NodesApiService } from '@alfresco/adf-core';
import { ContentNodeDialogService } from '@alfresco/adf-content-services';
var AttachFolderWidgetComponent = /** @class */ (function (_super) {
    tslib_1.__extends(AttachFolderWidgetComponent, _super);
    function AttachFolderWidgetComponent(contentDialog, formService, nodeService) {
        var _this = _super.call(this) || this;
        _this.contentDialog = contentDialog;
        _this.formService = formService;
        _this.nodeService = nodeService;
        _this.hasFolder = false;
        _this.selectedFolderName = '';
        return _this;
    }
    /**
     * @return {?}
     */
    AttachFolderWidgetComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.field &&
            this.field.value) {
            this.hasFolder = true;
            this.nodeService.getNode(this.field.value).subscribe((/**
             * @param {?} node
             * @return {?}
             */
            function (node) {
                _this.selectedFolderName = node.name;
            }));
        }
    };
    /**
     * @return {?}
     */
    AttachFolderWidgetComponent.prototype.isDefinedSourceFolder = /**
     * @return {?}
     */
    function () {
        return !!this.field.params &&
            !!this.field.params.folderSource &&
            !!this.field.params.folderSource.selectedFolder;
    };
    /**
     * @return {?}
     */
    AttachFolderWidgetComponent.prototype.openSelectDialogFromFileSource = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var params = this.field.params;
        if (this.isDefinedSourceFolder()) {
            this.contentDialog.openFolderBrowseDialogByFolderId(params.folderSource.selectedFolder.pathId).subscribe((/**
             * @param {?} selections
             * @return {?}
             */
            function (selections) {
                _this.selectedFolderName = selections[0].name;
                _this.field.value = selections[0].id;
                _this.hasFolder = true;
            }));
        }
        else {
            this.contentDialog.openFolderBrowseDialogBySite().subscribe((/**
             * @param {?} selections
             * @return {?}
             */
            function (selections) {
                _this.selectedFolderName = selections[0].name;
                _this.field.value = selections[0].id;
                _this.hasFolder = true;
            }));
        }
    };
    /**
     * @return {?}
     */
    AttachFolderWidgetComponent.prototype.removeFolder = /**
     * @return {?}
     */
    function () {
        this.field.value = null;
        this.selectedFolderName = '';
        this.hasFolder = false;
    };
    AttachFolderWidgetComponent.decorators = [
        { type: Component, args: [{
                    selector: 'attach-folder-widget',
                    template: "<div class=\"adf-attach-folder-widget {{field.className}}\"\n     [class.adf-invalid]=\"!field.isValid\"\n     [class.adf-readonly]=\"field.readOnly\">\n    <label class=\"adf-label\" [attr.for]=\"field.id\">{{field.name}}<span *ngIf=\"isRequired()\">*</span></label>\n    <div class=\"adf-attach-folder-widget-container\">\n        <div *ngIf=\"hasFolder\" class=\"adf-attach-folder-result\">\n            <mat-icon>folder</mat-icon>\n            <div class=\"adf-attach-folder-files-row\">\n                <span matLine id=\"{{'folder-'+field?.id}}\"\n                        role=\"button\" tabindex=\"0\" class=\"adf-folder\">{{selectedFolderName}}</span>\n                <button *ngIf=\"!field.readOnly\" mat-icon-button [id]=\"'folder-'+field?.id+'-remove'\"\n                        (click)=\"removeFolder();\">\n                    <mat-icon class=\"mat-24\">highlight_off</mat-icon>\n                </button>\n            </div>\n        </div>\n\n        <div class=\"button-row\" *ngIf=\"!hasFolder && !field.readOnly\">\n            <button mat-raised-button\n                color=\"primary\"\n                (click)=\"openSelectDialogFromFileSource()\"\n                [id]=\"'folder-'+field?.id+'-button'\">\n                    {{ 'FORM.FIELD.UPLOAD' | translate }}\n                <mat-icon>cloud_upload</mat-icon>\n            </button>\n        </div>\n    </div>\n    <error-widget [error]=\"field.validationSummary\"></error-widget>\n    <error-widget *ngIf=\"isInvalidFieldRequired()\" required=\"{{ 'FORM.FIELD.REQUIRED' | translate }}\"></error-widget>\n</div>\n",
                    host: {
                        '(click)': 'event($event)',
                        '(blur)': 'event($event)',
                        '(change)': 'event($event)',
                        '(focus)': 'event($event)',
                        '(focusin)': 'event($event)',
                        '(focusout)': 'event($event)',
                        '(input)': 'event($event)',
                        '(invalid)': 'event($event)',
                        '(select)': 'event($event)'
                    },
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-attach-folder-widget-container{margin-bottom:15px;display:flex;align-items:center}.adf-attach-folder-widget-container input{cursor:pointer;height:100%;right:0;opacity:0;position:absolute;top:0;width:300px;z-index:4}.adf-attach-folder-widget{width:100%;word-break:break-all;padding:.4375em 0;border-top:.84375em solid transparent}.adf-attach-folder-files-row{padding-left:8px}.adf-attach-folder-files-row .mat-line{margin-bottom:0}.adf-attach-folder-result{display:flex;align-items:center}"]
                }] }
    ];
    /** @nocollapse */
    AttachFolderWidgetComponent.ctorParameters = function () { return [
        { type: ContentNodeDialogService },
        { type: FormService },
        { type: NodesApiService }
    ]; };
    return AttachFolderWidgetComponent;
}(WidgetComponent));
export { AttachFolderWidgetComponent };
if (false) {
    /** @type {?} */
    AttachFolderWidgetComponent.prototype.hasFolder;
    /** @type {?} */
    AttachFolderWidgetComponent.prototype.selectedFolderName;
    /**
     * @type {?}
     * @private
     */
    AttachFolderWidgetComponent.prototype.contentDialog;
    /** @type {?} */
    AttachFolderWidgetComponent.prototype.formService;
    /**
     * @type {?}
     * @private
     */
    AttachFolderWidgetComponent.prototype.nodeService;
}
