// @alfresco/adf-process-services/esm5/content-widget/attach-file-widget-dialog.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Inject, ViewEncapsulation, ViewChild } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material';
import { ExternalAlfrescoApiService, AlfrescoApiService, AuthenticationService, LoginDialogPanelComponent, SitesService, SearchService } from '@alfresco/adf-core';
import { DocumentListService, ContentNodeSelectorService } from '@alfresco/adf-content-services';
var AttachFileWidgetDialogComponent = /** @class */ (function () {
    function AttachFileWidgetDialogComponent(data, externalApiService) {
        this.data = data;
        this.externalApiService = externalApiService;
        ((/** @type {?} */ (externalApiService))).init(data.ecmHost, data.context);
        this.buttonActionName = data.actionName ? "ATTACH-FILE.ACTIONS." + data.actionName.toUpperCase() : 'ATTACH-FILE.ACTIONS.CHOOSE';
    }
    /**
     * @return {?}
     */
    AttachFileWidgetDialogComponent.prototype.isLoggedIn = /**
     * @return {?}
     */
    function () {
        return this.externalApiService.getInstance().isLoggedIn();
    };
    /**
     * @return {?}
     */
    AttachFileWidgetDialogComponent.prototype.performLogin = /**
     * @return {?}
     */
    function () {
        this.loginPanel.submitForm();
    };
    /**
     * @return {?}
     */
    AttachFileWidgetDialogComponent.prototype.close = /**
     * @return {?}
     */
    function () {
        this.data.selected.complete();
    };
    /**
     * @param {?} nodeList
     * @return {?}
     */
    AttachFileWidgetDialogComponent.prototype.onSelect = /**
     * @param {?} nodeList
     * @return {?}
     */
    function (nodeList) {
        if (nodeList && nodeList[0].isFile) {
            this.chosenNode = nodeList;
        }
        else {
            this.chosenNode = null;
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    AttachFileWidgetDialogComponent.prototype.onClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.data.selected.next(this.chosenNode);
        this.data.selected.complete();
    };
    AttachFileWidgetDialogComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-attach-file-widget-dialog',
                    template: "<header\n    mat-dialog-title\n    data-automation-id=\"content-node-selector-title\">{{data?.title}}\n</header>\n\n<mat-dialog-content class=\"adf-login-dialog-content\">\n    <adf-login-dialog-panel id=\"attach-file-login-panel\" #adfLoginPanel *ngIf=\"!isLoggedIn()\">\n    </adf-login-dialog-panel>\n    <adf-content-node-selector-panel *ngIf=\"isLoggedIn()\"\n                                     id=\"attach-file-content-node\"\n                                     [isSelectionValid]=\"data?.isSelectionValid\"\n                                     (select)=\"onSelect($event)\">\n    </adf-content-node-selector-panel>\n</mat-dialog-content>\n\n<mat-dialog-actions align=\"end\">\n    <button\n        mat-button\n        (click)=\"close()\"\n        data-automation-id=\"attach-file-dialog-actions-cancel\">{{ 'ATTACH-FILE.ACTIONS.CANCEL' | translate }}\n    </button>\n\n    <button *ngIf=\"!isLoggedIn()\"\n        mat-button\n        (click)=\"performLogin()\"\n        data-automation-id=\"attach-file-dialog-actions-login\">{{ 'ATTACH-FILE.ACTIONS.LOGIN' | translate }}\n    </button>\n\n    <button *ngIf=\"isLoggedIn()\"\n        mat-button\n        [disabled]=\"!chosenNode\"\n        class=\"adf-choose-action\"\n        (click)=\"onClick($event)\"\n        data-automation-id=\"attach-file-dialog-actions-choose\">{{ buttonActionName | translate }}\n    </button>\n\n</mat-dialog-actions>\n",
                    encapsulation: ViewEncapsulation.None,
                    providers: [AuthenticationService,
                        DocumentListService,
                        SitesService,
                        ContentNodeSelectorService,
                        SearchService,
                        { provide: AlfrescoApiService, useClass: ExternalAlfrescoApiService }],
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    AttachFileWidgetDialogComponent.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] },
        { type: AlfrescoApiService }
    ]; };
    AttachFileWidgetDialogComponent.propDecorators = {
        loginPanel: [{ type: ViewChild, args: ['adfLoginPanel',] }]
    };
    return AttachFileWidgetDialogComponent;
}());
export { AttachFileWidgetDialogComponent };
if (false) {
    /** @type {?} */
    AttachFileWidgetDialogComponent.prototype.loginPanel;
    /** @type {?} */
    AttachFileWidgetDialogComponent.prototype.chosenNode;
    /** @type {?} */
    AttachFileWidgetDialogComponent.prototype.buttonActionName;
    /** @type {?} */
    AttachFileWidgetDialogComponent.prototype.data;
    /**
     * @type {?}
     * @private
     */
    AttachFileWidgetDialogComponent.prototype.externalApiService;
}
