// @alfresco/adf-process-services/esm5/people/components/people-search/people-search.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
var PeopleSearchComponent = /** @class */ (function () {
    function PeopleSearchComponent() {
        /**
         * Emitted when a search is performed with a new keyword.
         */
        this.searchPeople = new EventEmitter();
        /**
         * Emitted when a user is selected and the action button is clicked.
         */
        this.success = new EventEmitter();
        /**
         * Emitted when the "close" button is clicked.
         */
        this.closeSearch = new EventEmitter();
        this.selectedUser = {};
    }
    /**
     * @return {?}
     */
    PeopleSearchComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.filteredResults$ = this.results
            .pipe(map((/**
         * @param {?} users
         * @return {?}
         */
        function (users) {
            return users.filter((/**
             * @param {?} user
             * @return {?}
             */
            function (user) { return user.id !== _this.selectedUser.id; }));
        })));
        this.performSearch = this.performSearchCallback.bind(this);
    };
    /**
     * @private
     * @param {?} event
     * @return {?}
     */
    PeopleSearchComponent.prototype.performSearchCallback = /**
     * @private
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.searchPeople.emit(event);
        return this.filteredResults$;
    };
    /**
     * @param {?} user
     * @return {?}
     */
    PeopleSearchComponent.prototype.onRowClick = /**
     * @param {?} user
     * @return {?}
     */
    function (user) {
        this.selectedUser = user;
    };
    /**
     * @return {?}
     */
    PeopleSearchComponent.prototype.closeSearchList = /**
     * @return {?}
     */
    function () {
        this.closeSearch.emit();
    };
    /**
     * @return {?}
     */
    PeopleSearchComponent.prototype.involveUserAndClose = /**
     * @return {?}
     */
    function () {
        this.involveUser();
        this.closeSearchList();
    };
    /**
     * @return {?}
     */
    PeopleSearchComponent.prototype.involveUser = /**
     * @return {?}
     */
    function () {
        if (this.selectedUser === undefined) {
            return;
        }
        this.success.emit(this.selectedUser);
    };
    PeopleSearchComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-people-search',
                    template: "<div class=\"adf-search-text-header\">\n    <ng-content select=\"[adf-people-search-title], [people-search-title]\"></ng-content>\n</div>\n\n<adf-people-search-field [performSearch]=\"performSearch\" (rowClick)=\"onRowClick($event)\"></adf-people-search-field>\n\n<div class=\"search-list-action-container\">\n    <button mat-button type=\"button\" id=\"close-people-search\" (click)=\"closeSearchList()\">\n        {{'ADF_TASK_LIST.PEOPLE.DIALOG_CLOSE' | translate }}\n    </button>\n    <button mat-button type=\"button\" id=\"add-people\" (click)=\"involveUserAndClose()\">\n        <ng-content select=\"[adf-people-search-action-label], [people-search-action-label]\"></ng-content>\n    </button>\n</div>\n",
                    host: {
                        'class': 'adf-people-search'
                    },
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PeopleSearchComponent.ctorParameters = function () { return []; };
    PeopleSearchComponent.propDecorators = {
        results: [{ type: Input }],
        searchPeople: [{ type: Output }],
        success: [{ type: Output }],
        closeSearch: [{ type: Output }]
    };
    return PeopleSearchComponent;
}());
export { PeopleSearchComponent };
if (false) {
    /**
     * Parameters for displaying the list.
     * @type {?}
     */
    PeopleSearchComponent.prototype.results;
    /**
     * Emitted when a search is performed with a new keyword.
     * @type {?}
     */
    PeopleSearchComponent.prototype.searchPeople;
    /**
     * Emitted when a user is selected and the action button is clicked.
     * @type {?}
     */
    PeopleSearchComponent.prototype.success;
    /**
     * Emitted when the "close" button is clicked.
     * @type {?}
     */
    PeopleSearchComponent.prototype.closeSearch;
    /** @type {?} */
    PeopleSearchComponent.prototype.filteredResults$;
    /** @type {?} */
    PeopleSearchComponent.prototype.selectedUser;
    /** @type {?} */
    PeopleSearchComponent.prototype.performSearch;
}
