// @alfresco/adf-core/esm5/directives/node-download.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, Input, HostListener } from '@angular/core';
import { MatDialog } from '@angular/material';
import { AlfrescoApiService } from '../services/alfresco-api.service';
import { DownloadZipDialogComponent } from '../dialogs/download-zip.dialog';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var NodeDownloadDirective = /** @class */ (function () {
    function NodeDownloadDirective(apiService, dialog) {
        this.apiService = apiService;
        this.dialog = dialog;
    }
    /**
     * @return {?}
     */
    NodeDownloadDirective.prototype.onClick = /**
     * @return {?}
     */
    function () {
        this.downloadNodes(this.nodes);
    };
    /**
     * Downloads multiple selected nodes.
     * Packs result into a .ZIP archive if there is more than one node selected.
     * @param selection Multiple selected nodes to download
     */
    /**
     * Downloads multiple selected nodes.
     * Packs result into a .ZIP archive if there is more than one node selected.
     * @param {?} selection Multiple selected nodes to download
     * @return {?}
     */
    NodeDownloadDirective.prototype.downloadNodes = /**
     * Downloads multiple selected nodes.
     * Packs result into a .ZIP archive if there is more than one node selected.
     * @param {?} selection Multiple selected nodes to download
     * @return {?}
     */
    function (selection) {
        if (!this.isSelectionValid(selection)) {
            return;
        }
        if (selection instanceof Array) {
            if (selection.length === 1) {
                this.downloadNode(selection[0]);
            }
            else {
                this.downloadZip(selection);
            }
        }
        else {
            this.downloadNode(selection);
        }
    };
    /**
     * Downloads a single node.
     * Packs result into a .ZIP archive is the node is a Folder.
     * @param node Node to download
     */
    /**
     * Downloads a single node.
     * Packs result into a .ZIP archive is the node is a Folder.
     * @param {?} node Node to download
     * @return {?}
     */
    NodeDownloadDirective.prototype.downloadNode = /**
     * Downloads a single node.
     * Packs result into a .ZIP archive is the node is a Folder.
     * @param {?} node Node to download
     * @return {?}
     */
    function (node) {
        if (node && node.entry) {
            /** @type {?} */
            var entry = node.entry;
            if (entry.isFile) {
                this.downloadFile(node);
            }
            if (entry.isFolder) {
                this.downloadZip([node]);
            }
            // Check if there's nodeId for Shared Files
            if (!entry.isFile && !entry.isFolder && ((/** @type {?} */ (entry))).nodeId) {
                this.downloadFile(node);
            }
        }
    };
    /**
     * @private
     * @param {?} selection
     * @return {?}
     */
    NodeDownloadDirective.prototype.isSelectionValid = /**
     * @private
     * @param {?} selection
     * @return {?}
     */
    function (selection) {
        return selection || (selection instanceof Array && selection.length > 0);
    };
    /**
     * @private
     * @param {?} node
     * @return {?}
     */
    NodeDownloadDirective.prototype.downloadFile = /**
     * @private
     * @param {?} node
     * @return {?}
     */
    function (node) {
        if (node && node.entry) {
            /** @type {?} */
            var contentApi = this.apiService.getInstance().content;
            // nodeId for Shared node
            /** @type {?} */
            var id = ((/** @type {?} */ (node.entry))).nodeId || node.entry.id;
            /** @type {?} */
            var url = contentApi.getContentUrl(id, true);
            /** @type {?} */
            var fileName = node.entry.name;
            this.download(url, fileName);
        }
    };
    /**
     * @private
     * @param {?} selection
     * @return {?}
     */
    NodeDownloadDirective.prototype.downloadZip = /**
     * @private
     * @param {?} selection
     * @return {?}
     */
    function (selection) {
        if (selection && selection.length > 0) {
            // nodeId for Shared node
            /** @type {?} */
            var nodeIds = selection.map((/**
             * @param {?} node
             * @return {?}
             */
            function (node) { return (node.entry.nodeId || node.entry.id); }));
            this.dialog.open(DownloadZipDialogComponent, {
                width: '600px',
                disableClose: true,
                data: {
                    nodeIds: nodeIds
                }
            });
        }
    };
    /**
     * @private
     * @param {?} url
     * @param {?} fileName
     * @return {?}
     */
    NodeDownloadDirective.prototype.download = /**
     * @private
     * @param {?} url
     * @param {?} fileName
     * @return {?}
     */
    function (url, fileName) {
        if (url && fileName) {
            /** @type {?} */
            var link = document.createElement('a');
            link.style.display = 'none';
            link.download = fileName;
            link.href = url;
            document.body.appendChild(link);
            link.click();
            document.body.removeChild(link);
        }
    };
    NodeDownloadDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-node-download], [adfNodeDownload]'
                },] }
    ];
    /** @nocollapse */
    NodeDownloadDirective.ctorParameters = function () { return [
        { type: AlfrescoApiService },
        { type: MatDialog }
    ]; };
    NodeDownloadDirective.propDecorators = {
        nodes: [{ type: Input, args: ['adfNodeDownload',] }],
        onClick: [{ type: HostListener, args: ['click',] }]
    };
    return NodeDownloadDirective;
}());
export { NodeDownloadDirective };
if (false) {
    /**
     * Nodes to download.
     * @type {?}
     */
    NodeDownloadDirective.prototype.nodes;
    /**
     * @type {?}
     * @private
     */
    NodeDownloadDirective.prototype.apiService;
    /**
     * @type {?}
     * @private
     */
    NodeDownloadDirective.prototype.dialog;
}
