// @alfresco/adf-content-services/esm5/permission-manager/public-api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { PermissionListComponent } from './components/permission-list/permission-list.component';
export { NoPermissionTemplateComponent } from './components/permission-list/no-permission.component';
export { InheritPermissionDirective } from './components/inherited-button.directive';
export { PermissionDisplayModel } from './models/permission.model';
export { NodePermissionDialogService } from './services/node-permission-dialog.service';
export { NodePermissionService } from './services/node-permission.service';
export {} from './components/add-permission/add-permission-dialog-data.interface';
export { AddPermissionPanelComponent } from './components/add-permission/add-permission-panel.component';
export { AddPermissionComponent } from './components/add-permission/add-permission.component';
export { AddPermissionDialogComponent } from './components/add-permission/add-permission-dialog.component';
export { PermissionManagerModule } from './permission-manager.module';
