// @alfresco/adf-process-services/esm5/process-comments/process-comments.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { CommentProcessService } from '@alfresco/adf-core';
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Observable } from 'rxjs';
import { share } from 'rxjs/operators';
var ProcessCommentsComponent = /** @class */ (function () {
    function ProcessCommentsComponent(commentProcessService) {
        var _this = this;
        this.commentProcessService = commentProcessService;
        /**
         * Should the comments be read-only?
         */
        this.readOnly = true;
        /**
         * Emitted when an error occurs.
         */
        this.error = new EventEmitter();
        this.comments = [];
        this.beingAdded = false;
        this.comment$ = new Observable((/**
         * @param {?} observer
         * @return {?}
         */
        function (observer) { return _this.commentObserver = observer; }))
            .pipe(share());
        this.comment$.subscribe((/**
         * @param {?} comment
         * @return {?}
         */
        function (comment) {
            _this.comments.push(comment);
        }));
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ProcessCommentsComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        /** @type {?} */
        var processInstanceId = changes['processInstanceId'];
        if (processInstanceId) {
            if (processInstanceId.currentValue) {
                this.getProcessInstanceComments(processInstanceId.currentValue);
            }
            else {
                this.resetComments();
            }
        }
    };
    /**
     * @private
     * @param {?} processInstanceId
     * @return {?}
     */
    ProcessCommentsComponent.prototype.getProcessInstanceComments = /**
     * @private
     * @param {?} processInstanceId
     * @return {?}
     */
    function (processInstanceId) {
        var _this = this;
        this.resetComments();
        if (processInstanceId) {
            this.commentProcessService.getProcessInstanceComments(processInstanceId).subscribe((/**
             * @param {?} res
             * @return {?}
             */
            function (res) {
                res = res.sort((/**
                 * @param {?} comment1
                 * @param {?} comment2
                 * @return {?}
                 */
                function (comment1, comment2) {
                    /** @type {?} */
                    var date1 = new Date(comment1.created);
                    /** @type {?} */
                    var date2 = new Date(comment2.created);
                    return date1 > date2 ? -1 : date1 < date2 ? 1 : 0;
                }));
                res.forEach((/**
                 * @param {?} comment
                 * @return {?}
                 */
                function (comment) {
                    _this.commentObserver.next(comment);
                }));
            }), (/**
             * @param {?} err
             * @return {?}
             */
            function (err) {
                _this.error.emit(err);
            }));
        }
    };
    /**
     * @private
     * @return {?}
     */
    ProcessCommentsComponent.prototype.resetComments = /**
     * @private
     * @return {?}
     */
    function () {
        this.comments = [];
    };
    /**
     * @return {?}
     */
    ProcessCommentsComponent.prototype.add = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.message && this.message.trim() && !this.beingAdded) {
            this.beingAdded = true;
            this.commentProcessService.addProcessInstanceComment(this.processInstanceId, this.message)
                .subscribe((/**
             * @param {?} res
             * @return {?}
             */
            function (res) {
                _this.comments.unshift(res);
                _this.message = '';
                _this.beingAdded = false;
            }), (/**
             * @param {?} err
             * @return {?}
             */
            function (err) {
                _this.error.emit(err);
                _this.beingAdded = false;
            }));
        }
    };
    /**
     * @return {?}
     */
    ProcessCommentsComponent.prototype.clear = /**
     * @return {?}
     */
    function () {
        this.message = '';
    };
    /**
     * @return {?}
     */
    ProcessCommentsComponent.prototype.isReadOnly = /**
     * @return {?}
     */
    function () {
        return this.readOnly;
    };
    /**
     * @param {?} error
     * @return {?}
     */
    ProcessCommentsComponent.prototype.onError = /**
     * @param {?} error
     * @return {?}
     */
    function (error) {
        this.error.emit(error);
    };
    ProcessCommentsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-process-instance-comments',
                    template: "<div class=\"adf-comments-container\">\n    <div id=\"comment-header\" class=\"adf-comments-header\">\n        {{'ADF_PROCESS_LIST.DETAILS.COMMENTS.HEADER' | translate: { count: comments?.length} }}\n    </div>\n    <div class=\"adf-comments-input-container\" *ngIf=\"!isReadOnly()\">\n        <mat-form-field class=\"adf-full-width\">\n            <input matInput id=\"comment-input\" placeholder=\"{{'ADF_PROCESS_LIST.DETAILS.COMMENTS.ADD' | translate}}\" [(ngModel)]=\"message\" (keyup.enter)=\"add()\" (keyup.esc)=\"clear()\">\n        </mat-form-field>\n    </div>\n\n    <div *ngIf=\"comments.length > 0\">\n        <adf-comment-list [comments]=\"comments\">\n        </adf-comment-list>\n    </div>\n</div>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    ProcessCommentsComponent.ctorParameters = function () { return [
        { type: CommentProcessService }
    ]; };
    ProcessCommentsComponent.propDecorators = {
        processInstanceId: [{ type: Input }],
        readOnly: [{ type: Input }],
        error: [{ type: Output }]
    };
    return ProcessCommentsComponent;
}());
export { ProcessCommentsComponent };
if (false) {
    /**
     * (**required**) The numeric ID of the process instance to display comments for.
     * @type {?}
     */
    ProcessCommentsComponent.prototype.processInstanceId;
    /**
     * Should the comments be read-only?
     * @type {?}
     */
    ProcessCommentsComponent.prototype.readOnly;
    /**
     * Emitted when an error occurs.
     * @type {?}
     */
    ProcessCommentsComponent.prototype.error;
    /** @type {?} */
    ProcessCommentsComponent.prototype.comments;
    /**
     * @type {?}
     * @private
     */
    ProcessCommentsComponent.prototype.commentObserver;
    /** @type {?} */
    ProcessCommentsComponent.prototype.comment$;
    /** @type {?} */
    ProcessCommentsComponent.prototype.message;
    /** @type {?} */
    ProcessCommentsComponent.prototype.beingAdded;
    /**
     * @type {?}
     * @private
     */
    ProcessCommentsComponent.prototype.commentProcessService;
}
