// @alfresco/adf-core/esm5/utils/file-utils.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * @record
 */
export function FileInfo() { }
if (false) {
    /** @type {?|undefined} */
    FileInfo.prototype.entry;
    /** @type {?|undefined} */
    FileInfo.prototype.file;
    /** @type {?|undefined} */
    FileInfo.prototype.relativeFolder;
}
var FileUtils = /** @class */ (function () {
    function FileUtils() {
    }
    /**
     * @param {?} folder
     * @return {?}
     */
    FileUtils.flatten = /**
     * @param {?} folder
     * @return {?}
     */
    function (folder) {
        /** @type {?} */
        var reader = folder.createReader();
        /** @type {?} */
        var files = [];
        return new Promise((/**
         * @param {?} resolve
         * @return {?}
         */
        function (resolve) {
            /** @type {?} */
            var iterations = [];
            ((/**
             * @return {?}
             */
            function traverse() {
                reader.readEntries((/**
                 * @param {?} entries
                 * @return {?}
                 */
                function (entries) {
                    if (!entries.length) {
                        Promise.all(iterations).then((/**
                         * @return {?}
                         */
                        function () { return resolve(files); }));
                    }
                    else {
                        iterations.push(Promise.all(entries.map((/**
                         * @param {?} entry
                         * @return {?}
                         */
                        function (entry) {
                            if (entry.isFile) {
                                return new Promise((/**
                                 * @param {?} resolveFile
                                 * @return {?}
                                 */
                                function (resolveFile) {
                                    entry.file((/**
                                     * @param {?} file
                                     * @return {?}
                                     */
                                    function (file) {
                                        files.push({
                                            entry: entry,
                                            file: file,
                                            relativeFolder: entry.fullPath.replace(/\/[^\/]*$/, '')
                                        });
                                        resolveFile();
                                    }));
                                }));
                            }
                            else {
                                return FileUtils.flatten(entry).then((/**
                                 * @param {?} result
                                 * @return {?}
                                 */
                                function (result) {
                                    files.push.apply(files, tslib_1.__spread(result));
                                }));
                            }
                        }))));
                        // Try calling traverse() again for the same dir, according to spec
                        traverse();
                    }
                }));
            }))();
        }));
    };
    /**
     * @param {?} fileList
     * @return {?}
     */
    FileUtils.toFileArray = /**
     * @param {?} fileList
     * @return {?}
     */
    function (fileList) {
        /** @type {?} */
        var result = [];
        if (fileList && fileList.length > 0) {
            for (var i = 0; i < fileList.length; i++) {
                result.push(fileList[i]);
            }
        }
        return result;
    };
    return FileUtils;
}());
export { FileUtils };
