// @alfresco/adf-core/esm5/info-drawer/info-drawer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ContentChildren, EventEmitter, Input, Output, QueryList, TemplateRef, ViewChild, ViewEncapsulation } from '@angular/core';
var InfoDrawerTabComponent = /** @class */ (function () {
    function InfoDrawerTabComponent() {
        /**
         * The title of the tab.
         */
        this.label = '';
        /**
         * Icon to render for the tab.
         */
        this.icon = null;
    }
    InfoDrawerTabComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-info-drawer-tab',
                    template: '<ng-template><ng-content></ng-content></ng-template>'
                }] }
    ];
    InfoDrawerTabComponent.propDecorators = {
        label: [{ type: Input }],
        icon: [{ type: Input }],
        content: [{ type: ViewChild, args: [TemplateRef,] }]
    };
    return InfoDrawerTabComponent;
}());
export { InfoDrawerTabComponent };
if (false) {
    /**
     * The title of the tab.
     * @type {?}
     */
    InfoDrawerTabComponent.prototype.label;
    /**
     * Icon to render for the tab.
     * @type {?}
     */
    InfoDrawerTabComponent.prototype.icon;
    /** @type {?} */
    InfoDrawerTabComponent.prototype.content;
}
var InfoDrawerComponent = /** @class */ (function () {
    function InfoDrawerComponent() {
        /**
         * The title of the info drawer.
         */
        this.title = null;
        /**
         * The selected index tab.
         */
        this.selectedIndex = 0;
        /**
         * Emitted when the currently active tab changes.
         */
        this.currentTab = new EventEmitter();
    }
    /**
     * @return {?}
     */
    InfoDrawerComponent.prototype.showTabLayout = /**
     * @return {?}
     */
    function () {
        return this.contentBlocks.length > 0;
    };
    /**
     * @param {?} event
     * @return {?}
     */
    InfoDrawerComponent.prototype.onTabChange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.currentTab.emit(event.index);
    };
    InfoDrawerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-info-drawer',
                    template: "<adf-info-drawer-layout>\n    <div *ngIf=\"title\" info-drawer-title>{{title}}</div>\n    <ng-content *ngIf=\"!title\" info-drawer-title select=\"[info-drawer-title]\"></ng-content>\n\n    <ng-content info-drawer-buttons select=\"[info-drawer-buttons]\"></ng-content>\n\n    <ng-container info-drawer-content *ngIf=\"showTabLayout(); then tabLayout else singleLayout\"></ng-container>\n\n    <ng-template #tabLayout>\n        <mat-tab-group [(selectedIndex)]=\"selectedIndex\" class=\"adf-info-drawer-tabs\" (selectedTabChange)=\"onTabChange($event)\">\n            <mat-tab *ngFor=\"let contentBlock of contentBlocks\"\n                [label]=\"contentBlock.label\"\n                class=\"adf-info-drawer-tab\">\n\n                <ng-template mat-tab-label>\n                    <mat-icon *ngIf=\"contentBlock.icon\">{{ contentBlock.icon }}</mat-icon>\n                    <span *ngIf=\"contentBlock.label\">{{ contentBlock.label }}</span>\n                </ng-template>\n\n                <ng-container *ngTemplateOutlet=\"contentBlock.content\"></ng-container>\n            </mat-tab>\n        </mat-tab-group>\n    </ng-template>\n\n    <ng-template #singleLayout>\n        <ng-content select=\"[info-drawer-content]\"></ng-content>\n    </ng-template>\n</adf-info-drawer-layout>\n",
                    encapsulation: ViewEncapsulation.None,
                    host: { 'class': 'adf-info-drawer' },
                    styles: [".adf-info-drawer{display:block}.adf-info-drawer .mat-tab-label{min-width:0}.adf-info-drawer .adf-info-drawer-layout-content{padding:0}.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs){padding:10px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>*,.adf-info-drawer .adf-info-drawer-layout-content>:not(.adf-info-drawer-tabs)>*{margin-bottom:20px;display:block}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content>:last-child{margin-bottom:0}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label{flex-grow:1}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-label .mat-icon+span{padding-left:5px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-ink-bar{height:4px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body{padding:10px}.adf-info-drawer .adf-info-drawer-layout-content .adf-info-drawer-tabs .mat-tab-body-content{overflow:initial}"]
                }] }
    ];
    InfoDrawerComponent.propDecorators = {
        title: [{ type: Input }],
        selectedIndex: [{ type: Input }],
        currentTab: [{ type: Output }],
        contentBlocks: [{ type: ContentChildren, args: [InfoDrawerTabComponent,] }]
    };
    return InfoDrawerComponent;
}());
export { InfoDrawerComponent };
if (false) {
    /**
     * The title of the info drawer.
     * @type {?}
     */
    InfoDrawerComponent.prototype.title;
    /**
     * The selected index tab.
     * @type {?}
     */
    InfoDrawerComponent.prototype.selectedIndex;
    /**
     * Emitted when the currently active tab changes.
     * @type {?}
     */
    InfoDrawerComponent.prototype.currentTab;
    /** @type {?} */
    InfoDrawerComponent.prototype.contentBlocks;
}
