// @alfresco/adf-core/esm2015/services/automation.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { AppConfigService } from '../app-config/app-config.service';
import { AlfrescoApiService } from '../services/alfresco-api.service';
import { StorageService } from './storage.service';
import * as i0 from "@angular/core";
import * as i1 from "../app-config/app-config.service";
import * as i2 from "./alfresco-api.service";
import * as i3 from "./storage.service";
export class CoreAutomationService {
    /**
     * @param {?} appConfigService
     * @param {?} alfrescoApiService
     * @param {?} storageService
     */
    constructor(appConfigService, alfrescoApiService, storageService) {
        this.appConfigService = appConfigService;
        this.alfrescoApiService = alfrescoApiService;
        this.storageService = storageService;
    }
    /**
     * @return {?}
     */
    setup() {
        /** @type {?} */
        const adfProxy = window['adf'] || {};
        adfProxy.setConfigField = (/**
         * @param {?} field
         * @param {?} value
         * @return {?}
         */
        (field, value) => {
            this.appConfigService.config[field] = JSON.parse(value);
        });
        adfProxy.setStorageItem = (/**
         * @param {?} key
         * @param {?} data
         * @return {?}
         */
        (key, data) => {
            this.storageService.setItem(key, data);
        });
        adfProxy.clearStorage = (/**
         * @return {?}
         */
        () => {
            this.storageService.clear();
        });
        adfProxy.apiReset = (/**
         * @return {?}
         */
        () => {
            this.alfrescoApiService.reset();
        });
        window['adf'] = adfProxy;
    }
}
CoreAutomationService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
CoreAutomationService.ctorParameters = () => [
    { type: AppConfigService },
    { type: AlfrescoApiService },
    { type: StorageService }
];
/** @nocollapse */ CoreAutomationService.ngInjectableDef = i0.defineInjectable({ factory: function CoreAutomationService_Factory() { return new CoreAutomationService(i0.inject(i1.AppConfigService), i0.inject(i2.AlfrescoApiService), i0.inject(i3.StorageService)); }, token: CoreAutomationService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    CoreAutomationService.prototype.appConfigService;
    /**
     * @type {?}
     * @private
     */
    CoreAutomationService.prototype.alfrescoApiService;
    /**
     * @type {?}
     * @private
     */
    CoreAutomationService.prototype.storageService;
}
