// @alfresco/adf-core/esm2015/services/page-title.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { Title } from '@angular/platform-browser';
import { AppConfigService } from '../app-config/app-config.service';
import { TranslationService } from './translation.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
import * as i2 from "../app-config/app-config.service";
import * as i3 from "./translation.service";
export class PageTitleService {
    /**
     * @param {?} titleService
     * @param {?} appConfig
     * @param {?} translationService
     */
    constructor(titleService, appConfig, translationService) {
        this.titleService = titleService;
        this.appConfig = appConfig;
        this.translationService = translationService;
        this.originalTitle = '';
        this.translatedTitle = '';
        translationService.translate.onLangChange.subscribe((/**
         * @return {?}
         */
        () => this.onLanguageChanged()));
        translationService.translate.onTranslationChange.subscribe((/**
         * @return {?}
         */
        () => this.onLanguageChanged()));
    }
    /**
     * Sets the page title.
     * @param {?=} value The new title
     * @return {?}
     */
    setTitle(value = '') {
        this.originalTitle = value;
        this.translatedTitle = this.translationService.instant(value);
        this.updateTitle();
    }
    /**
     * @private
     * @return {?}
     */
    onLanguageChanged() {
        this.translatedTitle = this.translationService.instant(this.originalTitle);
        this.updateTitle();
    }
    /**
     * @private
     * @return {?}
     */
    updateTitle() {
        /** @type {?} */
        const name = this.appConfig.get('application.name') || 'Alfresco ADF Application';
        /** @type {?} */
        const title = this.translatedTitle ? `${this.translatedTitle} - ${name}` : `${name}`;
        this.titleService.setTitle(title);
    }
}
PageTitleService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
PageTitleService.ctorParameters = () => [
    { type: Title },
    { type: AppConfigService },
    { type: TranslationService }
];
/** @nocollapse */ PageTitleService.ngInjectableDef = i0.defineInjectable({ factory: function PageTitleService_Factory() { return new PageTitleService(i0.inject(i1.Title), i0.inject(i2.AppConfigService), i0.inject(i3.TranslationService)); }, token: PageTitleService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    PageTitleService.prototype.originalTitle;
    /**
     * @type {?}
     * @private
     */
    PageTitleService.prototype.translatedTitle;
    /**
     * @type {?}
     * @private
     */
    PageTitleService.prototype.titleService;
    /**
     * @type {?}
     * @private
     */
    PageTitleService.prototype.appConfig;
    /**
     * @type {?}
     * @private
     */
    PageTitleService.prototype.translationService;
}
