// @alfresco/adf-core/esm5/templates/error-content/error-content.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ChangeDetectionStrategy, Input, ViewEncapsulation } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { TranslationService } from '../../services/translation.service';
var ErrorContentComponent = /** @class */ (function () {
    function ErrorContentComponent(route, router, translateService) {
        this.route = route;
        this.router = router;
        this.translateService = translateService;
        /**
         * Target URL for the secondary button.
         */
        this.secondaryButtonUrl = 'report-issue';
        /**
         * Target URL for the return button.
         */
        this.returnButtonUrl = '/';
        /**
         * Error code associated with this error.
         */
        this.errorCode = ErrorContentComponent.UNKNOWN_ERROR;
    }
    /**
     * @return {?}
     */
    ErrorContentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.route) {
            this.route.params.forEach((/**
             * @param {?} params
             * @return {?}
             */
            function (params) {
                if (params['id']) {
                    _this.errorCode = _this.checkErrorExists(params['id']) ? params['id'] : ErrorContentComponent.UNKNOWN_ERROR;
                }
            }));
        }
    };
    /**
     * @param {?} errorCode
     * @return {?}
     */
    ErrorContentComponent.prototype.checkErrorExists = /**
     * @param {?} errorCode
     * @return {?}
     */
    function (errorCode) {
        /** @type {?} */
        var errorMessage = this.translateService.instant('ERROR_CONTENT.' + errorCode);
        return errorMessage !== ('ERROR_CONTENT.' + errorCode);
    };
    /**
     * @return {?}
     */
    ErrorContentComponent.prototype.getTranslations = /**
     * @return {?}
     */
    function () {
        this.hasSecondButton = this.translateService.instant('ERROR_CONTENT.' + this.errorCode + '.SECONDARY_BUTTON.TEXT') ? true : false;
    };
    /**
     * @return {?}
     */
    ErrorContentComponent.prototype.ngAfterContentChecked = /**
     * @return {?}
     */
    function () {
        this.getTranslations();
    };
    /**
     * @return {?}
     */
    ErrorContentComponent.prototype.onSecondButton = /**
     * @return {?}
     */
    function () {
        this.router.navigate(['/' + this.secondaryButtonUrl]);
    };
    /**
     * @return {?}
     */
    ErrorContentComponent.prototype.onReturnButton = /**
     * @return {?}
     */
    function () {
        this.router.navigate(['/' + this.returnButtonUrl]);
    };
    ErrorContentComponent.UNKNOWN_ERROR = 'UNKNOWN';
    ErrorContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-error-content',
                    template: "<div class=\"adf-error-content\">\n    <p class=\"adf-error-content-code\">\n        {{ errorCode }}\n    </p>\n    <div class=\"adf-error-content-shadow\"></div>\n    <p class=\"adf-error-content-title\">\n        {{ 'ERROR_CONTENT.' + errorCode + '.TITLE' | translate }}\n    </p>\n    <p class=\"adf-error-content-description\">\n        {{ 'ERROR_CONTENT.' + errorCode + '.DESCRIPTION' | translate }}\n    </p>\n    <div class=\"adf-error-content-buttons\">\n        <a a id=\"adf-secondary-button\" mat-raised-button color=\"primary\"\n            *ngIf=\"hasSecondButton\" (click)=\"onSecondButton()\"\n            class=\"adf-error-content-description-link\">\n            {{ 'ERROR_CONTENT.' + errorCode + '.SECONDARY_BUTTON.TEXT' | translate | uppercase }}\n        </a>\n        <a id=\"adf-return-button\" mat-raised-button color=\"primary\" (click)=\"onReturnButton()\">\n            {{ 'ERROR_CONTENT.' + this.errorCode + '.RETURN_BUTTON.TEXT' | translate | uppercase }}\n        </a>\n    </div>\n</div>\n",
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-error-content' },
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    ErrorContentComponent.ctorParameters = function () { return [
        { type: ActivatedRoute },
        { type: Router },
        { type: TranslationService }
    ]; };
    ErrorContentComponent.propDecorators = {
        secondaryButtonUrl: [{ type: Input }],
        returnButtonUrl: [{ type: Input }],
        errorCode: [{ type: Input }]
    };
    return ErrorContentComponent;
}());
export { ErrorContentComponent };
if (false) {
    /** @type {?} */
    ErrorContentComponent.UNKNOWN_ERROR;
    /**
     * Target URL for the secondary button.
     * @type {?}
     */
    ErrorContentComponent.prototype.secondaryButtonUrl;
    /**
     * Target URL for the return button.
     * @type {?}
     */
    ErrorContentComponent.prototype.returnButtonUrl;
    /**
     * Error code associated with this error.
     * @type {?}
     */
    ErrorContentComponent.prototype.errorCode;
    /** @type {?} */
    ErrorContentComponent.prototype.hasSecondButton;
    /**
     * @type {?}
     * @private
     */
    ErrorContentComponent.prototype.route;
    /**
     * @type {?}
     * @private
     */
    ErrorContentComponent.prototype.router;
    /**
     * @type {?}
     * @private
     */
    ErrorContentComponent.prototype.translateService;
}
