// @alfresco/adf-process-services/esm5/people/components/people-search-field/people-search-field.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { TranslationService, PeopleProcessService } from '@alfresco/adf-core';
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { FormControl } from '@angular/forms';
import { debounceTime, switchMap } from 'rxjs/operators';
import { of } from 'rxjs';
import { getDisplayUser } from '../../helpers/getDisplayUser';
var PeopleSearchFieldComponent = /** @class */ (function () {
    function PeopleSearchFieldComponent(peopleProcessService, translationService) {
        var _this = this;
        this.peopleProcessService = peopleProcessService;
        this.translationService = translationService;
        this.rowClick = new EventEmitter();
        this.searchUser = new FormControl();
        this.defaultPlaceholder = 'ADF_TASK_LIST.PEOPLE.SEARCH_USER';
        this.users$ = this.searchUser.valueChanges
            .pipe(debounceTime(200), switchMap((/**
         * @param {?} searchWord
         * @return {?}
         */
        function (searchWord) {
            if (searchWord && searchWord.trim()) {
                return _this.performSearch(searchWord);
            }
            else {
                return of([]);
            }
        })));
        this.defaultPlaceholder = this.translationService.instant(this.defaultPlaceholder);
    }
    /**
     * @return {?}
     */
    PeopleSearchFieldComponent.prototype.reset = /**
     * @return {?}
     */
    function () {
        this.searchUser.reset();
    };
    Object.defineProperty(PeopleSearchFieldComponent.prototype, "searchPlaceholder", {
        get: /**
         * @return {?}
         */
        function () {
            return this.placeholder || this.defaultPlaceholder;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    PeopleSearchFieldComponent.prototype.onRowClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.rowClick.emit(event);
    };
    /**
     * @param {?} firstName
     * @param {?} lastName
     * @param {?=} delimiter
     * @return {?}
     */
    PeopleSearchFieldComponent.prototype.getDisplayUser = /**
     * @param {?} firstName
     * @param {?} lastName
     * @param {?=} delimiter
     * @return {?}
     */
    function (firstName, lastName, delimiter) {
        if (delimiter === void 0) { delimiter = '-'; }
        return getDisplayUser(firstName, lastName, delimiter);
    };
    /**
     * @param {?} firstName
     * @param {?} lastName
     * @return {?}
     */
    PeopleSearchFieldComponent.prototype.getInitialUserName = /**
     * @param {?} firstName
     * @param {?} lastName
     * @return {?}
     */
    function (firstName, lastName) {
        firstName = (firstName !== null && firstName !== '' ? firstName[0] : '');
        lastName = (lastName !== null && lastName !== '' ? lastName[0] : '');
        return this.getDisplayUser(firstName, lastName, '');
    };
    PeopleSearchFieldComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-people-search-field',
                    template: "<mat-form-field class=\"adf-search-text-container\">\n    <input\n        matInput\n        placeholder=\"{{ searchPlaceholder }}\"\n        type=\"text\"\n        id=\"userSearchText\"\n        [value]=\"\"\n        [formControl]=\"searchUser\"\n        data-automation-id=\"adf-people-search-input\">\n</mat-form-field>\n<ng-container *ngIf=\"users$ | async; let users\">\n    <div class=\"adf-search-list-container\" id=\"search-people-list\" *ngIf=\"users.length > 0\">\n        <adf-people-list [users]=\"users\" (clickRow)=\"onRowClick($event)\">\n            <data-columns>\n                <data-column key=\"firstName\">\n                    <ng-template let-entry=\"$implicit\">\n                            <div *ngIf=\"!entry.row.obj.pictureId\" class=\"adf-people-pic\">\n                                {{getInitialUserName(entry.row.obj.firstName, entry.row.obj.lastName)}}</div>\n                            <div>\n                                <img [alt]=\"getDisplayUser(entry.row.obj.firstName, entry.row.obj.lastName, ' ')\" *ngIf=\"entry.row.obj.pictureId\" class=\"adf-people-img\"\n                                    [src]=\"peopleProcessService.getUserImage(entry.row.obj)\"/>\n                            </div>\n                    </ng-template>\n                </data-column>\n                <data-column key=\"email\" class=\"adf-full-width\">\n                    <ng-template let-entry=\"$implicit\">\n                        <div class=\"adf-people-full-name\">{{ getDisplayUser(entry.row.obj.firstName, entry.row.obj.lastName, ' ') }}</div>\n                    </ng-template>\n                </data-column>\n            </data-columns>\n        </adf-people-list>\n    </div>\n</ng-container>\n",
                    host: { 'class': 'adf-people-search-field' },
                    encapsulation: ViewEncapsulation.None,
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PeopleSearchFieldComponent.ctorParameters = function () { return [
        { type: PeopleProcessService },
        { type: TranslationService }
    ]; };
    PeopleSearchFieldComponent.propDecorators = {
        performSearch: [{ type: Input }],
        placeholder: [{ type: Input }],
        rowClick: [{ type: Output }]
    };
    return PeopleSearchFieldComponent;
}());
export { PeopleSearchFieldComponent };
if (false) {
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.performSearch;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.placeholder;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.rowClick;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.users$;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.searchUser;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.defaultPlaceholder;
    /** @type {?} */
    PeopleSearchFieldComponent.prototype.peopleProcessService;
    /**
     * @type {?}
     * @private
     */
    PeopleSearchFieldComponent.prototype.translationService;
}
