// @alfresco/adf-content-services/esm5/directives/node-lock.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:no-input-rename  */
import { Directive, ElementRef, Renderer2, HostListener, Input } from '@angular/core';
import { Node } from '@alfresco/js-api';
import { AllowableOperationsEnum, ContentService } from '@alfresco/adf-core';
import { ContentNodeDialogService } from '../content-node-selector/content-node-dialog.service';
var NodeLockDirective = /** @class */ (function () {
    function NodeLockDirective(element, renderer, contentService, contentNodeDialogService) {
        this.element = element;
        this.renderer = renderer;
        this.contentService = contentService;
        this.contentNodeDialogService = contentNodeDialogService;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    NodeLockDirective.prototype.onClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        this.contentNodeDialogService.openLockNodeDialog(this.node);
    };
    /**
     * @return {?}
     */
    NodeLockDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var hasAllowableOperations = this.contentService.hasAllowableOperations(this.node, AllowableOperationsEnum.LOCK);
        this.renderer.setProperty(this.element.nativeElement, 'disabled', !hasAllowableOperations);
    };
    NodeLockDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-node-lock]'
                },] }
    ];
    /** @nocollapse */
    NodeLockDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 },
        { type: ContentService },
        { type: ContentNodeDialogService }
    ]; };
    NodeLockDirective.propDecorators = {
        node: [{ type: Input, args: ['adf-node-lock',] }],
        onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
    };
    return NodeLockDirective;
}());
export { NodeLockDirective };
if (false) {
    /**
     * Node to lock/unlock.
     * @type {?}
     */
    NodeLockDirective.prototype.node;
    /** @type {?} */
    NodeLockDirective.prototype.element;
    /**
     * @type {?}
     * @private
     */
    NodeLockDirective.prototype.renderer;
    /**
     * @type {?}
     * @private
     */
    NodeLockDirective.prototype.contentService;
    /**
     * @type {?}
     * @private
     */
    NodeLockDirective.prototype.contentNodeDialogService;
}
