// @alfresco/adf-core/esm5/card-view/components/card-view-selectitem/card-view-selectitem.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input } from '@angular/core';
import { CardViewSelectItemModel } from '../../models/card-view-selectitem.model';
import { CardViewUpdateService } from '../../services/card-view-update.service';
import { Observable } from 'rxjs';
var CardViewSelectItemComponent = /** @class */ (function () {
    function CardViewSelectItemComponent(cardViewUpdateService) {
        this.cardViewUpdateService = cardViewUpdateService;
        this.editable = false;
    }
    /**
     * @return {?}
     */
    CardViewSelectItemComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.value = this.property.value;
    };
    /**
     * @return {?}
     */
    CardViewSelectItemComponent.prototype.isEditable = /**
     * @return {?}
     */
    function () {
        return this.editable && this.property.editable;
    };
    /**
     * @return {?}
     */
    CardViewSelectItemComponent.prototype.getOptions = /**
     * @return {?}
     */
    function () {
        return this.options$ || this.property.options$;
    };
    /**
     * @param {?} event
     * @return {?}
     */
    CardViewSelectItemComponent.prototype.onChange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.cardViewUpdateService.update(this.property, event.value);
        this.property.value = event.value;
    };
    CardViewSelectItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-card-view-selectitem',
                    template: "<div [attr.data-automation-id]=\"'card-select-label-' + property.key\" class=\"adf-property-label\">{{ property.label | translate }}</div>\n<div class=\"adf-property-value\">\n    <div *ngIf=\"!isEditable()\" data-automation-class=\"read-only-value\">{{ property.displayValue | async }}</div>\n    <div *ngIf=\"isEditable()\">\n        <mat-form-field>\n            <mat-select [(value)]=\"value\" (selectionChange)=\"onChange($event)\" data-automation-class=\"select-box\">\n              <mat-option *ngFor=\"let option of getOptions() | async\" [value]=\"option.key\">\n                {{ option.label | translate }}\n              </mat-option>\n            </mat-select>\n          </mat-form-field>\n    </div>\n</div>\n",
                    styles: [".mat-form-field-type-mat-select{width:100%}"]
                }] }
    ];
    /** @nocollapse */
    CardViewSelectItemComponent.ctorParameters = function () { return [
        { type: CardViewUpdateService }
    ]; };
    CardViewSelectItemComponent.propDecorators = {
        property: [{ type: Input }],
        editable: [{ type: Input }],
        options$: [{ type: Input }]
    };
    return CardViewSelectItemComponent;
}());
export { CardViewSelectItemComponent };
if (false) {
    /** @type {?} */
    CardViewSelectItemComponent.prototype.property;
    /** @type {?} */
    CardViewSelectItemComponent.prototype.editable;
    /** @type {?} */
    CardViewSelectItemComponent.prototype.options$;
    /** @type {?} */
    CardViewSelectItemComponent.prototype.value;
    /**
     * @type {?}
     * @private
     */
    CardViewSelectItemComponent.prototype.cardViewUpdateService;
}
