// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-text.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelTextDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelTextDirective, _super);
    function RaphaelTextDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelTextDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.text === null || this.text === undefined) {
            this.text = '';
        }
        this.draw(this.position, this.text);
    };
    /**
     * @param {?} position
     * @param {?} text
     * @return {?}
     */
    RaphaelTextDirective.prototype.draw = /**
     * @param {?} position
     * @param {?} text
     * @return {?}
     */
    function (position, text) {
        /** @type {?} */
        var textPaper = this.paper.text(position.x, position.y, text).attr({
            'text-anchor': 'middle',
            'font-family': 'Arial',
            'font-size': '11',
            'fill': '#373e48'
        });
        textPaper.transform(this.transform);
        return textPaper;
    };
    RaphaelTextDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-text, raphael-text' },] }
    ];
    /** @nocollapse */
    RaphaelTextDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelTextDirective.propDecorators = {
        paper: [{ type: Input }],
        position: [{ type: Input }],
        transform: [{ type: Input }],
        text: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelTextDirective;
}(RaphaelBase));
export { RaphaelTextDirective };
if (false) {
    /** @type {?} */
    RaphaelTextDirective.prototype.paper;
    /** @type {?} */
    RaphaelTextDirective.prototype.position;
    /** @type {?} */
    RaphaelTextDirective.prototype.transform;
    /** @type {?} */
    RaphaelTextDirective.prototype.text;
    /** @type {?} */
    RaphaelTextDirective.prototype.error;
    /** @type {?} */
    RaphaelTextDirective.prototype.elementRef;
}
