// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-rect.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelRectDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelRectDirective, _super);
    function RaphaelRectDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.radius = 0;
        _this.error = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelRectDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var opts = {
            'stroke-width': this.strokeWidth,
            'fill': this.fillColors,
            'stroke': this.stroke,
            'fill-opacity': this.fillOpacity
        };
        /** @type {?} */
        var elementDraw = this.draw(this.leftCorner, this.width, this.height, this.radius, opts);
        elementDraw.node.id = this.elementId;
    };
    /**
     * @param {?} leftCorner
     * @param {?} width
     * @param {?} height
     * @param {?} radius
     * @param {?} opts
     * @return {?}
     */
    RaphaelRectDirective.prototype.draw = /**
     * @param {?} leftCorner
     * @param {?} width
     * @param {?} height
     * @param {?} radius
     * @param {?} opts
     * @return {?}
     */
    function (leftCorner, width, height, radius, opts) {
        return this.paper.rect(leftCorner.x, leftCorner.y, width, height, radius).attr(opts);
    };
    RaphaelRectDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-rect, raphael-rect' },] }
    ];
    /** @nocollapse */
    RaphaelRectDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelRectDirective.propDecorators = {
        paper: [{ type: Input }],
        leftCorner: [{ type: Input }],
        width: [{ type: Input }],
        height: [{ type: Input }],
        radius: [{ type: Input }],
        fillColors: [{ type: Input }],
        stroke: [{ type: Input }],
        strokeWidth: [{ type: Input }],
        fillOpacity: [{ type: Input }],
        elementId: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelRectDirective;
}(RaphaelBase));
export { RaphaelRectDirective };
if (false) {
    /** @type {?} */
    RaphaelRectDirective.prototype.paper;
    /** @type {?} */
    RaphaelRectDirective.prototype.leftCorner;
    /** @type {?} */
    RaphaelRectDirective.prototype.width;
    /** @type {?} */
    RaphaelRectDirective.prototype.height;
    /** @type {?} */
    RaphaelRectDirective.prototype.radius;
    /** @type {?} */
    RaphaelRectDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelRectDirective.prototype.stroke;
    /** @type {?} */
    RaphaelRectDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelRectDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelRectDirective.prototype.elementId;
    /** @type {?} */
    RaphaelRectDirective.prototype.error;
    /** @type {?} */
    RaphaelRectDirective.prototype.elementRef;
}
