// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
var RaphaelService = /** @class */ (function () {
    function RaphaelService() {
        this.width = 300;
        this.height = 400;
    }
    /**
     * @param {?} element
     * @return {?}
     */
    RaphaelService.prototype.getInstance = /**
     * @param {?} element
     * @return {?}
     */
    function (element) {
        if (!this.paper) {
            this.ctx = element.nativeElement;
            this.refresh();
        }
        return this.paper;
    };
    /**
     * @private
     * @return {?}
     */
    RaphaelService.prototype.refresh = /**
     * @private
     * @return {?}
     */
    function () {
        this.ngOnDestroy();
        this.paper = this.getPaperBuilder(this.ctx);
    };
    /**
     * @param {?} ctx
     * @return {?}
     */
    RaphaelService.prototype.getPaperBuilder = /**
     * @param {?} ctx
     * @return {?}
     */
    function (ctx) {
        if (typeof Raphael === 'undefined') {
            throw new Error('insights configuration issue: Embedding Chart.js lib is mandatory');
        }
        /** @type {?} */
        var paper = new Raphael(ctx, this.width, this.height);
        return paper;
    };
    /**
     * @return {?}
     */
    RaphaelService.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.paper) {
            this.paper.clear();
            this.paper = void 0;
        }
    };
    /**
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    RaphaelService.prototype.setting = /**
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    function (width, height) {
        this.width = width;
        this.height = height;
    };
    /**
     * @return {?}
     */
    RaphaelService.prototype.reset = /**
     * @return {?}
     */
    function () {
        this.ngOnDestroy();
    };
    RaphaelService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    RaphaelService.ctorParameters = function () { return []; };
    return RaphaelService;
}());
export { RaphaelService };
if (false) {
    /** @type {?} */
    RaphaelService.prototype.paper;
    /** @type {?} */
    RaphaelService.prototype.width;
    /** @type {?} */
    RaphaelService.prototype.height;
    /**
     * @type {?}
     * @private
     */
    RaphaelService.prototype.ctx;
}
