// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-circle.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelCircleDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelCircleDirective, _super);
    function RaphaelCircleDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelCircleDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var opts = { 'stroke-width': this.strokeWidth, 'fill': this.fillColors, 'stroke': this.stroke, 'fill-opacity': this.fillOpacity };
        /** @type {?} */
        var drawElement = this.draw(this.center, this.radius, opts);
        drawElement.node.id = this.elementId;
    };
    /**
     * @param {?} center
     * @param {?} radius
     * @param {?} opts
     * @return {?}
     */
    RaphaelCircleDirective.prototype.draw = /**
     * @param {?} center
     * @param {?} radius
     * @param {?} opts
     * @return {?}
     */
    function (center, radius, opts) {
        /** @type {?} */
        var circle = this.paper.circle(center.x, center.y, radius).attr(opts);
        return circle;
    };
    RaphaelCircleDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-circle, raphael-circle' },] }
    ];
    /** @nocollapse */
    RaphaelCircleDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelCircleDirective.propDecorators = {
        paper: [{ type: Input }],
        center: [{ type: Input }],
        radius: [{ type: Input }],
        strokeWidth: [{ type: Input }],
        fillColors: [{ type: Input }],
        stroke: [{ type: Input }],
        fillOpacity: [{ type: Input }],
        elementId: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelCircleDirective;
}(RaphaelBase));
export { RaphaelCircleDirective };
if (false) {
    /** @type {?} */
    RaphaelCircleDirective.prototype.paper;
    /** @type {?} */
    RaphaelCircleDirective.prototype.center;
    /** @type {?} */
    RaphaelCircleDirective.prototype.radius;
    /** @type {?} */
    RaphaelCircleDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelCircleDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelCircleDirective.prototype.stroke;
    /** @type {?} */
    RaphaelCircleDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelCircleDirective.prototype.elementId;
    /** @type {?} */
    RaphaelCircleDirective.prototype.error;
    /** @type {?} */
    RaphaelCircleDirective.prototype.elementRef;
}
