// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-cross.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelCrossDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelCrossDirective, _super);
    function RaphaelCrossDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelCrossDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var opts = { 'stroke-width': this.strokeWidth, 'fill': this.fillColors, 'stroke': this.stroke, 'fill-opacity': this.fillOpacity };
        this.draw(this.center, this.width, this.height, opts);
    };
    /**
     * @param {?} center
     * @param {?} width
     * @param {?} height
     * @param {?=} opts
     * @return {?}
     */
    RaphaelCrossDirective.prototype.draw = /**
     * @param {?} center
     * @param {?} width
     * @param {?} height
     * @param {?=} opts
     * @return {?}
     */
    function (center, width, height, opts) {
        /** @type {?} */
        var quarterWidth = width / 4;
        /** @type {?} */
        var quarterHeight = height / 4;
        return this.paper.path('M' + (center.x + quarterWidth + 3) + ' ' + (center.y + quarterHeight + 3) +
            'L' + (center.x + 3 * quarterWidth - 3) + ' ' + (center.y + 3 * quarterHeight - 3) +
            'M' + (center.x + quarterWidth + 3) + ' ' + (center.y + 3 * quarterHeight - 3) +
            'L' + (center.x + 3 * quarterWidth - 3) + ' ' + (center.y + quarterHeight + 3)).attr(opts);
    };
    RaphaelCrossDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-cross, raphael-cross' },] }
    ];
    /** @nocollapse */
    RaphaelCrossDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelCrossDirective.propDecorators = {
        center: [{ type: Input }],
        width: [{ type: Input }],
        height: [{ type: Input }],
        fillColors: [{ type: Input }],
        stroke: [{ type: Input }],
        strokeWidth: [{ type: Input }],
        fillOpacity: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelCrossDirective;
}(RaphaelBase));
export { RaphaelCrossDirective };
if (false) {
    /** @type {?} */
    RaphaelCrossDirective.prototype.center;
    /** @type {?} */
    RaphaelCrossDirective.prototype.width;
    /** @type {?} */
    RaphaelCrossDirective.prototype.height;
    /** @type {?} */
    RaphaelCrossDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelCrossDirective.prototype.stroke;
    /** @type {?} */
    RaphaelCrossDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelCrossDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelCrossDirective.prototype.error;
    /** @type {?} */
    RaphaelCrossDirective.prototype.elementRef;
}
