// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-flow-arrow.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Polyline } from './polyline';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelFlowArrowDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelFlowArrowDirective, _super);
    function RaphaelFlowArrowDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        _this.ARROW_WIDTH = 4;
        _this.SEQUENCE_FLOW_STROKE = 1.5;
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelFlowArrowDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.draw(this.flow);
    };
    /**
     * @param {?} flow
     * @return {?}
     */
    RaphaelFlowArrowDirective.prototype.draw = /**
     * @param {?} flow
     * @return {?}
     */
    function (flow) {
        /** @type {?} */
        var line = this.drawLine(flow);
        this.drawArrow(line);
    };
    /**
     * @param {?} flow
     * @return {?}
     */
    RaphaelFlowArrowDirective.prototype.drawLine = /**
     * @param {?} flow
     * @return {?}
     */
    function (flow) {
        /** @type {?} */
        var polyline = new Polyline(flow.id, flow.waypoints, this.SEQUENCE_FLOW_STROKE, this.paper);
        polyline.element = this.paper.path(polyline.path);
        polyline.element.attr({ 'stroke-width': this.SEQUENCE_FLOW_STROKE });
        polyline.element.attr({ 'stroke': '#585858' });
        polyline.element.node.id = this.flow.id;
        /** @type {?} */
        var lastLineIndex = polyline.getLinesCount() - 1;
        /** @type {?} */
        var line = polyline.getLine(lastLineIndex);
        return line;
    };
    /**
     * @param {?} line
     * @return {?}
     */
    RaphaelFlowArrowDirective.prototype.drawArrow = /**
     * @param {?} line
     * @return {?}
     */
    function (line) {
        /** @type {?} */
        var doubleArrowWidth = 2 * this.ARROW_WIDTH;
        /** @type {?} */
        var width = this.ARROW_WIDTH / 2 + .5;
        /** @type {?} */
        var arrowHead = this.paper.path('M0 0L-' + width + '-' + doubleArrowWidth + 'L' + width + ' -' + doubleArrowWidth + 'z');
        arrowHead.transform('t' + line.x2 + ',' + line.y2);
        /** @type {?} */
        var angle = Raphael.deg(line.angle - Math.PI / 2);
        arrowHead.transform('...r' + angle + ' 0 0');
        arrowHead.attr('fill', '#585858');
        arrowHead.attr('stroke-width', this.SEQUENCE_FLOW_STROKE);
        arrowHead.attr('stroke', '#585858');
    };
    RaphaelFlowArrowDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-flow-arrow, raphael-flow-arrow' },] }
    ];
    /** @nocollapse */
    RaphaelFlowArrowDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelFlowArrowDirective.propDecorators = {
        paper: [{ type: Input }],
        flow: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelFlowArrowDirective;
}(RaphaelBase));
export { RaphaelFlowArrowDirective };
if (false) {
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.paper;
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.flow;
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.error;
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.ARROW_WIDTH;
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.SEQUENCE_FLOW_STROKE;
    /** @type {?} */
    RaphaelFlowArrowDirective.prototype.elementRef;
}
