// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-plus.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelPlusDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelPlusDirective, _super);
    function RaphaelPlusDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelPlusDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var opts = { 'stroke-width': this.strokeWidth, 'fill': this.fillColors, 'stroke': this.stroke, 'fill-opacity': this.fillOpacity };
        this.draw(this.center, opts);
    };
    /**
     * @param {?} center
     * @param {?=} opts
     * @return {?}
     */
    RaphaelPlusDirective.prototype.draw = /**
     * @param {?} center
     * @param {?=} opts
     * @return {?}
     */
    function (center, opts) {
        /** @type {?} */
        var path = this.paper.path('M 6.75,16 L 25.75,16 M 16,6.75 L 16,25.75').attr(opts);
        return path.transform('T' + (center.x + 4) + ',' + (center.y + 4));
    };
    RaphaelPlusDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-plus, raphael-plus' },] }
    ];
    /** @nocollapse */
    RaphaelPlusDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelPlusDirective.propDecorators = {
        center: [{ type: Input }],
        fillColors: [{ type: Input }],
        stroke: [{ type: Input }],
        strokeWidth: [{ type: Input }],
        fillOpacity: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelPlusDirective;
}(RaphaelBase));
export { RaphaelPlusDirective };
if (false) {
    /** @type {?} */
    RaphaelPlusDirective.prototype.center;
    /** @type {?} */
    RaphaelPlusDirective.prototype.fillColors;
    /** @type {?} */
    RaphaelPlusDirective.prototype.stroke;
    /** @type {?} */
    RaphaelPlusDirective.prototype.strokeWidth;
    /** @type {?} */
    RaphaelPlusDirective.prototype.fillOpacity;
    /** @type {?} */
    RaphaelPlusDirective.prototype.error;
    /** @type {?} */
    RaphaelPlusDirective.prototype.elementRef;
}
