// @alfresco/adf-insights/esm5/diagram/components/raphael/raphael-multiline-text.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { Point } from './models/point';
import { RaphaelBase } from './raphael-base';
import { RaphaelService } from './raphael.service';
/**
 * Directive selectors without adf- prefix will be deprecated on 3.0.0
 */
var RaphaelMultilineTextDirective = /** @class */ (function (_super) {
    tslib_1.__extends(RaphaelMultilineTextDirective, _super);
    function RaphaelMultilineTextDirective(elementRef, raphaelService) {
        var _this = _super.call(this, elementRef, raphaelService) || this;
        _this.elementRef = elementRef;
        _this.error = new EventEmitter();
        _this.TEXT_PADDING = 3;
        return _this;
    }
    /**
     * @return {?}
     */
    RaphaelMultilineTextDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.text === null || this.text === undefined) {
            this.text = '';
        }
        this.draw(this.position, this.text);
    };
    /**
     * @param {?} position
     * @param {?} text
     * @return {?}
     */
    RaphaelMultilineTextDirective.prototype.draw = /**
     * @param {?} position
     * @param {?} text
     * @return {?}
     */
    function (position, text) {
        /** @type {?} */
        var textPaper = this.paper.text(position.x + this.TEXT_PADDING, position.y + this.TEXT_PADDING, text).attr({
            'text-anchor': 'middle',
            'font-family': 'Arial',
            'font-size': '11',
            'fill': '#373e48'
        });
        /** @type {?} */
        var formattedText = this.formatText(textPaper, text, this.elementWidth);
        textPaper.attr({
            'text': formattedText
        });
        textPaper.transform(this.transform);
        return textPaper;
    };
    /**
     * @private
     * @param {?} textPaper
     * @param {?} text
     * @param {?} elementWidth
     * @return {?}
     */
    RaphaelMultilineTextDirective.prototype.formatText = /**
     * @private
     * @param {?} textPaper
     * @param {?} text
     * @param {?} elementWidth
     * @return {?}
     */
    function (textPaper, text, elementWidth) {
        /** @type {?} */
        var pText = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
        textPaper.attr({
            'text': pText
        });
        /** @type {?} */
        var letterWidth = textPaper.getBBox().width / text.length;
        /** @type {?} */
        var removedLineBreaks = text.split('\n');
        /** @type {?} */
        var actualRowLength = 0;
        /** @type {?} */
        var formattedText = [];
        removedLineBreaks.forEach((/**
         * @param {?} sentence
         * @return {?}
         */
        function (sentence) {
            /** @type {?} */
            var words = sentence.split(' ');
            words.forEach((/**
             * @param {?} word
             * @return {?}
             */
            function (word) {
                /** @type {?} */
                var length = word.length;
                if (actualRowLength + (length * letterWidth) > elementWidth) {
                    formattedText.push('\n');
                    actualRowLength = 0;
                }
                actualRowLength += length * letterWidth;
                formattedText.push(word + ' ');
            }));
            formattedText.push('\n');
            actualRowLength = 0;
        }));
        return formattedText.join('');
    };
    RaphaelMultilineTextDirective.decorators = [
        { type: Directive, args: [{ selector: 'adf-raphael-multiline-text, raphael-multiline-text' },] }
    ];
    /** @nocollapse */
    RaphaelMultilineTextDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: RaphaelService }
    ]; };
    RaphaelMultilineTextDirective.propDecorators = {
        paper: [{ type: Input }],
        position: [{ type: Input }],
        transform: [{ type: Input }],
        text: [{ type: Input }],
        elementWidth: [{ type: Input }],
        error: [{ type: Output }]
    };
    return RaphaelMultilineTextDirective;
}(RaphaelBase));
export { RaphaelMultilineTextDirective };
if (false) {
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.paper;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.position;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.transform;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.text;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.elementWidth;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.error;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.TEXT_PADDING;
    /** @type {?} */
    RaphaelMultilineTextDirective.prototype.elementRef;
}
