// @alfresco/adf-insights/esm5/analytics-process/components/widgets/checkbox/checkbox.widget.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector no-access-missing-member no-input-rename  */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { WidgetComponent } from './../widget.component';
var CheckboxWidgetAnalyticsComponent = /** @class */ (function (_super) {
    tslib_1.__extends(CheckboxWidgetAnalyticsComponent, _super);
    function CheckboxWidgetAnalyticsComponent(elementRef) {
        var _this = _super.call(this) || this;
        _this.elementRef = elementRef;
        return _this;
    }
    CheckboxWidgetAnalyticsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'analytics-checkbox-widget',
                    template: "<div [formGroup]=\"formGroup\">\n        <mat-checkbox\n        color=\"primary\"\n        formControlName=\"{{controllerName}}\"\n        [id]=\"field.id\"\n        [checked]=\"field.value\"\n        [(ngModel)]=\"field.value\"\n        (ngModelChange)=\"changeValue(field)\">{{field.nameKey | translate}}</mat-checkbox>\n</div>",
                    encapsulation: ViewEncapsulation.None
                }] }
    ];
    /** @nocollapse */
    CheckboxWidgetAnalyticsComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    CheckboxWidgetAnalyticsComponent.propDecorators = {
        field: [{ type: Input }],
        formGroup: [{ type: Input, args: ['group',] }],
        controllerName: [{ type: Input, args: ['controllerName',] }]
    };
    return CheckboxWidgetAnalyticsComponent;
}(WidgetComponent));
export { CheckboxWidgetAnalyticsComponent };
if (false) {
    /** @type {?} */
    CheckboxWidgetAnalyticsComponent.prototype.field;
    /** @type {?} */
    CheckboxWidgetAnalyticsComponent.prototype.formGroup;
    /** @type {?} */
    CheckboxWidgetAnalyticsComponent.prototype.controllerName;
    /** @type {?} */
    CheckboxWidgetAnalyticsComponent.prototype.elementRef;
}
