// @alfresco/adf-extensions/esm5/lib/components/dynamic-column/dynamic-column.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewChild, ViewContainerRef, ComponentFactoryResolver, ViewEncapsulation, ChangeDetectionStrategy } from '@angular/core';
import { ExtensionService } from '../../services/extension.service';
var DynamicColumnComponent = /** @class */ (function () {
    function DynamicColumnComponent(extensions, componentFactoryResolver) {
        this.extensions = extensions;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    /**
     * @return {?}
     */
    DynamicColumnComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var componentType = this.extensions.getComponentById(this.id);
        if (componentType) {
            /** @type {?} */
            var factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
            if (factory) {
                this.content.clear();
                this.componentRef = this.content.createComponent(factory, 0);
                this.updateInstance();
            }
        }
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    DynamicColumnComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes.node) {
            this.updateInstance();
        }
    };
    /**
     * @return {?}
     */
    DynamicColumnComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.componentRef) {
            this.componentRef.destroy();
            this.componentRef = null;
        }
    };
    /**
     * @private
     * @return {?}
     */
    DynamicColumnComponent.prototype.updateInstance = /**
     * @private
     * @return {?}
     */
    function () {
        if (this.componentRef && this.componentRef.instance) {
            this.componentRef.instance.context = this.context;
        }
    };
    DynamicColumnComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-dynamic-column',
                    template: "\n    <ng-container #content></ng-container>\n  ",
                    encapsulation: ViewEncapsulation.None,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'adf-dynamic-column' },
                    styles: ["\n      .adf-dynamic-column {\n        display: flex;\n        align-items: center;\n      }\n    "]
                }] }
    ];
    /** @nocollapse */
    DynamicColumnComponent.ctorParameters = function () { return [
        { type: ExtensionService },
        { type: ComponentFactoryResolver }
    ]; };
    DynamicColumnComponent.propDecorators = {
        content: [{ type: ViewChild, args: ['content', { read: ViewContainerRef },] }],
        id: [{ type: Input }],
        context: [{ type: Input }]
    };
    return DynamicColumnComponent;
}());
export { DynamicColumnComponent };
if (false) {
    /** @type {?} */
    DynamicColumnComponent.prototype.content;
    /** @type {?} */
    DynamicColumnComponent.prototype.id;
    /** @type {?} */
    DynamicColumnComponent.prototype.context;
    /**
     * @type {?}
     * @private
     */
    DynamicColumnComponent.prototype.componentRef;
    /**
     * @type {?}
     * @private
     */
    DynamicColumnComponent.prototype.extensions;
    /**
     * @type {?}
     * @private
     */
    DynamicColumnComponent.prototype.componentFactoryResolver;
}
