// @alfresco/adf-process-services/esm5/app-list/select-apps-dialog-component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AppsProcessService } from '@alfresco/adf-core';
import { Component, Inject } from '@angular/core';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
var SelectAppsDialogComponent = /** @class */ (function () {
    function SelectAppsDialogComponent(appsProcessService, dialogRef, data) {
        var _this = this;
        this.appsProcessService = appsProcessService;
        this.dialogRef = dialogRef;
        this.data = data;
        this.appsProcessService.getDeployedApplications().subscribe((/**
         * @param {?} apps
         * @return {?}
         */
        function (apps) {
            _this.processApps = apps.filter((/**
             * @param {?} currentApp
             * @return {?}
             */
            function (currentApp) {
                return currentApp.id;
            }));
        }));
    }
    /**
     * @return {?}
     */
    SelectAppsDialogComponent.prototype.onStart = /**
     * @return {?}
     */
    function () {
        this.dialogRef.close(this.selectedProcess);
    };
    SelectAppsDialogComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-select-apps-dialog',
                    template: "<header mat-dialog-title id=\"adf-selet-app-dialog-title\">{{'APP.DIALOG.TITLE' | translate}}</header>\n<section mat-dialog-content>\n\n    <mat-select id=\"adf-selet-app-dialog-dropdown\" placeholder=\"{{'APP.DIALOG.LIST' | translate}}\" [(value)]=\"selectedProcess\" >\n        <mat-option *ngFor=\"let currentProcessApp of processApps\" [value]=\"currentProcessApp\">\n            {{ currentProcessApp.name }}\n        </mat-option>\n    </mat-select>\n\n</section>\n<footer mat-dialog-actions fxLayout=\"row\" fxLayoutAlign=\"end center\">\n    <button mat-button (click)=\"onStart()\">{{'APP.DIALOG.START' | translate}}</button>\n</footer>\n"
                }] }
    ];
    /** @nocollapse */
    SelectAppsDialogComponent.ctorParameters = function () { return [
        { type: AppsProcessService },
        { type: MatDialogRef },
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] }
    ]; };
    return SelectAppsDialogComponent;
}());
export { SelectAppsDialogComponent };
if (false) {
    /** @type {?} */
    SelectAppsDialogComponent.prototype.processApps;
    /** @type {?} */
    SelectAppsDialogComponent.prototype.selectedProcess;
    /**
     * @type {?}
     * @private
     */
    SelectAppsDialogComponent.prototype.appsProcessService;
    /** @type {?} */
    SelectAppsDialogComponent.prototype.dialogRef;
    /** @type {?} */
    SelectAppsDialogComponent.prototype.data;
}
