// @alfresco/adf-core/esm5/datatable/components/datatable/drop-zone.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Directive, Input, ElementRef, NgZone } from '@angular/core';
var DropZoneDirective = /** @class */ (function () {
    function DropZoneDirective(elementRef, ngZone) {
        this.ngZone = ngZone;
        this.dropTarget = 'cell';
        this.element = elementRef.nativeElement;
    }
    /**
     * @return {?}
     */
    DropZoneDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.ngZone.runOutsideAngular((/**
         * @return {?}
         */
        function () {
            _this.element.addEventListener('dragover', _this.onDragOver.bind(_this));
            _this.element.addEventListener('drop', _this.onDrop.bind(_this));
        }));
    };
    /**
     * @return {?}
     */
    DropZoneDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.element.removeEventListener('dragover', this.onDragOver);
        this.element.removeEventListener('drop', this.onDrop);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    DropZoneDirective.prototype.onDragOver = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        /** @type {?} */
        var domEvent = new CustomEvent(this.dropTarget + "-dragover", {
            detail: {
                target: this.dropTarget,
                event: event,
                column: this.dropColumn,
                row: this.dropRow
            },
            bubbles: true
        });
        this.element.dispatchEvent(domEvent);
        if (domEvent.defaultPrevented) {
            event.preventDefault();
            event.stopPropagation();
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    DropZoneDirective.prototype.onDrop = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        /** @type {?} */
        var domEvent = new CustomEvent(this.dropTarget + "-drop", {
            detail: {
                target: this.dropTarget,
                event: event,
                column: this.dropColumn,
                row: this.dropRow
            },
            bubbles: true
        });
        this.element.dispatchEvent(domEvent);
        if (domEvent.defaultPrevented) {
            event.preventDefault();
            event.stopPropagation();
        }
    };
    DropZoneDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-drop-zone]'
                },] }
    ];
    /** @nocollapse */
    DropZoneDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: NgZone }
    ]; };
    DropZoneDirective.propDecorators = {
        dropTarget: [{ type: Input }],
        dropRow: [{ type: Input }],
        dropColumn: [{ type: Input }]
    };
    return DropZoneDirective;
}());
export { DropZoneDirective };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DropZoneDirective.prototype.element;
    /** @type {?} */
    DropZoneDirective.prototype.dropTarget;
    /** @type {?} */
    DropZoneDirective.prototype.dropRow;
    /** @type {?} */
    DropZoneDirective.prototype.dropColumn;
    /**
     * @type {?}
     * @private
     */
    DropZoneDirective.prototype.ngZone;
}
