// @alfresco/adf-core/esm5/datatable/components/datatable/empty-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Directive, ViewEncapsulation } from '@angular/core';
var EmptyListComponent = /** @class */ (function () {
    function EmptyListComponent() {
    }
    EmptyListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-empty-list',
                    template: "<div class=\"adf-empty-list_template\">\n    <ng-content select=\"[adf-empty-list-header]\"></ng-content>\n    <ng-content select=\"[adf-empty-list-body]\"></ng-content>\n    <ng-content select=\"[adf-empty-list-footer]\"></ng-content>\n    <ng-content></ng-content>\n</div>",
                    encapsulation: ViewEncapsulation.None,
                    styles: [".adf-empty-list_template{text-align:center;margin-top:20px;margin-bottom:20px}"]
                }] }
    ];
    return EmptyListComponent;
}());
export { EmptyListComponent };
var EmptyListHeaderDirective = /** @class */ (function () {
    function EmptyListHeaderDirective() {
    }
    EmptyListHeaderDirective.decorators = [
        { type: Directive, args: [{ selector: '[adf-empty-list-header]' },] }
    ];
    return EmptyListHeaderDirective;
}());
export { EmptyListHeaderDirective };
var EmptyListBodyDirective = /** @class */ (function () {
    function EmptyListBodyDirective() {
    }
    EmptyListBodyDirective.decorators = [
        { type: Directive, args: [{ selector: '[adf-empty-list-body]' },] }
    ];
    return EmptyListBodyDirective;
}());
export { EmptyListBodyDirective };
var EmptyListFooterDirective = /** @class */ (function () {
    function EmptyListFooterDirective() {
    }
    EmptyListFooterDirective.decorators = [
        { type: Directive, args: [{ selector: '[adf-empty-list-footer]' },] }
    ];
    return EmptyListFooterDirective;
}());
export { EmptyListFooterDirective };
