// @alfresco/adf-core/esm5/datatable/components/datatable/location-cell.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectionStrategy, Component, Input, ViewEncapsulation } from '@angular/core';
import { DataTableCellComponent } from './datatable-cell.component';
import { AlfrescoApiService } from '../../../services/alfresco-api.service';
var LocationCellComponent = /** @class */ (function (_super) {
    tslib_1.__extends(LocationCellComponent, _super);
    function LocationCellComponent(alfrescoApiService) {
        return _super.call(this, alfrescoApiService) || this;
    }
    /** @override */
    /**
     * @override
     * @return {?}
     */
    LocationCellComponent.prototype.ngOnInit = /**
     * @override
     * @return {?}
     */
    function () {
        if (this.column && this.column.key && this.row && this.data) {
            /** @type {?} */
            var path = this.data.getValue(this.row, this.column);
            if (path && path.name && path.elements) {
                this.value$.next(path.name.split('/').pop());
                if (!this.tooltip) {
                    this.tooltip = path.name;
                }
                /** @type {?} */
                var parent_1 = path.elements[path.elements.length - 1];
                this.link = [this.column.format, parent_1.id];
            }
        }
    };
    LocationCellComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-location-cell',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: "\n        <ng-container>\n            <a href=\"\" [title]=\"tooltip\" [routerLink]=\"link\">\n                {{ value$ | async }}\n            </a>\n        </ng-container>\n    ",
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-location-cell adf-datatable-content-cell' }
                }] }
    ];
    /** @nocollapse */
    LocationCellComponent.ctorParameters = function () { return [
        { type: AlfrescoApiService }
    ]; };
    LocationCellComponent.propDecorators = {
        link: [{ type: Input }]
    };
    return LocationCellComponent;
}(DataTableCellComponent));
export { LocationCellComponent };
if (false) {
    /** @type {?} */
    LocationCellComponent.prototype.link;
}
