// @alfresco/adf-core/esm5/layout/components/header/header.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, Output, EventEmitter, ViewEncapsulation } from '@angular/core';
var HeaderLayoutComponent = /** @class */ (function () {
    function HeaderLayoutComponent() {
        /**
         * The router link for the application logo, when clicked.
         */
        this.redirectUrl = '/';
        /**
         * Toggles whether the sidenav button will be displayed in the header
         * or not.
         */
        this.showSidenavToggle = true;
        /**
         * Emitted when the sidenav button is clicked.
         */
        this.clicked = new EventEmitter();
        /**
         * The side of the page that the drawer is attached to (can be 'start' or 'end')
         */
        this.position = 'start';
    }
    /**
     * @return {?}
     */
    HeaderLayoutComponent.prototype.toggleMenu = /**
     * @return {?}
     */
    function () {
        this.clicked.emit(true);
    };
    /**
     * @return {?}
     */
    HeaderLayoutComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.logo) {
            this.logo = './assets/images/logo.png';
        }
    };
    HeaderLayoutComponent.decorators = [
        { type: Component, args: [{
                    selector: 'adf-layout-header',
                    template: "<mat-toolbar\n    [color]=\"color\"\n    [style.background-color]=\"color\"\n    role=\"heading\"\n    aria-level=\"1\">\n    <button\n        *ngIf=\"showSidenavToggle && position === 'start'\"\n        id=\"adf-sidebar-toggle-start\"\n        data-automation-id=\"adf-menu-icon\"\n        class=\"mat-icon-button adf-menu-icon\"\n        mat-icon-button\n        (click)=\"toggleMenu()\"\n        aria-label=\"Toggle Menu\">\n        <mat-icon\n            class=\"mat-icon material-icon\"\n            role=\"img\"\n            aria-hidden=\"true\">menu</mat-icon>\n    </button>\n\n    <a [routerLink]=\"redirectUrl\" title=\"{{ tooltip }}\">\n        <img\n            src=\"{{ logo }}\"\n            class=\"adf-app-logo\"\n            alt=\"{{ 'CORE.HEADER.LOGO_ARIA' | translate }}\"\n        />\n    </a>\n\n    <span\n        [routerLink]=\"redirectUrl\"\n        fxFlex=\"1 1 auto\"\n        fxShow\n        fxHide.lt-sm=\"true\"\n        class=\"adf-app-title\"\n        >{{ title }}</span>\n    <ng-content></ng-content>\n\n    <button\n        *ngIf=\"showSidenavToggle && position === 'end'\"\n        id=\"adf-sidebar-toggle-end\"\n        data-automation-id=\"adf-menu-icon\"\n        class=\"mat-icon-button adf-menu-icon\"\n        mat-icon-button\n        (click)=\"toggleMenu()\"\n        aria-label=\"Toggle Menu\">\n        <mat-icon\n            class=\"mat-icon material-icon\"\n            role=\"img\"\n            aria-hidden=\"true\">menu</mat-icon>\n    </button>\n</mat-toolbar>\n",
                    encapsulation: ViewEncapsulation.None,
                    host: { class: 'adf-layout-header' }
                }] }
    ];
    HeaderLayoutComponent.propDecorators = {
        title: [{ type: Input }],
        logo: [{ type: Input }],
        redirectUrl: [{ type: Input }],
        tooltip: [{ type: Input }],
        color: [{ type: Input }],
        showSidenavToggle: [{ type: Input }],
        clicked: [{ type: Output }],
        position: [{ type: Input }]
    };
    return HeaderLayoutComponent;
}());
export { HeaderLayoutComponent };
if (false) {
    /**
     * Title of the application.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.title;
    /**
     * Path to an image file for the application logo.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.logo;
    /**
     * The router link for the application logo, when clicked.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.redirectUrl;
    /**
     * The tooltip text for the application logo.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.tooltip;
    /**
     * Background color for the header. It can be any hex color code or one
     * of the Material theme colors: 'primary', 'accent' or 'warn'.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.color;
    /**
     * Toggles whether the sidenav button will be displayed in the header
     * or not.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.showSidenavToggle;
    /**
     * Emitted when the sidenav button is clicked.
     * @type {?}
     */
    HeaderLayoutComponent.prototype.clicked;
    /**
     * The side of the page that the drawer is attached to (can be 'start' or 'end')
     * @type {?}
     */
    HeaderLayoutComponent.prototype.position;
}
