// @alfresco/adf-testing/esm2015/lib/process-services-cloud/actions/process-instances.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class ProcessInstancesService {
    /**
     * @param {?} api
     */
    constructor(api) {
        this.api = api;
    }
    /**
     * @param {?} processDefKey
     * @param {?} appName
     * @param {?=} options
     * @return {?}
     */
    createProcessInstance(processDefKey, appName, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const path = '/' + appName + '/rb/v1/process-instances';
            /** @type {?} */
            const method = 'POST';
            /** @type {?} */
            const queryParams = {};
            /** @type {?} */
            const postBody = Object.assign({ 'processDefinitionKey': processDefKey, 'payloadType': 'StartProcessPayload' }, options);
            return yield this.api.performBpmOperation(path, method, queryParams, postBody);
        });
    }
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    suspendProcessInstance(processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId + '/suspend';
            /** @type {?} */
            const method = 'POST';
            /** @type {?} */
            const queryParams = {};
            /** @type {?} */
            const postBody = {};
            return yield this.api.performBpmOperation(path, method, queryParams, postBody);
        });
    }
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    deleteProcessInstance(processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId;
            /** @type {?} */
            const method = 'DELETE';
            /** @type {?} */
            const queryParams = {};
            /** @type {?} */
            const postBody = {};
            return yield this.api.performBpmOperation(path, method, queryParams, postBody);
        });
    }
    /**
     * @param {?} processInstanceId
     * @param {?} appName
     * @return {?}
     */
    completeProcessInstance(processInstanceId, appName) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            /** @type {?} */
            const path = '/' + appName + '/rb/v1/process-instances/' + processInstanceId + '/complete';
            /** @type {?} */
            const method = 'POST';
            /** @type {?} */
            const queryParams = {};
            /** @type {?} */
            const postBody = {};
            return yield this.api.performBpmOperation(path, method, queryParams, postBody);
        });
    }
}
if (false) {
    /** @type {?} */
    ProcessInstancesService.prototype.api;
}
