// @alfresco/adf-content-services/esm5/folder-directive/folder-edit.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:no-input-rename  */
import { Directive, ElementRef, HostListener, Input, Output, EventEmitter } from '@angular/core';
import { MatDialog } from '@angular/material';
import { Node } from '@alfresco/js-api';
import { FolderDialogComponent } from '../dialogs/folder.dialog';
import { ContentService } from '@alfresco/adf-core';
var FolderEditDirective = /** @class */ (function () {
    function FolderEditDirective(dialogRef, elementRef, content) {
        this.dialogRef = dialogRef;
        this.elementRef = elementRef;
        this.content = content;
        /**
         * Emitted when an error occurs (eg, a folder with same name already exists).
         */
        this.error = new EventEmitter();
        /**
         * Title of folder edit dialog.
         */
        this.title = null;
        /**
         * Emitted when the folder has been edited successfully.
         */
        this.success = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    FolderEditDirective.prototype.onClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        if (this.folder) {
            this.openDialog();
        }
    };
    Object.defineProperty(FolderEditDirective.prototype, "dialogConfig", {
        get: /**
         * @private
         * @return {?}
         */
        function () {
            var width = FolderEditDirective.DIALOG_WIDTH;
            var folder = this.folder;
            return {
                data: {
                    folder: folder,
                    editTitle: this.title
                },
                width: width + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @private
     * @return {?}
     */
    FolderEditDirective.prototype.openDialog = /**
     * @private
     * @return {?}
     */
    function () {
        var _this = this;
        var _a = this, dialogRef = _a.dialogRef, dialogConfig = _a.dialogConfig, content = _a.content;
        /** @type {?} */
        var dialogInstance = dialogRef.open(FolderDialogComponent, dialogConfig);
        dialogInstance.componentInstance.error.subscribe((/**
         * @param {?} error
         * @return {?}
         */
        function (error) {
            _this.error.emit(error);
        }));
        dialogInstance.componentInstance.success.subscribe((/**
         * @param {?} node
         * @return {?}
         */
        function (node) {
            _this.success.emit(node);
        }));
        dialogInstance.afterClosed().subscribe((/**
         * @param {?} node
         * @return {?}
         */
        function (node) {
            if (node) {
                content.folderEdit.next(node);
            }
        }));
    };
    FolderEditDirective.DIALOG_WIDTH = 400;
    FolderEditDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[adf-edit-folder]'
                },] }
    ];
    /** @nocollapse */
    FolderEditDirective.ctorParameters = function () { return [
        { type: MatDialog },
        { type: ElementRef },
        { type: ContentService }
    ]; };
    FolderEditDirective.propDecorators = {
        folder: [{ type: Input, args: ['adf-edit-folder',] }],
        error: [{ type: Output }],
        title: [{ type: Input }],
        success: [{ type: Output }],
        onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
    };
    return FolderEditDirective;
}());
export { FolderEditDirective };
if (false) {
    /** @type {?} */
    FolderEditDirective.DIALOG_WIDTH;
    /**
     * Folder node to edit.
     * @type {?}
     */
    FolderEditDirective.prototype.folder;
    /**
     * Emitted when an error occurs (eg, a folder with same name already exists).
     * @type {?}
     */
    FolderEditDirective.prototype.error;
    /**
     * Title of folder edit dialog.
     * @type {?}
     */
    FolderEditDirective.prototype.title;
    /**
     * Emitted when the folder has been edited successfully.
     * @type {?}
     */
    FolderEditDirective.prototype.success;
    /** @type {?} */
    FolderEditDirective.prototype.dialogRef;
    /** @type {?} */
    FolderEditDirective.prototype.elementRef;
    /** @type {?} */
    FolderEditDirective.prototype.content;
}
