// @alfresco/js-api/_esm5/src/api/activiti-rest-api/api/adminEndpoints.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { EndpointBasicAuthRepresentation } from '../model/endpointBasicAuthRepresentation';
import { EndpointConfigurationRepresentation } from '../model/endpointConfigurationRepresentation';
import { BaseApi } from './base.api';
var AdminEndpointsApi = (function (_super) {
    tslib_1.__extends(AdminEndpointsApi, _super);
    function AdminEndpointsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AdminEndpointsApi.prototype.createBasicAuthConfiguration = function (createRepresentation) {
        var postBody = createRepresentation;
        if (createRepresentation === undefined || createRepresentation === null) {
            throw new Error("Required param 'createRepresentation' in createBasicAuthConfiguration");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    };
    AdminEndpointsApi.prototype.createEndpointConfiguration = function (representation) {
        var postBody = representation;
        if (representation === undefined || representation === null) {
            throw new Error("Required param 'representation' in createEndpointConfiguration");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    };
    AdminEndpointsApi.prototype.getBasicAuthConfiguration = function (basicAuthId, tenantId) {
        var postBody = null;
        if (basicAuthId === undefined || basicAuthId === null) {
            throw new Error("Required param 'basicAuthId' in getBasicAuthConfiguration");
        }
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in getBasicAuthConfiguration");
        }
        var pathParams = {
            'basicAuthId': basicAuthId
        };
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    };
    AdminEndpointsApi.prototype.getBasicAuthConfigurations = function (tenantId) {
        var postBody = null;
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in getBasicAuthConfigurations");
        }
        var pathParams = {};
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    };
    AdminEndpointsApi.prototype.getEndpointConfiguration = function (endpointConfigurationId, tenantId) {
        var postBody = null;
        if (endpointConfigurationId === undefined || endpointConfigurationId === null) {
            throw new Error("Required param 'endpointConfigurationId' in getEndpointConfiguration");
        }
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in getEndpointConfiguration");
        }
        var pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    };
    AdminEndpointsApi.prototype.getEndpointConfigurations = function (tenantId) {
        var postBody = null;
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in getEndpointConfigurations");
        }
        var pathParams = {};
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    };
    AdminEndpointsApi.prototype.removeBasicAuthConfiguration = function (basicAuthId, tenantId) {
        var postBody = null;
        if (basicAuthId === undefined || basicAuthId === null) {
            throw new Error("Required param 'basicAuthId' in removeBasicAuthonfiguration");
        }
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in removeBasicAuthonfiguration");
        }
        var pathParams = {
            'basicAuthId': basicAuthId
        };
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    AdminEndpointsApi.prototype.removeEndpointConfiguration = function (endpointConfigurationId, tenantId) {
        var postBody = null;
        if (endpointConfigurationId === undefined || endpointConfigurationId === null) {
            throw new Error("Required param 'endpointConfigurationId' in removeEndpointConfiguration");
        }
        if (tenantId === undefined || tenantId === null) {
            throw new Error("Required param 'tenantId' in removeEndpointConfiguration");
        }
        var pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        var queryParams = {
            'tenantId': tenantId
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    AdminEndpointsApi.prototype.updateBasicAuthConfiguration = function (basicAuthId, createRepresentation) {
        var postBody = createRepresentation;
        if (basicAuthId === undefined || basicAuthId === null) {
            throw new Error("Required param 'basicAuthId' in updateBasicAuthConfiguration");
        }
        if (createRepresentation === undefined || createRepresentation === null) {
            throw new Error("Required param 'createRepresentation' in updateBasicAuthConfiguration");
        }
        var pathParams = {
            'basicAuthId': basicAuthId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    };
    AdminEndpointsApi.prototype.updateEndpointConfiguration = function (endpointConfigurationId, representation) {
        var postBody = representation;
        if (endpointConfigurationId === undefined || endpointConfigurationId === null) {
            throw new Error("Required param 'endpointConfigurationId' in updateEndpointConfiguration");
        }
        if (representation === undefined || representation === null) {
            throw new Error("Required param 'representation' in updateEndpointConfiguration");
        }
        var pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    };
    return AdminEndpointsApi;
}(BaseApi));
export { AdminEndpointsApi };
