// @alfresco/js-api/_esm5/src/api/activiti-rest-api/api/content.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { RelatedContentRepresentation } from '../model/relatedContentRepresentation';
import { ResultListDataRepresentationRelatedContentRepresentation } from '../model/resultListDataRepresentationRelatedContentRepresentation';
import { BaseApi } from './base.api';
var ContentApi = (function (_super) {
    tslib_1.__extends(ContentApi, _super);
    function ContentApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ContentApi.prototype.createRelatedContentOnProcessInstance = function (processInstanceId, relatedContent, opts) {
        opts = opts || {};
        if (processInstanceId === undefined || processInstanceId === null) {
            throw new Error("Required param 'processInstanceId' in createRelatedContentOnProcessInstance");
        }
        if (relatedContent === undefined || relatedContent === null) {
            throw new Error("Required param 'relatedContent' in createRelatedContentOnProcessInstance");
        }
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var accepts = ['application/json'];
        if (relatedContent instanceof RelatedContentRepresentation) {
            var postBody = relatedContent;
            var contentTypes = ['application/json'];
            return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
        else {
            var postBody = null;
            formParams = {
                'file': relatedContent
            };
            var contentTypes = ['multipart/form-data'];
            return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/raw-content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
    };
    ContentApi.prototype.createRelatedContentOnTask = function (taskId, relatedContent, opts) {
        opts = opts || {};
        if (taskId === undefined || taskId === null) {
            throw new Error("Required param 'taskId' in createRelatedContentOnTask");
        }
        if (relatedContent === undefined || relatedContent === null) {
            throw new Error("Required param 'relatedContent' in createRelatedContentOnTask");
        }
        var pathParams = {
            'taskId': taskId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var accepts = ['application/json'];
        if (relatedContent instanceof RelatedContentRepresentation) {
            var postBody = relatedContent;
            var contentTypes = ['application/json'];
            return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
        else {
            var postBody = null;
            formParams = {
                'file': relatedContent
            };
            var contentTypes = ['multipart/form-data'];
            return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/raw-content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
    };
    ContentApi.prototype.createTemporaryRawRelatedContent = function (file) {
        var postBody = null;
        if (file === undefined || file === null) {
            throw new Error("Required param 'file' in createTemporaryRawRelatedContent");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {
            'file': file
        };
        var contentTypes = ['multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/raw', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.createTemporaryRelatedContent = function (relatedContent) {
        var postBody = relatedContent;
        if (relatedContent === undefined || relatedContent === null) {
            throw new Error("Required param 'relatedContent' in createTemporaryRelatedContent");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.deleteContent = function (contentId) {
        var postBody = null;
        if (contentId === undefined || contentId === null) {
            throw new Error("Required param 'contentId' in deleteContent");
        }
        var pathParams = {
            'contentId': contentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/{contentId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ContentApi.prototype.getContent = function (contentId) {
        var postBody = null;
        if (contentId === undefined || contentId === null) {
            throw new Error("Required param 'contentId' in getContent");
        }
        var pathParams = {
            'contentId': contentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/{contentId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.getRawContentUrl = function (contentId) {
        return this.apiClient.basePath + "/api/enterprise/content/" + contentId + "/raw";
    };
    ContentApi.prototype.getRawContent = function (contentId, renditionType) {
        var postBody = null;
        if (contentId === undefined || contentId === null) {
            throw new Error("Required param 'contentId' in getRawContent");
        }
        var pathParams = {
            'contentId': contentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json', '*/*'];
        if (renditionType) {
            return this.apiClient.callApi('/api/enterprise/content/{contentId}/rendition/{renditionType}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
        }
        else {
            return this.apiClient.callApi('/api/enterprise/content/{contentId}/raw', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, undefined, undefined, 'blob');
        }
    };
    ContentApi.prototype.getRelatedContentForProcessInstance = function (processInstanceId, opts) {
        opts = opts || {};
        var postBody = null;
        if (processInstanceId === undefined || processInstanceId === null) {
            throw new Error("Required param 'processInstanceId' in getRelatedContentForProcessInstance");
        }
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationRelatedContentRepresentation);
    };
    ContentApi.prototype.getRelatedContentForTask = function (taskId, opts) {
        opts = opts || {};
        var postBody = null;
        if (taskId === undefined || taskId === null) {
            throw new Error("Required param 'taskId' in getRelatedContentForTask");
        }
        var pathParams = {
            'taskId': taskId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationRelatedContentRepresentation);
    };
    return ContentApi;
}(BaseApi));
export { ContentApi };
