// @alfresco/js-api/_esm5/src/api/activiti-rest-api/api/integrationAlfrescoOnPremise.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { ResultListDataRepresentationAlfrescoContentRepresentation } from '../model/resultListDataRepresentationAlfrescoContentRepresentation';
import { ResultListDataRepresentationAlfrescoEndpointRepresentation } from '../model/resultListDataRepresentationAlfrescoEndpointRepresentation';
import { ResultListDataRepresentationAlfrescoSiteRepresenation } from '../model/resultListDataRepresentationAlfrescoSiteRepresenation';
import { BaseApi } from './base.api';
var IntegrationAlfrescoOnPremiseApi = (function (_super) {
    tslib_1.__extends(IntegrationAlfrescoOnPremiseApi, _super);
    function IntegrationAlfrescoOnPremiseApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    IntegrationAlfrescoOnPremiseApi.prototype.getAllSites = function (repositoryId) {
        var postBody = null;
        if (repositoryId === undefined || repositoryId === null) {
            throw new Error("Required param 'repositoryId' in getAllSites");
        }
        var pathParams = {
            'repositoryId': repositoryId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco/{repositoryId}/sites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoSiteRepresenation);
    };
    IntegrationAlfrescoOnPremiseApi.prototype.getContentInFolderPath = function (repositoryId, siteId, folderPath) {
        var postBody = null;
        if (repositoryId === undefined || repositoryId === null) {
            throw new Error("Required param 'repositoryId' in getContentInFolderPath");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getContentInFolderPath");
        }
        if (folderPath === undefined || folderPath === null) {
            throw new Error("Required param 'folderPath' in getContentInFolderPath");
        }
        var pathParams = {
            'repositoryId': repositoryId, 'siteId': siteId, 'folderPath': folderPath
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/rest/integration/alfresco/{repositoryId}/sites/{siteId}/folderpath/{folderPath}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    IntegrationAlfrescoOnPremiseApi.prototype.getContentInFolder = function (repositoryId, folderId) {
        var postBody = null;
        if (repositoryId === undefined || repositoryId === null) {
            throw new Error("Required param 'repositoryId' in getContentInFolder");
        }
        if (folderId === undefined || folderId === null) {
            throw new Error("Required param 'folderId' in getContentInFolder");
        }
        var pathParams = {
            'repositoryId': repositoryId, 'folderId': folderId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco/{repositoryId}/folders/{folderId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    IntegrationAlfrescoOnPremiseApi.prototype.getContentInSite = function (repositoryId, siteId) {
        var postBody = null;
        if (repositoryId === undefined || repositoryId === null) {
            throw new Error("Required param 'repositoryId' in getContentInSite");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getContentInSite");
        }
        var pathParams = {
            'repositoryId': repositoryId, 'siteId': siteId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco/{repositoryId}/sites/{siteId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    IntegrationAlfrescoOnPremiseApi.prototype.getRepositories = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'tenantId': opts['tenantId'],
            'includeAccounts': opts['includeAccounts']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/profile/accounts/alfresco', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoEndpointRepresentation);
    };
    return IntegrationAlfrescoOnPremiseApi;
}(BaseApi));
export { IntegrationAlfrescoOnPremiseApi };
