// @alfresco/js-api/_esm5/src/api/activiti-rest-api/api/integrationAlfrescoCloud.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { ResultListDataRepresentationAlfrescoContentRepresentation } from '../model/resultListDataRepresentationAlfrescoContentRepresentation';
import { ResultListDataRepresentationAlfrescoNetworkRepresenation } from '../model/resultListDataRepresentationAlfrescoNetworkRepresenation';
import { ResultListDataRepresentationAlfrescoSiteRepresenation } from '../model/resultListDataRepresentationAlfrescoSiteRepresenation';
import { BaseApi } from './base.api';
var IntegrationAlfrescoCloudApi = (function (_super) {
    tslib_1.__extends(IntegrationAlfrescoCloudApi, _super);
    function IntegrationAlfrescoCloudApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    IntegrationAlfrescoCloudApi.prototype.confirmAuthorisation = function (code) {
        var postBody = null;
        if (code === undefined || code === null) {
            throw new Error("Required param 'code' in confirmAuthorisation");
        }
        var pathParams = {};
        var queryParams = {
            'code': code
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['text/html'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/confirm-auth-request', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    IntegrationAlfrescoCloudApi.prototype.getAllNetworks = function () {
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/networks', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoNetworkRepresenation);
    };
    IntegrationAlfrescoCloudApi.prototype.getAllSites = function (networkId) {
        var postBody = null;
        if (networkId === undefined || networkId === null) {
            throw new Error("Required param 'networkId' in getAllSites");
        }
        var pathParams = {
            'networkId': networkId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/networks/{networkId}/sites', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoSiteRepresenation);
    };
    IntegrationAlfrescoCloudApi.prototype.getContentInFolderPath = function (networkId, opts) {
        opts = opts || {};
        var postBody = null;
        if (networkId === undefined || networkId === null) {
            throw new Error("Required param 'networkId' in getContentInFolderPath");
        }
        var pathParams = {
            'networkId': networkId
        };
        var queryParams = {
            'siteId': opts['siteId'],
            'path': opts['path']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/networks/{networkId}/sites/{siteId}/folderpath/{folderPath}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    IntegrationAlfrescoCloudApi.prototype.getContentInFolder = function (networkId, folderId) {
        var postBody = null;
        if (networkId === undefined || networkId === null) {
            throw new Error("Required param 'networkId' in getContentInFolder");
        }
        if (folderId === undefined || folderId === null) {
            throw new Error("Required param 'folderId' in getContentInFolder");
        }
        var pathParams = {
            'networkId': networkId, 'folderId': folderId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/networks/{networkId}/folders/{folderId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    IntegrationAlfrescoCloudApi.prototype.getContentInSite = function (networkId, siteId) {
        var postBody = null;
        if (networkId === undefined || networkId === null) {
            throw new Error("Required param 'networkId' in getContentInSite");
        }
        if (siteId === undefined || siteId === null) {
            throw new Error("Required param 'siteId' in getContentInSite");
        }
        var pathParams = {
            'networkId': networkId, 'siteId': siteId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/integration/alfresco-cloud/networks/{networkId}/sites/{siteId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAlfrescoContentRepresentation);
    };
    return IntegrationAlfrescoCloudApi;
}(BaseApi));
export { IntegrationAlfrescoCloudApi };
