// @alfresco/js-api/_esm5/src/api/activiti-rest-api/api/models.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import * as tslib_1 from "tslib";
import { ModelRepresentation } from '../model/modelRepresentation';
import { ObjectNode } from '../model/objectNode';
import { ResultListDataRepresentationModelRepresentation } from '../model/resultListDataRepresentationModelRepresentation';
import { ValidationErrorRepresentation } from '../model/validationErrorRepresentation';
import { BaseApi } from './base.api';
var ModelsApi = (function (_super) {
    tslib_1.__extends(ModelsApi, _super);
    function ModelsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ModelsApi.prototype.createModel = function (modelRepresentation) {
        var postBody = modelRepresentation;
        if (modelRepresentation === undefined || modelRepresentation === null) {
            throw new Error("Required param 'modelRepresentation' in createModel");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.deleteModel = function (modelId, opts) {
        opts = opts || {};
        var postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in deleteModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {
            'cascade': opts['cascade'],
            'deleteRuntimeApp': opts['deleteRuntimeApp']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ModelsApi.prototype.duplicateModel = function (modelId, modelRepresentation) {
        var postBody = modelRepresentation;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in duplicateModel");
        }
        if (modelRepresentation === undefined || modelRepresentation === null) {
            throw new Error("Required param 'modelRepresentation' in duplicateModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/clone', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.getModelJSON = function (modelId) {
        var postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in getModelJSON");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/editor/json', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ObjectNode);
    };
    ModelsApi.prototype.getModelThumbnail = function (modelId) {
        var postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in getModelThumbnail");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['image/png'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/thumbnail', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ModelsApi.prototype.getModel = function (modelId, opts) {
        opts = opts || {};
        var postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in getModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {
            'includePermissions': opts['includePermissions']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.getModelsToIncludeInAppDefinition = function () {
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models-for-app-definition', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationModelRepresentation);
    };
    ModelsApi.prototype.getModels = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'filter': opts['filter'],
            'filterText': opts['filterText'],
            'sort': opts['sort'],
            'modelType': opts['modelType'],
            'referenceId': opts['referenceId']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationModelRepresentation);
    };
    ModelsApi.prototype.importNewVersion = function (modelId, file) {
        var postBody = null;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in importNewVersion");
        }
        if (file === undefined || file === null) {
            throw new Error("Required param 'file' in importNewVersion");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {
            'file': file
        };
        var contentTypes = ['multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/newversion', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.importProcessModel = function (file) {
        var postBody = null;
        if (file === undefined || file === null) {
            throw new Error("Required param 'file' in importProcessModel");
        }
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {
            'file': file
        };
        var contentTypes = ['multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-models/import', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.saveModel = function (modelId, values) {
        var postBody = values;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in saveModel");
        }
        if (values === undefined || values === null) {
            throw new Error("Required param 'values' in saveModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/editor/json', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.updateModel = function (modelId, updatedModel) {
        var postBody = updatedModel;
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in updateModel");
        }
        if (updatedModel === undefined || updatedModel === null) {
            throw new Error("Required param 'updatedModel' in updateModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ModelRepresentation);
    };
    ModelsApi.prototype.validateModel = function (modelId, opts) {
        opts = opts || {};
        var postBody = opts['values'];
        if (modelId === undefined || modelId === null) {
            throw new Error("Required param 'modelId' in validateModel");
        }
        var pathParams = {
            'modelId': modelId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/x-www-form-urlencoded'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/models/{modelId}/editor/validate', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ValidationErrorRepresentation);
    };
    return ModelsApi;
}(BaseApi));
export { ModelsApi };
