// @alfresco/adf-process-services/esm2015/task-list/services/task-upload.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AlfrescoApiService, AppConfigService, UploadService } from '@alfresco/adf-core';
import { Injectable } from '@angular/core';
import { throwError } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@alfresco/adf-core";
export class TaskUploadService extends UploadService {
    /**
     * @param {?} apiService
     * @param {?} appConfigService
     */
    constructor(apiService, appConfigService) {
        super(apiService, appConfigService);
    }
    /**
     * @param {?} file
     * @return {?}
     */
    getUploadPromise(file) {
        /** @type {?} */
        const opts = {
            isRelatedContent: true
        };
        /** @type {?} */
        const taskId = file.options.parentId;
        /** @type {?} */
        const promise = this.apiService.getInstance().activiti.contentApi.createRelatedContentOnTask(taskId, file.file, opts);
        promise.catch((/**
         * @param {?} err
         * @return {?}
         */
        (err) => this.handleError(err)));
        return promise;
    }
    /**
     * @private
     * @param {?} error
     * @return {?}
     */
    handleError(error) {
        return throwError(error || 'Server error');
    }
}
TaskUploadService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
TaskUploadService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: AppConfigService }
];
/** @nocollapse */ TaskUploadService.ngInjectableDef = i0.defineInjectable({ factory: function TaskUploadService_Factory() { return new TaskUploadService(i0.inject(i1.AlfrescoApiService), i0.inject(i1.AppConfigService)); }, token: TaskUploadService, providedIn: "root" });
