// @alfresco/adf-content-services/esm5/document-list/components/content-action/content-action-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component } from '@angular/core';
import { DocumentListComponent } from './../document-list.component';
var ContentActionListComponent = /** @class */ (function () {
    function ContentActionListComponent(documentList) {
        this.documentList = documentList;
    }
    /**
     * Registers action handler within the parent document list component.
     * @param action Action model to register.
     */
    /**
     * Registers action handler within the parent document list component.
     * @param {?} action Action model to register.
     * @return {?}
     */
    ContentActionListComponent.prototype.registerAction = /**
     * Registers action handler within the parent document list component.
     * @param {?} action Action model to register.
     * @return {?}
     */
    function (action) {
        if (this.documentList && action) {
            this.documentList.actions.push(action);
            return true;
        }
        return false;
    };
    /**
     * @param {?} action
     * @return {?}
     */
    ContentActionListComponent.prototype.unregisterAction = /**
     * @param {?} action
     * @return {?}
     */
    function (action) {
        if (this.documentList && action) {
            /** @type {?} */
            var idx = this.documentList.actions.indexOf(action);
            if (idx >= 0) {
                this.documentList.actions.splice(idx, 1);
                return true;
            }
        }
        return false;
    };
    ContentActionListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'content-actions',
                    template: ''
                }] }
    ];
    /** @nocollapse */
    ContentActionListComponent.ctorParameters = function () { return [
        { type: DocumentListComponent }
    ]; };
    return ContentActionListComponent;
}());
export { ContentActionListComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ContentActionListComponent.prototype.documentList;
}
