// @alfresco/adf-core/esm2015/form/components/widgets/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ContainerWidgetComponent } from './container/container.widget';
import { TabsWidgetComponent } from './tabs/tabs.widget';
import { UnknownWidgetComponent } from './unknown/unknown.widget';
import { AmountWidgetComponent } from './amount/amount.widget';
import { CheckboxWidgetComponent } from './checkbox/checkbox.widget';
import { DateWidgetComponent } from './date/date.widget';
import { DisplayTextWidgetComponentComponent } from './display-text/display-text.widget';
import { DocumentWidgetComponent } from './document/document.widget';
import { DropdownWidgetComponent } from './dropdown/dropdown.widget';
import { DynamicTableWidgetComponent } from './dynamic-table/dynamic-table.widget';
import { BooleanEditorComponent } from './dynamic-table/editors/boolean/boolean.editor';
import { DateEditorComponent } from './dynamic-table/editors/date/date.editor';
import { DateTimeEditorComponent } from './dynamic-table/editors/datetime/datetime.editor';
import { DropdownEditorComponent } from './dynamic-table/editors/dropdown/dropdown.editor';
import { RowEditorComponent } from './dynamic-table/editors/row.editor';
import { TextEditorComponent } from './dynamic-table/editors/text/text.editor';
import { ErrorWidgetComponent } from './error/error.component';
import { FunctionalGroupWidgetComponent } from './functional-group/functional-group.widget';
import { HyperlinkWidgetComponent } from './hyperlink/hyperlink.widget';
import { MultilineTextWidgetComponentComponent } from './multiline-text/multiline-text.widget';
import { NumberWidgetComponent } from './number/number.widget';
import { PeopleWidgetComponent } from './people/people.widget';
import { RadioButtonsWidgetComponent } from './radio-buttons/radio-buttons.widget';
import { InputMaskDirective } from './text/text-mask.component';
import { TextWidgetComponent } from './text/text.widget';
import { TypeaheadWidgetComponent } from './typeahead/typeahead.widget';
import { UploadWidgetComponent } from './upload/upload.widget';
import { DateTimeWidgetComponent } from './date-time/date-time.widget';
// core
export { baseHost, WidgetComponent } from './widget.component';
export { FormFieldTypes, FormWidgetModel, FormFieldModel, FormModel, ContainerModel, ContainerColumnModel, TabModel, FormOutcomeModel, FormOutcomeEvent, RequiredFieldValidator, NumberFieldValidator, DateFieldValidator, MinDateFieldValidator, MaxDateFieldValidator, MinDateTimeFieldValidator, MaxDateTimeFieldValidator, MinLengthFieldValidator, MaxLengthFieldValidator, MinValueFieldValidator, MaxValueFieldValidator, RegExFieldValidator, FixedValueFieldValidator, FORM_FIELD_VALIDATORS, ContentLinkModel, ErrorMessageModel } from './core/index';
// containers
export { TabsWidgetComponent } from './tabs/tabs.widget';
export { ContainerWidgetComponent } from './container/container.widget';
// primitives
export { UnknownWidgetComponent } from './unknown/unknown.widget';
export { TextWidgetComponent } from './text/text.widget';
export { NumberWidgetComponent } from './number/number.widget';
export { CheckboxWidgetComponent } from './checkbox/checkbox.widget';
export { MultilineTextWidgetComponentComponent } from './multiline-text/multiline-text.widget';
export { DropdownWidgetComponent } from './dropdown/dropdown.widget';
export { HyperlinkWidgetComponent } from './hyperlink/hyperlink.widget';
export { RadioButtonsWidgetComponent } from './radio-buttons/radio-buttons.widget';
export { DisplayTextWidgetComponentComponent } from './display-text/display-text.widget';
export { UploadWidgetComponent } from './upload/upload.widget';
export { TypeaheadWidgetComponent } from './typeahead/typeahead.widget';
export { FunctionalGroupWidgetComponent } from './functional-group/functional-group.widget';
export { PeopleWidgetComponent } from './people/people.widget';
export { DateWidgetComponent } from './date/date.widget';
export { AmountWidgetComponent } from './amount/amount.widget';
export { DynamicTableWidgetComponent } from './dynamic-table/dynamic-table.widget';
export { ErrorWidgetComponent } from './error/error.component';
export { DocumentWidgetComponent } from './document/document.widget';
export { DateTimeWidgetComponent } from './date-time/date-time.widget';
// editors (dynamic table)
export { DynamicTableModel } from './dynamic-table/dynamic-table.widget.model';
export { RowEditorComponent } from './dynamic-table/editors/row.editor';
export { DateEditorComponent } from './dynamic-table/editors/date/date.editor';
export { DropdownEditorComponent } from './dynamic-table/editors/dropdown/dropdown.editor';
export { BooleanEditorComponent } from './dynamic-table/editors/boolean/boolean.editor';
export { TextEditorComponent } from './dynamic-table/editors/text/text.editor';
export { DateTimeEditorComponent } from './dynamic-table/editors/datetime/datetime.editor';
export { CUSTOM_INPUT_CONTROL_VALUE_ACCESSOR, InputMaskDirective } from './text/text-mask.component';
/** @type {?} */
export const WIDGET_DIRECTIVES = [
    UnknownWidgetComponent,
    TabsWidgetComponent,
    ContainerWidgetComponent,
    TextWidgetComponent,
    NumberWidgetComponent,
    CheckboxWidgetComponent,
    MultilineTextWidgetComponentComponent,
    DropdownWidgetComponent,
    HyperlinkWidgetComponent,
    RadioButtonsWidgetComponent,
    DisplayTextWidgetComponentComponent,
    UploadWidgetComponent,
    TypeaheadWidgetComponent,
    FunctionalGroupWidgetComponent,
    PeopleWidgetComponent,
    DateWidgetComponent,
    AmountWidgetComponent,
    DynamicTableWidgetComponent,
    DateEditorComponent,
    DropdownEditorComponent,
    BooleanEditorComponent,
    TextEditorComponent,
    RowEditorComponent,
    ErrorWidgetComponent,
    DocumentWidgetComponent,
    DateTimeWidgetComponent,
    DateTimeEditorComponent
];
/** @type {?} */
export const MASK_DIRECTIVE = [
    InputMaskDirective
];
