// @alfresco/adf-core/esm2015/form/components/widgets/core/tab.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { FormWidgetModel } from './form-widget.model';
export class TabModel extends FormWidgetModel {
    /**
     * @param {?} form
     * @param {?=} json
     */
    constructor(form, json) {
        super(form, json);
        this.isVisible = true;
        this.fields = [];
        if (json) {
            this.title = json.title;
            this.visibilityCondition = (/** @type {?} */ (json.visibilityCondition));
        }
    }
    /**
     * @return {?}
     */
    hasContent() {
        return this.fields && this.fields.length > 0;
    }
}
if (false) {
    /** @type {?} */
    TabModel.prototype.title;
    /** @type {?} */
    TabModel.prototype.isVisible;
    /** @type {?} */
    TabModel.prototype.visibilityCondition;
    /** @type {?} */
    TabModel.prototype.fields;
}
