// @alfresco/adf-insights/esm5/diagram/components/structural/diagram-subprocess.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* tslint:disable:component-selector  */
import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
import { DiagramColorService } from '../../services/diagram-color.service';
var DiagramSubprocessComponent = /** @class */ (function () {
    function DiagramSubprocessComponent(elementRef, diagramColorService) {
        this.elementRef = elementRef;
        this.diagramColorService = diagramColorService;
        this.error = new EventEmitter();
        this.options = { stroke: '', fillColors: '', fillOpacity: '', strokeWidth: '', radius: 4 };
    }
    /**
     * @return {?}
     */
    DiagramSubprocessComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.rectLeftCorner = { x: this.data.x, y: this.data.y };
        this.width = this.data.width;
        this.height = this.data.height;
        this.options.fillColors = 'none';
        this.options.stroke = this.diagramColorService.getBpmnColor(this.data, DiagramColorService.MAIN_STROKE_COLOR);
        this.options.strokeWidth = 1;
    };
    DiagramSubprocessComponent.decorators = [
        { type: Component, args: [{
                    selector: 'diagram-subprocess',
                    template: "<raphael-rect [leftCorner]=\"rectLeftCorner\" [width]=\"width\" [height]=\"height\" [radius]=\"options.radius\"\n              [stroke]=\"options.stroke\" [strokeWidth]=\"options.strokeWidth\"\n              [fillColors]=\"options.fillColors\" [fillOpacity]=\"options.fillOpacity\"></raphael-rect>\n<diagram-tooltip [data]=\"data\"></diagram-tooltip>\n"
                }] }
    ];
    /** @nocollapse */
    DiagramSubprocessComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: DiagramColorService }
    ]; };
    DiagramSubprocessComponent.propDecorators = {
        data: [{ type: Input }],
        error: [{ type: Output }]
    };
    return DiagramSubprocessComponent;
}());
export { DiagramSubprocessComponent };
if (false) {
    /** @type {?} */
    DiagramSubprocessComponent.prototype.data;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.error;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.rectLeftCorner;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.width;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.height;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.options;
    /** @type {?} */
    DiagramSubprocessComponent.prototype.elementRef;
    /**
     * @type {?}
     * @private
     */
    DiagramSubprocessComponent.prototype.diagramColorService;
}
