// @alfresco/adf-testing/esm2015/lib/core/utils/string.util.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class StringUtil {
    /**
     * @param {?=} length
     * @return {?}
     */
    static generatePasswordString(length = 8) {
        /** @type {?} */
        let text = '';
        /** @type {?} */
        const possibleUpperCase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
        /** @type {?} */
        const possibleLowerCase = 'abcdefghijklmnopqrstuvwxyz';
        /** @type {?} */
        const lowerCaseLimit = Math.floor(length / 2);
        for (let i = 0; i < lowerCaseLimit; i++) {
            text += possibleLowerCase.charAt(Math.floor(Math.random() * possibleLowerCase.length));
        }
        for (let i = 0; i < length - lowerCaseLimit; i++) {
            text += possibleUpperCase.charAt(Math.floor(Math.random() * possibleUpperCase.length));
        }
        return text;
    }
    /**
     * Generates a random string.
     *
     * \@method generateRandomString
     * @param {?=} length If this parameter is not provided the length is set to 8 by default.
     * @return {?}
     */
    static generateRandomString(length = 8) {
        /** @type {?} */
        let text = '';
        /** @type {?} */
        const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        for (let i = 0; i < length; i++) {
            text += possible.charAt(Math.floor(Math.random() * possible.length));
        }
        return text;
    }
    /**
     * Generates a random email address following the format: abcdef\@activiti.test.com
     *
     * \@method generateRandomEmail
     * @param {?} domain
     * @param {?=} length
     * @return {?}
     */
    static generateRandomEmail(domain, length = 5) {
        /** @type {?} */
        let email = '';
        /** @type {?} */
        const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        for (let i = 0; i < length; i++) {
            email += possible.charAt(Math.floor(Math.random() * possible.length));
        }
        email += domain;
        return email.toLowerCase();
    }
    /**
     * Generates a random string - digits only.
     *
     * \@method generateRandomString
     * @param {?=} length {int} If this parameter is not provided the length is set to 8 by default.
     * @return {?}
     */
    static generateRandomStringDigits(length = 8) {
        /** @type {?} */
        let text = '';
        /** @type {?} */
        const possible = '0123456789';
        for (let i = 0; i < length; i++) {
            text += possible.charAt(Math.floor(Math.random() * possible.length));
        }
        return text;
    }
    /**
     * Generates a random string - non-latin characters only.
     *
     * \@method generateRandomString
     * @param {?=} length {int} If this parameter is not provided the length is set to 3 by default.
     * @return {?}
     */
    static generateRandomStringNonLatin(length = 3) {
        /** @type {?} */
        let text = '';
        /** @type {?} */
        const possible = '密码你好𠮷';
        for (let i = 0; i < length; i++) {
            text += possible.charAt(Math.floor(Math.random() * possible.length));
        }
        return text;
    }
}
