// @alfresco/adf-testing/esm2015/lib/core/pages/error.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { BrowserVisibility } from '../utils/browser-visibility';
import { element, by } from 'protractor';
export class ErrorPage {
    constructor() {
        this.errorPageCode = element(by.css('adf-error-content .adf-error-content-code'));
        this.errorPageTitle = element(by.css('adf-error-content .adf-error-content-title'));
        this.errorPageDescription = element(by.css('adf-error-content .adf-error-content-description'));
        this.backButton = element(by.id('adf-return-button'));
        this.secondButton = element(by.id('adf-secondary-button'));
    }
    /**
     * @return {?}
     */
    clickBackButton() {
        BrowserVisibility.waitUntilElementIsVisible(this.backButton);
        this.backButton.click();
    }
    /**
     * @return {?}
     */
    clickSecondButton() {
        BrowserVisibility.waitUntilElementIsVisible(this.secondButton);
        this.secondButton.click();
    }
    /**
     * @return {?}
     */
    checkErrorCode() {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageCode);
    }
    /**
     * @return {?}
     */
    getErrorCode() {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageCode);
        return this.errorPageCode.getText();
    }
    /**
     * @return {?}
     */
    getErrorTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageTitle);
        return this.errorPageTitle.getText();
    }
    /**
     * @return {?}
     */
    getErrorDescription() {
        BrowserVisibility.waitUntilElementIsVisible(this.errorPageDescription);
        return this.errorPageDescription.getText();
    }
}
if (false) {
    /** @type {?} */
    ErrorPage.prototype.errorPageCode;
    /** @type {?} */
    ErrorPage.prototype.errorPageTitle;
    /** @type {?} */
    ErrorPage.prototype.errorPageDescription;
    /** @type {?} */
    ErrorPage.prototype.backButton;
    /** @type {?} */
    ErrorPage.prototype.secondButton;
}
