// @alfresco/adf-testing/esm2015/lib/core/pages/login-sso.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by, browser, protractor } from 'protractor';
import { BrowserVisibility } from '../utils/browser-visibility';
export class LoginSSOPage {
    constructor() {
        this.ssoButton = element(by.css(`[data-automation-id="login-button-sso"]`));
        this.usernameField = element(by.id('username'));
        this.passwordField = element(by.id('password'));
        this.loginButton = element(by.css('input[class="submit"]'));
        this.header = element(by.id('adf-header'));
        this.loginError = element(by.css(`div[data-automation-id="login-error"]`));
    }
    /**
     * @param {?} username
     * @param {?} password
     * @return {?}
     */
    loginSSOIdentityService(username, password) {
        browser.ignoreSynchronization = true;
        BrowserVisibility.waitUntilElementIsVisible(this.usernameField);
        this.enterUsername(username);
        this.enterPassword(password);
        this.clickLoginButton();
        browser.actions().sendKeys(protractor.Key.ENTER).perform();
        return BrowserVisibility.waitUntilElementIsVisible(this.header);
    }
    /**
     * @return {?}
     */
    clickOnSSOButton() {
        BrowserVisibility.waitUntilElementIsVisible(this.ssoButton);
        this.ssoButton.click();
    }
    /**
     * @param {?} username
     * @return {?}
     */
    enterUsername(username) {
        BrowserVisibility.waitUntilElementIsVisible(this.usernameField);
        this.usernameField.clear();
        this.usernameField.sendKeys(username);
    }
    /**
     * @param {?} password
     * @return {?}
     */
    enterPassword(password) {
        BrowserVisibility.waitUntilElementIsVisible(this.passwordField);
        this.passwordField.clear();
        this.passwordField.sendKeys(password);
    }
    /**
     * @return {?}
     */
    clickLoginButton() {
        BrowserVisibility.waitUntilElementIsVisible(this.loginButton);
        return this.loginButton.click();
    }
    /**
     * @return {?}
     */
    checkLoginErrorIsDisplayed() {
        BrowserVisibility.waitUntilElementIsVisible(this.loginError);
    }
    /**
     * @return {?}
     */
    getLoginErrorMessage() {
        BrowserVisibility.waitUntilElementIsVisible(this.loginError);
        return this.loginError.getText();
    }
}
if (false) {
    /** @type {?} */
    LoginSSOPage.prototype.ssoButton;
    /** @type {?} */
    LoginSSOPage.prototype.usernameField;
    /** @type {?} */
    LoginSSOPage.prototype.passwordField;
    /** @type {?} */
    LoginSSOPage.prototype.loginButton;
    /** @type {?} */
    LoginSSOPage.prototype.header;
    /** @type {?} */
    LoginSSOPage.prototype.loginError;
}
