// @alfresco/adf-testing/esm2015/lib/core/pages/user-info.page.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { element, by, browser, protractor } from 'protractor';
import { BrowserVisibility } from '../utils/browser-visibility';
import { TabsPage } from '../../material/pages/tabs.page';
export class UserInfoPage {
    constructor() {
        this.dialog = element.all(by.css('mat-card[class*="adf-userinfo-card"]')).first();
        this.userImage = element(by.css('div[id="user-initial-image"]'));
        this.userInfoEcmHeaderTitle = element(by.css('div[id="ecm-username"]'));
        this.userInfoEcmTitle = element(by.css('mat-card-content span[id="ecm-full-name"]'));
        this.ecmEmail = element(by.css('span[id="ecm-email"]'));
        this.ecmJobTitle = element(by.css('span[id="ecm-job-title"]'));
        this.userInfoProcessHeaderTitle = element(by.css('div[id="bpm-username"]'));
        this.userInfoProcessTitle = element(by.css('mat-card-content span[id="bpm-full-name"]'));
        this.processEmail = element(by.css('span[id="bpm-email"]'));
        this.processTenant = element(by.css('span[class="detail-profile"]'));
        this.apsImage = element(by.css('img[id="bpm-user-detail-image"]'));
        this.acsImage = element(by.css('img[id="ecm-user-detail-image"]'));
        this.initialImage = element.all(by.css('div[id="user-initials-image"]')).first();
        this.userInfoSsoHeaderTitle = this.dialog.element(by.css('div[id="identity-username"]'));
        this.userInfoSsoTitle = element(by.css('.adf-userinfo__detail-title'));
        this.ssoEmail = element(by.id('identity-email'));
        this.userProfileButton = element(by.css('button[data-automation-id="adf-user-profile"]'));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    dialogIsDisplayed() {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).dialog);
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    dialogIsNotDisplayed() {
        BrowserVisibility.waitUntilElementIsNotOnPage((/** @type {?} */ (this)).dialog);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    clickUserProfile() {
        BrowserVisibility.waitUntilElementIsVisible(this.userProfileButton);
        this.userProfileButton.click();
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    clickOnContentServicesTab() {
        /** @type {?} */
        const tabsPage = new TabsPage();
        tabsPage.clickTabByTitle('Content Services');
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    checkProcessServicesTabIsSelected() {
        /** @type {?} */
        const tabsPage = new TabsPage;
        tabsPage.checkTabIsSelectedByTitle('Process Services');
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    clickOnProcessServicesTab() {
        /** @type {?} */
        const tabsPage = new TabsPage;
        tabsPage.clickTabByTitle('Process Services');
        return (/** @type {?} */ (this));
    }
    /**
     * @template THIS
     * @this {THIS}
     * @return {THIS}
     */
    userImageIsDisplayed() {
        BrowserVisibility.waitUntilElementIsVisible((/** @type {?} */ (this)).userImage);
        return (/** @type {?} */ (this));
    }
    /**
     * @return {?}
     */
    getContentHeaderTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.dialog);
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoEcmHeaderTitle);
        return this.userInfoEcmHeaderTitle.getText();
    }
    /**
     * @return {?}
     */
    getContentTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoEcmTitle);
        return this.userInfoEcmTitle.getText();
    }
    /**
     * @return {?}
     */
    getContentEmail() {
        BrowserVisibility.waitUntilElementIsVisible(this.ecmEmail);
        return this.ecmEmail.getText();
    }
    /**
     * @return {?}
     */
    getContentJobTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.ecmJobTitle);
        return this.ecmJobTitle.getText();
    }
    /**
     * @return {?}
     */
    getProcessHeaderTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoProcessHeaderTitle);
        return this.userInfoProcessHeaderTitle.getText();
    }
    /**
     * @return {?}
     */
    getProcessTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoProcessTitle);
        return this.userInfoProcessTitle.getText();
    }
    /**
     * @return {?}
     */
    getProcessEmail() {
        BrowserVisibility.waitUntilElementIsVisible(this.processEmail);
        return this.processEmail.getText();
    }
    /**
     * @return {?}
     */
    getProcessTenant() {
        BrowserVisibility.waitUntilElementIsVisible(this.processTenant);
        return this.processTenant.getText();
    }
    /**
     * @return {?}
     */
    getSsoHeaderTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoSsoHeaderTitle);
        return this.userInfoSsoHeaderTitle.getText();
    }
    /**
     * @return {?}
     */
    getSsoTitle() {
        BrowserVisibility.waitUntilElementIsVisible(this.userInfoSsoTitle);
        return this.userInfoSsoTitle.getText();
    }
    /**
     * @return {?}
     */
    getSsoEmail() {
        BrowserVisibility.waitUntilElementIsVisible(this.ssoEmail);
        return this.ssoEmail.getText();
    }
    /**
     * @return {?}
     */
    closeUserProfile() {
        BrowserVisibility.waitUntilElementIsVisible(this.dialog);
        browser.actions().sendKeys(protractor.Key.ESCAPE).perform();
    }
    /**
     * @return {?}
     */
    checkACSProfileImage() {
        BrowserVisibility.waitUntilElementIsVisible(this.acsImage);
    }
    /**
     * @return {?}
     */
    checkAPSProfileImage() {
        BrowserVisibility.waitUntilElementIsVisible(this.apsImage);
    }
    /**
     * @return {?}
     */
    checkInitialImage() {
        BrowserVisibility.waitUntilElementIsVisible(this.initialImage);
    }
    /**
     * @return {?}
     */
    initialImageNotDisplayed() {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.initialImage);
    }
    /**
     * @return {?}
     */
    ACSProfileImageNotDisplayed() {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.acsImage);
    }
    /**
     * @return {?}
     */
    APSProfileImageNotDisplayed() {
        BrowserVisibility.waitUntilElementIsNotOnPage(this.apsImage);
    }
}
if (false) {
    /** @type {?} */
    UserInfoPage.prototype.dialog;
    /** @type {?} */
    UserInfoPage.prototype.userImage;
    /** @type {?} */
    UserInfoPage.prototype.userInfoEcmHeaderTitle;
    /** @type {?} */
    UserInfoPage.prototype.userInfoEcmTitle;
    /** @type {?} */
    UserInfoPage.prototype.ecmEmail;
    /** @type {?} */
    UserInfoPage.prototype.ecmJobTitle;
    /** @type {?} */
    UserInfoPage.prototype.userInfoProcessHeaderTitle;
    /** @type {?} */
    UserInfoPage.prototype.userInfoProcessTitle;
    /** @type {?} */
    UserInfoPage.prototype.processEmail;
    /** @type {?} */
    UserInfoPage.prototype.processTenant;
    /** @type {?} */
    UserInfoPage.prototype.apsImage;
    /** @type {?} */
    UserInfoPage.prototype.acsImage;
    /** @type {?} */
    UserInfoPage.prototype.initialImage;
    /** @type {?} */
    UserInfoPage.prototype.userInfoSsoHeaderTitle;
    /** @type {?} */
    UserInfoPage.prototype.userInfoSsoTitle;
    /** @type {?} */
    UserInfoPage.prototype.ssoEmail;
    /** @type {?} */
    UserInfoPage.prototype.userProfileButton;
}
