// @alfresco/adf-testing/esm2015/lib/core/models/user.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { StringUtil } from '../utils/string.util';
export class UserModel {
    /**
     * @param {?=} details
     */
    constructor(details) {
        this.firstName = StringUtil.generateRandomString();
        this.lastName = StringUtil.generateRandomString() + 'LastName';
        this.password = StringUtil.generateRandomString();
        this.email = StringUtil.generateRandomEmail('@alfresco.com');
        this.username = StringUtil.generateRandomString().toLowerCase();
        Object.assign(this, details);
    }
    /**
     * @return {?}
     */
    get id() {
        return this.email;
    }
}
if (false) {
    /** @type {?} */
    UserModel.prototype.firstName;
    /** @type {?} */
    UserModel.prototype.lastName;
    /** @type {?} */
    UserModel.prototype.password;
    /** @type {?} */
    UserModel.prototype.email;
    /** @type {?} */
    UserModel.prototype.username;
    /** @type {?} */
    UserModel.prototype.idIdentityService;
}
